#+title: Bisonex
#+category: bisonex
* Idées
** Validation analytique
mail Yannis : données patients +/- simulées
*** Utiliser données GCAT et uploader le notre ?
https://www.nature.com/articles/ncomms7275
*** [#A] Variant calling : Genome in a bottle : NA12878 + autres
Résumé : https://www.nist.gov/programs-projects/genome-bottle
Manuscript : https://www.nature.com/articles/s41587-019-0054-x.epdf?author_access_token=E_1bL0MtBBwZr91xEsy6B9RgN0jAjWel9jnR3ZoTv0OLNnFBR7rUIZNDXq0DIKdg3w6KhBF8Rz2RWQFFc0St45kC6CZs3cDYc87HNHovbWSOubJHDa9CeJV-pN0BW_mQ0n7cM13KF2JRr_wAAn524w%3D%3D

Article comparant les variant calling : https://www.biorxiv.org/content/10.1101/2020.12.11.422022v1.full.pdf
**** Tester le séquencage aussi
Depuis un fastq correspondant à Illumina  https://github.com/genome-in-a-bottle/giab_data_indexes
   puis on compare le VCF avec les "high confidence"
On séquence directement NA12878 -> inutile pour le pipeline seul
**** Tester seul la partie bioinformatique
   Tout résumé ici : https://www.nist.gov/programs-projects/genome-bottle

- methode https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/giab/data/NA12878/analysis/Illumina_PlatinumGenomes_NA12877_NA12878_09162015/IlluminaPlatinumGenomes-user-guide.pdf
- vcf
     https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/giab/release/NA12878_HG001/latest/GRCh38/

NB: à quoi correspond https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/giab/data/NA12878/analysis/Illumina_PlatinumGenomes_NA12877_NA12878_09162015/hg38/2.0.1/NA12878/ ??

   Article comparant les variant calling : https://www.biorxiv.org/content/10.1101/2020.12.11.422022v1.full.pdf
   Article pour vcfeval : https://www.nature.com/articles/s41587-019-0054-x

   La version 4 ajoute 273 gènes "clinically relevant" https://www.biorxiv.org/content/10.1101/2021.06.07.444885v3.full.pdf

   Ajout des zones "difficiles"
   https://www.biorxiv.org/content/10.1101/2020.07.24.212712v5.full.pdf

*** [#B] Pipeline : générer patient avec tous les variants retrouvés à Cento
Comparaison de génération ADN (2019)
https://academic.oup.com/bfg/article/19/1/49/5680294
**** SimuSCop (exome)
https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03665-5
https://github.com/qasimyu/simuscop
1. Crééer un modèle depuis bam + vcf : Setoprofile
2. Génerer données NGS
** Annotation :
*** Comparaison vep / snpeff et annovar

* Changement nouvelle version
- Dernière version du génome (la version "prête à l'emploi" est seulement GRCh38 sans les version patchées)
* Notes
** Nextflow
*** afficher les résultats d'un process/workflow
#+begin_src
lol.out.view()
#+end_src
Attention, ne fonctionne pas si plusieurs sortie:
#+begin_src
lol.out[0].view()
#+end_src
ou si /a/ est le nom de la sortie
#+begin_src
lol.out.a.view()
#+end_src
** Quelle version du génome ?
- T2T: notation chromose = chR1,2 : ok genome, clinvar, dbSNP
- GRCh38: notation chromose = NC_... : ok genome, clinvar, dbSNP

** Performances
Ordinateur de Carine (WSL2) : 4h dont 1h15 alignement (parallélisé) et 1h15 haplotypecaller (séquentiel)
** Chromosomes NC, NT, NW
Correspondance :
https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&chromInfoPage=
Signification
https://genome.ucsc.edu/FAQ/FAQdownloads.html#downloadAlt
- alt = séquences alternatives (utilisables)
- fix = patch (correction ou amélioration)
- random = séquence connue sur un chromosome mais non encore utilisée
** Pipelines prêt-à-l’emploi nextflow
Problème : nécessite singularity ou docker (ou conda)
Potentiellement utilisable avec nix...
** Validation : Quelles données de référence ?
Discussion avec Alexis
- Platinum genomes = génome seul
*** [[https://github.com/genome-in-a-bottle/giab_data_indexes][Genome in a bottle]]
  - NA12878 :
    - Illumina HiSeq Exome : fastq + capture en hg37
    - Illumina TruSeq Exome : bam, pas de capture
    - Exomes en hg37 https://zenodo.org/record/3597727 avec capture
      - HiSeq2000
      - NextSeq 500
      - HiSeq 2500
  - HG002,3,4
    - Illumina Whole Exome  : bam. le kit de capture est "Agilent SureSelect Human All Exon V5 kit" selon [[https://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/analysis/OsloUniversityHospital_Exome_GATK_jointVC_11242015/README.txt][README]]. On il faut les régions [[https://kb.10xgenomics.com/hc/en-us/articles/115004150923-Where-can-I-find-the-Agilent-Target-BED-files-][selon ce site]]

      Un autre fichier est disponible (capture ???)
    https://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/analysis/OsloUniversityHospital_Exome_GATK_jointVC_11242015/wex_Agilent_SureSelect_v05_b37.baits.slop50.merged.list
  "target region" +/- 50bp
    testé sur chr311780-312086 : ok

Autres technologies non adaptées au pipeline (vu avec Alexis)
*** [[https://www.illumina.com/platinumgenomes.html][Platinum genome
]] Que du génome « sequenced to 50x depth on a HiSeq 2000 system”
Genome possible
*** 1000 genomes
- intersection des capture + CCDS  [[id:b77e64fa-06a8-4ffa-8b5b-ab3fda684b61][Données brutes exome 1000 Genomes (fastq + capture)]]
- Broad instute : SureSelect human all exon v2 target capture kit : non disponible sur le site d'agilent (V6 ou plus)
*** Zone de capture
GIAB fourni le .bed pour l'exome . INfo : https://support.illumina.com/sequencing/sequencing_kits/nextera-rapid-capture-exome-kit/downloads.html
*** Valider la méthode
- 1000 genomes + SureSelect human all exon v2 target capture kit : non disponible sur le site d'agilent (V6 ou plus)
  https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-2928-9
- GIAB + liftover du fichire de capture en hg38
Ce qui est aussi fait par
https://bcbio-nextgen.readthedocs.io/en/stable/contents/germline_variants.html
Mais avec UCSC liftover
** Centogène
https://www.twistbioscience.com/node/23906
Bed non fourni pour exactement cette capture
On prend https://www.twistbioscience.com/resources/data-files/twist-alliance-vcgs-exome-401mb-bed-files
qui content la majeure partie
* Réunion
** <2023-08-10 Thu> Alexis
Ok pour bloquer le développment d'ici mardi prochain

Dév:
- pipeline jusque VEP en T2T + GRCh38
- ok pour valider spip T2T sur quelques variant => à intégrer au pipeline
- annotation :
  - ok pour mobidetails hg38
  - +OMIM T2T+ non
  - +franklin hg38+ non pour le moment
- métriques (fastq a minima) + rapport multiqc
- optionnel
  - reformater la sortie
- on abandonne
  - XAMScissors ave indel
  - parallélisation haplotype caller
  - spliceai à la vollée
  - pangolin

Test
- GIAB:
  - hg38: ok pour refaire les tests NA12878 avec données cento, sinon ok pour "c'est difficile" sur les 3 fichiers de capture
  - T2T: ok pour faire des tests rapides mais probablement pas assez de temps !
- patient de synthèse : variant cento confirém par sanger seuls

Résultats
- ok pour scale up bwa mem et haplotyecaller

Manuscrit
- validation de méthode : laisser tomber la version actuelle et faire comme strasbourg (cf ngs diag) dans la présentatino
- a envoyé le powerponit avec les références des différsences articles
- ok pour robo4 si résultat
- architecture cible = VM : 78 coeurs 54Go RAUM et 1To espace disque

Passage en production : ok pour présentation rapide du code
* Nixpkgs :nix:
** DONE GATK
CLOSED: [2023-05-06 Sat 08:51]
*** DONE [[https://github.com/NixOS/nixpkgs/pull/185819][Binaire]]
CLOSED: [2022-09-10 Sat 23:53] SCHEDULED: <2022-08-10 Wed>
/Entered on/ [2022-08-09 Tue 10:57]
PR submitted
*** KILL Corriger code pour utiliser source
CLOSED: [2022-09-11 Sun 22:05]
*** DONE Corriger PATH pour include java et python
CLOSED: [2022-10-11 Tue 11:46]
https://github.com/NixOS/nixpkgs/pull/191548
Review <2022-10-10 Mon> , corrigé dans la journée
*** DONE Update 4.3.0.0
CLOSED: [2023-04-13 Thu 09:01]
** HOLD Nextflow
*** KILL version script seule
CLOSED: [2023-04-01 Sat 18:29]
Fix pour SGE et nextflow
https://github.com/NixOS/nixpkgs/issues/192396
*** KILL Version avec gradle
CLOSED: [2022-10-09 Sun 22:51]
*** HOLD [[https://github.com/NixOS/nixpkgs/issues/192396][Bug report Version 22.10.6]]
**** Notes
Erreur :
ERROR: Cannot download nextflow required file -- make sure you can connect to the internet

Alternatively you can try to download this file:
    https://www.nextflow.io/releases/v22.10.6/nextflow-22.10.6-all.jar

and save it as:
    .//nix/store/md2b1ah4d7ivj82k8xxap30dmdci00pa-nextflow-22.10.6/bin/.nextflow-wrapped

Dans la mise à jour, il y a la création d'un environnement virtuel qui casse l'exécution de nextflow (besoin de télécharger)
Fix = désactiver
**** KILL Patch NXF_OFFLINE=true
CLOSED: [2023-07-02 Sun 11:02] SCHEDULED: <2023-06-11 Sun>
** WAIT [[https://github.com/NixOS/nixpkgs/pull/249329][Multiqc]]
HG002,sanger-chr20,data/HG002-sanger-inserted-chr20_1.fq.gz,data/HG002-sanger-inserted-chr20_2.fq.gz
** KILL Mutalyzer
CLOSED: [2023-08-16 Wed 19:07] SCHEDULED: <2023-08-13 Sun>
Packaging faisable mais nombreux paquet python
** TODO Variant validator -> hgvs
C'est juste une interface autour d'hgvs mais il faut
- postgresql
- un accès ou télécharger des bases de données

  Dépendences
  s: wcwidth, pyee, pure-eval, ptyprocess, pickleshare, parsley, parse, fake-useragent, executing, backcall, appdirs, zipp, websockets, w3lib, urllib3, traitlets, tqdm, tabulate, sqlparse, soupsieve, six, pygments, psycopg2, prompt-toolkit, pexpect, parso, lxml, idna, humanfriendly, decorator, cython, cssselect, configparser, charset-normalizer, certifi, attrs, requests, pysam, pyquery, matplotlib-inline, jedi, importlib-metadata, coloredlogs, beautifulsoup4, asttokens, yoyo-migrations, stack-data, pyppeteer, bs4, bioutils, requests-html, ipython, biocommons.seqrepo, hgvs
** TODO SPIP T2T
*** DONE PR upstream
CLOSED: [2023-08-12 Sat 18:23] SCHEDULED: <2023-08-12 Sat 18:00>
*** DONE Mail R. Lemann
CLOSED: [2023-08-12 Sat 18:23] SCHEDULED: <2023-08-12 Sat 18:00>
*** TODO Mise à jour packages nix
** TODO VEP :vep:
*** DONE [[https://github.com/NixOS/nixpkgs/pull/185691][BioPerl]]
SCHEDULED: <2022-08-10 Wed>
/Entered on/ [2022-08-09 Tue 10:57]
PR submitted
*** TODO BioDBBBigFile
:PROPERTIES:
:ORDERED:  t
:END:
/Entered on/ [2022-08-10 Wed 14:28]
On utilise la dernière version de kent, donc plus de problème.
PRête à être mergé. Rebase faite<2023-07-02 Sun>
**** DONE Version de kent déjà packagée : forcer version  335
CLOSED: [2023-07-02 Sun 11:20]
***** KILL [[https://github.com/NixOS/nixpkgs/pull/206991][Restore building kent 404]]
CLOSED: [2023-05-06 Sat 17:40]
Review faite <2023-03-26 Sun> , atteinte merge]
Relancé <2023-05-06 Sat>
Kent 446 n'a pas ce problème donc PR inutile
***** DONE [[https://github.com/NixOS/nixpkgs/pull/223411][Ajouter les header to package]] (inc folder)
CLOSED: [2023-05-08 Mon 10:18] SCHEDULED: <2023-05-07 Sun>
Review à faire
https://github.com/NixOS/nixpkgs/pull/223411
Corrigé et plus besoin de la PR précédente
***** KILL [[https://github.com/NixOS/nixpkgs/pull/186462][BioDBBBigFile]] avec ces 2 changements
CLOSED: [2023-07-02 Sun 11:20]
**** KILL Version de kent déjà packagée : 404
CLOSED: [2023-03-27 Mon 16:43]
Compile mais les tests de passent pas
**** DONE Modifier selon PR https://github.com/NixOS/nixpkgs/pull/186462
CLOSED: [2023-07-30 Sun 22:01] SCHEDULED: <2023-07-30 Sun 20:00>
:LOGBOOK:
CLOCK: [2023-07-30 Sun 19:13]--[2023-07-30 Sun 20:50] =>  1:37
:END:
Modification nécessaire pour kent :
- plus de patch
- suppression d'une boucle dans postPatch
On supprime aussi NIX_BUILD_TOP
**** TODO Corriger PR biobigfile
SCHEDULED: <2023-10-21 Sat>
/Entered on/ [2023-10-15 Sun 17:21]
*** DONE [[https://github.com/NixOS/nixpkgs/pull/186459][BioDBHTS]]
CLOSED: [2023-05-06 Sat 08:49] SCHEDULED: <2023-04-15 Sat>
/Entered on/ [2022-08-10 Wed 14:28]
Correction pour review faites <2022-10-10 Mon>
*** DONE [[https://github.com/NixOS/nixpkgs/pull/186464][BioExtAlign]]
CLOSED: [2022-10-22 Sat 12:43] SCHEDULED: <2022-08-10 Wed>
/Entered on/ [2022-08-10 Wed 14:28]
Review <2022-10-10 Mon>, correction dans la journée.
Correction 2e passe, attente
Impossible de faire marcher les tests Car il ne trouve pas le module Bio::Tools::Align, qui est dans un dossier ailleurs dans le dépôt. Même en compilant tout le dépôt, cela ne fonctionne pas... On skip les tests.

*** TODO VEP

** WAIT [[https://github.com/NixOS/nixpkgs/pull/230394][rtg-tools]] :vcfeval:
Soumis
** WAIT Package Spip https://github.com/NixOS/nixpkgs/pull/247476
** TODO Happy :happy:
*** TODO PR python 3 upstream
SCHEDULED: <2023-10-22 Sun>
*** TODO nixpkgs en l'état
SCHEDULED: <2023-10-22 Sun>
** PROJ SpliceAI
** TODO Bamsurgeon
/Entered on/ [2023-05-13 Sat 19:11]
*** TODO Velvet

** TODO PR Picard avec option pour gérer la mémoire
Similaire à
https://github.com/bioconda/bioconda-recipes/blob/master/recipes/picard/picard.sh


* Julia :julia:
** KILL XAM.jl: PR pour modification record :julia:
CLOSED: [2023-05-29 Mon 15:40] SCHEDULED: <2023-05-28 Sun>
/Entered on/ [2023-05-27 Sat 22:39]
** TODO XAMscissors.jl :xamscissors:
Modification de la séquence dans BAM.
*Pas de mise à jour de CIGAR*
On convertit en fastq et on lance le pipeline pour "corriger"


#+begin_src sh
cd /home/alex/code/bisonex/out/63003856/preprocessing/mapped
samtools view 63003856_S135.bam NC_000022.11 -o 63003856_S135_chr22.bam
cd /home/alex/recherche/bisonex/code/BamScissors.jl
cp ~/code/bisonex/out/63003856/preprocessing/mapped/63003856_S135_chr22.bam .
samtools index 63003856_chr22.bam
#+end_src

Le script va modifier le bam, le trier et générer le fastq. !!!
Attention: ne pas oublier l'option -n !!!

#+begin_src sh
time julia --project=.. insertVariant.jl
scp 63003856_S135_chr22_{1,2}.fq.gz meso:/Work/Users/apraga/bisonex/tests/bamscissors/
#+end_src
*** WAIT Implémenter les SNV avec VAF :snv:
Stratégie :
1. calculer la profondeur sur les positions
2. créer un dictionnaire { nom du reads : position dataframe }
3. itérer sur tous les reads et changer ceux marqués
**** DONE VAF = 1
CLOSED: [2023-05-29 Mon 15:34]
**** DONE VAF selon loi normale
CLOSED: [2023-05-29 Mon 15:35]
Tronquée si > 1
**** WAIT Tests unitaires
***** DONE NA12878: 1 gène sur chromosome 22
CLOSED: [2023-05-30 Tue 23:55]
root = "https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/giab/data/NA12878/Garvan_NA12878_HG001_HiSeq_Exome/"
#+begin_src sh
samtools view project.NIST_NIST7035_H7AP8ADXX_NA12878.bwa.markDuplicates.bam  chr22 -o project.NIST_NIST7035_H7AP8ADXX_NA12878_chr22.bam
samtools view project.NIST_NIST7035_H7AP8ADXX_NA12878_chr22.bam chr22:19419700-19424000 -o NIST7035_H7AP8ADXX_NA12878_chr22_MRPL40_hg19.bam
#+end_src
***** WAIT Pull request formatspeciment
https://github.com/BioJulia/FormatSpecimens.jl/pull/8
***** DONE Formatspecimens
CLOSED: [2023-05-29 Mon 23:03]
****** DONE 1 read
CLOSED: [2023-05-29 Mon 23:02]
****** DONE VAF sur 1 exon
CLOSED: [2023-05-29 Mon 23:03]
**** DONE [#A] Bug: perte de nombreux reads avec NA12878
CLOSED: [2023-08-19 Sat 20:45] SCHEDULED: <2023-08-18 Fri>
:PROPERTIES:
:ID:       5c1c36f3-f68e-4e6d-a7b6-61dca89abc37
:END:
Ex: chrX:g.124056226 : on passe de 65 reads à 1

Test xamscissors: pas de soucis...
On teste sur cette position +/- 200bp
#+begin_src sh :dir /home/alex/roam/research/bisonex/code/sanger
samtools view   /home/alex/code/bisonex/out/2300346867_NA12878-63118093_S260-GRCh38/preprocessing/mapped/2300346867_NA12878-63118093_S260-GRCh38.bam chrX:124056026-124056426 -o chrXsmall.bam
#+end_src

#+RESULTS:

***** DONE Vérifier profondeur avec dernière version :
CLOSED: [2023-08-19 Sat 20:34] SCHEDULED: <2023-08-19 Sat>
****** DONE chr20: profondeur ok
SCHEDULED: <2023-08-19 Sat>
****** DONE toutes les données
CLOSED: [2023-08-19 Sat 20:34] SCHEDULED: <2023-08-19 Sat>
Ok pour 7 variants (IGV) notament chromosome X

*** TODO Implémenter les indel avec VAF :indel:
*** TODO Soumission paquet
* Données
:PROPERTIES:
:CATEGORY: data
:END:
** DONE Remplacer bam par fastq sur mesocentre
CLOSED: [2023-04-16 Sun 16:33]
Commande
*** DONE Supprimer les fastq non "paired"
CLOSED: [2023-04-16 Sun 16:33]
nushell
Liste des fastq avec "paired-end" manquant
#+begin_src nu
ls **/*.fastq.gz | get name | path basename | split column "_" | get column1 | uniq -u | save single.txt
#+end_src

#+RESULTS:
: 62907927
: 62907970
: 62899606
: 62911287
: 62913201
: 62914084
: 62915905
: 62921595
: 62923065
: 62925220
: 62926503
: 62926502
: 62926500
: 62926499
: 62926498
: 62931719
: 62943423
: 62943400
: 62948290
: 62949205
: 62949206
: 62949118
: 62951284
: 62960792
: 62960785
: 62960787
: 62960617
: 62962561
: 62962692
: 62967473
: 62972194
: 62979102

On vérifie
#+begin_src nu
open single.txt  | lines | each {|e| ls $"fastq/*_($in)/*" | get 0  }
open single.txt  | lines | each {|e| ls $"fastq/*_($in)/*" | get 0.name }  | path basename | split column "_" | get column1 | uniq -c
#+end_src
On met tous dans un dossier (pas de suppression )
#+begin_src
open single.txt  | lines | each {|e| ls $"fastq/*_($in)/*" | get 0  }  | each {|e| ^mv $e.name bad-fastq/}
#+end_src

On vérifie que les dossiier sont videsj
 open single.txt  | lines | each {|e| ls $"fastq/*_($in)" | get 0.name } | ^ls -l $in

 Puis on supprime
 open single.txt  | lines | each {|e| ls $"fastq/*_($in)" | get 0.name } | ^rm -r $in
*** DONE Supprimer bam qui ont des fastq
CLOSED: [2023-04-16 Sun 16:33]
On liste les identifiants des fastq et bam dans un tableau avec leur type :
#+begin_src
let fastq = (ls fastq/*/*.fastq.gz | get name | parse "{dir}/{full_id}/{id}_{R}_001.fastq.gz"  | select dir id | uniq )
let bam = (ls bam/*/*.bam | get name | parse "{dir}/{full_id}/{id}_{S}.bqrt.bam"  | select dir id)
#+end_src
On groupe les résultat par identifiant (résultats = liste de records qui doit être convertie en table)
et on trie ceux qui n'ont qu'un fastq ou un bam
#+begin_src
let single = ( $bam | append $fastq | group-by id | transpose id files | get files | where {|x| ($x | length) == 1})
#+end_src
On convertit en table et on récupère seulement les bam
#+begin_src
$single | reduce {|it, acc| $acc | append $it} | where dir == bam | get id | each {|e| ^ls $"bam/*_($e)/*.bam"}
#+end_src

#+RESULTS:
: bam/2100656174_62913201/62913201_S52.bqrt.bam
: bam/2100733271_62925220/62925220_S33.bqrt.bam
: bam/2100738763_62926502/62926502_S108.bqrt.bam
: bam/2100746726_62926498/62926498_S105.bqrt.bam
: bam/2100787936_62931955/62931955_S4.bqrt.bam
: bam/2200066374_62948290/62948290_S130.bqrt.bam
: bam/2200074722_62948298/62948298_S131.bqrt.bam
: bam/2200074990_62948306/62948306_S218.bqrt.bam
: bam/2200214581_62967331/62967331_S267.bqrt.bam
: bam/2200225399_62972187/62972187_S85.bqrt.bam
: bam/2200293962_62979117/62979117_S63.bqrt.bam
: bam/2200423985_62999352/62999352_S1.bqrt.bam
: bam/2200495073_63010427/63010427_S20.bqrt.bam
: bam/2200511274_63012586/63012586_S114.bqrt.bam
: bam/2200669188_63036688/63036688_S150.bqrt.bam

* Nouveau workflow :workflow:
** TODO Bases de données
*** KILL Nix pour télécharger les données brutes
**** Conclusion
Non viable sur cluster car en dehors de /nix/store
On peut utiliser des symlink mais trop compliqué
**** KILL Axel au lieu de curl pour gérer les timeout?
CLOSED: [2022-08-19 Fri 15:18]
*** DONE Tester patch de @pennae pour gros fichiers
SCHEDULED: <2022-08-19 Fri>
*** KILL Télécharger les données avec nextflow: hg38
CLOSED: [2023-06-12 Mon 23:29]
**** DONE Genome de référence
**** DONE dbSNP
**** DONE VEP 20G
CLOSED: [2023-06-12 Mon 23:29]
Ajout vérification checksum -> à vérifier
**** DONE VEP version 1.10
CLOSED: [2023-08-06 Sun 09:45] SCHEDULED: <2023-08-06 Sun>
**** DONE transcriptome (spip)
CLOSED: [2023-06-12 Mon 23:29]
Rajouter checksum manuel
**** KILL Refseq
**** KILL OMIM
CLOSED: [2023-06-12 Mon 23:29]
codé, à vérifier
**** KILL ACMG incidental
CLOSED: [2023-06-12 Mon 23:29]
*** DONE Données :T2T:
CLOSED: [2023-09-10 Sun 16:45]
:PROPERTIES:
:ID:       5d915178-ca96-44ef-87f1-6702af114f2b
:END:
**** DONE fasta
CLOSED: [2023-06-12 Mon 23:30]
***** DONE compatibilité hg38
CLOSED: [2023-06-12 Mon 23:30]
**** DONE fasta index
CLOSED: [2023-06-13 Tue 00:07]
***** DONE compatibilité hg38
CLOSED: [2023-06-13 Tue 00:07]
**** DONE fasta dictionnaire
CLOSED: [2023-06-13 Tue 00:07]
**** DONE dbSNP
CLOSED: [2023-06-12 Mon 23:30]
***** DONE compatibilité hg38
CLOSED: [2023-06-12 Mon 23:30]
**** DONE commonSNP
CLOSED: [2023-06-14 Wed 22:32]
***** DONE compatibilité hg38
CLOSED: [2023-06-14 Wed 22:32]
cd /Work/Groups/bisonex/data/dbsnp/GRCh38.p14
❯ ga@mesointeractive GRCh38.p14]$ zgrep -c '^NC' dbSNP_common.vcf.gz
21340485
[apraga@mesointeractive GRCh38.p14]$ pwd
[apraga@mesointeractive GRCh38.p14]$ zgrep -c '^NC'
dbSNP_common.vcf.gz                     ID_of_common_snp_not_clinvar_patho.txt
dbSNP_common.vcf.gz.tbi                 ID_of_common_snp.txt

[apraga@mesointeractive dbsnp]$ cd chm13v2.0/
[apraga@mesointeractive chm13v2.0]$ ls
chm13v2.0_dbSNPv155.vcf.gz      dbSNP_common.vcf.gz.tbi                 versions.yml
chm13v2.0_dbSNPv155.vcf.gz.tbi  ID_of_common_snp_not_clinvar_patho.txt
dbSNP_common.vcf.gz             ID_of_common_snp.txt
[apraga@mesointeractive chm13v2.0]$ zgrep -c '^chr' dbSNP_common.vcf.gz
19433713
[apraga@mesointeractive chm13v2.0] $
❯ man tmux
**** DONE commonSNP non patho
CLOSED: [2023-06-14 Wed 22:35]
***** DONE compatibilité hg38
CLOSED: [2023-06-14 Wed 22:35]
**** DONE cache vep
CLOSED: [2023-06-30 Sun 14:20] SCHEDULED: <2023-07-25 Tue>
*** HOLD Processing bases de données
**** DONE dbSNP common
**** DONE Seulement les ID dans dbSNP common !
CLOSED: [2022-11-19 Sat 21:42]
172G au lieu de 253M...
**** HOLD common dbSNP not clinvar patho
***** DONE Conclusion partielle
CLOSED: [2022-12-12 Mon 22:25]
- vcfeval : prometteur mais n'arrive pas à traiter toutes les régions
- isec : trop de problèmes avec
- classif clinvar directement dans dbSNP: le plus simple
  Et ça permet de rattraper quelques erreurs dans le script d'Alexis
***** KILL Utiliser directement le numéro dbSNP dans clinvar ? Non
CLOSED: [2022-11-20 Sun 19:51]
Ex: chr20
#+begin_src sh :dir ~/code/bisonex/test_isec
bcftools query -f 'rs%INFO/RS \n' -i 'INFO/RS != "." & INFO/CLNSIG="Pathogenic"' clinvar_chr20.vcf.gz | sort > ID_clinvar_patho.txt
bcftools query -f '%ID\n' dbSNP_common_chr20.vcf.gz | sort > ID_of_common_snp.txt
comm -23 ID_of_common_snp.txt ID_clinvar_patho.txt > ID_of_common_snp_not_clinvar_patho.txt
wc -l ID_of_common_snp_not_clinvar_patho.txt
# sort ID
#+end_src

#+RESULTS:
: 518846 ID_of_common_snp_not_clinvar_patho.txt

Version d'alexis
#+begin_src sh :dir ~/code/bisonex/test_isec
snp=dbSNP_common_chr20.vcf.gz
clinvar=clinvar_chr20_notremapped.vcf.gz
python ../script/pythonScript/clinvar_sbSNP.py \
    --clinvar $clinvar \
    --chrm_name_table ../database/RefSeq/refseq_to_number_only_consensual.txt \
    --dbSNP $snp --output prod.txt
wc -l prod.txt
zgrep '^NC' dbSNP_common_chr20.vcf.gz | wc -l
#+end_src

#+RESULTS:
| 518832 | prod.txt |
| 518846 |          |
***** KILL classification clinvar codée dbSNP ?
CLOSED: [2022-12-04 Sun 14:38]
Sur le chromosome 20
*Attention* CLNSIG a plusieurs champs (séparé par une virgule)
On y accède avec INFO/CLNSIG[*]
Ensuite, chaque item peut avoir plusieurs haploïdie (séparé par un |). IL faut donc utiliser une regexp
NB: *ne pas mettre la condition* dans une variable !!

Pour avoir les clinvar patho, on veut 5 mais pas 255 (= autre) pour la classification !`
Il faut également les likely patho et conflicting
#+begin_src sh :dir ~/code/bisonex/test_isec
bcftools query -f '%INFO/CLNSIG\n' dbSNP_common_chr20.vcf.gz -i \
'INFO/CLNSIG[*]~"^5|" | INFO/CLNSIG[*]=="5" | INFO/CLNSIG[*]~"|5" | INFO/CLNSIG[*]~"^4|" | INFO/CLNSIG[*]=="4" | INFO/CLNSIG[*]~"|4" | INFO/CLNSIG[*]~"^12|" | INFO/CLNSIG[*]=="12" | INFO/CLNSIG[*]~"|12"' | sort

#+end_src

#+RESULTS:
| . |  . | 12 |    |   |   |   |   |   |   |   |
| . | 12 |  0 |  2 |   |   |   |   |   |   |   |
| 2 |  3 |  2 |  2 | 2 | 5 | . |   |   |   |   |
| . |  2 |  3 |  2 | 2 | 4 |   |   |   |   |   |
| . |  . |  3 | 12 | 3 |   |   |   |   |   |   |
| . |  5 |  2 |  . |   |   |   |   |   |   |   |
| . |  . |  . |  5 | 2 | 2 |   |   |   |   |   |
| . |  9 |  9 |  9 | 5 | 5 | 2 | 3 | 2 | 3 | 2 |


Si on les exclut :
#+begin_src sh :dir ~/code/bisonex/test_isec
bcftools query -f '%ID\n' dbSNP_common_chr20.vcf.gz -e \
'INFO/CLNSIG[*]~"^5|" | INFO/CLNSIG[*]=="5" | INFO/CLNSIG[*]~"|5" | INFO/CLNSIG[*]~"4" | INFO/CLNSIG[*]~"12"' | sort | uniq > common-notpatho.txt
#+end_src

#+RESULTS:

 #+begin_src sh :dir ~/code/bisonex/test_isec
snp=dbSNP_common_chr20.vcf.gz
clinvar=clinvar_chr20_notremapped.vcf.gz
python ../script/pythonScript/clinvar_sbSNP.py \
    --clinvar $clinvar \
    --chrm_name_table ../database/RefSeq/refseq_to_number_only_consensual.txt \
    --dbSNP $snp --output tmp.txt
sort tmp.txt | uniq > common-notpatho-alexis.txt
wc -l common-notpatho-alexis.txt
 #+end_src

 #+RESULTS:
 : 518832 common-notpatho-alexis.txt

On en a 6 de plus que la version d'Alexis mais quelques différences

Ceux d'Alexis qui manquent:
#+begin_src sh :dir ~/code/bisonex/test_isec
comm -23 common-notpatho-alexis.txt common-notpatho.txt > alexis-only.txt
cat alexis-only.txt
#+end_src

#+RESULTS:
| rs1064039  |
| rs3833341  |
| rs73598374 |

On les teste dans clinvar et dbSNP
#+begin_src sh :dir ~/code/bisonex/test_isec
bcftools query -f '%POS %REF %ALT %INFO/CLNSIG\n' -i 'ID=@alexis-only.txt' dbSNP_common_chr20.vcf.gz
bcftools query -f '%POS\n' -i 'ID=@alexis-only.txt' dbSNP_common_chr20.vcf.gz > alexis-only-pos.txt
while read  -r line; do
bcftools query -f '%POS %REF %ALT %INFO/CLNSIG\n' -i 'POS='$line clinvar_chr20.vcf.gz
done < alexis-only-pos.txt
# bcftools query -f '%POS %REF %ALT %INFO/CLNSIG\n' -i 'POS=23637790' clinvar_chr20.vcf.gz
#+end_src

#+RESULTS:
|   764018 | A | ACAGGTCAAT,ACAGGT | .,5     | 2,. |   |
| 23637790 | C | G,T               | .,.,12  |     |   |
| 44651586 | C | A,G,T             | .,.,.,5 |   2 | 2 |
|   764018 | A | ACAGGTCAAT        | Benign  |     |   |
| 23637790 | C | T                 | Benign  |     |   |
| 44651586 | C | T                 | Benign  |     |   |

On a donc une discordance entre clinvar et dbSNP.
On dirait qu'ils ont mal fait l'intersection avec clinvar.

Par exemple https://www.ncbi.nlm.nih.gov/snp/rs3833341#clinical_significance

Tu as l'impression qu'il y a un 1 clinvar bénin et 1 patho.

En cherchant par NM, tu vois qu'il est bénin sur clinvar car il y a d'autres soumissions ! https://www.ncbi.nlm.nih.gov/clinvar/variation/262235/


Confirmation sur nos bases de données :

$ bcftools query -f '%POS %REF %ALT %INFO/CLNSIG\n' -i 'POS=764018' dbSNP_common_chr20.vcf.gz

764018 A ACAGGTCAAT,ACAGGT .,5|2,.
$ bcftools query -f '%POS %REF %ALT %INFO/CLNSIG\n' -i 'POS=764018' clinvar_chr20.vcf.gz
764018 A ACAGGTCAAT Benign

***** KILL Corriger script alexi
CLOSED: [2022-12-04 Sun 13:03]
Gère clinvar patho, probablement patho ou conflicting !
***** HOLD Rtg tools
****** Test
1. Générer SDf file
   #+begin_src sh
rtg format genomeRef.fna  -o genomeRef.sdf
   #+end_src
2. Pour les bases de donnés, il faut l'option --sample ALT sinon on a
 #+begin_src
$ rtg vcfeval -b dbSNP_common.vcf.gz -c clinvar.vcf.gz -o test -t genomeRef.sdf/^C
VCF header does not contain a FORMAT field named GQ
Error: Record did not contain enough samples: NC_000001.11	10001	rs1570391677	A,C	.	PASS	RS=1570391677;dbSNPBuildID=154;SSR=0;PSEUDOGENEINFO=DDX11L1:100287102;VC=SNV;R5;GNO;FREQ=KOREAN:0.9891,0.0109,.|SGDP_PRJ:0,1,.|dbGaP_PopFreq:1,.,0;COMMON
 #+end_src

 Essai intersection clinvar (patho ou non) dbSNP
   - faux négatif = dbSNP common qui ne sont pas dans clinvar
   - faux positif = clinvar qui ne sont pas dbSNP common
   - vrai positif = clinvar qui sont dans dbSNP common
   - vrai positif baseline = dbSNP common qui sont dans clinvar
 On calcule le nombre de lignes
 #+begin_src ssh
zgrep '^[^#]' /Work/Groups/bisonex/data/clinvar/GRCh38/clinvar.vcf.gz | wc -l
for i in *.vcf.gz; do echo $i; zgrep '^[^#]' $i | wc -l; done
 #+end_src
 | clinvar            |  1493470 |
 | fn.vcf.gz          | 22330220 |
 | fp.vcf.gz          |  1222529 |
 | tp-baseline.vcf.gz |   131040 |
 | tp.vcf.gz          |   136638 |
À noter qu'on ne retrouve pas tout clinvar...
1222529 + 131040 = 1353569 < 1493470
certains régions ne sont pas traitées :
#+begin_quote
Evaluation too complex (50002 unresolved paths, 34891 iterations) at reference region NC_000001.11:790930-790970. Variants in this region will not be included in results
#+end_quote
#+begin_src sh
grep 'not be included' vcfeval.log | wc -l
56192
#+end_src
Le total est quand même inférieur

On veut les clinvar non patho dans dbSNP soit les faux négatif (dbSNP common not contenu dans clinvar patho)
#+begin_src sh
bcftools filter -i 'INFO/CLNSIG="Pathogenic"' /Work/Groups/bisonex/data/clinvar/GRCh38/clinvar.vcf.gz -o /Work/Groups/bisonex/data/clinvar/GRCh38/clinvar-patho.vcf.gz
tabix /Work/Groups/bisonex/data/clinvar/GRCh38/clinvar-patho.vcf.gz
#+end_src
On lance le script (dbSNP common et clinvar = 9h)
#+begin_src sh
#!/bin/bash
#SBATCH --nodes=1
#SBATCH -p smp
#SBATCH --time=12:00:00
#SBATCH --mem=12G

dir=/Work/Groups/bisonex/data
dbSNP=$dir/dbSNP/GRCh38.p13/dbSNP_common.vcf.gz
clinvar=$dir/clinvar/GRCh38/clinvar-patho.vcf.gz
genome=$dir/genome/GRCh38.p13/genomeRef.sdf
srun rtg vcfeval -b $dbSNP -c $clinvar -o common-not-patho -t $genome --sample ALT

#+end_src
****** HOLD Voir pour régions complexes non traitées
***** DONE bcftools isec : non
CLOSED: [2022-11-27 Sun 00:38]
#+begin_src sh
bcftools isec dbSNP_common.vcf.gz clinvar.vcf.gz -p common
#+end_src
On vérifie bien que les 2 fichiers commons on le même nombre de lignes
#+begin_src sh
$ grep -e '^NC'  0002.vcf | wc -l
74302
alex@gentoo ~/code/bisonex/data/common $ grep -e '^NC'  0003.vcf | wc -l
74302
#+end_src
****** DONE Impact option -n
CLOSED: [2022-10-23 Sun 13:56]
Mais en spécifiant -n =2:
#+begin_src sh
$ bedtools intersect -a  dbSNP_common.vcf.gz -b clinvar.vcf.gz
74978
#+end_src
Si on ne regarde que les variants, on retrouve bien 74302
#+begin_src sh
rg "^NC" none_sorted.vcf  | wc -l
#+end_src
NB : test fait avec
#+begin_src
bcftools isec dbSNP_common.vcf.gz clinvar.vcf.gz -c none -n =2 -w 1 | sort > none.vcf
sort common/0003.vcf > common/0003_sorted.vcf
comm -13 common/0003_sorted.vcf none_sorted.vcf
#+end_src

****** DONE Géstion des duplicates: -c none
CLOSED: [2022-10-23 Sun 13:56]
Si on ne garde que ceux avec REF et ALT identiques
#+begin_src sh
bcftools isec dbSNP_common.vcf.gz clinvar.vcf.gz -c none -n =2 -w 1 | wc -l
74978
#+end_src
Si on garde tout
#+begin_src sh
bcftools isec dbSNP_common.vcf.gz clinvar.vcf.gz -c all -n =2 -w 1 | wc -l
137777
#+end_src
Pour regarder la différence :
#+begin_src sh
bcftools isec dbSNP_common.vcf.gz clinvar.vcf.gz -c none -n =2 -w 1 | sort > none_sorted.vcf
bcftools isec dbSNP_common.vcf.gz clinvar.vcf.gz -c all -n =2 -w 1 | sort > all_sorted.vcf
comm -13 none_sorted.vcf all_sorted.vcf | head
#+end_src
Sur un exemple,on a bien des variants différents
****** DONE Suppression des clinvar patho
CLOSED: [2022-10-23 Sun 18:55]
Semble faire le travail vu que dbSNP_commo a 23194960 lignes (donc ~80 000 de moins)
 #+begin_src sh
$ bcftools isec -e 'INFO/CLNSIG="Pathogenic" & INFO/CLNSIG="Pathogenic/Likely_pathogenic"' -c none -n~10  dbSNP_common.vcf.gz clinvar.vcf.gz | wc -l
Note: -w option not given, printing list of sites...
23119984
 #+end_src
 Par contre, l'o'ption -w ou -p fait des ficher "data"...
Après un nouvel essai, plus de problème
#+begin_src
$ bcftools isec -e 'INFO/CLNSIG="Pathogenic" & INFO/CLNSIG="Pathogenic/Likely_pathogenic"' -c none -n=1 dbSNP_common.vcf.gz clinvar.vcf.gz -w 1 -o lol.vcf.gz
$ zcat lol.vcf.gz | wc -l
23120660
#+end_src
À noter le choix de l'option -n qui change entre "=1" et "~10"...
En effet "=1" = au moins 1 fichier et "~10" fait exactement dans le premier et non dans le second
#+begin_src
$ bcftools isec -e 'INFO/CLNSIG="Pathogenic" & INFO/CLNSIG="Pathogenic/Likely_pathogenic"' -c none -n~10 dbSNP_common.vcf.gz clinvar.vcf.gz -w 1 -o lol.vcf.gz
$ zcat lol.vcf.gz | wc -l
23120660
#+end_src
****** DONE Valider avec Alexis : bcftool isec
CLOSED: [2022-11-07 Mon 21:42   ]
****** DONE Pourquoi nombre de lignes différentes avec la version d'Alexis -> isec ne gère pas plusieurs ALT
CLOSED: [2022-11-26 Sat 23:36]
Grosse différence !
#+begin_src
$ wc -l ID_of_common_snp_not_clinvar_patho.txt
23119915 ID_of_common_snp_not_clinvar_patho.txt
$ wc -l /Work/Users/apraga/bisonex/database/dbSNP/ID_of_common_snp_not_clinvar_patho.txt
85820 /Work/Users/apraga/bisonex/database/dbSNP/ID_of_common_snp_not_clinvar_patho.txt
#+end_src
À noter que tout dbSNP = 23194960
******* Clinvar classe 4 ? Moins mais toujours trop
#+begin_src
$ zgrep '^NC' tmp.vcf.gz  | wc -l
21081654
#+end_src
******* Comparer les ID et regarder ceux en plus
#+begin_src sh
bcftools isec -e 'INFO/CLNSIG="Pathogenic"' -c none -n~10 /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/dbSNP_common.vcf.gz /Work/Groups/bisonex/data/clinvar/GRCh38/clinvar.vcf.gz -w 1 -o tmp.vcf.gz

zgrep -o -e 'rs[[:digit:]]\' tmp.vcf.gz | sort | id_sorted.txt
sort ../database/dbSNP/ID_of_common_snp_not_clinvar_patho.txt  > reference_sorted.txt
comm -23 id_sorted.txt reference_sorted.txt > unique1.txt
#+end_src
Par exemple
#+begin_src sh
zgrep rs1000000561 ../database/dbSNP/dbSNP_common.vcf.gz
#+end_src
NC_000002.12	136732859	rs1000000561	ACG	A,ACGCG	.	PASS	RS=1000000561;dbSNPBuildID=151;SSR=0;VC=INDEL;GNO;FREQ=ALSPAC:0.2506,0.7494,.|TOMMO:0.9971,0.002865,.|TWINSUK:0.2473,0.7527,.|dbGaP_PopFreq:0.993,0.006943,8.902e-05;COMMON

Attention, clinvar est en numéro de chromosomoe et dbSNP en NC...
Normalement, géré lors du calcul d'intersection !
Ce SNP n'est pas dans clinvar (vérifié dans UCSC)
******* Tester sur chromosome 20
#+begin_src sh :dir ~/code/bisonex/test_isec
bcftools view --regions NC_000020.11 ../database/dbSNP/dbSNP_common.vcf.gz -o dbSNP_common_chr20.vcf.gz
bcftools view --regions 20 ../database/clinvar/clinvar.vcf.gz -o clinvar_chr20.vcf.gz
tabix dbSNP_common_chr20.vcf.gz
tabix clinvar_chr20.vcf.gz
#+end_src

#+RESULTS:

Attention à bien renommer clinvar !

#+begin_src sh :dir ~/code/bisonex/test_isec
mv clinvar_chr20.vcf.gz clinvar_chr20_notremapped.vcf.gz
bcftools annotate --rename-chrs chromosome_mapping.txt clinvar_chr20_notremapped.vcf.gz -o clinvar_chr20.vcf.gz
#+end_src

#+RESULTS:

*ATTENTION*: sans indexer les vcf, les fichiers seront *VIDES*
*ATTENTION*: par défaut les filtres s'appliquent sur les 2. Cela est un problème si on joue sur l'inclusion et non l'exclusion
Attention: vérifier la conventdion de nommage des chromosomes
******** Test pathogene: ne prend pas en compte les multi-allèles ????
On teste l'intersection dbsnp et clinvar patho ainsi que le complémentaire
#+begin_src sh :dir ~/code/bisonex/test_isec
clinvar=clinvar_chr20_patho.vcf.gz
snp=dbSNP_common_chr20.vcf.gz
bcftools index $clinvar
bcftools index $snp
bcftools filter -i 'INFO/CLNSIG="Pathogenic"' clinvar_chr20.vcf.gz -o $clinvar
bcftools isec  $snp $clinvar -p tmp
for i in tmp/*.vcf ; do echo $i; grep '^[^#]'  $i | wc -l; done
#+end_src

#+RESULTS:
| tmp/0000.vcf |
|       518846 |
| tmp/0001.vcf |
|            0 |
| tmp/0002.vcf |
|            0 |
| tmp/0003.vcf |
|            0 |


Aucun clinvar patho... Clairement faux !
Autre méthode : on inclut tous les SNP et clinvar patho et on regarde ceux uniquement dans dbsnp

#+begin_src sh :dir ~/code/bisonex/test_isec
snp=dbSNP_common_chr20.vcf.gz
clinvar=clinvar_chr20.vcf.gz
bcftools isec -n=2 -i - -i 'INFO/CLNSIG="Pathogenic"' $snp $clinvar -p tmp
 # grep '^[^#]' tmp/0000.vcf | wc -l
#+end_src

#+RESULTS:
Soit tout dbsnp donc rien

Note : on ne peut pas exclure les clinvar patho directement
#+begin_src sh :dir ~/code/bisonex/test_isec
snp=dbSNP_common_chr20.vcf.gz
clinvar=clinvar_chr20.vcf.gz
bcftools isec -i - -e 'INFO/CLNSIG="Pathogenic"' $snp $clinvar -p tmp
for i in tmp/*.vcf ; do echo $i; grep '^[^#]'  $i | wc -l; done
#+end_src
Car on ne peut plus faire la différence !


Si on utilise la version d'Alexis
#+begin_src sh :dir ~/code/bisonex/test_isec
snp=dbSNP_common_chr20.vcf.gz
clinvar=clinvar_chr20_notremapped.vcf.gz
python ../script/pythonScript/clinvar_sbSNP.py \
    --clinvar $clinvar \
    --chrm_name_table ../database/RefSeq/refseq_to_number_only_consensual.txt \
    --dbSNP $snp --output tmp.txt
sort tmp.txt > common-notpatho-alexis.txt
wc -l common-notpatho-alexis.txt
#+end_src

#+RESULTS:
: 518832 common-notpatho-alexis.txt

Si on cherche les clinvar patho (donc non présent dans la sortie)
#+begin_src sh :dir ~/code/bisonex/test_isec
  bcftools query -f '%ID\n' dbSNP_common_chr20.vcf.gz | sort > all.txt
  sort common-notpatho-alexis.txt > alexis.txt
  comm -23 all.txt alexis.txt > patho.txt
#+end_src

#+begin_src sh :dir ~/code/bisonex/test_isec
bcftools query -f '%POS\n' -i 'ID=@patho.txt' dbSNP_common_chr20.vcf.gz -o pos.txt
for pos in $(cat pos.txt); do
  bcftools query -f '%CHROM %POS %ID %REF %ALT\n' -i 'POS='$pos dbSNP_common_chr20.vcf.gz
  bcftools query -f '%CHROM %POS %ID %REF %ALT %INFO/CLNSIG\n' -i 'POS='$pos  clinvar_chr20.vcf.gz
  echo "------"
done
#+end_src

#+RESULTS:
| NC_000020.11 |  3234173 |   rs3827075 | T         | A,C,G     |                                              |
| NC_000020.11 |  3234173 |      262001 | T         | G         | Conflicting_interpretations_of_pathogenicity |
| NC_000020.11 |  3234173 |     1072511 | T         | TGGCGAAGC | Pathogenic                                   |
| NC_000020.11 |  3234173 |      208613 | TGGCGAAGC | G         | Pathogenic                                   |
| NC_000020.11 |  3234173 |        1312 | TGGCGAAGC | T         | Pathogenic                                   |
| ------       |          |             |           |           |                                              |
| NC_000020.11 |  4699605 |   rs1799990 | A         | G         |                                              |
| NC_000020.11 |  4699605 |       13397 | A         | G         | Benign/Likely_benign                         |
| ------       |          |             |           |           |                                              |
| NC_000020.11 | 10652589 |   rs1131695 | G         | A,C,T     |                                              |
| NC_000020.11 | 10652589 |      163705 | G         | .         | Benign                                       |
| NC_000020.11 | 10652589 |      143063 | G         | A         | Benign                                       |
| NC_000020.11 | 10652589 |      234555 | G         | C         | Pathogenic                                   |
| ------       |          |             |           |           |                                              |
| NC_000020.11 | 10658574 |   rs1801138 | G         | A,T       |                                              |
| NC_000020.11 | 10658574 |       42481 | G         | A         | Benign                                       |
| NC_000020.11 | 10658574 |      992651 | G         | T         | Likely_pathogenic                            |
| NC_000020.11 | 10658574 |      213550 | GC        | A         | Pathogenic                                   |
| ------       |          |             |           |           |                                              |
| NC_000020.11 | 10672794 |  rs79338570 | G         | A,C       |                                              |
| NC_000020.11 | 10672794 |      255557 | G         | A         | Benign/Likely_benign                         |
| NC_000020.11 | 10672794 |      594067 | G         | C         | Conflicting_interpretations_of_pathogenicity |
| NC_000020.11 | 10672794 |     1324603 | G         | GGA       | Likely_pathogenic                            |
| ------       |          |             |           |           |                                              |
| NC_000020.11 | 18525868 | rs146917730 | C         | T         |                                              |
| NC_000020.11 | 18525868 |      811603 | C         | T         | Conflicting_interpretations_of_pathogenicity |
| ------       |          |             |           |           |                                              |
| NC_000020.11 | 25390747 | rs373200654 | G         | C         |                                              |
| NC_000020.11 | 25390747 |      338000 | G         | C         | Conflicting_interpretations_of_pathogenicity |
| ------       |          |             |           |           |                                              |
| NC_000020.11 | 32800145 |   rs2424926 | C         | G,T       |                                              |
| NC_000020.11 | 32800145 |      338173 | C         | G         | Benign                                       |
| NC_000020.11 | 32800145 |      338174 | C         | T         | Conflicting_interpretations_of_pathogenicity |
| ------       |          |             |           |           |                                              |
| NC_000020.11 | 33412656 |  rs35938843 | C         | G,T       |                                              |
| NC_000020.11 | 33412656 |      220958 | C         | T         | Conflicting_interpretations_of_pathogenicity |
| ------       |          |             |           |           |                                              |
| NC_000020.11 | 45891622 | rs181943893 | G         | A,C,T     |                                              |
| NC_000020.11 | 45891622 |      459632 | G         | C         | Conflicting_interpretations_of_pathogenicity |
| NC_000020.11 | 45891622 |      797035 | G         | T         | Likely_benign                                |
| NC_000020.11 | 45891622 |     1572689 | GCTA      | G         | Likely_benign                                |
| ------       |          |             |           |           |                                              |
| NC_000020.11 | 54171651 |  rs35873579 | G         | A,T       |                                              |
| NC_000020.11 | 54171651 |      285894 | G         | A         | Conflicting_interpretations_of_pathogenicity |
| NC_000020.11 | 54171651 |     1373583 | G         | C         | Uncertain_significance                       |
| NC_000020.11 | 54171651 |      895614 | G         | T         | Benign/Likely_benign                         |
| ------       |          |             |           |           |                                              |
| NC_000020.11 | 62172726 |  rs36106901 | G         | A         |                                              |
| NC_000020.11 | 62172726 |      981031 | G         | A         | Conflicting_interpretations_of_pathogenicity |
| ------       |          |             |           |           |                                              |
| NC_000020.11 | 63349782 |   rs1044396 | G         | A,C       |                                              |
| NC_000020.11 | 63349782 |       93427 | G         | A         | Benign                                       |
| NC_000020.11 | 63349782 |      857384 | G         | C         | Conflicting_interpretations_of_pathogenicity |
| ------       |          |             |           |           |                                              |
| NC_000020.11 | 63414925 |   rs1801545 | G         | A,C,T     |                                              |
| NC_000020.11 | 63414925 |      194284 | G         | A         | Conflicting_interpretations_of_pathogenicity |
| NC_000020.11 | 63414925 |      129337 | G         | C         | Benign                                       |
| NC_000020.11 | 63414925 |      851545 | GG        | CA        | Uncertain_significance                       |
| ------       |          |             |           |           |                                              |

On a donc plusieurs problèmes :
1. isec devrait fonctionner au moins sur
| NC_000020.11 | 25390747 | rs373200654 | G         | C         |                                              |
| NC_000020.11 | 25390747 |      338000 | G         | C         | Conflicting_interpretations_of_pathogenicity |

On teste juste sur cette ligne
#+begin_src sh :dir ~/code/bisonex/test_isec
bcftools filter -i 'POS=25390747' clinvar_chr20.vcf.gz -o clinvar_test.vcf.gz
bcftools filter -i 'POS=25390747' dbSNP_common_chr20.vcf.gz -o dbSNP_test.vcf.gz
#+end_src
On retrouve bien la ligne dans l'intersection...
#+begin_src sh :dir ~/code/bisonex/test_isec
bcftools filter -i 'POS=25390747' clinvar_chr20.vcf.gz -o clinvar_test.vcf.gz
bcftools index dbSNP_test.vcf.gz dbSNP_test.vcf.gz
bcftools index dbSNP_test.vcf.gz clinvar_test.vcf.gz
bcftools isec dbSNP_test.vcf.gz clinvar_test.vcf.gz -p test
#+end_src

#+RESULTS:

2. isec ne semble pas fonctionner sur en cas d'ALT multiples
| NC_000020.11 | 32800145 | rs2424926 | C | G,T |                                              |
| NC_000020.11 | 32800145 |    338173 | C | G   | Benign                                       |
| NC_000020.11 | 32800145 |    338174 | C | T   | Conflicting_interpretations_of_pathogenicity |
|              |          |           |   |     |                                              |
3. s'il y a plusieurs variantions à une position, il faut bien vérifier que tous ne sont pas patho.
   La version d'Alexis le fait bien
| NC_000020.11 | 3234173 | rs3827075 | T         | A,C,G     |                                              |
| NC_000020.11 | 3234173 |    262001 | T         | G         | Conflicting_interpretations_of_pathogenicity |
| NC_000020.11 | 3234173 |   1072511 | T         | TGGCGAAGC | Pathogenic                                   |
| NC_000020.11 | 3234173 |    208613 | TGGCGAAGC | G         | Pathogenic                                   |
| NC_000020.11 | 3234173 |      1312 | TGGCGAAGC | T         | Pathogenic                                   |

****** DONE Voir si isec gère les multiallélique (chr20) : non, impossible de faire marcher
CLOSED: [2022-11-27 Sun 00:37]
******* DONE chr20 en prenant un patho clinvar aussi dans dbSNP
CLOSED: [2022-11-27 Sun 00:37]
#+begin_src sh :dir ~/code/bisonex/test_isec
bcftools filter dbSNP_common_chr20.vcf.gz -i 'POS=10652589' -o test_dbsnp.vcf.gz
bcftools filter clinvar_chr20.vcf.gz -i 'POS=10652589' -o test_clinvar.vcf.gz
bcftools index test_dbsnp.vcf.gz
bcftools index test_clinvar.vcf.gz
#+end_src

#+RESULTS:

#+begin_src sh :dir ~/code/bisonex/test_isec
bcftools isec test_dbsnp.vcf.gz test_clinvar.vcf.gz -p tmp
grep '^[^#]' tmp/0002.vcf
grep '^[^#]' tmp/0003.vcf
#+end_src

#+RESULTS:

Même en biallélique, ne fonctionne pas.
Testé en modifiant test_dbsnp !
Fonctionne avec un variant par ligne

****** DONE isec en coupant les sites multialléliques: non
CLOSED: [2022-11-27 Sun 00:37]
******* DONE Exemple simple ok
CLOSED: [2022-11-27 Sun 00:34]
#+begin_src sh :dir ~/code/bisonex/test_isec
bcftools filter -i 'POS=10652589' dbSNP_common_chr20.vcf.gz -o dbsnp_mwi.vcf.gz
bcftools filter -i 'POS=10652589' clinvar_chr20.vcf.gz -o clinvar_mwi.vcf.gz
bcftools index -f dbsnp_mwi.vcf.gz
bcftools index -f clinvar_mwi.vcf.gz
bcftools isec dbsnp_mwi.vcf.gz clinvar_mwi.vcf.gz -n=2
#+end_src

#+RESULTS:

Même en biallélique, ne fonctionne pas.
Chr 20
Avec les fichiers du teste précédent
#+begin_src sh :dir ~/code/bisonex/test_isec

bcftools norm -m -any dbsnp_mwi.vcf.gz -o dbsnp_mwi_norm.vcf.gz
bcftools index dbsnp_mwi_norm.vcf.gz
bcftools isec dbsnp_mwi_norm.vcf.gz clinvar_mwi.vcf.gz -n=2
#+end_src

#+RESULTS:
| NC_000020.11 | 10652589 | G | A | 11 |
| NC_000020.11 | 10652589 | G | C | 11 |
******* DONE Sur dbSNP chr20 non
CLOSED: [2023-10-07 Sat 17:57]
#+begin_src sh :dir ~/code/bisonex/test_isec
bcftools norm -m -any dbSNP_common_chr20 -o dbSNP_common_chr20_norm.vcf.gz
#+end_src
#+begin_src sh :dir ~/code/bisonex/test_isec
bcftools isec -i 'INFO/CLNSIG="Pathogenic"' dbSNP_common_chr20_norm.vcf.gz clinvar_chr20.vcf.gz -p tmp
#+end_src

#+RESULTS:

***** DONE Essai bedtools intersect
#+begin_src sh
bedtools intersect -a  dbSNP_common.vcf.gz -b clinvar.vcf.gz
#+end_src
$ wc -l intersect.vcf
220206 intersect.vcf
** TODO Dépendences avec Nix
*** DONE GATK
CLOSED: [2022-10-21 Fri 21:59]
*** WAIT BioDBHTS
Contribuer pull request
*** DONE BioExtAlign
CLOSED: [2022-10-22 Sat 00:38]
*** WAIT BioBigFile
Revoir si on peut utliser kent dernière version
Contribuer pull request
*** HOLD rtg-tools
Convertir clinvar NC
*** DONE simuscop
CLOSED: [2022-12-30 Fri 22:31]
*** DONE Spip
CLOSED: [2022-12-04 Sun 12:49]
Pas de pull request
*** DONE R + packages
CLOSED: [2022-11-19 Sat 21:05]
*** TODO hap.py
https://github.com/Illumina/hap.py
**** DONE Version sans rtgtools avec python 3
CLOSED: [2023-02-02 Thu 22:15]
Procédure pour tester
#+begin_src
nix develop .#hap-py
$ genericBuild
#+end_src

1. Supprimer l’appel à make_dependencies dans cmakelist.txt : on peut tout installer avec nix
2. Patch Roc.cpp pour avoir numeric_limits ( error: 'numeric_limits' is not a member of 'std')
3. ajout de flags de link (essai, error)
set(ZLIB_LIBRARIES -lz -lbz2 -lcurl -lcrypto -llzma)
4. Changer les appels à print en print() dans le code python et suppression de quelques import
[nix-shell:~/source]$ sed -i.orig 's/print \"\(.*\)"/print(\1)/' src/python/*.py
**** DONE Sérialiser json pour écrire données de sorties
CLOSED: [2023-02-17 Fri 19:25]
**** DONE Tester sur example
CLOSED: [2023-02-04 Sat 00:25]
#+begin_src sh
$ cd hap.py
$ ../result/bin/hap.py example/happy/PG_NA12878_chr21.vcf.gz       example/happy/NA12878_chr21.vcf.gz       -f example/happy/PG_Conf_chr21.bed.gz       -o test -r example/chr21.fa
#+end_src

#+RESULTS:
| Type  | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision | METRIC.Frac_NA | METRIC.F1_Score |
| INDEL | ALL    |        8937 |     7839 |     1098 |       11812 |      343 |      3520 |    45 |   283 |      0.877140 |         0.958635 |       0.298002 |        0.916079 |
| INDEL | PASS   |        8937 |     7550 |     1387 |        9971 |      283 |      1964 |    30 |   242 |      0.844803 |         0.964656 |       0.196971 |        0.900760 |
| SNP   | ALL    |       52494 |    52125 |      369 |       90092 |      582 |     37348 |   107 |   354 |      0.992971 |         0.988966 |       0.414554 |        0.990964 |
| SNP   | PASS   |       52494 |    46920 |     5574 |       48078 |      143 |       992 |     8 |    97 |      0.893816 |         0.996963 |       0.020633 |        0.942576 |

**** KILL Version avec rtg-tools
CLOSED: [2023-07-30 Sun 14:38]
**** HOLD Faire fonctionner Tests
***** HOLD Essai 2 : depuis nix develop:
#+begin_src
nix develop .#hap-py
genericBuild
#+end_src
Lancé initialement à la main, mais on peut maintenant utiliser run_tests
#+begin_src
HCDIR=bin/ ../src/sh/run_tests.sha
#+end_src
- [X] test boost
- [X] multimerge
- [X] hapenum
- [X] fp accuracy
- [X] faulty variant
- leftshift fails
- [X] other vcf
- [X] chr prefix
- [X] gvcf
- [X] decomp
- [X] contig lengt
- [X]  integration test
- [ ] scmp fails sur le type
- [X] giab
- [X] performance
- [ ] quantify fails sur le type
- [ ] stratified échec sur les résultats !
- [X] pg counting
- [ ] sompy: ne trouve pas Strelka dans somatic
phases="buildPhase checkPhase installPhase fixupPhase" genericBuild
#+end_src
**** KILL Reproduire les performances precisionchallenge : attention à HG002 et HG001!
CLOSED: [2023-04-01 Sat 19:43]
https://www.nist.gov/programs-projects/genome-bottle
***** KILL 0GOOR
CLOSED: [2023-04-01 Sat 19:40]
Le problème venait 1. de l'ADN et 2. du renommage des chromosomes qui était faux
****** DONE HG002
CLOSED: [2023-02-17 Fri 19:31]
 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision  METRIC.Frac_NA  METRIC.F1_Score
INDEL    ALL       525466    491355     34111      1156702     57724     605307   9384  25027       0.935084          0.895313        0.523304         0.914766
INDEL   PASS       525466    491355     34111      1156702     57724     605307   9384  25027       0.935084          0.895313        0.523304         0.914766
  SNP    ALL      3365115   3358399      6716      5666020     21995    2284364   4194   1125       0.998004          0.993496        0.403169         0.995745
  SNP   PASS      3365115   3358399      6716      5666020     21995    2284364   4194   1125       0.998004          0.993496        0.403169         0.995745

 TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
                    NaN                     NaN                   1.528276                   2.752637
                    NaN                     NaN                   1.528276                   2.752637
               2.100129                1.473519                   1.581196                   1.795603
               2.100129                1.473519                   1.581196                   1.795603
***** KILL Avec python2
CLOSED: [2023-02-17 Fri 19:25]
****** KILL avec nix
CLOSED: [2023-02-17 Fri 19:25]
conda create -n python2 python=2.7 anaconda
****** KILL avec conda
CLOSED: [2023-02-17 Fri 19:25]
******* Gentoo: regex_error sur test...
Ok avec bash !
#+begin_src

anaconda3/bin/conda create --name py2 python=2.7
conda activate py2
conda install -c bioconda hap.py
#+end_src
******** Faire tourner les tests.
Il faut remplace bin/test_haplotypes par test_haplotypes dans src/sh/run_tests.sh
#+begin_src sh
 HGREF=../genome/GRCh38/GCA_000001405.15_GRCh38_no_alt_analysis_set.fasta HCDIR=~/anaconda3/envs/py2/bin bash src/sh/run_tests.sh
#+end_src
Echec:

test_haplotypes: /opt/conda/conda-bld/work/hap.py-0.3.7/src/c++/lib/tools/Fasta.cpp:81: MMappedFastaFile::MMappedFastaFile(const string&): Assertion `fd != -1' failed.
unknown location(0): fatal error in "testVariantPrimitiveSplitter": signal: SIGABRT (application abort requested)
/opt/conda/conda-bld/work/hap.py-0.3.7/src/c++/test/test_align.cpp(298): last checkpoint
******** Chr21
HGREF=../genome/GRCh38/GCA_000001405.15_GRCh38_no_alt_analysis_set.fasta hap.py        example/happy/PG_NA12878_chr21.vcf.gz       example/happy/NA12878_chr21.vcf.gz       -f example/happy/PG_Conf_chr21.bed.gz       -o test
******* Helios
échec
** TODO T2T :T2T:
Toutes les ressourcs sont décrites ici
https://github.com/marbl/CHM13
Détails sur le pipeline
https://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hub_3267197_GCA_009914755.4&c=CP068277.2&g=hub_3267197_hgLiftOver
*** Liftover pipelines
:PROPERTIES:
:ID:       d2280207-3f65-4a31-a291-41fa9a9658c2
:END:
Contient les chain files
*** DONE Alignement
CLOSED: [2023-06-26 Mon 19:42]
NXF_OPTS=-D"user.name=${USER}" nextflow run main.nf -profile standard,helios  --input="/Work/Groups/bisonex/data/giab/*_R{1,2}_001.fastq.gz" --id=NA12878-T2T -bg
SCHEDULED: <2023-06-14 Wed>
*** DONE Haplotypecaller
CLOSED: [2023-06-26 Mon 19:42] SCHEDULED: <2023-06-15 Thu>
*** DONE Faire fonctionner le filtre technical variant
CLOSED: [2023-08-03 Thu 14:24] SCHEDULED: <2023-08-03 Wed 10:30>
*** DONE Annotation vep seule
CLOSED: [2023-08-05 Sat 08:59] SCHEDULED: <2023-08-05 Sat>
T2T n'a pas
- de version merged
- polyphen
- gnomAD
On désactive l'annotation spip pour le moment
*** DONE Générer la base de donnée spip :spip:
CLOSED: [2023-08-09 Wed 21:41] SCHEDULED: <2023-08-03 Thu 11:30>
**** KILL Vérifier la génération du transcriptome en hg38: checksum différent
CLOSED: [2023-08-09 Wed 21:41]
- [X] Nettoyer et vérifier sur hg38 avec ediff les RData : différent
- [X] Sinon, ne pas nettoyer et générer: idem
**** DONE Récupérer ncbi RefSeq curated
CLOSED: [2023-08-07 Mon 22:59] SCHEDULED: <2023-08-06 Sun>
.txt sur UCSC mais pas en T2T: http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/
Format: https://genome.ucsc.edu/cgi-bin/hgTables?hgsid=1173061381_UepaHnvaOKFZKMOV4o7DtcNUHGVa&hgta_doSchemaDb=chlSab2&hgta_doSchemaTable=ncbiRefSeqCurated
Ancient format vs nouveau
|  1 | bin          |  1 | chrom        |
|  2 | name         |  2 | chromStart   |
|  3 | chrom        |  3 | chromEnd     |
|  4 | strand       |  4 | name         |
|  5 | txStart      |  5 | score        |
|  6 | txEnd        |  6 | strand       |
|  7 | cdsStart     |  7 | thickStart   |
|  8 | cdsEnd       |  8 | thickEnd     |
|  9 | exonCount    |  9 | reserved     |
| 10 | exonStarts   | 10 | blockCount   |
| 11 | exonEnds     | 11 | blockSizes   |
| 12 | score        | 12 | chromStarts  |
| 13 | name2        | 13 | name2        |
| 14 | cdsStartStat | 14 | cdsStartStat |
| 15 | cdsEndStat   | 15 | cdsEndStat   |
| 16 | exonFrames   | 16 | exonFrames   |
|    |              | 17 | type         |
|    |              | 18 | geneName     |
|    |              | 19 | geneName2    |
|    |              | 20 | geneType     |


En T2T, seulement au format bigBed : https://hgdownload.soe.ucsc.edu/gbdb/hs1/ncbiRefSeq/
Il y a un exécutable pour convertir en bed : http://hgdownload.soe.ucsc.edu/admin/exe/
Sous gentoo, il faut instaler mit-krb5 (pour libkrb5)
#+begin_src
./bigBedToBed ncbiRefSeqCurated.bb ncbiRefSeqCurated.bed
#+end_src
Exemple:
chr1    7505    13582   NR_182076.1     0       -       13582   13582   0       2       5477,138,       0,5939, LOC127239154    none    none    -1,-1,          NR_182076.1     LOC127239154

Dans R:
   V1        V2   V3 V4    V5    V6    V7    V8 V9                V10
1 585 NR_046018 chr1  + 11873 14409 14409 14409  3 11873,12612,13220,
                 V11 V12     V13  V14  V15       V16           V17         V18
1 12227,12721,14409,   0 DDX11L1 none none -1,-1,-1, 354,109,1189, 0,739,1347,

Ne pas oublier les headers car ils sont dans un ordre différent:
Colonnes en GRGh38 =
3, 5, 6, 2, 12, 4, 7,  8, 12, 9, 17, 18, 13
Correspondance en T2T
1, 7, 8, 4, 5,  6, 14, 15, 5, ?,  ?,  ?, 13

En fait, il suffit d'avoir
- le gène
- le début du transcrit
- la fin du transcrit
- le brin
  pour générer
***** KILL Tester correspondance partielle ?
CLOSED: [2023-08-07 Mon 22:58]
pas de CDS et pas de colonne 17 et 18
seules les colonnes (dans la nouvelle dataframe) 10,11,12 causent problèmes (9,17,18 dans les ancienne)

NB: on peut retrouver le nombre d'exons colonnes 9 à partir de la lons
***** DONE Correspondance totale
CLOSED: [2023-08-07 Mon 22:59]
> dataRefSeq[1,]
    V1    V2    V3        V4 V5 V6    V7    V8 V9 V10           V11         V12
1 chr1 11873 14409 NR_046018  0  + 14409 14409  0   3 354,109,1189, 0,739,1347,
      V13
1 DDX11L1
> source("pkgs/getRefSeqDatabaseT2T.r")
Use the URL: http://hgdownload.cse.ucsc.edu/goldenPath/
read files...
> dataRefSeq[1,]
    V1   V2    V3        V4 V5 V6    V7    V8 V5.1 V10       V11     V12
1 chr1 7505 13582 NR_182076  0  - 13582 13582    0   2 5477,138, 0,5939,
           V13
1 LOC127239154
*** DONE Inclure génération refseq + transcriptom T2T dans dérivation nix :spip:
CLOSED: [2023-08-12 Sat 18:08] SCHEDULED: <2023-08-12 Sat 15:00>
**** DONE [#A] Tester nouvelle dérivation
CLOSED: [2023-08-12 Sat 18:08] SCHEDULED: <2023-08-12 Sat 15:00>
***** DONE En hg38
CLOSED: [2023-08-12 Sat 17:59]
test-spip-hg38.txt
varID
NM_000051:c.2251-10T>G
NM_000267:c.889-12T>A
NM_000059:c.8488-9T>G
NM_000249:c.589-10T>A
NM_000249:c.791-7T>A
#+begin_src
❯ ../result/bin/spip -I test-spip-hg38.txt -O test-spip-hg38.out -g hg38
#+end_src
#+RESULTS:
| varID                  | Interpretation | InterConfident | SPiPscore | strand | gNomen | varType | ntChange | ExonInfo | exonSize | transcript |  gene | NearestSS |    DistSS | RegType      | SPiCEproba | SPiCEinter_2thr | deltaMES |    BP | mutInPBarea | deltaESRscore | posCryptMut | sstypeCryptMut | probaCryptMut | classProbaCryptMut | nearestSStoCrypt | nearestPosSStoCrypt | nearestDistSStoCrypt | posCryptWT | probaCryptWT | classProbaCryptWT | posSSPhysio | probaSSPhysio | classProbaSSPhysio | probaSSPhysioMut | classProbaSSPhysioMut |   |   |    |           |             |    |                |    |
| NM_000051:c.2251-10T>G | Alter          | by             | SPiCE     |  98.41 | %      | [91.47  | %        | -        |    99.96 | %]         | 0.986 | +         | 108257471 | substitution | T>G        | Intron          |       14 |  1140 | NM_000051   | ATM           | acceptor    |            -10 | IntronCons    |                  1 | high             |                   0 | 0No                  |         10 |    108257471 | Acc               | 0.024836003 | No            | Acc                |        108257480 |                   -10 | 0 | 0 | No | 108257480 | 0.006489079 | No | 0.000004368542 | No |
| NM_000267:c.889-12T>A  | Alter          | by             | SPiCE     |  98.41 | %      | [91.47  | %        | -        |    99.96 | %]         | 1.000 | +         |  31200410 | substitution | T>A        | Intron          |        8 | 17756 | NM_000267   | NF1           | acceptor    |            -12 | IntronCons    |                  1 | high             |                   0 | 0No                  |         10 |     31200411 | Acc               | 0.009082899 | No            | Acc                |         31200421 |                   -11 | 0 | 0 | No |  31200421 | 0.005160854 | No | 0.000003718518 | No |
| NM_000059:c.8488-9T>G  | Alter          | by             | SPiCE     |  98.41 | %      | [91.47  | %        | -        |    99.96 | %]         | 0.994 | +         |  32370947 | substitution | T>G        | Intron          |       19 |   398 | NM_000059   | BRCA2         | acceptor    |             -9 | IntronCons    |                  1 | high             |                   0 | 0No                  |         10 |     32370947 | Acc               | 0.004449623 | No            | Acc                |         32370955 |                    -9 | 0 | 0 | No |  32370955 | 0.005060308 | No | 0.000005609419 | No |
| NM_000249:c.589-10T>A  | Alter          | by             | SPiCE     |  98.41 | %      | [91.47  | %        | -        |    99.96 | %]         | 0.978 | +         |  37012001 | substitution | T>A        | Intron          |        7 |   148 | NM_000249   | MLH1          | acceptor    |            -10 | IntronCons    |                  1 | high             |                   0 | 0No                  |         10 |     37012002 | Acc               | 0.009529819 | No            | Acc                |         37012010 |                    -9 | 0 | 0 | No |  37012010 | 0.028437574 | No | 0.000009275960 | No |
| NM_000249:c.791-7T>A   | Alter          | by             | SPiCE     |  98.41 | %      | [91.47  | %        | -        |    99.96 | %]         | 0.988 | +         |  37017499 | substitution | T>A        | Intron          |        9 |  2961 | NM_000249   | MLH1          | acceptor    |             -7 | IntronCons    |                  1 | high             |                   0 | 0No                  |         10 |     37017500 | Acc               | 0.015564917 | No            | Acc                |         37017505 |                    -6 | 0 | 0 | No |  37017505 | 0.023995855 | No | 0.000022606476 | No |
|                        |                |                |           |        |        |         |          |          |          |            |       |           |           |              |            |                 |          |       |             |               |             |                |               |                    |                  |                     |                      |            |              |                   |             |               |                    |                  |                       |   |   |    |           |             |    |                |    |
|                        |                |                |           |        |        |         |          |          |          |            |       |           |           |              |            |                 |          |       |             |               |             |                |               |                    |                  |                     |                      |            |              |                   |             |               |                    |                  |                       |   |   |    |           |             |    |                |    |
***** DONE En T2T
CLOSED: [2023-08-12 Sat 18:08]
****** DONE Sur  1 variant avec score 35.01%: ok
CLOSED: [2023-08-12 Sat 18:05]
##fileformat=VCFv4.0
##assembly=CHM18v2.0/hs1
##ALT=<ID=*,Description="Represents allele(s) other than observed.">
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO
chr10	89894645	lol	A	G	.	.	.

 #+begin_src
 ../result/bin/spip -I test-spip-T2T.vcf -O test-spip-T2T.out -g hs1
 #+end_src
#+RESULTS:
| CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	varID	Interpretation	InterConfident	SPiPscore	strand	gNomen	varType	ntChange	ExonInfo	exonSize	transcript	gene	NearestSS	DistSS	RegType	SPiCEproba	SPiCEinter_2thr	deltaMES	BP	mutInPBarea	deltaESRscore	posCryptMut	sstypeCryptMut	probaCryptMut	classProbaCryptMut	nearestSStoCrypt	nearestPosSStoCrypt	nearestDistSStoCrypt	posCryptWT	probaCryptWT	classProbaCryptWT	posSSPhysio	probaSSPhysio	classProbaSSPhysio	probaSSPhysioMut	classProbaSSPhysioMut |
| chr10	89894645	lol	A	G	.	.	.	NM_000043:g.89894645:A>G	Alter ESR	35.81 % [28.11 % - 44.1 %]	0.288	+	89894645	substitution	A>G	Exon 6	  63	NM_000043	FAS	acceptor	  8	ExonESR	0	Outside SPiCE Interpretation	0	0	No	-1.67753	89894644	Acc	0.0000003317384	No	Acc	89894637	7	89894644	0.0000002205815	No	89894637	0.02545572	No	0.02545572	No |
| chr10	89894645	lol	A	G	.	.	.	NM_001320619:g.89894645:A>G	Alter ESR	35.81 % [28.11 % - 44.1 %]	0.288	+	89894645	substitution	A>G	Exon 6	  63	NM_001320619	FAS	acceptor	  8	ExonESR	0	Outside SPiCE Interpretation	0	0	No	-1.67753	89894644	Acc	0.0000003317384	No	Acc	89894637	7	89894644	0.0000002205815	No	89894637	0.02545572	No	0.02545572	No |
| chr10	89894645	lol	A	G	.	.	.	NM_152871:g.89894645:A>G	NTR	00 % [00 % - 00.92 %]	0.000	+	89894645	substitution	A>G	Intron 5	1398	NM_152871	FAS	donor	160	DeepIntron	0	Outside SPiCE Interpretation	0	0	No	10.00000	89894644	Don	0.0001360257829	No	Don	89894485	159	0	0.0000000000000	No	89894485	0.07177992	Yes	0.07177992	Yes |
| chr10	89894645	lol	A	G	.	.	.	NM_152872:g.89894645:A>G	Alter ESR	35.81 % [28.11 % - 44.1 %]	0.288	+	89894645	substitution	A>G	Exon 6	  63	NM_152872	FAS	acceptor	  8	ExonESR	0	Outside SPiCE Interpretation	0	0	No	-1.67753	89894644	Acc	0.0000003317384	No	Acc	89894637	7	89894644	0.0000002205815	No	89894637	0.02545572	No	0.02545572	No |
| chr10	89894645	lol	A	G	.	.	.	NR_028033:g.89894645:A>G	NTR	00 % [00 % - 00.92 %]	0.000	+	89894645	substitution	A>G	Intron 4	1398	NR_028033	FAS	donor	160	DeepIntron	0	Outside SPiCE Interpretation	0	0	No	10.00000	89894644	Don	0.0001360257829	No	Don	89894485	159	0	0.0000000000000	No	89894485	0.07177992	Yes	0.07177992	Yes |
| chr10	89894645	lol	A	G	.	.	.	NR_028034:g.89894645:A>G	NTR	00 % [00 % - 00.92 %]	0.000	+	89894645	substitution	A>G	Intron 3	1398	NR_028034	FAS	donor	160	DeepIntron	0	Outside SPiCE Interpretation	0	0	No	10.00000	89894644	Don	0.0001360257829	No	Don	89894485	159	0	0.0000000000000	No	89894485	0.07177992	Yes	0.07177992	Yes |
| chr10	89894645	lol	A	G	.	.	.	NR_028035:g.89894645:A>G	Alter ESR	35.81 % [28.11 % - 44.1 %]	0.288	+	89894645	substitution	A>G	Exon 4	  63	NR_028035	FAS	acceptor	  8	ExonESR	0	Outside SPiCE Interpretation	0	0	No	-1.67753	89894644	Acc	0.0000003317384	No	Acc	89894637	7	89894644	0.0000002205815	No	89894637	0.02545572	No	0.02545572	No |
| chr10	89894645	lol	A	G	.	.	.	NR_028036:g.89894645:A>G	Alter ESR	35.81 % [28.11 % - 44.1 %]	0.288	+	89894645	substitution	A>G	Exon 5	  63	NR_028036	FAS	acceptor	  8	ExonESR	0	Outside SPiCE Interpretation	0	0	No	-1.67753	89894644	Acc	0.0000003317384	No	Acc	89894637	7	89894644	0.0000002205815	No	89894637	0.02545572	No	0.02545572	No |
| chr10	89894645	lol	A	G	.	.	.	NR_135313:g.89894645:A>G	Alter ESR	35.81 % [28.11 % - 44.1 %]	0.288	+	89894645	substitution	A>G	Exon 5	  63	NR_135313	FAS	acceptor	  8	ExonESR	0	Outside SPiCE Interpretation	0	0	No	-1.67753	89894644	Acc	0.0000003317384	No	Acc	89894637	7	89894644	0.0000002205815	No	89894637	0.02545572	No	0.02545572	No |
| chr10	89894645	lol	A	G	.	.	.	NM_001410956:g.89894645:A>G	Alter ESR	35.81 % [28.11 % - 44.1 %]	0.288	+	89894645	substitution	A>G	Exon 6	  63	NM_001410956	FAS	acceptor	  8	ExonESR	0	Outside SPiCE Interpretation	0	0	No	-1.67753	89894644	Acc	0.0000003317384	No	Acc	89894637	7	89894644	0.0000002205815	No	89894637	0.02545572	No	0.02545572	No |
| chr10	89894645	lol	A	G	.	.	.	NR_135314:g.89894645:A>G	Alter ESR	35.81 % [28.11 % - 44.1 %]	0.288	+	89894645	substitution	A>G	Exon 6	  63	NR_135314	FAS	acceptor	  8	ExonESR	0	Outside SPiCE Interpretation	0	0	No	-1.67753	89894644	Acc	0.0000003317384	No	Acc	89894637	7	89894644	0.0000002205815	No	89894637	0.02545572	No	0.02545572	No |
| chr10	89894645	lol	A	G	.	.	.	NR_135315:g.89894645:A>G	Alter ESR	35.81 % [28.11 % - 44.1 %]	0.288	+	89894645	substitution	A>G	Exon 4	  63	NR_135315	FAS	acceptor	  8	ExonESR	0	Outside SPiCE Interpretation	0	0	No	-1.67753	89894644	Acc	0.0000003317384	No	Acc	89894637	7	89894644	0.0000002205815	No	89894637	0.02545572	No	0.02545572	No |
|                                                                                                                      |
****** DONE 5 variants patho
CLOSED: [2023-08-12 Sat 18:08]
##fileformat=VCFv4.0
##assembly=CHM13v2.0/hs1
##ALT=<ID=*,Description="Represents allele(s) other than observed.">
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO
chr11	108264994	lol	T	G	.	.	.
chr17	32146124	lol	T	A	.	.	.
chr3	31588185	lol	T	G	.	.	.
chr3	37013346	lol	T	A	.	.	.
chr3	37018844	lol	T	A	.	.	.

 #+begin_src
 ../result/bin/spip -I test-spip-T2T.vcf -O test-spip-T2T.out -g hs1
 #+end_src

 #+RESULTS:
| CHROM |       POS | ID  | REF | ALT | QUAL | FILTER | INFO | varID                        | Interpretation | InterConfident              | SPiPscore | strand |    gNomen | varType      | ntChange | ExonInfo  | exonSize | transcript   | gene  | NearestSS | DistSS | RegType    | SPiCEproba | SPiCEinter_2thr              | deltaMES | BP | mutInPBarea | deltaESRscore | posCryptMut | sstypeCryptMut |    probaCryptMut | classProbaCryptMut | nearestSStoCrypt | nearestPosSStoCrypt | nearestDistSStoCrypt | posCryptWT |     probaCryptWT | classProbaCryptWT | posSSPhysio | probaSSPhysio | classProbaSSPhysio | probaSSPhysioMut | classProbaSSPhysioMut |
| chr11 | 108264994 | lol | T   | G   | .    | .      | .    | NM_000051:g.108264994:T>G    | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.986 | +      | 108264994 | substitution | T>G      | Intron 14 |     1140 | NM_000051    | ATM   | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |   108264994 | Acc            | 0.02483600258856 | No                 | Acc              |           108265003 |                  -10 |          0 | 0.00000000000000 | No                |   108265003 |   0.006489079 | No                 |   0.000004368542 | No                    |
| chr11 | 108264994 | lol | T   | G   | .    | .      | .    | NM_001351834:g.108264994:T>G | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.986 | +      | 108264994 | substitution | T>G      | Intron 15 |     1140 | NM_001351834 | ATM   | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |   108264994 | Acc            | 0.02483600258856 | No                 | Acc              |           108265003 |                  -10 |          0 | 0.00000000000000 | No                |   108265003 |   0.006489079 | No                 |   0.000004368542 | No                    |
| chr17 |  32146124 | lol | T   | A   | .    | .      | .    | NM_000267:g.32146124:T>A     | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     1.000 | +      |  32146124 | substitution | T>A      | Intron 8  |    17755 | NM_000267    | NF1   | acceptor  |    -12 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    32146125 | Acc            | 0.00908289924996 | No                 | Acc              |            32146135 |                  -11 |          0 | 0.00000000000000 | No                |    32146135 |   0.005160854 | No                 |   0.000003718518 | No                    |
| chr17 |  32146124 | lol | T   | A   | .    | .      | .    | NM_001128147:g.32146124:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     1.000 | +      |  32146124 | substitution | T>A      | Intron 8  |    17755 | NM_001128147 | NF1   | acceptor  |    -12 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    32146125 | Acc            | 0.00908289924996 | No                 | Acc              |            32146135 |                  -11 |          0 | 0.00000000000000 | No                |    32146135 |   0.005160854 | No                 |   0.000003718518 | No                    |
| chr17 |  32146124 | lol | T   | A   | .    | .      | .    | NM_001042492:g.32146124:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     1.000 | +      |  32146124 | substitution | T>A      | Intron 8  |    17755 | NM_001042492 | NF1   | acceptor  |    -12 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    32146125 | Acc            | 0.00908289924996 | No                 | Acc              |            32146135 |                  -11 |          0 | 0.00000000000000 | No                |    32146135 |   0.005160854 | No                 |   0.000003718518 | No                    |
| chr3  |  31588185 | lol | T   | G   | .    | .      | .    | NM_178862:g.31588185:T>G     | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  31588185 | substitution | T>G      | Intron 3  |    16710 | NM_178862    | STT3B | donor     |   5663 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    31588184 | Don            | 0.00000002308527 | No                 | Don              |            31582522 |                 5662 |   31588181 | 0.00000000129731 | No                |    31582522 |   0.071835522 | Yes                |   0.071835521718 | Yes                   |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_000249:g.37013346:T>A     | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_000249    | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001167617:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001167617 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001167618:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001167618 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001167619:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 6  |      148 | NM_001167619 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001258271:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001258271 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001258273:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 6  |      148 | NM_001258273 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001258274:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 8  |      148 | NM_001258274 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354615:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 6  |      148 | NM_001354615 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354616:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 6  |      148 | NM_001354616 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354617:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354617 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354618:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354618 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354619:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 8  |      148 | NM_001354619 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354620:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354620 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354621:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354621 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354622:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354622 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354623:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 6  |      148 | NM_001354623 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354624:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354624 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354625:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 6  |      148 | NM_001354625 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354626:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354626 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354627:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354627 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354628:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354628 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354629:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 6  |      148 | NM_001354629 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354630:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354630 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_000249:g.37018844:T>A     | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_000249    | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001167617:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001167617 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001167618:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001167618 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001167619:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 8  |     2961 | NM_001167619 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001258271:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001258271 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001258273:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 8  |     2961 | NM_001258273 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001258274:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 10 |     2961 | NM_001258274 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354615:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 8  |     2961 | NM_001354615 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354616:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 8  |     2961 | NM_001354616 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354617:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001354617 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354618:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001354618 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354619:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 10 |     2961 | NM_001354619 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354620:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001354620 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354621:g.37018844:T>A  | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  37018844 | substitution | T>A      | Intron 8  |     8209 | NM_001354621 | MLH1  | acceptor  |  -2810 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37021653 |                -2809 |   37018850 | 0.02399585516738 | No                |    37021653 |   0.241893494 | Yes                |   0.241893494461 | Yes                   |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354622:g.37018844:T>A  | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  37018844 | substitution | T>A      | Intron 9  |     5764 | NM_001354622 | MLH1  | acceptor  |  -2810 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37021653 |                -2809 |   37018850 | 0.02399585516738 | No                |    37021653 |   0.241893494 | Yes                |   0.241893494461 | Yes                   |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354623:g.37018844:T>A  | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  37018844 | substitution | T>A      | Intron 8  |     5764 | NM_001354623 | MLH1  | acceptor  |  -2810 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37021653 |                -2809 |   37018850 | 0.02399585516738 | No                |    37021653 |   0.241893494 | Yes                |   0.241893494461 | Yes                   |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354624:g.37018844:T>A  | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  37018844 | substitution | T>A      | Intron 9  |    11091 | NM_001354624 | MLH1  | donor     |   2955 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Don              |            37015889 |                 2956 |   37018850 | 0.02399585516738 | No                |    37026980 |   0.028353019 | No                 |   0.028353019047 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354625:g.37018844:T>A  | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  37018844 | substitution | T>A      | Intron 8  |    11091 | NM_001354625 | MLH1  | donor     |   2955 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Don              |            37015889 |                 2956 |   37018850 | 0.02399585516738 | No                |    37026980 |   0.028353019 | No                 |   0.028353019047 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354626:g.37018844:T>A  | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  37018844 | substitution | T>A      | Intron 9  |    11091 | NM_001354626 | MLH1  | donor     |   2955 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Don              |            37015889 |                 2956 |   37018850 | 0.02399585516738 | No                |    37026980 |   0.028353019 | No                 |   0.028353019047 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354627:g.37018844:T>A  | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  37018844 | substitution | T>A      | Intron 9  |    11091 | NM_001354627 | MLH1  | donor     |   2955 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Don              |            37015889 |                 2956 |   37018850 | 0.02399585516738 | No                |    37026980 |   0.028353019 | No                 |   0.028353019047 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354628:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001354628 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354629:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 8  |     2961 | NM_001354629 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354630:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001354630 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |

*** DONE Vérifier annotation SPIP sur variants confirmé
CLOSED: [2023-08-10 Thu 23:16] SCHEDULED: <2023-08-09 Wed>
**** DONE 5 variants patho tirés de l'article princips
CLOSED: [2023-08-10 Thu 23:00]
On trié par SQUIRLS décroissant
#+begin_src sh
varID
NM_000051:c.2251-10T>G
NM_000267:c.889-12T>A
NM_000059:c.8488-9T>G
NM_000249:c.589-10T>A
NM_000249:c.791-7T>A
#+end_src
***** DONE En hg38
CLOSED: [2023-08-09 Wed 22:01]
#+begin_src
spip --input test-spip.txt --output test-spip.out --GenomeAssenbly hg38 --threads 1 --maxLines 1000
#+end_src

#+RESULTS:
| varID                  | Interpretation | InterConfident              | SPiPscore | strand |    gNomen | varType      | ntChange | ExonInfo  | exonSize | transcript | gene  | NearestSS | DistSS | RegType    | SPiCEproba | SPiCEinter_2thr | deltaMES | BP  | mutInPBarea | deltaESRscore | posCryptMut | sstypeCryptMut | probaCryptMut | classProbaCryptMut | nearestSStoCrypt | nearestPosSStoCrypt | nearestDistSStoCrypt | posCryptWT | probaCryptWT | classProbaCryptWT | posSSPhysio | probaSSPhysio | classProbaSSPhysio | probaSSPhysioMut | classProbaSSPhysioMut |
| NM_000051:c.2251-10T>G | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.986 | +      | 108257471 | substitution | T>G      | Intron 14 |     1140 | NM_000051  | ATM   | acceptor  |    -10 | IntronCons |          1 | high            |        0 | 0No |          10 |     108257471 | Acc         |    0.024836003 | No            | Acc                |        108257480 |                 -10 |                    0 |          0 | No           |         108257480 | 0.006489079 | No            |     0.000004368542 | No               |                       |
| NM_000267:c.889-12T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     1.000 | +      |  31200410 | substitution | T>A      | Intron 8  |    17756 | NM_000267  | NF1   | acceptor  |    -12 | IntronCons |          1 | high            |        0 | 0No |          10 |      31200411 | Acc         |    0.009082899 | No            | Acc                |         31200421 |                 -11 |                    0 |          0 | No           |          31200421 | 0.005160854 | No            |     0.000003718518 | No               |                       |
| NM_000059:c.8488-9T>G  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.994 | +      |  32370947 | substitution | T>G      | Intron 19 |      398 | NM_000059  | BRCA2 | acceptor  |     -9 | IntronCons |          1 | high            |        0 | 0No |          10 |      32370947 | Acc         |    0.004449623 | No            | Acc                |         32370955 |                  -9 |                    0 |          0 | No           |          32370955 | 0.005060308 | No            |     0.000005609419 | No               |                       |
| NM_000249:c.589-10T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37012001 | substitution | T>A      | Intron 7  |      148 | NM_000249  | MLH1  | acceptor  |    -10 | IntronCons |          1 | high            |        0 | 0No |          10 |      37012002 | Acc         |    0.009529819 | No            | Acc                |         37012010 |                  -9 |                    0 |          0 | No           |          37012010 | 0.028437574 | No            |     0.000009275960 | No               |                       |
| NM_000249:c.791-7T>A   | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37017499 | substitution | T>A      | Intron 9  |     2961 | NM_000249  | MLH1  | acceptor  |     -7 | IntronCons |          1 | high            |        0 | 0No |          10 |      37017500 | Acc         |    0.015564917 | No            | Acc                |         37017505 |                  -6 |                    0 |          0 | No           |          37017505 | 0.023995855 | No            |     0.000022606476 | No               |                       |

Test sur mobidetails : 98% pour spip (! différent du fichier excel...)

Second variant: ok en VCf également

***** DONE Lifter les variants T2T : ok ! mais multiples trnascrits en génomique ...
CLOSED: [2023-08-09 Wed 23:23] SCHEDULED: <2023-08-09 Wed>
##fileformat=VCFv4.0
##assembly=GRCh38/hg38
##ALT=<ID=*,Description="Represents allele(s) other than observed.">
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO
chr11	108264994	lol	T	G	.	.	.
chr17	32146124	lol	T	A	.	.	.
chr3	31588185	lol	T	G	.	.	.
chr3	37013346	lol	T	A	.	.	.
chr3	37018844	lol	T	A	.	.	.

❯ ./result/bin/spip -I test-spip-T2T.vcf -O test-spip-T2T.out -g hs1 --refseq dataRefSeqhs1.RData --transcriptome transcriptome_hs1.RData

        #+RESULTS:
| CHROM |       POS | ID  | REF | ALT | QUAL | FILTER | INFO | varID                        | Interpretation | InterConfident              | SPiPscore | strand |    gNomen | varType      | ntChange | ExonInfo  | exonSize | transcript   | gene  | NearestSS | DistSS | RegType    | SPiCEproba | SPiCEinter_2thr              | deltaMES | BP | mutInPBarea | deltaESRscore | posCryptMut | sstypeCryptMut |    probaCryptMut | classProbaCryptMut | nearestSStoCrypt | nearestPosSStoCrypt | nearestDistSStoCrypt | posCryptWT |     probaCryptWT | classProbaCryptWT | posSSPhysio | probaSSPhysio | classProbaSSPhysio | probaSSPhysioMut | classProbaSSPhysioMut |
| chr11 | 108264994 | lol | T   | G   | .    | .      | .    | NM_000051:g.108264994:T>G    | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.986 | +      | 108264994 | substitution | T>G      | Intron 14 |     1140 | NM_000051    | ATM   | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |   108264994 | Acc            | 0.02483600258856 | No                 | Acc              |           108265003 |                  -10 |          0 | 0.00000000000000 | No                |   108265003 |   0.006489079 | No                 |   0.000004368542 | No                    |
| chr11 | 108264994 | lol | T   | G   | .    | .      | .    | NM_001351834:g.108264994:T>G | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.986 | +      | 108264994 | substitution | T>G      | Intron 15 |     1140 | NM_001351834 | ATM   | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |   108264994 | Acc            | 0.02483600258856 | No                 | Acc              |           108265003 |                  -10 |          0 | 0.00000000000000 | No                |   108265003 |   0.006489079 | No                 |   0.000004368542 | No                    |
| chr17 |  32146124 | lol | T   | A   | .    | .      | .    | NM_000267:g.32146124:T>A     | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     1.000 | +      |  32146124 | substitution | T>A      | Intron 8  |    17755 | NM_000267    | NF1   | acceptor  |    -12 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    32146125 | Acc            | 0.00908289924996 | No                 | Acc              |            32146135 |                  -11 |          0 | 0.00000000000000 | No                |    32146135 |   0.005160854 | No                 |   0.000003718518 | No                    |
| chr17 |  32146124 | lol | T   | A   | .    | .      | .    | NM_001128147:g.32146124:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     1.000 | +      |  32146124 | substitution | T>A      | Intron 8  |    17755 | NM_001128147 | NF1   | acceptor  |    -12 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    32146125 | Acc            | 0.00908289924996 | No                 | Acc              |            32146135 |                  -11 |          0 | 0.00000000000000 | No                |    32146135 |   0.005160854 | No                 |   0.000003718518 | No                    |
| chr17 |  32146124 | lol | T   | A   | .    | .      | .    | NM_001042492:g.32146124:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     1.000 | +      |  32146124 | substitution | T>A      | Intron 8  |    17755 | NM_001042492 | NF1   | acceptor  |    -12 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    32146125 | Acc            | 0.00908289924996 | No                 | Acc              |            32146135 |                  -11 |          0 | 0.00000000000000 | No                |    32146135 |   0.005160854 | No                 |   0.000003718518 | No                    |
| chr3  |  31588185 | lol | T   | G   | .    | .      | .    | NM_178862:g.31588185:T>G     | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  31588185 | substitution | T>G      | Intron 3  |    16710 | NM_178862    | STT3B | donor     |   5663 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    31588184 | Don            | 0.00000002308527 | No                 | Don              |            31582522 |                 5662 |   31588181 | 0.00000000129731 | No                |    31582522 |   0.071835522 | Yes                |   0.071835521718 | Yes                   |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_000249:g.37013346:T>A     | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_000249    | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001167617:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001167617 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001167618:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001167618 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001167619:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 6  |      148 | NM_001167619 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001258271:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001258271 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001258273:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 6  |      148 | NM_001258273 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001258274:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 8  |      148 | NM_001258274 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354615:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 6  |      148 | NM_001354615 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354616:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 6  |      148 | NM_001354616 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354617:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354617 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354618:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354618 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354619:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 8  |      148 | NM_001354619 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354620:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354620 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354621:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354621 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354622:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354622 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354623:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 6  |      148 | NM_001354623 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354624:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354624 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354625:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 6  |      148 | NM_001354625 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354626:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354626 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354627:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354627 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354628:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354628 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354629:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 6  |      148 | NM_001354629 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37013346 | lol | T   | A   | .    | .      | .    | NM_001354630:g.37013346:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.978 | +      |  37013346 | substitution | T>A      | Intron 7  |      148 | NM_001354630 | MLH1  | acceptor  |    -10 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37013347 | Acc            | 0.00952981867149 | No                 | Acc              |            37013355 |                   -9 |          0 | 0.00000000000000 | No                |    37013355 |   0.028437574 | No                 |   0.000009275960 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_000249:g.37018844:T>A     | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_000249    | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001167617:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001167617 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001167618:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001167618 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001167619:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 8  |     2961 | NM_001167619 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001258271:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001258271 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001258273:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 8  |     2961 | NM_001258273 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001258274:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 10 |     2961 | NM_001258274 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354615:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 8  |     2961 | NM_001354615 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354616:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 8  |     2961 | NM_001354616 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354617:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001354617 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354618:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001354618 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354619:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 10 |     2961 | NM_001354619 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354620:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001354620 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354621:g.37018844:T>A  | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  37018844 | substitution | T>A      | Intron 8  |     8209 | NM_001354621 | MLH1  | acceptor  |  -2810 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37021653 |                -2809 |   37018850 | 0.02399585516738 | No                |    37021653 |   0.241893494 | Yes                |   0.241893494461 | Yes                   |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354622:g.37018844:T>A  | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  37018844 | substitution | T>A      | Intron 9  |     5764 | NM_001354622 | MLH1  | acceptor  |  -2810 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37021653 |                -2809 |   37018850 | 0.02399585516738 | No                |    37021653 |   0.241893494 | Yes                |   0.241893494461 | Yes                   |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354623:g.37018844:T>A  | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  37018844 | substitution | T>A      | Intron 8  |     5764 | NM_001354623 | MLH1  | acceptor  |  -2810 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37021653 |                -2809 |   37018850 | 0.02399585516738 | No                |    37021653 |   0.241893494 | Yes                |   0.241893494461 | Yes                   |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354624:g.37018844:T>A  | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  37018844 | substitution | T>A      | Intron 9  |    11091 | NM_001354624 | MLH1  | donor     |   2955 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Don              |            37015889 |                 2956 |   37018850 | 0.02399585516738 | No                |    37026980 |   0.028353019 | No                 |   0.028353019047 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354625:g.37018844:T>A  | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  37018844 | substitution | T>A      | Intron 8  |    11091 | NM_001354625 | MLH1  | donor     |   2955 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Don              |            37015889 |                 2956 |   37018850 | 0.02399585516738 | No                |    37026980 |   0.028353019 | No                 |   0.028353019047 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354626:g.37018844:T>A  | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  37018844 | substitution | T>A      | Intron 9  |    11091 | NM_001354626 | MLH1  | donor     |   2955 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Don              |            37015889 |                 2956 |   37018850 | 0.02399585516738 | No                |    37026980 |   0.028353019 | No                 |   0.028353019047 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354627:g.37018844:T>A  | NTR            | 00 % [00 % - 00.92 %]       |     0.000 | +      |  37018844 | substitution | T>A      | Intron 9  |    11091 | NM_001354627 | MLH1  | donor     |   2955 | DeepIntron |          0 | Outside SPiCE Interpretation |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Don              |            37015889 |                 2956 |   37018850 | 0.02399585516738 | No                |    37026980 |   0.028353019 | No                 |   0.028353019047 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354628:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001354628 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354629:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 8  |     2961 | NM_001354629 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
| chr3  |  37018844 | lol | T   | A   | .    | .      | .    | NM_001354630:g.37018844:T>A  | Alter by SPiCE | 98.41 % [91.47 % - 99.96 %] |     0.988 | +      |  37018844 | substitution | T>A      | Intron 9  |     2961 | NM_001354630 | MLH1  | acceptor  |     -7 | IntronCons |          1 | high                         |        0 |  0 | No          |            10 |    37018845 | Acc            | 0.01556491731833 | No                 | Acc              |            37018850 |                   -6 |          0 | 0.00000000000000 | No                |    37018850 |   0.023995855 | No                 |   0.000022606476 | No                    |
|       |           |     |     |     |      |        |      |                              |                |                             |           |        |           |              |          |           |          |              |       |           |        |            |            |                              |          |    |             |               |             |                |                  |                    |                  |                     |                      |            |                  |                   |             |               |                    |                  |                       |
**** DONE 1 variant avec score = 38% (mobidetails)
CLOSED: [2023-08-10 Thu 23:16]
chr10:g.89010760A>G
Avec
#+begin_src sh
varID
NM_000043.4:c.513A>G
#+end_src
On a bien 35.81%
| varID                | Interpretation | InterConfident             | SPiPscore | strand |   gNomen | varType      | ntChange | ExonInfo | exonSize | transcript | gene | NearestSS | DistSS | RegType | SPiCEproba | SPiCEinter_2thr              | deltaMES | BP | mutInPBarea | deltaESRscore | posCryptMut |  sstypeCryptMut | probaCryptMut | classProbaCryptMut | nearestSStoCrypt | nearestPosSStoCrypt | nearestDistSStoCrypt |      posCryptWT | probaCryptWT | classProbaCryptWT | posSSPhysio | probaSSPhysio | classProbaSSPhysio | probaSSPhysioMut | classProbaSSPhysioMut |
| NM_000043.4:c.513A>G | Alter ESR      | 35.81 % [28.11 % - 44.1 %] |     0.288 | +      | 89010760 | substitution | A>G      | Exon 6   |       63 | NM_000043  | FAS  | acceptor  |      8 | ExonESR |          0 | Outside SPiCE Interpretation |       00 | No |    -1.67753 |      89010759 | Acc         | 0.0000003317384 | No            | Acc                |         89010752 |                   7 |             89010759 | 0.0000002205815 | No           |          89010752 |  0.02545572 | No            |         0.02545572 | No               |                       |

Avec les coordonnées génomiques en hg38, on ne retrouve pas le transcrit (non visible sur UCSCS...) parmi les nombreux transcrit.

Mais en T2T oui !
##fileformat=VCFv4.0
##assembly=CHM18v2.0/hs1
##ALT=<ID=*,Description="Represents allele(s) other than observed.">
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO
chr10	89894645	lol	A	G	.	.	.

❯ ./result/bin/spip -I test-spip-T2T.vcf -O test-spip-T2T.out -g hs1 --refseq dataRefSeqhs1.RData --transcriptome transcriptome_hs1.RData

| CHROM |      POS | ID  | REF | ALT | QUAL | FILTER | INFO | varID                       | Interpretation | InterConfident             | SPiPscore | strand |   gNomen | varType      | ntChange | ExonInfo | exonSize | transcript   | gene | NearestSS | DistSSRegType | SPiCEproba | SPiCEinter_2thr | deltaMES                     | BP | mutInPBarea | deltaESRscore | posCryptMut | sstypeCryptMut | probaCryptMut | classProbaCryptMut | nearestSStoCrypt | nearestPosSStoCrypt | nearestDistSStoCrypt | posCryptWT | probaCryptWT | classProbaCryptWT | posSSPhysio | probaSSPhysio | classProbaSSPhysio | probaSSPhysioMut | classProbaSSPhysioMut |     |
| chr10 | 89894645 | lol | A   | G   | .    | .      | .    | NM_000043:g.89894645:A>G    | Alter ESR      | 35.81 % [28.11 % - 44.1 %] |     0.288 | +      | 89894645 | substitution | A>G      | Exon 6   |       63 | NM_000043    | FAS  | acceptor  |             8 | ExonESR    |               0 | Outside SPiCE Interpretation |  0 |           0 | No            |    -1.67753 |       89894644 | Acc           |    0.0000003317384 | No               | Acc                 |             89894637 |          7 |     89894644 |   0.0000002205815 | No          |      89894637 |         0.02545572 | No               |            0.02545572 | No  |
| chr10 | 89894645 | lol | A   | G   | .    | .      | .    | NM_001320619:g.89894645:A>G | Alter ESR      | 35.81 % [28.11 % - 44.1 %] |     0.288 | +      | 89894645 | substitution | A>G      | Exon 6   |       63 | NM_001320619 | FAS  | acceptor  |             8 | ExonESR    |               0 | Outside SPiCE Interpretation |  0 |           0 | No            |    -1.67753 |       89894644 | Acc           |    0.0000003317384 | No               | Acc                 |             89894637 |          7 |     89894644 |   0.0000002205815 | No          |      89894637 |         0.02545572 | No               |            0.02545572 | No  |
| chr10 | 89894645 | lol | A   | G   | .    | .      | .    | NM_152871:g.89894645:A>G    | NTR            | 00 % [00 % - 00.92 %]      |     0.000 | +      | 89894645 | substitution | A>G      | Intron 5 |     1398 | NM_152871    | FAS  | donor     |           160 | DeepIntron |               0 | Outside SPiCE Interpretation |  0 |           0 | No            |    10.00000 |       89894644 | Don           |    0.0001360257829 | No               | Don                 |             89894485 |        159 |            0 |   0.0000000000000 | No          |      89894485 |         0.07177992 | Yes              |            0.07177992 | Yes |
| chr10 | 89894645 | lol | A   | G   | .    | .      | .    | NM_152872:g.89894645:A>G    | Alter ESR      | 35.81 % [28.11 % - 44.1 %] |     0.288 | +      | 89894645 | substitution | A>G      | Exon 6   |       63 | NM_152872    | FAS  | acceptor  |             8 | ExonESR    |               0 | Outside SPiCE Interpretation |  0 |           0 | No            |    -1.67753 |       89894644 | Acc           |    0.0000003317384 | No               | Acc                 |             89894637 |          7 |     89894644 |   0.0000002205815 | No          |      89894637 |         0.02545572 | No               |            0.02545572 | No  |
| chr10 | 89894645 | lol | A   | G   | .    | .      | .    | NR_028033:g.89894645:A>G    | NTR            | 00 % [00 % - 00.92 %]      |     0.000 | +      | 89894645 | substitution | A>G      | Intron 4 |     1398 | NR_028033    | FAS  | donor     |           160 | DeepIntron |               0 | Outside SPiCE Interpretation |  0 |           0 | No            |    10.00000 |       89894644 | Don           |    0.0001360257829 | No               | Don                 |             89894485 |        159 |            0 |   0.0000000000000 | No          |      89894485 |         0.07177992 | Yes              |            0.07177992 | Yes |
|chr10129957338-T-C chr10 | 89894645 | lol | A   | G   | .    | .      | .    | NR_028034:g.89894645:A>G    | NTR            | 00 % [00 % - 00.92 %]      |     0.000 | +      | 89894645 | substitution | A>G      | Intron 3 |     1398 | NR_028034    | FAS  | donor     |           160 | DeepIntron |               0 | Outside SPiCE Interpretation |  0 |           0 | No            |    10.00000 |       89894644 | Don           |    0.0001360257829 | No               | Don                 |             89894485 |        159 |            0 |   0.0000000000000 | No          |      89894485 |         0.07177992 | Yes              |            0.07177992 | Yes |
| chr10 | 89894645 | lol | A   | G   | .    | .      | .    | NR_028035:g.89894645:A>G    | Alter ESR      | 35.81 % [28.11 % - 44.1 %] |     0.288 | +      | 89894645 | substitution | A>G      | Exon 4   |       63 | NR_028035    | FAS  | acceptor  |             8 | ExonESR    |               0 | Outside SPiCE Interpretation |  0 |           0 | No            |    -1.67753 |       89894644 | Acc           |    0.0000003317384 | No               | Acc                 |             89894637 |          7 |     89894644 |   0.0000002205815 | No          |      89894637 |         0.02545572 | No               |            0.02545572 | No  |
| chr10 | 89894645 | lol | A   | G   | .    | .      | .    | NR_028036:g.89894645:A>G    | Alter ESR      | 35.81 % [28.11 % - 44.1 %] |     0.288 | +      | 89894645 | substitution | A>G      | Exon 5   |       63 | NR_028036    | FAS  | acceptor  |             8 | ExonESR    |               0 | Outside SPiCE Interpretation |  0 |           0 | No            |    -1.67753 |       89894644 | Acc           |    0.0000003317384 | No               | Acc                 |             89894637 |          7 |     89894644 |   0.0000002205815 | No          |      89894637 |         0.02545572 | No               |            0.02545572 | No  |
| chr10 | 89894645 | lol | A   | G   | .    | .      | .    | NR_135313:g.89894645:A>G    | Alter ESR      | 35.81 % [28.11 % - 44.1 %] |     0.288 | +      | 89894645 | substitution | A>G      | Exon 5   |       63 | NR_135313    | FAS  | acceptor  |             8 | ExonESR    |               0 | Outside SPiCE Interpretation |  0 |           0 | No            |    -1.67753 |       89894644 | Acc           |    0.0000003317384 | No               | Acc                 |             89894637 |          7 |     89894644 |   0.0000002205815 | No          |      89894637 |         0.02545572 | No               |            0.02545572 | No  |
| chr10 | 89894645 | lol | A   | G   | .    | .      | .    | NM_001410956:g.89894645:A>G | Alter ESR      | 35.81 % [28.11 % - 44.1 %] |     0.288 | +      | 89894645 | substitution | A>G      | Exon 6   |       63 | NM_001410956 | FAS  | acceptor  |             8 | ExonESR    |               0 | Outside SPiCE Interpretation |  0 |           0 | No            |    -1.67753 |       89894644 | Acc           |    0.0000003317384 | No               | Acc                 |             89894637 |          7 |     89894644 |   0.0000002205815 | No          |      89894637 |         0.02545572 | No               |            0.02545572 | No  |
| chr10 | 89894645 | lol | A   | G   | .    | .      | .    | NR_135314:g.89894645:A>G    | Alter ESR      | 35.81 % [28.11 % - 44.1 %] |     0.288 | +      | 89894645 | substitution | A>G      | Exon 6   |       63 | NR_135314    | FAS  | acceptor  |             8 | ExonESR    |               0 | Outside SPiCE Interpretation |  0 |           0 | No            |    -1.67753 |       89894644 | Acc           |    0.0000003317384 | No               | Acc                 |             89894637 |          7 |     89894644 |   0.0000002205815 | No          |      89894637 |         0.02545572 | No               |            0.02545572 | No  |
| chr10 | 89894645 | lol | A   | G   | .    | .      | .    | NR_135315:g.89894645:A>G    | Alter ESR      | 35.81 % [28.11 % - 44.1 %] |     0.288 | +      | 89894645 | substitution | A>G      | Exon 4   |       63 | NR_135315    | FAS  | acceptor  |             8 | ExonESR    |               0 | Outside SPiCE Interpretation |  0 |           0 | No            |    -1.67753 |       89894644 | Acc           |    0.0000003317384 | No               | Acc                 |             89894637 |          7 |     89894644 |   0.0000002205815 | No          |      89894637 |         0.02545572 | No               |            0.02545572 | No  |
|       |          |     |     |     |      |        |      |                             |                |                            |           |        |          |              |          |          |          |              |      |           |               |            |                 |                              |    |             |               |             |                |               |                    |                  |                     |                      |            |              |                   |             |               |                    |                  |                       |     |
**** DONE Vérifier multiples transcripts en hg38 avec coordonées génomiquues: ok
CLOSED: [2023-08-10 Thu 23:00]
Beaucoup plus de transcrits en T2T
Ex: 1 transcrit refseq curated
http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr11%3A108257446%2D108257496&hgsid=1672963428_J5aWAqack2FpJ7mvhFTNVw7bKzxo
vs 2 transcrits en T2T
http://genome.ucsc.edu/cgi-bin/hgTracks?db=hub_3671779_hs1&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr11%3A108264969%2D108265019&hgsid=1672963612_Eso9frdQ7z6RkKkcKsIf2Waq3pec
C'est bien ce qu'on retrouve avec spip
*** DONE [#A] Filtre vep avec spip
CLOSED: [2023-08-13 Sun 00:39] SCHEDULED: <2023-08-12 Sat 19:00>
*** DONE Annotation CADD + spliceAI GRCh38 avec nouvelle version :annotation:
CLOSED: [2023-08-28 Mon 17:21] SCHEDULED: <2023-08-20 Sun>
*** DONE OMIM: possible seulement sur nom du gènes:annotation:
CLOSED: [2023-08-13 Sun 11:57] SCHEDULED: <2023-08-13 Sun 16:00>
Base de données non disponible et compliqué de faire la mise à jour nous.

Si on essaie de prendre les gènes de GRCH38, ils ne sont pas forcément en T2T
Ex: DDX11L17 n'existe pas dans T2T à ces coordonées
zgrep DDX11L17 GCF_009914755.1_T2T-CHM13v2.0_genomic.gff.gz
Note: c'est un pseudogene
https://www.genecards.org/cgi-bin/carddisp.pl?gene=DDX11L17

Si on prend les gènes de T2T, il y en a des nouveaux.
Ex: le premier est LOC101928626.
À cette position, rien en GRCh38

Si on essaye avec ENSEMBL: non car n'ont pas le même identifiant
Ex: ACHE
Idéalement, il faudrait l'identifiant NCBI (disponible dans OMIM) mais n'est pas en sortie de VEP
Et cela demande la version "merged" donc impossible en T2T

Est-ce faisable de faire une chr10129957338-T-Ccorrespondance sur le nom du gène ?
Tous les gènes de T2T:
#+begin_src sh :dir ~/Downloads
 zgrep -o "ID=gene[^;]*;"  GCF_009914755.1_T2T-CHM13v2.0_genomic.gff.gz | sed 's/ID=gene-//;s/;//' | sort | uniq > t2t-genes.txt
 wc -l t2t-genes.txt
#+end_src

#+RESULTS:
: 57660 t2t-genes.txt

#+begin_src sh :dir ~/Downloads
zgrep -o "ID=gene[^;]*;"  GCF_000001405.40_GRCh38.p14_genomic.gff.gz | sed 's/ID=gene-//;s/;//' | sort | uniq > hg38-genes.txt
wc -l hg38-genes.txt
#+end_src

#+RESULTS:
: 67127 hg38-genes.txt

Gènes communs aux 2
#+begin_src sh :dir ~/Downloads
comm -12 t2t-genes.txt hg38-genes.txt | wc -l
#+end_src

#+RESULTS:
: 54506

Gènes uniquements dans t2t
#+begin_src sh :dir ~/Downloads
comm -23 t2t-genes.txt hg38-genes.txt | wc -l
#+end_src

#+RESULTS:
: 3154

Gènes uniquements dans GRCh38
#+begin_src sh :dir ~/Downloads
comm -13 t2t-genes.txt hg38-genes.txt | wc -l
#+end_src

#+RESULTS:
: 12621
*** HOLD OMIM sur nom du gène :annotation:
*** DONE Mobidetails API
CLOSED: [2023-09-10 Sun 16:44]
Trop long ... 1h à 1h30 d'exécution
Disponible dans module
*** DONE Filtre vep avec spip for T2T et spliceAI pour GRCh38
CLOSED: [2023-09-16 Sat 22:47]
*** DONE Repasser tests en GRCh38 avec nouveau filtre (spip ou splice ai) :sanger:
CLOSED: [2023-09-17 Sun 09:07] SCHEDULED: <2023-09-16 Sat>
*** HOLD Franklin API
https://www.postman.com/genoox-ps/workspace/franklin-api-documentation-s-public-workspace/documentation/6621518-4335389d-12e3-445f-8182-339df95b2a09
*** KILL Regarder si clinique disponible avec vep :annotation:
CLOSED: [2023-09-10 Sun 16:44]
*** TODO Filtrer suffisament après annotation
**** DONE Tester filtre sans splice: 6130 mais il en manque 4
CLOSED: [2023-10-18 Wed 22:50] SCHEDULED: <2023-09-27 Wed>
Mail Paul: Exome donc hors splice, peu intéressant
***** DONE Enlever complètement condition splice: 6130 variants restants...
CLOSED: [2023-09-27 Wed 19:37] SCHEDULED: <2023-09-26 Tue>
Cf [[id:c9b2009a-503b-4561-94c6-29ae21a3188d][Filtre vep avec spliceAI: 37365 -> 6130]]

Dans tests/splicai
#+begin_src sh
 filter_vep -i output-all-gpu.vcf --format vcf --filter "        not(Consequence matches non_coding_transcript or Consequence matches stream                 or Consequence matches intergenic_variant                 or Consequence matches UTR                 or Consequence matches intron_variant                 or Consequence matches synonymous                 or BIOTYPE  matches pseudogene                 or BIOTYPE  matches misc_RNA)"                --only_matched         -o test.vcf

 grep -c -v '^#' test.vcf
6130
#+end_src
***** DONE Remplacer par impact fonctionnel: peu d'impact : majorité = MODERATE
CLOSED: [2023-09-27 Wed 19:45] SCHEDULED: <2023-09-26 Tue>

 filter_vep -i output-all-gpu-filtered.vcf --format vcf --filter "IMPACT is HIGH"  --only_matched | grep -c -v '^#'
258

filter_vep -i output-all-gpu-filtered.vcf --format vcf --filter "IMPACT is LOW"  --only_matched | grep -c -v '^#'
11

filter_vep -i output-all-gpu-filtered.vcf --format vcf --filter "IMPACT is MODERATE"  --only_matched | grep -c -v '^#'
5824
***** DONE Regarder les conséquences pour tes les transcripts
CLOSED: [2023-09-27 Wed 21:04]
/Work/Users/apraga/bisonex/out/annotate/vep/NA12878-sanger-all-T2T

 filter_vep -i NA12878-sanger-all-T2T.vep.vcf.gz --format vcf --filter "        not(Consequence matches non_coding_transcript or Consequence matches stream                 or Consequence matches intergenic_variant                 or Consequence matches UTR                 or Consequence matches intron_variant                 or Consequence matches synonymous                 or BIOTYPE  matches pseudogene                 or BIOTYPE  matches misc_RNA)"                --only_matched         -o filtered.vcf

 bcftools +split-vep filtered.vcf -f '%Consequence\n' -d | sort | uniq -c
     94 coding_sequence_variant
     13 coding_sequence_variant&NMD_transcript_variant
    257 frameshift_variant
     21 frameshift_variant&NMD_transcript_variant
      2 frameshift_variant&splice_donor_region_variant
     20 frameshift_variant&splice_region_variant
      1 frameshift_variant&splice_region_variant&NMD_transcript_variant
      1 incomplete_terminal_codon_variant&coding_sequence_variant
    211 inframe_deletion
     18 inframe_deletion&NMD_transcript_variant
      6 inframe_deletion&splice_region_variant
    242 inframe_insertion
     22 inframe_insertion&NMD_transcript_variant
      4 inframe_insertion&splice_region_variant
  14689 missense_variant
   1416 missense_variant&NMD_transcript_variant
      6 missense_variant&splice_donor_5th_base_variant
    374 missense_variant&splice_region_variant
     34 missense_variant&splice_region_variant&NMD_transcript_variant
     53 splice_acceptor_variant
     11 splice_acceptor_variant&NMD_transcript_variant
     79 splice_donor_variant
      6 splice_donor_variant&NMD_transcript_variant
     30 start_lost
      5 start_lost&NMD_transcript_variant
    135 stop_gained
     13 stop_gained&frameshift_variant
      3 stop_gained&frameshift_variant&NMD_transcript_variant
      2 stop_gained&frameshift_variant&splice_region_variant
     14 stop_gained&NMD_transcript_variant
      5 stop_gained&splice_region_variant
      2 stop_gained&splice_region_variant&NMD_transcript_variant
      4 stop_lost
      1 stop_lost&NMD_transcript_variant
      9 stop_retained_variant
      6 stop_retained_variant&NMD_transcript_variant
      1 transcript_ablation


Idem tests/spliceai

bcftools +split-vep output-all-gpu-filtered.vcf  -f '%Consequence\n' -d | sort | uniq -c
     94 coding_sequence_variant
     13 coding_sequence_variant&NMD_transcript_variant
    257 frameshift_variant
     21 frameshift_variant&NMD_transcript_variant
      2 frameshift_variant&splice_donor_region_variant
     20 frameshift_variant&splice_region_variant
      1 frameshift_variant&splice_region_variant&NMD_transcript_variant
      1 incomplete_terminal_codon_variant&coding_sequence_variant
    211 inframe_deletion
     18 inframe_deletion&NMD_transcript_variant
      6 inframe_deletion&splice_region_variant
    242 inframe_insertion
     22 inframe_insertion&NMD_transcript_variant
      4 inframe_insertion&splice_region_variant
  14689 missense_variant
   1416 missense_variant&NMD_transcript_variant
      6 missense_variant&splice_donor_5th_base_variant
    374 missense_variant&splice_region_variant
     34 missense_variant&splice_region_variant&NMD_transcript_variant
     53 splice_acceptor_variant
     11 splice_acceptor_variant&NMD_transcript_variant
     79 splice_donor_variant
      6 splice_donor_variant&NMD_transcript_variant
     30 start_lost
      5 start_lost&NMD_transcript_variant
    135 stop_gained
     13 stop_gained&frameshift_variant
      3 stop_gained&frameshift_variant&NMD_transcript_variant
      2 stop_gained&frameshift_variant&splice_region_variant
     14 stop_gained&NMD_transcript_variant
      5 stop_gained&splice_region_variant
      2 stop_gained&splice_region_variant&NMD_transcript_variant
      4 stop_lost
      1 stop_lost&NMD_transcript_variant
      9 stop_retained_variant
      6 stop_retained_variant&NMD_transcript_variant
      1 transcript_ablation
***** DONE Regarder les conséquences pour -s worst
CLOSED: [2023-09-27 Wed 21:04]
/Work/Users/apraga/bisonex/out/annotate/vep/NA12878-sanger-all-T2T
Après filtre_vep sans splice

]$ bcftools +split-vep filtered.vcf -f '%Consequence\n' -d -s worst | sort | uniq -c
     48 coding_sequence_variant
      6 coding_sequence_variant&nmd_transcript_variant
    121 frameshift_variant
      9 frameshift_variant&nmd_transcript_variant
      1 frameshift_variant&splice_donor_region_variant
      9 frameshift_variant&splice_region_variant
     79 inframe_deletion
      3 inframe_deletion&nmd_transcript_variant
      2 inframe_deletion&splice_region_variant
     85 inframe_insertion
      2 inframe_insertion&nmd_transcript_variant
      1 inframe_insertion&splice_region_variant
   5309 missense_variant
    207 missense_variant&nmd_transcript_variant
      3 missense_variant&splice_donor_5th_base_variant
    110 missense_variant&splice_region_variant
      9 missense_variant&splice_region_variant&nmd_transcript_variant
     19 splice_acceptor_variant
      1 splice_acceptor_variant&nmd_transcript_variant
     21 splice_donor_variant
      1 splice_donor_variant&nmd_transcript_variant
     14 start_lost
     44 stop_gained
      4 stop_gained&frameshift_variant
      2 stop_gained&frameshift_variant&splice_region_variant
      3 stop_gained&nmd_transcript_variant
      3 stop_gained&splice_region_variant
      2 stop_gained&splice_region_variant&nmd_transcript_variant
      2 stop_lost
      1 stop_lost&nmd_transcript_variant
      6 stop_retained_variant
      2 stop_retained_variant&nmd_transcript_variant
      1 transcript_ablation

Dans tests/spliceai
$ bcftools +split-vep output-all-gpu-filtered.vcf  -f '%Consequence\n' -s worst -d | sort | uniq -c
     48 coding_sequence_variant
      6 coding_sequence_variant&nmd_transcript_variant
    121 frameshift_variant
      9 frameshift_variant&nmd_transcript_variant
      1 frameshift_variant&splice_donor_region_variant
      9 frameshift_variant&splice_region_variant
     79 inframe_deletion
      3 inframe_deletion&nmd_transcript_variant
      2 inframe_deletion&splice_region_variant
     85 inframe_insertion
      2 inframe_insertion&nmd_transcript_variant
      1 inframe_insertion&splice_region_variant
   5309 missense_variant
    207 missense_variant&nmd_transcript_variant
      3 missense_variant&splice_donor_5th_base_variant
    110 missense_variant&splice_region_variant
      9 missense_variant&splice_region_variant&nmd_transcript_variant
     19 splice_acceptor_variant
      1 splice_acceptor_variant&nmd_transcript_variant
     21 splice_donor_variant
      1 splice_donor_variant&nmd_transcript_variant
     14 start_lost
     44 stop_gained
      4 stop_gained&frameshift_variant
      2 stop_gained&frameshift_variant&splice_region_variant
      3 stop_gained&nmd_transcript_variant
      3 stop_gained&splice_region_variant
      2 stop_gained&splice_region_variant&nmd_transcript_variant
      2 stop_lost
      1 stop_lost&nmd_transcript_variant
      6 stop_retained_variant
      2 stop_retained_variant&nmd_transcript_variant
      1 transcript_ablation
***** KILL Vérifier si tests sanger passent: non
CLOSED: [2023-09-28 Thu 01:33] SCHEDULED: <2023-09-27 Wed>
     │ String                Float64   Int64
─────┼───────────────────────────────────────
   1 │ chr10:g.130884530      60.0     67
   2 │ chr10:g.240362         60.0     79
   3 │ chr14:g.52665581       60.0     51
   4 │ chr19:g.41325390       60.0    180
***** DONE Comparer aux filtres en GRCh38:  ce sont bien les filtres hors splice...
CLOSED: [2023-10-17 Tue 21:12]
T2T:
 filter_vep -i  2300346867_NA12878-63118093_S260-T2T/2300346867_NA12878-63118093_S260-T2T.vep.vcf.gz  --format vcf --filter "        not(Consequence matches non_coding_transcript or Consequence matches stream or Consequence matches intergenic_variant or Consequence matches UTR or Consequence matches intron_variant or Consequence matches synonymous or BIOTYPE  matches pseudogene or BIOTYPE  matches misc_RNA)"                --only_matched       | bcftools +counts

Number of samples: 1
Number of SNPs:    5362
Number of INDELs:  325
Number of MNPs:    323
Number of others:  0
Number of sites:   5991

GRCh38
filter_vep -i  2300346867_NA12878-63118093_S260-GRCh38/2300346867_NA12878-63118093_S260-GRCh38.vep.vcf.gz  --format vcf --filter "        not(Consequence matches non_coding_transcript or Consequence matches stream or Consequence matches intergenic_variant or Consequence matches UTR or Consequence matches intron_variant or Consequence matches synonymous or BIOTYPE  matches pseudogene or BIOTYPE  matches misc_RNA)"                --only_matched       | bcftools +counts

Number of samples: 1
Number of SNPs:    1182
Number of INDELs:  143
Number of MNPs:    535
Number of others:  0
Number of sites:   1840


***** DONE Proportions de conséquence : T2T vs GRCh38 avec multiqc: idem
CLOSED: [2023-10-17 Tue 21:00]
À l'oeil

***** Réexaminer les conséquences
****** DONE Impact fonctionnel: plus de LOW et de MODIFIER++
CLOSED: [2023-10-17 Tue 21:22]
T2T
bcftools +split-vep  2300346867_NA12878-63118093_S260-T2T/2300346867_NA12878-63118093_S260-T2T.filtervep.vcf -f '%IMPACT\n' -d | sort | uniq -c
    596 HIGH
   2828 LOW
  16314 MODERATE
  11261 MODIFIER

  GRCh38
 bcftools +split-vep  2300346867_NA12878-63118093_S260-GRCh38/2300346867_NA12878-63118093_S260-GRCh38.filtervep.vcf -f '%IMPACT\n' -d | sort | uniq -c
    414 HIGH
    466 LOW
  10054 MODERATE
    550 MODIFIER
****** DONE Pire conséquence: trop de missense
CLOSED: [2023-10-17 Tue 21:23]
GRCh38
$  bcftools +split-vep  2300346867_NA12878-63118093_S260-GRCh38/2300346867_NA12878-63118093_S260-GRCh38.filtervep.vcf -f '%Consequence\n' -d -s worst | sort | uniq -c
      2 3_prime_utr_variant&nmd_transcript_variant
      1 5_prime_utr_variant
      2 coding_sequence_variant
     47 frameshift_variant
      6 frameshift_variant&nmd_transcript_variant
      1 frameshift_variant&splice_donor_region_variant
      1 frameshift_variant&splice_region_variant
      1 frameshift_variant&start_lost&start_retained_variant
     37 inframe_deletion
      9 inframe_deletion&nmd_transcript_variant
     27 inframe_insertion
      5 inframe_insertion&nmd_transcript_variant
     21 intron_variant
   1593 missense_variant
     37 missense_variant&nmd_transcript_variant
     17 missense_variant&splice_region_variant
      1 missense_variant&splice_region_variant&nmd_transcript_variant
      1 protein_altering_variant
      1 splice_acceptor_variant
      1 splice_acceptor_variant&frameshift_variant
      2 splice_acceptor_variant&nmd_transcript_variant
      3 splice_donor_5th_base_variant&intron_variant
      1 splice_donor_5th_base_variant&intron_variant&non_coding_transcript_variant
      2 splice_donor_region_variant&intron_variant
      1 splice_donor_region_variant&intron_variant&nmd_transcript_variant
      1 splice_donor_region_variant&intron_variant&non_coding_transcript_variant
     10 splice_donor_variant
      1 splice_donor_variant&non_coding_transcript_variant
     11 splice_polypyrimidine_tract_variant&intron_variant
      1 splice_polypyrimidine_tract_variant&intron_variant&non_coding_transcript_variant
      1 splice_region_variant&intron_variant
      9 splice_region_variant&splice_polypyrimidine_tract_variant&intron_variant
      3 splice_region_variant&synonymous_variant
      1 splice_region_variant&synonymous_variant&nmd_transcript_variant
      4 start_lost
     19 stop_gained
      2 stop_gained&frameshift_variant
      2 stop_gained&nmd_transcript_variant
      1 stop_gained&splice_region_variant
      1 stop_gained&splice_region_variant&nmd_transcript_variant
      3 stop_lost
      2 stop_lost&nmd_transcript_variant
      1 stop_retained_variant
     18 synonymous_variant
      1 synonymous_variant&nmd_transcript_variant
      1 transcript_ablation

      T2T
      [apraga@mesointeractive filter]$  bcftools +split-vep  2300346867_NA12878-63118093_S260-T2T/2300346867_NA12878-63118093_S260-T2T.filtervep.vcf -f '%Consequence\n' -d -s worst | sort | uniq -c
     15 3_prime_utr_variant
     11 3_prime_utr_variant&nmd_transcript_variant
     51 5_prime_utr_variant
      3 5_prime_utr_variant&nmd_transcript_variant
     48 coding_sequence_variant
      5 coding_sequence_variant&nmd_transcript_variant
      3 downstream_gene_variant
    121 frameshift_variant
      9 frameshift_variant&nmd_transcript_variant
      1 frameshift_variant&splice_donor_region_variant
      9 frameshift_variant&splice_region_variant
     78 inframe_deletion
      2 inframe_deletion&nmd_transcript_variant
      2 inframe_deletion&splice_region_variant
     84 inframe_insertion
      2 inframe_insertion&nmd_transcript_variant
      1 inframe_insertion&splice_region_variant
     16 intergenic_variant
    368 intron_variant
     21 intron_variant&nmd_transcript_variant
     71 intron_variant&non_coding_transcript_variant
   5187 missense_variant
    207 missense_variant&nmd_transcript_variant
      3 missense_variant&splice_donor_5th_base_variant
    105 missense_variant&splice_region_variant
      9 missense_variant&splice_region_variant&nmd_transcript_variant
     33 non_coding_transcript_exon_variant
     12 splice_acceptor_variant
      1 splice_acceptor_variant&5_prime_utr_variant&intron_variant&nmd_transcript_variant
      1 splice_acceptor_variant&nmd_transcript_variant
      3 splice_acceptor_variant&non_coding_transcript_variant
      1 splice_acceptor_variant&splice_polypyrimidine_tract_variant&intron_variant&nmd_transcript_variant
     16 splice_donor_5th_base_variant&intron_variant
      2 splice_donor_5th_base_variant&intron_variant&non_coding_transcript_variant
     33 splice_donor_region_variant&intron_variant
      4 splice_donor_region_variant&intron_variant&nmd_transcript_variant
      7 splice_donor_region_variant&intron_variant&non_coding_transcript_variant
     19 splice_donor_variant
      1 splice_donor_variant&nmd_transcript_variant
      2 splice_donor_variant&non_coding_transcript_variant
      3 splice_donor_variant&splice_donor_5th_base_variant&coding_sequence_variant&intron_variant
     64 splice_polypyrimidine_tract_variant&intron_variant
      6 splice_polypyrimidine_tract_variant&intron_variant&nmd_transcript_variant
      8 splice_polypyrimidine_tract_variant&intron_variant&non_coding_transcript_variant
      2 splice_region_variant&3_prime_utr_variant
      2 splice_region_variant&5_prime_utr_variant
      4 splice_region_variant&intron_variant
      6 splice_region_variant&non_coding_transcript_exon_variant
     54 splice_region_variant&splice_polypyrimidine_tract_variant&intron_variant
      4 splice_region_variant&splice_polypyrimidine_tract_variant&intron_variant&nmd_transcript_variant
      5 splice_region_variant&splice_polypyrimidine_tract_variant&intron_variant&non_coding_transcript_variant
     27 splice_region_variant&synonymous_variant
     13 start_lost
     31 stop_gained
      4 stop_gained&frameshift_variant
      2 stop_gained&frameshift_variant&splice_region_variant
      3 stop_gained&nmd_transcript_variant
      2 stop_gained&splice_region_variant
      2 stop_gained&splice_region_variant&nmd_transcript_variant
      2 stop_lost
      1 stop_lost&nmd_transcript_variant
      6 stop_retained_variant
      2 stop_retained_variant&nmd_transcript_variant
    349 synonymous_variant
     17 synonymous_variant&nmd_transcript_variant
      1 transcript_ablation
      2 upstream_gene_variant

**** DONE Regarder annotation VEP des variants sur NA12878 non trataié :na12878:
CLOSED: [2023-10-18 Wed 22:50] SCHEDULED: <2023-10-16 Mon>
/Entered on/ [2023-10-16 Mon 19:39]
**** DONE Regarder si les variants sont dans des zones modifiées de T2T
CLOSED: [2023-10-19 Thu 17:19] SCHEDULED: <2023-10-18 Wed>
/Entered on/ [2023-10-18 Wed 22:49]
Liftover des variants de GRCh38 -> T2T

Cf ~/roam/research/bisonex/code/t2t/comparePositions.jl

#+begin_quote
Successfully converted 1896 records: View Conversions

Conversion failed on 17 records.
 #+end_quote

On utilise t2tOnly()

Proportion par chromosome
julia> @by d :Column1 $nrow
24×2 DataFrame
 Row │ Column1  nrow
     │ String7  Int64
─────┼────────────────
   1 │ chr1       678
   2 │ chr2       369
   3 │ chr3       287
   4 │ chr4       224
   5 │ chr5       258
   6 │ chr6       430
   7 │ chr7       321
   8 │ chr8       218
   9 │ chr9       251
  10 │ chr10      275
  11 │ chr11      489
  12 │ chr12      350
  13 │ chr13       74
  14 │ chr14      185
  15 │ chr15      171
  16 │ chr16      283
  17 │ chr17      364
  18 │ chr18       82
  19 │ chr19      550
  20 │ chr20      142
  21 │ chr21       93
  22 │ chr22      171
  23 │ chrX        98
  24 │ chrY         1

**** DONE Regarder si les variants sont sur des nouveaux gènes
CLOSED: [2023-10-19 Thu 20:18] SCHEDULED: <2023-10-19 Thu>

Cf ~/roam/reasearch/bisonex/code/t2t/compareGene.jl

Test de la fonction pour extraire les gènes:

Important: les chiffres ne sont pas données sur le nom du gène seul mais sur le nom + la position. On ne supprime donc pas les noms en double"
- T2T
   - refseq + liftoff, 20 008, The official webpage says 20 006 for v4 https://ccb.jhu.edu/T2T.shtml.
   - Gencode38 19490
- GRCh38
  - refseq we have 23 314, so 4 less than the official website : https://www.ncbi.nlm.nih.gov/datasets/gene/GCF_000001405.40/?gene_type=protein-coding

On choisit refseq:: Il y a 170 gènes (unique sur le nom du gène !) dans T2T non présents dans GRCh38-p14
À noter qu'il y a plus de gènes dans Refseq en GRC38-p14 qu'en T2T (20 080 > 19 748 )

Avec compareGene.j, on retrouve 0.73%
**** DONE Annotation avec vep + GTF (dernière versio): n'aide pas
CLOSED: [2023-10-20 Fri 00:16] SCHEDULED: <2023-10-19 Thu>
https://www.science.org/doi/10.1126/science.abj6987#core-R61
/Entered on/ [2023-10-19 Thu 10:41]

#+begin_src sh :dir "meso://Work/Users/apraga/bisonex/test/t2t"
wget https://s3-us-west-2.amazonaws.com/human-pangenomics/T2T/CHM13/assemblies/annotation/chm13v2.0_RefSeq_Liftoff_v5.1.gff3.gz
zgrep -v "#" chm13v2.0_RefSeq_Liftoff_v5.1.gff3.gz | sort -k1,1 -k4,4n -k5,5n -t$'\t' | bgzip -c > chm13v2.0_RefSeq_prepared.gff3.gz
tabix -p gff chm13v2.0_RefSeq_prepared.gff3.gz
#+end_src
# ./vep -i input.vcf --gff data.gff.gz --fasta genome.fa.gz

Dans ce dossier, on annote
#+begin_src sh
vep -i ../../out/annotate/vep/2300346867_NA12878-63118093_S260-T2T/2300346867_NA12878-63118093_S260-T2T.vep.vcf.gz --gff chm13v2.0_RefSeq_prepared.gff3.gz --fasta /Work/Groups/bisonex/data/fasta/chm13v2.0/chm13v2.0.fa --vcf -o na12878_annotated.vcf

#+end_src
Et on filtre
#+begin_src sh
 filter_vep -i  na12878_annotated.vcf  --format vcf --filter "        not(Consequence matches non_coding_transcript or Consequence matches stream or Consequence matches intergenic_variant or Consequence matches UTR or Consequence matches intron_variant or Consequence matches synonymous or BIOTYPE  matches pseudogene or BIOTYPE  matches misc_RNA)"                --only_matched  > na12878_annotated_filtered.vcf

#+end_src
Idem

grep -c -v '#' na12878_annotated_filtered.vcf
6368

**** DONE Figure propre pour position des variants
CLOSED: [2023-10-19 Thu 15:41] SCHEDULED: <2023-10-19 Thu>
**** DONE Nombre de variants dans les zones exclusives à T2T
CLOSED: [2023-10-19 Thu 16:39] SCHEDULED: <2023-10-19 Thu>
Zones unique à T2T données par : https://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=1735553302_OcJ6esPoUFcSykF6hKiRmIGU24KD&db=hub_3267197_GCA_009914755.4&c=CP068269.2&g=hub_3267197_hgUnique

Note: le .fai donné ( https://s3-us-west-2.amazonaws.com/human-pangenomics/T2T/CHM13/assemblies/analysis_set/chm13v2.0.fa.gz.gzi ) cause un problème aec bcftools :
Chromosome "" defined twice in chm13v2.0.fa.gz.gzi

On utilise donc l'index regénér sur le mésocentre
#+begin_src sh :dir "/home/alex/roam/research/bisonex/code/t2t"
wget https://s3-us-west-2.amazonaws.com/human-pangenomics/T2T/CHM13/assemblies/chain/v1_nflo/grch38-chm13v2.paf
scp meso:/Work/Projects/bisonex/data/fasta/chm13v2.0/chm13v2.0.fa.fai .

cut -f 1,3,4 grch38-chm13v2.paf | bedtools sort -i - -g chm13v2.0.fa.fai | bedtools merge | bedtools complement -g chm13v2.0.fa.fai -i - | bedtools merge | save T2T-CHM13v2.0_unique_regions_hg38.bed -f
#+end_src

#+RESULTS:

On génère le BED des variants supplémentaires en T2T avec

Puis
#+begin_src sh :dir "/home/alex/roam/research/bisonex/code/t2t"
bedtools intersect -a na12878-t2t-only.bed -b T2T-CHM13v2.0_unique_regions_hg38.bed > na12878-t2t-only-unique.bed
wc -l na12878-t2t-only.bed
wc -l na12878-t2t-only-unique.bed
#+end_src

#+RESULTS:
| 6364 | na12878-t2t-only.bed        |
|   47 | na12878-t2t-only-unique.bed |

Donc 0.73% sont dans des zones unique
**** KILL Comparer l'annotation sur 1 variant filtré en GRCh388 et non filtré en T2T
CLOSED: [2023-10-19 Thu 22:54] SCHEDULED: <2023-10-19 Thu>
**** KILL Snpeff
SCHEDULED: <2023-10-19 Thu>
CLOSED: [2023-10-19 Thu 10:42]
Base de données non disponible
**** DONE Garder les transcrits canonique: reste nombreux
CLOSED: [2023-10-20 Fri 23:46] SCHEDULED: <2023-10-19 Thu>
/Entered on/ [2023-10-19 Thu 11:23]

***** Test GRCH38
Dans /Work/Users/apraga/bisonex/test/t2t

On vérifie qu'afficher CANONICAL ne montre que les transcrits canoniques

#+begin_src sh
bcftools +split-vep  /Work/Users/apraga/bisonex/out/annotate/vep/2300346867_NA12878-63118093_S260-GRCh38/2300346867_NA12878-63118093_S260-GRCh38.vep.vcf.gz -f '%CHROM:%POS %CANONICAL\n' -d | wc -l
20366

bcftools +split-vep  /Work/Users/apraga/bisonex/out/annotate/vep/2300346867_NA12878-63118093_S260-GRCh38/2300346867_NA12878-63118093_S260-GRCh38.vep.vcf.gz -f '%CHROM:%POS\n' -d | wc -l
90865
#+end_src

On peut aussi filtrer les transcrits canoniques seuls
#+begin_src sh
bcftools +split-vep  /Work/Users/apraga/bisonex/out/annotate/vep/2300346867_NA12878-63118093_S260-GRCh38/2300346867_NA12878-63118093_S260-GRCh38.vep.vcf.gz -f '%CHROM:%POS %Consequence\n' -i 'CANONICAL="YES"' -d | wc -l
20366
#+end_src
****** Après VEP: canonique et missense ou pire: 1528 transcrits
#+begin_src sh
bcftools +split-vep  /Work/Users/apraga/bisonex/out/annotate/vep/2300346867_NA12878-63118093_S260-GRCh38/2300346867_NA12878-63118093_S260-GRCh38.vep.vcf.gz -f '%CHROM:%POS %Consequence\n' -i 'CANONICAL="YES"' -d -s worst:missense+ | wc -l
1528
****** Après filtre vep: canonique et missense ou pire: 1255 transcrits
#+begin_src sh
bcftools +split-vep  /Work/Users/apraga/bisonex/out/annotate/filter/2300346867_NA12878-63118093_S260-GRCh38/2300346867_NA12878-63118093_S260-GRCh38.filtervep.vcf -f '%CHROM:%POS %Consequence\n' -i 'CANONICAL="YES"'  -d | wc -l
3519

bcftools +split-vep  /Work/Users/apraga/bisonex/out/annotate/filter/2300346867_NA12878-63118093_S260-GRCh38/2300346867_NA12878-63118093_S260-GRCh38.filtervep.vcf -f '%CHROM:%POS %Consequence\n' -i 'CANONICAL="YES"'  -d  -s worst:missense+| wc -l
1255
#+end_src
****** Notre filtre vep + canonique: 2216 variants
#+begin_src
filter_vep -i  /Work/Users/apraga/bisonex/out/annotate/vep/2300346867_NA12878-63118093_S260-GRCh38/2300346867_NA12878-63118093_S260-GRCh38.vep.vcf.gz   --format vcf --filter "        not(Consequence matches non_coding_transcript or Consequence matches stream or Consequence matches intergenic_variant or Consequence matches UTR or Consequence matches intron_variant or Consequence matches synonymous or BIOTYPE  matches pseudogene or BIOTYPE  matches misc_RNA)" --filter "CANONICAL is YES"                --only_matched  | wc -l
2216
#+end_src
***** Test T2T
50 000 vs 30 000
 #+begin_src sh
 bcftools +split-vep  /Work/Users/apraga/bisonex/out/annotate/vep/2300346867_NA12878-63118093_S260-T2T/2300346867_NA12878-63118093_S260-T2T.vep.vcf.gz -f '%CHROM:%POS %Consequence\n' -i 'CANONICAL="YES"' -d | wc -l
50142
 #+end_src
****** Apres VEP: canonique et missense ou pire: 4615 transcrits
 #+begin_src sh
 bcftools +split-vep  /Work/Users/apraga/bisonex/out/annotate/vep/2300346867_NA12878-63118093_S260-T2T/2300346867_NA12878-63118093_S260-T2T.vep.vcf.gz -f '%CHROM:%POS %Consequence\n' -i 'CANONICAL="YES"' -d -s worst:missense+ | wc -l
 #+end_src
****** Après filtre vep: canonique et missense ou pire : 3734 transcrits (et 3734 variants)
 #+begin_src sh
 bcftools +split-vep  /Work/Users/apraga/bisonex/out/annotate/filter/2300346867_NA12878-63118093_S260-T2T/2300346867_NA12878-63118093_S260-T2T.filtervep.vcf -f '%CHROM:%POS %Consequence\n' -i 'CANONICAL="YES"' -d -s worst:missense+ | wc -l
 #+end_src
****** Notre filtre vep + canonique: 5539 variants
#+begin_src sh
filter_vep -i  /Work/Users/apraga/bisonex/out/annotate/vep/2300346867_NA12878-63118093_S260-T2T/2300346867_NA12878-63118093_S260-T2T.vep.vcf.gz   --format vcf --filter "        not(Consequence matches non_coding_transcript or Consequence matches stream or Consequence matches intergenic_variant or Consequence matches UTR or Consequence matches intron_variant or Consequence matches synonymous or BIOTYPE  matches pseudogene or BIOTYPE  matches misc_RNA)" --filter "CANONICAL is YES"                --only_matched  | wc -l
5539
#+end_src
**** KILL Vérifier fréquence gnomad: non disponible dans cache
CLOSED: [2023-10-20 Fri 23:24] SCHEDULED: <2023-10-20 Fri>
Pas de sortie gnomad ? On teste dans /Work/Users/apraga/bisonex/test/t2t/gnomad

On copie les smlink
#+begin_src  sh
cp -P /Work/Users/apraga/bisonex/work/98/967730e87f1bd228dbd68725a1ef7d/* .
 cp /Work/Users/apraga/bisonex/work/98/967730e87f1bd228dbd68725a1ef7d/.command.sh  .
rm *.spip.*
rm *.vep.*
#+end_src
** DONE [#B] Indicateurs qualité :qualité:
CLOSED: [2023-09-10 Sun 16:46]
*** Idée
Raredisease:
- FastQC : nombreuses statistiques. Non disponible Nix
- Mosdepth : calcule la profondeur (2x plus rapide que samtools depth). Nix
- MultiQC : fusionne juste les résultats des analyses. Non disponible nix
- Picard's CollectMutipleMetrics, CollectHsMetrics, and CollectWgsMetrics
- Qualimap : alternative fastqc ? Non disponible nix
- Sentieon's WgsMetricsAlgo : propriétaire
- TIDDIT's cov : TIDIT = remaninement chromosomique

Sarek:
- alignment statistics : samtools stats, mosdepth
- QC : MultiQC

MultiQC : non disponible Nix
*** DONE FastqQC
CLOSED: [2023-08-15 Tue 21:43] SCHEDULED: <2023-08-13 Sun>
*** DONE Mosdepth
CLOSED: [2023-08-15 Tue 21:43] SCHEDULED: <2023-08-13 Sun>
Pour exomple, il faut le fichier de capture
subworkflows/local/bam_markduplicates/
*** DONE Samtools stats
CLOSED: [2023-08-15 Tue 21:43] SCHEDULED: <2023-08-13 Sun>
*** DONE [#B] Compte-redu exécution avec MultiQC
CLOSED: [2023-08-15 Tue 21:43] SCHEDULED: <2023-08-13 Sun>
*** DONE Résultats sur NA12878 : 98% à 20x
CLOSED: [2023-08-19 Sat 20:45] SCHEDULED: <2023-08-17 Thu>
**** DONE Comprendre 91% à 20x seulement: SNVs inséré
CLOSED: [2023-08-18 Fri 22:25]
***** DONE Tester autre kit : Twist exome comprehensive
CLOSED: [2023-08-18 Fri 22:24]
Moins bon
***** DONE Tester génome sans alt
CLOSED: [2023-08-18 Fri 22:25]
Idem
***** DONE Tester NA12878 sans SNVs inséré: cause !!
CLOSED: [2023-08-18 Fri 22:25]
***** DONE Tester hg19 sur NA12878 non inséré
CLOSED: [2023-08-18 Fri 22:25]
**** DONE Comprendre pourquoi SNVs diminuent le score: reads manquants
CLOSED: [2023-08-19 Sat 20:34] SCHEDULED: <2023-08-18 Fri>
Voir [[id:5c1c36f3-f68e-4e6d-a7b6-61dca89abc37][Bug: perte de nombreux reads avec NA12878]]
*** DONE Relancer résultats avec NA1287 et NA12878 + sanger
CLOSED: [2023-08-29 Tue 10:30] SCHEDULED: <2023-08-29 Tue>
*** DONE Comparer avec hg19
CLOSED: [2023-08-28 Mon 17:22] SCHEDULED: <2023-08-20 Sun>
*** DONE Comparer avec autres kit de capture
CLOSED: [2023-08-28 Mon 17:22] SCHEDULED: <2023-08-20 Sun>
*** DONE Comparer avec no-alt
CLOSED: [2023-08-28 Mon 17:22] SCHEDULED: <2023-08-20 Sun>
** HOLD vérifier si normalisation
** KILL [#B] Vérification nomenclature hgvs :hgvs:
CLOSED: [2023-08-16 Wed 19:07] SCHEDULED: <2023-08-15 Tue>
*** KILL mutalyzer
CLOSED: [2023-08-16 Wed 19:07] SCHEDULED: <2023-08-13 Sun>
*** KILL API variantvalidator
CLOSED: [2023-08-16 Wed 19:07] SCHEDULED: <2023-08-13 Sun>

** DONE Exécution
CLOSED: [2022-09-13 Tue 21:37]
*** KILL test Bionix
*** KILL Implémenter execution avec Nix ?
Voir https://academic.oup.com/gigascience/article/9/11/giaa121/5987272?login=false
pour un exemple.
Probablement plus simple d’utiliser Nix pour gestion de l’environnement et snakemake pour l’exécution
Pas d’accès internet depuis le cluster
*** DONE nextflow
CLOSED: [2022-09-13 Tue 21:37]
**** TODO Bug scheduler SGE
Le job se fait tuer car l'utilisateur n'est pas passé correctement à nextflow
***** DONE Forcer l'utilisateur à l'exécution
CLOSED: [2023-04-01 Sat 17:57]
NXF_OPTS=-D"user.name=alex"
***** DONE Vérifier si le problème persiste avec 22.10.6
CLOSED: [2023-04-01 Sat 18:38] SCHEDULED: <2023-04-01 Sat>
oui
***** KILL Packager l'utilisateur dans le programme ?
Mauvaise idée..
*** DONE Diminuer mémoire pour haplotypecaller
CLOSED: [2023-09-20 Wed 21:44] SCHEDULED: <2023-09-19 Tue>
/Entered on/ [2023-09-19 Tue 15:30]
Medium = 32Go pour 6 coeurs => 4 jobs (donc tout le noeud) prend plus que les 96GB...
On essaie 16Gb
Puis commit
*** DONE Report multiqc avec 10 runs
CLOSED: [2023-09-19 Tue 15:31] SCHEDULED: <2023-09-19 Tue>
/Entered on/ [2023-09-19 Tue 15:31]
Cf mail 2023-09-19
*** DONE Bug: variant sur 7788314 pour patient 62982193 filtré : DP < 30
CLOSED: [2023-10-02 Mon 21:58] SCHEDULED: <2023-09-25 Mon>
/Entered on/ [2023-09-22 Fri 22:59]
35 selon IGV mais 27 en pratique dans le VCF.
VCF cento: 26 reads également...
VOUS, non confirmé sanger
Mail envoyé Alexis

Vu avec Paul : on laisse DP >= 30 si c'est la seule occurence
*** DONE Bug mésohelios: les jobs se font killer :bug:
CLOSED: [2023-10-13 Fri 11:44]
/Entered on/ [2023-10-11 Wed 12:06]
**** DONE Comprendre pourquoi
CLOSED: [2023-10-11 Wed 16:06] SCHEDULED: <2023-10-11 Wed>
Utilisateurs déconnectés à 4h du matin tous les jours
**** DONE Démarrer nextflow avec sbatch
CLOSED: [2023-10-13 Fri 11:07] SCHEDULED: <2023-10-11 Wed>
On retrouve le même bug avec squeue qui n'arrive pas à retrouver l'utilisateur en utilisant nextflow+nix
Même en forcant USER et export NXF_OPTS='-D"user.name=apraga"'

Test avec la version packagée sur mésocentre (il faut mettre à la main le dossier...): ok
#+begin_src sh
module load nextflow@23.04.3/gcc-12.1.0
# Force it
nextflow="/Softs/spack/opt/spack/linux-rocky8-x86_64/gcc-12.1.0/nextflow-23.04.3-qputqf2dmtvabpv76mizj63gbsa5irq3/bin/nextflow  -c /nix/store/q46gdjil6bbap0bcghpqimh1camzmlpx-nextflow.config"
#+end_src

Avec user.name= on a une exécution correcte :
#+begin_src sh
#!/bin/bash -l
# Fichier submission.SBATCH

#SBATCH --job-name="bisonex"
#SBATCH --output=%x.%J.out   ## %x=job name, %J=job id
#SBATCH --error=%x.%J.out
 # walltime (hh:mm::ss) max is 8 days
#SBATCH -t 24:00:00
#SBATCH --partition=smp
#SBATCH -c 1  ## request 16 cores (MAX is 32)
#SBATCH --mem=12G ## (MAX is 96G)
#SBATCH --mail-user=apraga@chu-besancon.fr
#SBATCH --mail-type=END,FAIL   # notify when job end/fail

module load nix/2.11.0

# Otherwise job fails as it cannot write to $HOME/.nextflow
export NXF_HOME=/Work/Users/apraga/.nextflow

# user.name must be forced (again... our fix does not seem  to work in nix)
nextflow -Duser.name=apraga run main.nf -profile standard,helios --input=samples.csv --genome=GRCh38
#+end_src
**** DONE Mettre à jour documentation
CLOSED: [2023-10-13 Fri 11:44] SCHEDULED: <2023-10-13 Fri>
** DONE Mettre à jour spip pour corriger bug 62982239 : variant trop long (?)
CLOSED: [2023-09-22 Fri 22:22] SCHEDULED: <2023-09-21 Thu>
/Entered on/ [2023-09-21 Thu 23:11]
Rapporté par https://github.com/raphaelleman/SPiP/issues/9
*** DONE Relance run
CLOSED: [2023-09-22 Fri 22:22] SCHEDULED: <2023-09-21 Thu>
*** DONE Mise à jour spip
CLOSED: [2023-09-21 Thu 23:41] SCHEDULED: <2023-09-21 Thu>
** DONE nixpkgs unstable -> 23.05
CLOSED: [2023-09-22 Fri 22:22] SCHEDULED: <2023-09-21 Thu>
*** DONE repasser tests sanger
CLOSED: [2023-09-22 Fri 22:22] SCHEDULED: <2023-09-21 Thu>
** TODO Preprocessing avec nextflow
*** TODO Map to reference
**** TODO Sample ID dans header
/Work/Users/apraga/bisonex/out/63003856_S135/preprocessing/baserecalibrator
*** DONE Mark duplicate
CLOSED: [2022-10-09 Sun 22:30]
*** DONE Recalibrate base quality score
CLOSED: [2022-10-09 Sun 22:30]
** DONE Variant calling avec Nextflow
CLOSED: [2022-11-19 Sat 21:34]
*** DONE Haplotype caller
CLOSED: [2022-10-09 Sun 22:40]
*** DONE Filter variants
CLOSED: [2022-10-09 Sun 22:40]
*** DONE Filter common snp not clinvar path
CLOSED: [2022-11-07 Mon 23:00]
Voir [[*common dbSNP not clinvar patho][common dbSNP not clinvar patho]]
*** DONE Filter variant only in consensual sequence
CLOSED: [2022-11-08 Tue 22:23]
*** DONE Filter technical variants
CLOSED: [2022-11-19 Sat 21:34]
*** DONE Utilise AVX pour accélerer l'exécution
CLOSED: [2023-04-29 Sat 15:46]
Sans cela, on a l'avertissement
#+begin_quote
17:28:00.720 INFO  PairHMM - OpenMP multi-threaded AVX-accelerated native PairHMM implementation is not supported
17:28:00.721 INFO  NativeLibraryLoader - Loading libgkl_utils.so from jar:file:/nix/store/cy9ckxqwrkifx7wf02hm4ww1p6lnbxg9-gatk-4.2.4.1/bin/gatk-package-4.2.4.1-local.jar!/com/intel/gkl/native/libgkl_utils.so
17:28:00.733 WARN  NativeLibraryLoader - Unable to load libgkl_utils.so from native/libgkl_utils.so (/Work/Users/apraga/bisonex/out/NA12878_NIST7035/preprocessing/applybqsr/libgkl_utils821485189051585397.so: libgomp.so.1: cannot open shared object file: No such file or directory)
17:28:00.733 WARN  IntelPairHmm - Intel GKL Utils not loaded
17:28:00.733 WARN  PairHMM - ***WARNING: Machine does not have the AVX instruction set support needed for the accelerated AVX PairHmm. Falling back to the MUCH slower LOGLESS_CACHING implementation!
17:28:00.763 INFO  ProgressMeter - Starting traversal
#+end_quote
libgomp.so est fourni par gcc donc il faut charger le module
 module load gcc@11.3.0/gcc-12.1.0
** KILL Utiliser subworkflow
CLOSED: [2023-04-02 Sun 18:08]
Notre version permet d'être plus souple
*** KILL Alignement
CLOSED: [2023-04-02 Sun 18:08] SCHEDULED: <2023-04-05 Wed>
*** KILL Vep
CLOSED: [2023-04-02 Sun 18:08] SCHEDULED: <2023-04-05 Wed>
vcf_annotate_ensemblvep
** TODO Annotation avec nextflow :annotation:
*** KILL VEP : --gene-phenotype ?
CLOSED: [2023-04-18 mar. 18:32]
Vu avec alexis : bases de données non à jour
https://www.ensembl.org/info/genome/variation/phenotype/sources_phenotype_documentation.html
*** DONE plugin VEP
CLOSED: [2023-04-18 mar. 18:32]
Cloner dépôt git avec plugin
Puis utiliser --dir_plugins
*** HOLD Utiliser code d’Alexis
*** TODO Nouvelle version avec VEP
Example avec --custom
https://www.ensembl.org/info/docs/tools/vep/script/vep_custom.html
**** DONE Ajout spliceAI
CLOSED: [2023-05-18 Thu 11:02] SCHEDULED: <2023-04-30 Sun>
plugin VEP
***** DONE Télécharger les données
CLOSED: [2023-05-11 Thu 19:01]
Difficile d'automatiser, le lien est temporaire...
***** DONE PLugin
CLOSED: [2023-05-11 Thu 20:16]
***** DONE Séparer score en plusieurs colonnes
CLOSED: [2023-05-11 Thu 20:16]
Test avec ce fichier pour avoir une ligne avec annotation et une ligne sans
#CHROM	POS	ID	REF	ALT
1	9091	.	A	C
1	69091	.	A	C

et
#+begin_src sh
rm -f postvep.tsv* && vep -i testspliceai.vcf.gz -o postvep.tsv --tab  --dir 109 --merged --pick --use_given_ref   --offline  --plugin SpliceAI,snv=spliceai_scores.raw.snv.hg38.vcf.gz,indel=spliceai_scores.raw.indel.hg38.vcf.gz
#+end_src

#+begin_src

$ bgzip postvep.tsv
$ python spliceai.py
$ cat postvep2.tsv
,variation,Location,Allele,Gene,Feature,Feature_type,Consequence,cDNA_position,CDS_position,Protein_position,Amino_acids,Codons,Existing_variation,IMPACT,DISTANCE,STRAND,FLAGS,REFSEQ_MATCH,SOURCE,REFSEQ_OFFSET,SpliceAI_AG,SpliceAI_AL,SpliceAI_DG,SpliceAI_DL
0,1_9091_A/C,1:9091,C,ENSG00000290825,ENST00000456328,Transcript,upstream_gene_variant,-,-,-,-,-,-,MODIFIER,2778,1,-,-,Ensembl,-,,,,
1,1_69091_A/C,1:69091,C,ENSG00000186092,ENST00000641515,Transcript,missense_variant,124,64,22,M/L,Atg/Ctg,-,MODERATE,-,1,-,-,Ensembl,-,0.01,0.00,0.00,0.01
#+end_src

Test

cp work/bf/437ae511958509e43072f032f4d495/small.tab.gz tests/vep-spip.tab.gz
cp work/d5/3b1244b5ae83d54409ee0d456e8c55/small_cadd.tab.gz tests/vep-cadd-splice.tab.gz
**** DONE Package Nix spliceAI
CLOSED: [2023-10-07 Sat 18:00]
On utilise le tensorflow fourni avec nix (branche spliceai)
Il faut LD_PRELOAD=/lib64/libcuda.so  pour l'exécution
***** DONE Version CPU non optimisé
CLOSED: [2023-09-23 Sat 21:34]
****** DONE Vérifier annotation hg19 et 38
CLOSED: [2023-09-23 Sat 18:49]
****** DONE Annotation maison T2T
CLOSED: [2023-09-23 Sat 21:33] SCHEDULED: <2023-09-23 Sat>
***** DONE Version CPU optismisée
CLOSED: [2023-09-23 Sat 21:34]
Activer flag dans le package nix
***** DONE Version CPU:  Test chr20
CLOSED: [2023-09-23 Sat 22:25] SCHEDULED: <2023-09-23 Sat>
10min ?
***** DONE Version CPU:  NA12878 sanger complet: kill car trop long
CLOSED: [2023-09-24 Sun 08:22] SCHEDULED: <2023-09-23 Sat>
10min ?
***** DONE GPU avec la version sur mésocentre: mail envoyé
CLOSED: [2023-09-26 Tue 11:49]
#+begin_src sh
module unload nix/2.11.0
module load anaconda3@2021.05/gcc-12.1.0
module load deep/tensorflow-gpu
pip install spliceai
#+end_src
Puis on teste sur la queue gpu
#+begin_src sh
srun -p gpu -t 4:00:00 --gres=gpu:1 --pty bash

cd /Work/Users/apraga/bisonex/tests/spliceai/
module load deep/tensorflow-gpu
module unload nix/2.11.0

time spliceai -I NA12878-sanger-chr20-T2T.vep.vcf.gz -O output-20-2.vcf -R /Work/Groups/bisonex/data/fasta/chm13v2.0/chm13v2.0.fa -A ~/t2t.txt
#+end_src

Échec: librarie DNN not found...

#+begin_quote
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-09-24 09:37:45.892545: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2023-09-24 09:37:47.759421: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1635] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 38188 MB memory:  -> device: 0, name: NVIDIA A100-PCIE-40GB, pci bus id: 0000:21:00.0, compute capability: 8.0
WARNING:tensorflow:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.
WARNING:tensorflow:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.
WARNING:tensorflow:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.
WARNING:tensorflow:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.
WARNING:tensorflow:No training configuration found in the save file, so the model was *not* compiled. Compile it manually.
2023-09-24 09:37:54.143021: I tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:637] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.
2023-09-24 09:37:54.217160: E tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:429] Could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED
2023-09-24 09:37:54.217220: E tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:438] Possibly insufficient driver version: 510.108.3
2023-09-24 09:37:54.217245: W tensorflow/core/framework/op_kernel.cc:1830] OP_REQUIRES failed at conv_ops.cc:1068 : UNIMPLEMENTED: DNN library is not found.
2023-09-24 09:37:54.217262: I tensorflow/core/common_runtime/executor.cc:1197] [/job:localhost/replica:0/task:0/device:GPU:0] (DEBUG INFO) Executor start aborting (this does not indicate an error and you can ignore this message): UNIMPLEMENTED: DNN library is not found.
         [[{{node model_1/conv1d_3/Conv1D}}]]
Traceback (most recent call last):
  File "/Home/Users/apraga/.local/bin/spliceai", line 8, in <module>
    sys.exit(main())
  File "/Home/Users/apraga/.local/lib/python3.9/site-packages/spliceai/__main__.py", line 72, in main
    scores = get_delta_scores(record, ann, args.D, args.M)
  File "/Home/Users/apraga/.local/lib/python3.9/site-packages/spliceai/utils.py", line 159, in get_delta_scores
    y_ref = np.mean([ann.models[m].predict(x_ref) for m in range(5)], axis=0)
  File "/Home/Users/apraga/.local/lib/python3.9/site-packages/spliceai/utils.py", line 159, in <listcomp>
    y_ref = np.mean([ann.models[m].predict(x_ref) for m in range(5)], axis=0)
  File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/utils/traceback_utils.py", line 70, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/tensorflow/python/eager/execute.py", line 52, in quick_execute
    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
tensorflow.python.framework.errors_impl.UnimplementedError: Graph execution error:

Detected at node 'model_1/conv1d_3/Conv1D' defined at (most recent call last):
    File "/Home/Users/apraga/.local/bin/spliceai", line 8, in <module>
      sys.exit(main())
    File "/Home/Users/apraga/.local/lib/python3.9/site-packages/spliceai/__main__.py", line 72, in main
      scores = get_delta_scores(record, ann, args.D, args.M)
    File "/Home/Users/apraga/.local/lib/python3.9/site-packages/spliceai/utils.py", line 159, in get_delta_scores
      y_ref = np.mean([ann.models[m].predict(x_ref) for m in range(5)], axis=0)
    File "/Home/Users/apraga/.local/lib/python3.9/site-packages/spliceai/utils.py", line 159, in <listcomp>
      y_ref = np.mean([ann.models[m].predict(x_ref) for m in range(5)], axis=0)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/utils/traceback_utils.py", line 65, in error_handler
      return fn(*args, **kwargs)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/engine/training.py", line 2382, in predict
      tmp_batch_outputs = self.predict_function(iterator)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/engine/training.py", line 2169, in predict_function
      return step_function(self, iterator)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/engine/training.py", line 2155, in step_function
      outputs = model.distribute_strategy.run(run_step, args=(data,))
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/engine/training.py", line 2143, in run_step
      outputs = model.predict_step(data)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/engine/training.py", line 2111, in predict_step
      return self(x, training=False)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/utils/traceback_utils.py", line 65, in error_handler
      return fn(*args, **kwargs)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/engine/training.py", line 558, in __call__
      return super().__call__(*args, **kwargs)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/utils/traceback_utils.py", line 65, in error_handler
      return fn(*args, **kwargs)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/engine/base_layer.py", line 1145, in __call__
      outputs = call_fn(inputs, *args, **kwargs)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/utils/traceback_utils.py", line 96, in error_handler
      return fn(*args, **kwargs)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/engine/functional.py", line 512, in call
      return self._run_internal_graph(inputs, training=training, mask=mask)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/engine/functional.py", line 669, in _run_internal_graph
      outputs = node.layer(*args, **kwargs)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/utils/traceback_utils.py", line 65, in error_handler
      return fn(*args, **kwargs)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/engine/base_layer.py", line 1145, in __call__
      outputs = call_fn(inputs, *args, **kwargs)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/utils/traceback_utils.py", line 96, in error_handler
      return fn(*args, **kwargs)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/layers/convolutional/base_conv.py", line 290, in call
      outputs = self.convolution_op(inputs, self.kernel)
    File "/Softs/helios/gpu/anaconda3/2023.03-1/envs/tensorflow-gpu-2.12.0+py3.9/lib/python3.9/site-packages/keras/layers/convolutional/base_conv.py", line 262, in convolution_op
      return tf.nn.convolution(
Node: 'model_1/conv1d_3/Conv1D'
DNN library is not found.
         [[{{node model_1/conv1d_3/Conv1D}}]] [Op:__inference_predict_function_22195]
#+end_quote

***** DONE GPU: chr20 ok
CLOSED: [2023-09-26 Tue 11:50]
LD_PRELOAD=/lib64/libcuda.so spliceai -I NA12878-sanger-20-2-T2T.vep.vcf.gz -O output-20-2-gpu.vcf -R /Work/Groups/bisonex/data/fasta/chm13v2.0/chm13v2.0.fa -A ~/t2t.txt
temps d'exécution : 5min

***** DONE GPU: toutes les données :GPU:spliceai:
CLOSED: [2023-09-28 Thu 01:34]
****** DONE Run : 70GB, 3h30
CLOSED: [2023-09-27 Wed 10:37] SCHEDULED: <2023-09-26 Tue>
32G insufissant ! Il faut 70GB :

Job ID: 17340
Cluster: mesoubfc
User/Group: apraga/mesousers
State: COMPLETED (exit code 0)
Cores: 1
CPU Utilized: 03:11:53
CPU Efficiency: 93.55% of 03:25:07 core-walltime
Job Wall-clock time: 03:25:07
Memory Utilized: 67.75 GB
Memory Efficiency: 52.93% of 128.00 GB
#+begin_src slurm
#!/bin/bash -l
# Fichier submission.SBATCH

#SBATCH --job-name="spliceai-gpu"
#SBATCH --output=%x.%J.out   ## %x=nom_du_job, %J=id du job
#SBATCH --error=%x.%J.out

# walltime (hh:mm::ss) max is 8 days
#SBATCH -t 24:00:00

#SBATCH --partition=gpu
#SBATCH --gres=gpu:1

## To request more memory, use --mem option.
## Please don't use more than 128g.
#SBATCH --mem=64G

## votre dresse mail pour les notifs
#SBATCH --mail-user=apraga@chu-besancon.fr
#SBATCH --mail-type=END,FAIL

nvidia-smi

module purge
module load nix/2.11.0

LD_PRELOAD=/lib64/libcuda.so spliceai -I NA12878-sanger-all-T2T.vep.vcf.gz -O output-all-gpu.vcf -R /Work/Groups/bisonex/data/fasta/chm13v2.0/chm13v2.0.fa -A ~/t2t.txt
#+end_src
****** DONE Annoter la sortie de VEP avec ce VCF
CLOSED: [2023-09-28 Thu 01:32]
Problème: SpliceAI est dans un INFO différent et donc pas avec les transcrits. filter_vep ne semble pas gérer ce cas.
On doit donc fusionner : le plus simple est de le convertir en fichier d'annotation et d'utiliser vep (comme pour Spip )
Générer un fichier d'annotation
#+begin_src
bcftools annotate -x INFO/CSQ output-all-gpu.vcf  -o spliceai.vcf.gz
bcftools index spliceai.vcf.gz
#+end_src

Annoter avec vep
#+begin_src sh
ln -s  /Work/Projects/bisonex/data/fasta/chm13v2.0/chm13v2.0.fa .
ln -s  /Work/Projects/bisonex/data/fasta/chm13v2.0/chm13v2.0.fa.fai .
ln -s  /Work/Projects/bisonex/data/vep/chm13v2.0/106 .
ln -s  /Work/Projects/bisonex/data/clinvar/chm13v2.0/clinvar.vcf.gz .
ln -s  /Work/Projects/bisonex/data/clinvar/chm13v2.0/clinvar.vcf.gz.tbi .
#+end_src

Essai 1: on coupe le fichier en 2 pour utiliser le plugin spliceai
filter -i 'POS<15000' spliceai.vcf.gz -o spliceai1.vcf.gz
bcftools filter -i 'POS>=15000' spliceai.vcf.gz -o spliceai2.vcf.gz

vep -i  output-all-gpu.vcf -o output-all-gpu-annotated.vcf.gz --appris --biotype --canonical --ccds --compress_output bgzip --domains --exclude_predicted --flag_pick --hgvs --hgvsg --gene_phenotype --numbers --mane --protein --offline --uniprot --symbol --tsl --use_given_ref --variant_class --vcf --plugin NMD --custom clinvar.vcf.gz,ClinVar,vcf,exact,0,CLNSIG,CLNREVSTAT,CLNDN --plugin SpliceAI,snv=spliceai1.vcf.gz,indel=spliceai2.vcf.gz --fasta chm13v2.0.fa --assembly T2T-CHM13v2.0 --species homo_sapiens_gca009914755v4/ --cache  --cache_version 106 --dir_cache 106

test
 filter_vep -i output-all-gpu-annotated.vcf.gz --format vcf --filter "        not(Consequence matches non_coding_transcript or Consequence matches stream                 or Consequence matches intergenic_variant                 or Consequence matches UTR                 or Consequence matches intron_variant                 or Consequence matches synonymous                 or BIOTYPE  matches pseudogene                 or BIOTYPE  matches misc_RNA) or (SpliceAI_pred_DS_AG and SpliceAI_pred_DS_AG >= 0.2)             or (SpliceAI_pred_DS_AL and SpliceAI_pred_DS_AL >= 0.2)             or (SpliceAI_pred_DS_DG and SpliceAI_pred_DS_DG >= 0.2)             or (SpliceAI_pred_DS_DL and SpliceAI_pred_DS_DL >= 0.2) "                --only_matched        -o output-all-gpu-filtered.vcf.gz

 Méthode un peu sale car on a des "." dans l'annotation spliceai

****** KILL Save
CLOSED: [2023-09-27 Wed 21:40]
# ****** DONE Filtre vep avec spliceAI: 37365 -> 6130. SpliceAI n'apporte rien
# CLOSED: [2023-09-27 Wed 19:37] SCHEDULED: <2023-09-27 Wed>
# :PROPERTIES:
# :ID:       c9b2009a-503b-4561-94c6-29ae21a3188d
# :END:
# #+begin_src sh
# filter_vep -i output-all-gpu.vcf --format vcf --filter "        not(Consequence matches non_coding_transcript or Consequence matches stream                 or Consequence matches intergenic_variant                 or Consequence matches UTR                 or Consequence matches intron_variant                 or Consequence matches synonymous                 or BIOTYPE  matches pseudogene                 or BIOTYPE  matches misc_RNA) or (SpliceAI_pred_DS_AG and SpliceAI_pred_DS_AG >= 0.2)             or (SpliceAI_pred_DS_AL and SpliceAI_pred_DS_AL >= 0.2)             or (SpliceAI_pred_DS_DG and SpliceAI_pred_DS_DG >= 0.2)             or (SpliceAI_pred_DS_DL and SpliceAI_pred_DS_DL >= 0.2) "                --only_matched         -o output-all-gpu-filtered.vcf
# #+end_src

# filter_vep -i output-all-gpu.vcf --format vcf --filter "        not(Consequence matches non_coding_transcript or Consequence matches stream                 or Consequence matches intergenic_variant                 or Consequence matches UTR                 or Consequence matches intron_variant                 or Consequence matches synonymous                 or BIOTYPE  matches pseudogene                 or BIOTYPE  matches misc_RNA)"                --only_matched        | grep -c -v '^#'
# 6130

# $ grep -c -v '^#' output-all-gpu-filtered.vcf
# 6130
# ****** DONE Re-vérifier filtre avec spip: 7730 -> probable problème avec spip
# CLOSED: [2023-09-27 Wed 20:54] SCHEDULED: <2023-09-27 Wed>
#  filter_vep -i NA12878-sanger-all-T2T.vep.vcf.gz --format vcf --filter "        not(Consequence matches non_coding_transcript or Consequence matches stream                 or Consequence matches intergenic_variant                 or Consequence matches UTR                 or Consequence matches intron_variant                 or Consequence matches synonymous                 or BIOTYPE  matches pseudogene                 or BIOTYPE  matches misc_RNA) or (SPIP_spipScore and SPIP_spipScore >= 20)"                --only_matched        | grep -c -v '^#'
# perl: warning: Setting locale failed.
# perl: warning: Please check that your locale settings:
#         LANGUAGE = (unset),
#         LC_ALL = (unset),
#         LANG = "en_US.utf8"
#     are supported and installed on your system.
# perl: warning: Falling back to the standard locale ("C").
# 7730

****** DONE vérifier si tests sanger passent: ok
CLOSED: [2023-09-28 Thu 01:32] SCHEDULED: <2023-09-27 Wed>

 ok !
 Haplotypecaller : /Work/Users/apraga/bisonex/out//call_variant/haplotypecaller/NA12878-sanger-all-T2T/NA12878-sanger-all-T2T.haplotypecaller.vcf.gz/Work/Users/apraga/bisonex/out//call_variant/haplotypecaller/NA12878-sanger-all-T2T/NA12878-sanger-all-T2T.haplotypecaller.vcf.gz
144 found over 146
2×3 DataFrame
 Row │ variant              meanQual  depth
     │ String               Float64   Int64
─────┼──────────────────────────────────────
   1 │ chr12:g.13594572C>T      60.0      1
   2 │ chr17:g.10204026T>A      60.0      1
/Work/Users/apraga/bisonex/tests/spliceai/output-all-gpu-filtered.vcf.gz
144 found over 146
spliceai : another 0 missed variants
0×3 DataFrame
 Row │ variant  meanQual  depth
     │ String   Float64   Int64
─────┴──────────────────────────
***** KILL Avec pip: echec
CLOSED: [2023-09-28 Thu 01:34]
2023-09-24 08:28:46.361434: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1956] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU.
***** DONE Tester conda: echec
CLOSED: [2023-09-23 Sat 21:43] SCHEDULED: <2023-09-23 Sat>
Ananconda: N'arrive pas à installer

#+begin_quote

  - feature:/linux-64::__glibc==2.28=0
  - python=3.11 -> libgcc-ng[version='>=11.2.0'] -> __glibc[version='>=2.17']
  - spliceai -> tensorflow[version='>=1.13.0'] -> __cuda
  - spliceai -> tensorflow[version='>=1.13.0'] -> __glibc[version='>=2.17']

Your installed version is: 2.28
#+end_quote
Il faut utiliser mamba
***** DONE Mail Paul
CLOSED: [2023-09-28 Thu 09:30] SCHEDULED: <2023-09-28 Thu>
Au total:
- pas de filtre sur l'épissage pour "rattraper" variants intronique : 6130 variants mais on en perd 4 (donner un exemple)
- avec spip: pas de variant perdu (hormis les 2 lié au bam) mais 7 332 variants au total et exécution lente (1H) mais possible sur serveur
- avec spliceai sur GPU (annotation "à la volée"): pas de variant perdu mais 6609 variants au total. 3h30 de calcul sur le mésocentre, impossible à faire chez nous car GPU

Rejoint ce que disait Yannis....
J'ai l'impression que c'est lié à un grand nombre de missense [1]
Note: j'ai plus confiance dans l'annottaion spliceAI pour T2T car moins compliqué à porter

[1] en prenant le transcrit avec la "pire" conséquences, on a 80% de missense (total = 6609)

     11 3_prime_utr_variant
      9 3_prime_utr_variant&nmd_transcript_variant
      6 5_prime_utr_variant
     48 coding_sequence_variant
      5 coding_sequence_variant&nmd_transcript_variant
    121 frameshift_variant
      9 frameshift_variant&nmd_transcript_variant
      1 frameshift_variant&splice_donor_region_variant
      9 frameshift_variant&splice_region_variant
     78 inframe_deletion
      3 inframe_deletion&nmd_transcript_variant
      2 inframe_deletion&splice_region_variant
     84 inframe_insertion
      2 inframe_insertion&nmd_transcript_variant
      1 inframe_insertion&splice_region_variant
    156 intron_variant
      8 intron_variant&nmd_transcript_variant
     24 intron_variant&non_coding_transcript_variant
   5305 missense_variant
    205 missense_variant&nmd_transcript_variant
      3 missense_variant&splice_donor_5th_base_variant
    110 missense_variant&splice_region_variant
      9 missense_variant&splice_region_variant&nmd_transcript_variant
     11 non_coding_transcript_exon_variant
     12 splice_acceptor_variant
      1 splice_acceptor_variant&nmd_transcript_variant
      2 splice_acceptor_variant&non_coding_transcript_variant
      9 splice_donor_5th_base_variant&intron_variant
      1 splice_donor_5th_base_variant&intron_variant&nmd_transcript_variant
     16 splice_donor_region_variant&intron_variant
      4 splice_donor_region_variant&intron_variant&non_coding_transcript_variant
     19 splice_donor_variant
      1 splice_donor_variant&nmd_transcript_variant
      3 splice_donor_variant&non_coding_transcript_variant
      1 splice_donor_variant&splice_donor_5th_base_variant&3_prime_utr_variant&intron_variant&nmd_transcript_variant
      3 splice_donor_variant&splice_donor_5th_base_variant&coding_sequence_variant&intron_variant
      1 splice_donor_variant&splice_donor_5th_base_variant&intron_variant
     39 splice_polypyrimidine_tract_variant&intron_variant
      5 splice_polypyrimidine_tract_variant&intron_variant&nmd_transcript_variant
     10 splice_polypyrimidine_tract_variant&intron_variant&non_coding_transcript_variant
      1 splice_region_variant&3_prime_utr_variant
      1 splice_region_variant&5_prime_utr_variant
      9 splice_region_variant&intron_variant
      1 splice_region_variant&intron_variant&nmd_transcript_variant
      2 splice_region_variant&intron_variant&non_coding_transcript_variant
      5 splice_region_variant&non_coding_transcript_exon_variant
      1 splice_region_variant&non_coding_transcript_variant
     43 splice_region_variant&splice_polypyrimidine_tract_variant&intron_variant
      2 splice_region_variant&splice_polypyrimidine_tract_variant&intron_variant&nmd_transcript_variant
      6 splice_region_variant&splice_polypyrimidine_tract_variant&intron_variant&non_coding_transcript_variant
     15 splice_region_variant&synonymous_variant
     14 start_lost
     44 stop_gained
      4 stop_gained&frameshift_variant
      2 stop_gained&frameshift_variant&splice_region_variant
      3 stop_gained&nmd_transcript_variant
      3 stop_gained&splice_region_variant
      2 stop_gained&splice_region_variant&nmd_transcript_variant
      2 stop_lost
      1 stop_lost&nmd_transcript_variant
      6 stop_retained_variant
      2 stop_retained_variant&nmd_transcript_variant
     89 synonymous_variant
      2 synonymous_variant&nmd_transcript_variant
      1 transcript_ablation
      1 upstream_gene_variant

En prenant tous les transcrits, 66% missense (total = 22085)
           39 3_prime_UTR_variant
    120 3_prime_UTR_variant&NMD_transcript_variant
     22 5_prime_UTR_variant
      2 5_prime_UTR_variant&NMD_transcript_variant
     94 coding_sequence_variant
     13 coding_sequence_variant&NMD_transcript_variant
    527 downstream_gene_variant
    257 frameshift_variant
     21 frameshift_variant&NMD_transcript_variant
      2 frameshift_variant&splice_donor_region_variant
     20 frameshift_variant&splice_region_variant
      1 frameshift_variant&splice_region_variant&NMD_transcript_variant
      1 incomplete_terminal_codon_variant&coding_sequence_variant
    211 inframe_deletion
     18 inframe_deletion&NMD_transcript_variant
      6 inframe_deletion&splice_region_variant
    242 inframe_insertion
     22 inframe_insertion&NMD_transcript_variant
      4 inframe_insertion&splice_region_variant
    983 intron_variant
    244 intron_variant&NMD_transcript_variant
    358 intron_variant&non_coding_transcript_variant
  14690 missense_variant
   1416 missense_variant&NMD_transcript_variant
      6 missense_variant&splice_donor_5th_base_variant
    374 missense_variant&splice_region_variant
     34 missense_variant&splice_region_variant&NMD_transcript_variant
    383 non_coding_transcript_exon_variant
     53 splice_acceptor_variant
     11 splice_acceptor_variant&NMD_transcript_variant
     11 splice_acceptor_variant&non_coding_transcript_variant
     20 splice_donor_5th_base_variant&intron_variant
      4 splice_donor_5th_base_variant&intron_variant&NMD_transcript_variant
      9 splice_donor_5th_base_variant&intron_variant&non_coding_transcript_variant
     59 splice_donor_region_variant&intron_variant
     11 splice_donor_region_variant&intron_variant&NMD_transcript_variant
     24 splice_donor_region_variant&intron_variant&non_coding_transcript_variant
     79 splice_donor_variant
      6 splice_donor_variant&NMD_transcript_variant
     17 splice_donor_variant&non_coding_transcript_variant
      1 splice_donor_variant&splice_donor_5th_base_variant&3_prime_UTR_variant&intron_variant&NMD_transcript_variant
     21 splice_donor_variant&splice_donor_5th_base_variant&coding_sequence_variant&intron_variant
      3 splice_donor_variant&splice_donor_5th_base_variant&intron_variant
      1 splice_donor_variant&splice_donor_5th_base_variant&non_coding_transcript_exon_variant&intron_variant
    176 splice_polypyrimidine_tract_variant&intron_variant
     27 splice_polypyrimidine_tract_variant&intron_variant&NMD_transcript_variant
     48 splice_polypyrimidine_tract_variant&intron_variant&non_coding_transcript_variant
      1 splice_region_variant&3_prime_UTR_variant
     24 splice_region_variant&3_prime_UTR_variant&NMD_transcript_variant
      9 splice_region_variant&5_prime_UTR_variant
     61 splice_region_variant&intron_variant
     23 splice_region_variant&intron_variant&NMD_transcript_variant
     37 splice_region_variant&intron_variant&non_coding_transcript_variant
     26 splice_region_variant&non_coding_transcript_exon_variant
      5 splice_region_variant&non_coding_transcript_variant
    145 splice_region_variant&splice_polypyrimidine_tract_variant&intron_variant
     27 splice_region_variant&splice_polypyrimidine_tract_variant&intron_variant&NMD_transcript_variant
     41 splice_region_variant&splice_polypyrimidine_tract_variant&intron_variant&non_coding_transcript_variant
     37 splice_region_variant&synonymous_variant
      3 splice_region_variant&synonymous_variant&NMD_transcript_variant
     30 start_lost
      5 start_lost&NMD_transcript_variant
    135 stop_gained
     13 stop_gained&frameshift_variant
      3 stop_gained&frameshift_variant&NMD_transcript_variant
      2 stop_gained&frameshift_variant&splice_region_variant
     14 stop_gained&NMD_transcript_variant
      5 stop_gained&splice_region_variant
      2 stop_gained&splice_region_variant&NMD_transcript_variant
      4 stop_lost
      1 stop_lost&NMD_transcript_variant
      9 stop_retained_variant
      6 stop_retained_variant&NMD_transcript_variant
    311 synonymous_variant
     24 synonymous_variant&NMD_transcript_variant
      1 transcript_ablation
    390 upstream_gene_variant

**** KILL Ajout LOEUF et pli
CLOSED: [2023-10-07 Sat 17:57]
plugin VEP
**** DONE NMD
CLOSED: [2023-10-07 Sat 17:57]
plugin VEP
**** KILL Ajout LOEUF
CLOSED: [2023-04-19 mer. 16:32]
plugin VEP
**** DONE Spip
CLOSED: [2023-05-01 Mon 23:07] SCHEDULED: <2023-04-30 Sun>
BED ne semble pas bien marcher (il faut définir une zone)
VCF : trop d’information
Attention, plusieurs transcripts mais résultats identiques. On supprimer les doublons
***** DONE interpretation + score + intervalle de confiance séparé
CLOSED: [2023-05-01 Mon 23:07] SCHEDULED: <2023-04-30 Sun>
Tests :
dans tests/
vep -i 63004925-small.vcf -o postvep.vcf --vcf --fasta genomeRef.fna --dir 109 --merged --pick  --offline --custom ../script/spip_annotation.vcf.gz,SPIP,vcf,exact,0,spipInterp,spipScore,spipConfidence
***** DONE Score
CLOSED: [2023-04-22 Sat 15:30]
**** DONE CADD: remplacer par plugin VEP
CLOSED: [2023-05-07 Sun 14:45] SCHEDULED: <2023-05-07 Sun>
***** Test
#+begin_src
vep  -i test.vcf  -o lol.vcf --offline --dir  /Work/Projects/bisonex/data/vep/GRCh38/ --merged --vcf --fasta /Work/Projects/bisonex/data/genome/GRCh38.p13/genomeRef.fna --plugin CADD,/Work/Users/apraga/bisonex/work/13/9287a7fef17ab9365f5696f20710cd/gnomad.genomes.r3.0.snv.tsv.gz,/Work/Users/apraga/bisonex/work/13/9287a7fef17ab9365f5696f20710cd/gnomad.genomes.r3.0.indel.tsv.gz  --dir_plugins ../VEP_plugins/ -v
#+end_src

Test
#+begin_src sh
vep --id "1  230710048 230710048 A/G 1"   --offline --dir  /Work/Projects/bisonex/data/vep/GRCh38/ --merged --vcf --fasta /Work/Projects/bisonex/data/genome/GRCh38.p13/genomeRef.fna --plugin CADD,/Work/Users/apraga/bisonex/work/13/9287a7fef17ab9365f5696f20710cd/gnomad.genomes.r3.0.snv.tsv.gz,/Work/Users/apraga/bisonex/work/13/9287a7fef17ab9365f5696f20710cd/gnomad.genomes.r3.0.indel.tsv.gz  --hgvsg --plugin pLI --plugin LOEUF -o lol
#+end_src

CSQ=G|missense_variant|MODERATE|AGT|ENSG00000135744|Transcript|ENST00000366667|protein_coding|2/5||||843|776|259|M/T|aTg/aCg|||-1||HGNC|HGNC:333||Ensembl||A|A||1:g.230710048A>G|0.347|-0.277922|
Correspond bien à https://www.ensembl.org/Homo_sapiens/Tools/VEP/Results?tl=I7ZsIbrj14P6lD43-9115494
***** DONE Utiliser whole genome
CLOSED: [2023-04-29 Sat 15:46]
***** KILL Renommer les chromosome avant ...
CLOSED: [2023-05-01 Mon 09:14] SCHEDULED: <2023-04-30 Sun>
Trop long !
- Téléchargement de CADD: 4h20
- renommer les chromosome pour SNV : 6h20
- tabix sur les SNV : job tué au bout de 21h....
***** DONE annoter séparément et fusionner les tableaux
CLOSED: [2023-05-07 Sun 14:45] SCHEDULED: <2023-05-01 Mon>
NB: on pourrait filtrer CADD avec tabix pour se restreindre à nos variants
**** DONE clinvar
CLOSED: [2023-04-22 Sat 15:31]
**** KILL Vérifier résultats HGVS avec mutalyzer
CLOSED: [2023-05-01 Mon 09:26]
**** HOLD Parallélisation
***** HOLD par chromosome avec workflow VEP
https://github.com/Ensembl/ensembl-vep/blob/release/109/nextflow/workflows/run_vep.nf
***** HOLD Avec option --fork
**** DONE Utiliser la version de nf-core de VEP
CLOSED: [2023-05-13 Sat 18:27] SCHEDULED: <2023-05-07 Sun>
**** DONE OMIM
CLOSED: [2023-08-31 Thu 10:38] SCHEDULED: <2023-08-29 Tue>
**** DONE plI et LOEUF depuis gnomad
CLOSED: [2023-08-31 Thu 10:38] SCHEDULED: <2023-08-29 Tue>
**** DONE Grantham
CLOSED: [2023-08-31 Thu 22:08] SCHEDULED: <2023-08-30 Wed>
**** DONE Corriger spliceAI
CLOSED: [2023-08-31 Thu 13:51] SCHEDULED: <2023-08-31 Thu>
Pas d'annotation
- chromosome ? essai 1 au lieu de chr1 : idem. Et fonctionne pour CADD
- index ?
  - retélécharger
  - indexer nous-meme
**** DONE Supprimer score spip en double
CLOSED: [2023-08-31 Thu 14:17] SCHEDULED: <2023-08-31 Thu>
**** DONE Vérifier variant 63126867
CLOSED: [2023-08-31 Thu 10:52] SCHEDULED: <2023-08-31 Thu>
**** DONE Ajouter tronquant ou non
CLOSED: [2023-08-31 Thu 22:08] SCHEDULED: <2023-08-31 Thu>
**** DONE Ajouter récessif
CLOSED: [2023-08-31 Thu 22:08] SCHEDULED: <2023-08-31 Thu>
**** KILL Corriger allelic depth
CLOSED: [2023-08-31 Thu 11:18] SCHEDULED: <2023-08-31 Thu>
Problème lié à libre office
**** DONE Regénérer annotation pour na12878, inserted et patient PEX1
CLOSED: [2023-08-31 Thu 22:08] SCHEDULED: <2023-08-31 Thu>
**** TODO ACMG incidental
**** DONE Sortie VCF (pour avoir la fraction allélique AF)
CLOSED: [2023-08-28 Mon 17:22]
**** DONE VCF -> tsv avec bcftools
CLOSED: [2023-08-29 Tue 11:03] SCHEDULED: <2023-08-28 Mon>
**** DONE Un seul transcrit après VEP avec filter_vep :filter:
CLOSED: [2023-08-29 Tue 11:03] SCHEDULED: <2023-08-28 Mon>
Avec mise à jour VEP 110, pick_flag semble fonctionner.
***** DONE Test chr20: Pas de variant "perdus"
CLOSED: [2023-08-28 Mon 17:31] SCHEDULED: <2023-08-28 Mon>
contrairement au résultat communiqué à alexis par mail

#+begin_src sh :dir out/annotate
bcftools +counts vep/NA12878-sanger-chr20-GRCh38/NA12878-sanger-chr20-GRCh38.vep.vcf.gz
#+end_src

Number of samples: 1
Number of SNPs:    123
Number of INDELs:  32
Number of MNPs:    53
Number of others:  0
Number of sites:   208

#+begin_src  sh
filter_vep -i vep/NA12878-sanger-chr20-GRCh38/NA12878-sanger-chr20-GRCh38.vep.vcf.gz  --filter 'PICK' | bcftools +counts
#+end_src
Number of samples: 1
Number of SNPs:    123
Number of INDELs:  32
Number of MNPs:    53
Number of others:  0
Number of sites:   208

2nd vérification

#+begin_src sh :dir out/annotate
 filter_vep -i vep/NA12878-sanger-chr20-GRCh38/NA12878-sanger-chr20-GRCh38.vep.vcf.gz  --filter 'PICK' --soft_filter | grep fail
#+end_src
***** DONE Test NA12878 + variants sanger : variants perdus avec --pick ?
CLOSED: [2023-08-29 Tue 10:36] SCHEDULED: <2023-08-28 Mon>

#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/out/annotate
~/.nix-profile/bin/bcftools +counts vep/NA12878-sanger-all-GRCh38/NA12878-sanger-all-GRCh38.vep.vcf.gz
#+end_src

#+RESULTS:
| Number | of | samples: |    1 |
| Number | of | SNPs:    | 6293 |
| Number | of | INDELs:  | 1515 |
| Number | of | MNPs:    | 1588 |
| Number | of | others:  |    0 |
| Number | of | sites:   | 9322 |

#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/out/annotate
~/.nix-profile/bin/filter_vep -i vep/NA12878-sanger-all-GRCh38/NA12878-sanger-all-GRCh38.vep.vcf.gz  --filter 'PICK' | bcftools +counts
#+end_src

| Number | of | samples: |    1 |
| Number | of | SNPs:    | 6293 |
| Number | of | INDELs:  | 1515 |
| Number | of | MNPs:    | 1588 |
| Number | of | others:  |    0 |
| Number | of | sites:   | 9322 |


***** DONE Test NA12878 + variants sanger: vérifier sortie avec julia : ok
CLOSED: [2023-08-29 Tue 10:21] SCHEDULED: <2023-08-28 Mon>
143 variants/146 comme avant
***** DONE Relancer en T2T pour vérifier compatibilité :T2T:
CLOSED: [2023-08-29 Tue 11:03] SCHEDULED: <2023-08-29 Tue>

**** DONE Repasser les tests sanger sur NA12878
CLOSED: [2023-09-01 Fri 10:32] SCHEDULED: <2023-08-31 Thu>
2 variants manquants après filter vep

**** DONE Choisir le meilleur transcript nous-meme
CLOSED: [2023-09-01 Fri 10:32] SCHEDULED: <2023-09-01 Fri>
**** DONE Vérifier T2T passe
CLOSED: [2023-08-31 Thu 22:10] SCHEDULED: <2023-08-31 Thu>
**** DONE Revoir choix du transcrit + filtre  avec paul
CLOSED: [2023-09-08 Fri 22:46] SCHEDULED: <2023-09-06 Wed>
**** DONE Filtrer les variants selon les filtres d'Alexis et garder tous les résultat
CLOSED: [2023-09-10 Sun 15:39] SCHEDULED: <2023-09-09 Sat>
**** DONE Ajout colonne MANE SELECT et garder les autres
CLOSED: [2023-09-10 Sun 15:39] SCHEDULED: <2023-09-09 Sat>

**** DONE v1.0
CLOSED: [2023-09-11 Mon 19:11] SCHEDULED: <2023-09-09 Sat>
***** DONE Branche prod
CLOSED: [2023-09-10 Sun 15:44] SCHEDULED: <2023-09-09 Sat>
Merge depuis debug
***** DONE Mail alexis
CLOSED: [2023-09-01 Fri 10:32] SCHEDULED: <2023-08-31 Thu>
***** DONE Relancer test sanger
CLOSED: [2023-09-11 Mon 19:11] SCHEDULED: <2023-09-10 Sun>
***** DONE Mail Paul pour validation
CLOSED: [2023-09-11 Mon 19:11] SCHEDULED: <2023-09-10 Sun>

**** DONE Utiliser spliceAI >= 0.2 pour filtre au lieu de spip
CLOSED: [2023-09-11 Mon 21:48] SCHEDULED: <2023-09-11 Mon>
**** DONE Repasser tests sanger avec spliceAI
CLOSED: [2023-09-14 Thu 22:45] SCHEDULED: <2023-09-11 Mon>
**** DONE Corriger colonne récessive
CLOSED: [2023-09-14 Thu 22:57] SCHEDULED: <2023-09-14 Thu>
soit 1/1, soit 1/2
soit 0/1 avec 2 variants par gene
*** KILL Comparer les annotations sur 63003856
CLOSED: [2023-08-28 Mon 17:28]
**** Relancer le nouveau pipeline
*** KILL Ancienne version
CLOSED: [2023-08-28 Mon 17:24]
**** KILL HGVS
CLOSED: [2023-08-28 Mon 17:24]
**** KILL Filtrer après VEP
CLOSED: [2023-08-28 Mon 17:24]
**** KILL OMIM
CLOSED: [2023-08-28 Mon 17:24]
**** KILL clinvar
CLOSED: [2023-08-28 Mon 17:24]
**** KILL ACMG incidental
CLOSED: [2023-08-28 Mon 17:24]
**** KILL Grantham
CLOSED: [2023-08-28 Mon 17:24]
**** KILL LRG
CLOSED: [2023-04-18 mar. 17:22] SCHEDULED: <2023-04-18 Tue>
Vu avec alexis, n’est plus à jour
**** KILL Gnomad
CLOSED: [2023-08-28 Mon 17:24]

*** DONE Réordonner les colonnes :annotation:
CLOSED: [2023-08-31 Thu 10:38] SCHEDULED: <2023-08-28 Mon>
Pas d'OMIM, pas de CADD, pas de spliceAI
*** DONE Ajouter gnomAD v3 :gnomadv3:
CLOSED: [2023-10-01 Sun 15:34] SCHEDULED: <2023-09-29 Fri>
/Entered on/ [2023-09-29 Fri 22:38]
Après discussion avec Mathieu sur le problème de certaines régions corrigées dans la v3 !
VEP utilise la v2 pour les exomes et v3 pour génomes
Il manquera pour les patients 1-107

On ne l'a pas dans la sortie de VEP jusque là
**** DONE Test sur 1 patient
CLOSED: [2023-09-30 Sat 23:54] SCHEDULED: <2023-09-29 Fri>
**** DONE Reprendre run batch
CLOSED: [2023-10-01 Sun 15:34]
** DONE Porter exactement la version d'Alexis sur Helios
CLOSED: [2023-01-14 Sat 17:56]
Branche "prod"
** KILL Tester version d'alexis avec Nix
CLOSED: [2023-06-14 Wed 22:37]
*** DONE Ajouter clinvar
CLOSED: [2022-11-13 Sun 19:37]
*** DONE Alignement
CLOSED: [2022-11-13 Sun 12:52]
*** DONE Haplotype caller
CLOSED: [2022-11-13 Sun 13:00]
*** KILL Filter
CLOSED: [2023-06-14 Wed 22:37]
- [X] depth
- [ ] comon snp not path
Problème avec liste des ID
**** KILL variant annotation
CLOSED: [2023-06-14 Wed 22:37]
Besoin de vep
*** KILL Variant calling
CLOSED: [2023-06-14 Wed 22:37]
** KILL Tester sarek
CLOSED: [2023-08-12 Sat 15:53]
#+begin_src sh
 module load apptainer/1.1.8
 nextflow run nf-core/sarek -profile test,singularity --outdir test-sarek
#+end_src
Les dépendences ne se téléchargent pas correctement, on les extrait à la main
#+begin_src sh

 rg -IN galaxyproject modules  | sed 's/ //g;s/:$//' | sort | uniq > deps.txt
#+end_src
 Nettoyage à la main
 Puis
 #+begin_src sh
 cat deps.txt | xargs -L1 singularity pull
 #+end_src
** DONE Support pour samplesheet
CLOSED: [2023-08-03 Thu 14:24] SCHEDULED: <2023-08-03 Thu 13:00>
/Entered on/ [2023-08-03 Thu 13:12]
** DONE Petit jeu de données : chr22 sur HG001
CLOSED: [2023-08-05 Sat 14:21] SCHEDULED: <2023-08-05 Sat>
** DONE Corriger OMIM annotation: manquant pour NMNAT1
CLOSED: [2023-09-16 Sat 22:47] SCHEDULED: <2023-09-16 Sat>
/Entered on/ [2023-09-16 Sat 19:32]
** PROJ Regarder la profondeur des variants rendus
/Entered on/ [2023-10-05 Thu 21:44]
* Documentation
:PROPERTIES:
:CATEGORY: doc
:END:
** DONE Procédure d'installation nix + dependences pour VM CHU
CLOSED: [2023-04-22 Sat 15:27] SCHEDULED: <2023-04-13 Thu>
* Bibliographie
** DONE Finir[cite:@alser2021]
CLOSED: [2023-09-26 Tue 11:26] SCHEDULED: <2023-09-22 Fri>
* Manuscript
:PROPERTIES:
:CATEGORY: manuscript
:END:
** DONE Flowchart pipeline (avec T2T)
CLOSED: [2023-09-17 Sun 23:15] SCHEDULED: <2023-09-17 Sun>
** Aligneur
*** DONE Figure: nombre de publication par aligneur
CLOSED: [2023-09-19 Tue 16:54] SCHEDULED: <2023-09-19 Tue>
/Entered on/ [2023-09-19 Tue 08:43]
*** DONE Biblio performance aligneur <(biblio aligneur)> <(aligneur)>
CLOSED: [2023-10-13 Fri 17:40] SCHEDULED: <2023-10-01 Sun>
*** DONE Figure: nombre d'articles citant les principaux aligneur par année
CLOSED: [2023-10-11 Wed 23:54] SCHEDULED: <2023-10-03 Tue>
Il faudrait utiliser pubmed en local, sinon c'est 10 000 requete par aligner !
*** DONE Figure: nombre d'articles citant les principaux aligneur
CLOSED: [2023-10-12 Thu 23:58] SCHEDULED: <2023-10-12 Thu>
Il faudrait utiliser pubmed en local, sinon c'est 10 000 requete par aligner !
On se base sur

** Appel de variant
*** TODO Biblio <(biblio appel variant)> <(appel variant)>
SCHEDULED: <2023-10-18 Wed>
*** TODO Figure: nombre de publication par appel de variant
SCHEDULED: <2023-10-21 Sat>
/Entered on/ [2023-09-19 Tue 08:43]
** TODO Figure: nombre d'exomes par années
SCHEDULED: <2023-10-25 Wed>
/Entered on/ [2023-09-19 Tue 08:43]
* Tests :tests:
** KILL Non régression : version prod
CLOSED: [2023-05-23 Tue 08:46]
*** DONE ID common snp
CLOSED: [2022-11-19 Sat 21:36]
#+begin_src
$ wc -l ID_of_common_snp.txt
23194290 ID_of_common_snp.txt
$ wc -l /Work/Users/apraga/bisonex/database/dbSNP/ID_of_common_snp.txt
23194290 /Work/Users/apraga/bisonex/database/dbSNP/ID_of_common_snp.txt
#+end_src
*** DONE ID common snp not clinvar patho
CLOSED: [2022-12-11 Sun 20:11]
**** DONE Vérification du problème
CLOSED: [2022-12-11 Sun 16:30]
Sur le J:
21155134 /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/ID_of_common_snp_not_clinvar_patho.txt.ref

Version de "non-régression"
21155076 database/dbSNP/ID_of_common_snp_not_clinvar_patho.txt

Nouvelle version
23193391 /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/ID_of_common_snp_not_clinvar_patho.txt

Si on enlève les doublons
$ sort database/dbSNP/ID_of_common_snp_not_clinvar_patho.txt | uniq > old.txt
$ wc -l old.txt
21107097 old.txt

$ sort /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/ID_of_common_snp_not_clinvar_patho.txt | uniq > new.txt
$ wc -l new.txt
21174578 new.txt

$ sort /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/ID_of_common_snp_not_clinvar_patho.txt.ref | uniq > ref.txt
$ wc -l ref.txt
21107155 ref.txt

Si on regarde la différence
 comm -23 ref.txt old.txt
rs1052692
rs1057518973
rs1057518973
rs11074121
rs112848754
rs12573787
rs145033890
rs147889095
rs1553904159
rs1560294695
rs1560296615
rs1560310926
rs1560325547
rs1560342418
rs1560356225
rs1578287542
...

On cherche le premier
bcftools query -i 'ID="rs1052692"' database/dbSNP/dbSNP_common.vcf.gz -f '%CHROM %POS %REF %ALT\n'
NC_000019.10 1619351 C A,T
Il est bien patho...
$ bcftools query -i 'POS=1619351' database/clinvar/clinvar.vcf.gz -f '%CHROM %POS %REF %ALT %INFO/CLNSIG\n'
19 1619351 C T Conflicting_interpretations_of_pathogenicity

On vérifie pour tous les autres

$ comm -23 ref.txt old.txt > tocheck.txt
On génère les régions à vérifier (chromosome number:position)
$ bcftools query -i 'ID=@tocheck.txt' database/dbSNP/dbSNP_common.vcf.gz -f '%CHROM\t%POS\n' > tocheck.pos

On génère le mapping inverse (chromosome number -> NC)
$ awk ' { t = $1; $1 = $2; $2 = t; print; } ' database/RefSeq/refseq_to_number_only_consensual.txt  > mapping.txt
On remap clinvar
$ bcftools annotate --rename-chrs mapping.txt database/clinvar/clinvar.vcf.gz -o clinvar_remapped.vcf.gz
$ tabix clinvar_remapped.vcf.gz

Enfin, on cherche dans clinvar la classification
$ bcftools query -R tocheck.pos clinvar_remapped.vcf.gz -f '%CHROM %POS %INFO/CLNSIG\n'
$ bcftools query -R tocheck.pos database/dbSNP/dbSNP_common.vcf.gz -f '%CHROM %POS %ID \n' | grep '^NC'
#+RESULTS:
**** DONE Comprendre pourquoi la nouvelle version donne un résultat différent
CLOSED: [2022-12-11 Sun 20:11]
***** DONE Même version dbsnp et clinvar ?
CLOSED: [2022-12-10 Sat 23:02]
Clinvar différent !
  $ bcftools stats clinvar.gz
  clinvar (Alexis)
SN	0	number of samples:	0
SN	0	number of records:	1492828
SN	0	number of no-ALTs:	965
SN	0	number of SNPs:	1338007
SN	0	number of MNPs:	5562
SN	0	number of indels:	144580
SN	0	number of others:	3714
SN	0	number of multiallelic sites:	0
SN	0	number of multiallelic SNP sites:	0
clinvar (new)
SN	0	number of samples:	0
SN	0	number of records:	1493470
SN	0	number of no-ALTs:	965
SN	0	number of SNPs:	1338561
SN	0	number of MNPs:	5565
SN	0	number of indels:	144663
SN	0	number of others:	3716
SN	0	number of multiallelic sites:	0
SN	0	number of multiallelic SNP sites:	0
***** DONE Mettre à jour clinvar et dbnSNP pour travailler sur les mêm bases
CLOSED: [2022-12-11 Sun 12:10]
Problème persiste
***** DONE Supprimer la conversion en int du chromosome
CLOSED: [2022-12-10 Sat 19:29]
***** KILL Même NC ?
CLOSED: [2022-12-10 Sat 19:29]
$  zgrep "contig=<ID=NC_\(.*\)" clinvar/GRCh38/clinvar.vcf.gz > contig.clinvar
$ diff contig.txt contig.clinvar
< ##contig=<ID=NC_012920.1>
***** DONE Tester sur chromosome 19: ok
CLOSED: [2022-12-11 Sun 13:53]
On prépare les données
#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/tests/debug-commonsnp
PATH=$PATH:$HOME/.nix-profile/bin
bcftools filter -i 'CHROM="NC_000019.10"' /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/dbSNP_common.vcf.gz -o dbSNP_common_19.vcf.gz
bcftools filter -i 'CHROM="NC_000019.10"' /Work/Groups/bisonex/data/clinvar/GRCh38/clinvar.vcf.gz -o clinvar_19.vcf.gz
bcftools filter -i 'CHROM="NC_000019.10"' /Work/Groups/bisonex/data-alexis/dbSNP/dbSNP_common.vcf.gz -o dbSNP_common_19_old.vcf.gz
 bcftools filter -i 'CHROM="19"' /Work/Groups/bisonex/data-alexis/clinvar/clinvar.vcf.gz -o clinvar_19_old.vcf.gz
#+end_src

On récupère les 2 versions du script
#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/tests/debug-commonsnp
PATH=$PATH:$HOME/.nix-profile/bin
git checkout regression ../../script/pythonScript/clinvar_sbSNP.py
cp ../../script/pythonScript/clinvar_sbSNP.py clinvar_sbSNP_old.py
git checkout HEAD ../../script/pythonScript/clinvar_sbSNP.py
#+end_src

#+RESULTS:

On compare
#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/tests/debug-commonsnp
PATH=$PATH:$HOME/.nix-profile/bin
python ../../script/pythonScript/clinvar_sbSNP.py clinvar_sbSNP.py --clinvar clinvar_19.vcf.gz --dbSNP dbSNP_common_19.vcf.gz --output tmp.txt
sort tmp.txt | uniq > new.txt
table=/Work/Groups/bisonex/data-alexis/RefSeq/refseq_to_number_only_consensual.txt
python clinvar_sbSNP_old.py --clinvar clinvar_19_old.vcf.gz --dbSNP dbSNP_common_19_old.vcf.gz --output tmp_old.txt --chrm_name_table $table
sort tmp_old.txt | uniq > old.txt
wc -l old.txt new.txt
#+end_src

#+RESULTS:
|  535155 | old.txt |
|  535194 | new.txt |
| 1070349 | total   |

Si on prend le premier manquant dans new, il est conflicting patho donc il ne devrait pas y être...

$ bcftools query -i 'ID="rs10418277"' dbSNP
_common_19.vcf.gz  -f '%CHROM %POS %REF %ALT\n'
NC_000019.10 54939682 C G,T
$ bcftools query -i 'ID="rs10418277"' dbSNP_common_19_old.vcf.gz  -f '%CHROM %POS %REF %ALT\n'
NC_000019.10 54939682 C G,T

$ bcftools query -i 'POS=54939682' clinvar_19.vcf.gz  -f '%POS %REF %ALT %INFO/CLNSIG\n'
54939682 C G Conflicting_interpretations_of_pathogenicity
54939682 C T Benign
$ bcftools query -i 'POS=54939682' clinvar_19_old.vcf.gz  -f '%POS %REF %ALT %INFO/CLNSIG\n'
54939682 C G Conflicting_interpretations_of_pathogenicity
54939682 C T Benign

$ grep rs10418277 *.txt
new.txt:rs10418277
tmp.txt:rs10418277

Le problème venait de la POS qui n'était plus convertie en int (suppression de la ligne par erreur ??)
On vérifie

#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/tests/debug-commonsnp
PATH=$PATH:$HOME/.nix-profile/bin
python ../../script/pythonScript/clinvar_sbSNP.py --clinvar clinvar_19.vcf.gz --dbSNP dbSNP_common_19.vcf.gz --output tmp.txt
sort tmp.txt | uniq > new.txt
table=/Work/Groups/bisonex/data-alexis/RefSeq/refseq_to_number_only_consensual.txt
python clinvar_sbSNP_old.py --clinvar clinvar_19_old.vcf.gz --dbSNP dbSNP_common_19_old.vcf.gz --output tmp_old.txt --chrm_name_table $table
sort tmp_old.txt | uniq > old.txt
wc -l old.txt new.txt
diff old.txt new.txt
#+end_src

#+RESULTS:
|  535155 | old.txt |
|  535155 | new.txt |
| 1070310 | total   |

***** DONE Tester sur chromosome 19 et 20: ok
CLOSED: [2022-12-11 Sun 15:56]
On prépare les données
#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/tests/debug-commonsnp
PATH=$PATH:$HOME/.nix-profile/bin
bcftools filter -i 'CHROM="NC_000019.10" | CHROM="NC_000020.11"' /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/dbSNP_common.vcf.gz -o dbSNP_common_19_20.vcf.gz
bcftools filter -i 'CHROM="NC_000019.10" | CHROM="NC_000020.11"' /Work/Groups/bisonex/data/clinvar/GRCh38/clinvar.vcf.gz -o clinvar_19_20.vcf.gz
bcftools filter -i 'CHROM="NC_000019.10" | CHROM="NC_000020.11"' /Work/Groups/bisonex/data-alexis/dbSNP/dbSNP_common.vcf.gz -o dbSNP_common_19_20_old.vcf.gz
bcftools filter -i 'CHROM="19" | CHROM="20"' /Work/Groups/bisonex/data-alexis/clinvar/clinvar.vcf.gz -o clinvar_19_20_old.vcf.gz
#+end_src

#+RESULTS:

On récupère les 2 versions du script
#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/tests/debug-commonsnp
PATH=$PATH:$HOME/.nix-profile/bin
git checkout regression ../../script/pythonScript/clinvar_sbSNP.py
cp ../../script/pythonScript/clinvar_sbSNP.py clinvar_sbSNP_old.py
git checkout HEAD ../../script/pythonScript/clinvar_sbSNP.py
#+end_src

#+RESULTS:

On compare
#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/tests/debug-commonsnp
PATH=$PATH:$HOME/.nix-profile/bin
python ../../script/pythonScript/clinvar_sbSNP.py clinvar_sbSNP.py --clinvar clinvar_19_20.vcf.gz --dbSNP dbSNP_common_19_20.vcf.gz --output tmp.txt
sort tmp.txt | uniq > new.txt
table=/Work/Groups/bisonex/data-alexis/RefSeq/refseq_to_number_only_consensual.txt
python clinvar_sbSNP_old.py --clinvar clinvar_19_20_old.vcf.gz --dbSNP dbSNP_common_19_20_old.vcf.gz --output tmp_old.txt --chrm_name_table $table
sort tmp_old.txt | uniq > old.txt
wc -l old.txt new.txt
#+end_src

***** DONE Regarder la répartition des différences
CLOSED: [2022-12-11 Sun 16:29]
#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/tests/debug-commonsnp
sort /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/ID_of_common_snp_not_clinvar_patho.txt  | uniq > notpatho.new
sort /Work/Groups/bisonex/data-alexis/dbSNP/ID_of_common_snp_not_clinvar_patho.txt  | uniq > notpatho.old
comm -23 notpatho.new notpatho.old > nopatho.diff
#+end_src
#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/tests/debug-commonsnp
PATH=$PATH:$HOME/.nix-profile/bin
 bcftools query -i 'ID=@nopatho.diff' /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/dbSNP_common.vcf.gz -f '%CHROM\n' | sort | uniq -c
 #+end_src

 On a principalement des coordonnées non consensuelles (non "NC_", voir notes)

 #+RESULTS:
  :     2 NC_000002.12
  :    18 NC_000003.12
  :     2 NC_000004.12
  :     2 NC_000005.10
  :    14 NC_000006.12
  :     6 NC_000007.14
  :     2 NC_000009.12
  :     1 NC_000010.11
  :     6 NC_000014.9
  :     1 NC_000015.10
  :     3 NC_000016.10
  :     3 NC_000017.11
  :     1 NC_000019.10
  :     1 NC_000020.11
  :     1 NC_000021.9
  :     2 NC_000022.11
  : 16018 NT_113793.3
  : 17010 NT_113796.3
  :    14 NT_113891.3
  :     1 NT_167244.2
  :    13 NT_167245.2
  :     2 NT_167246.2
  :    13 NT_167247.2
  :     7 NT_167248.2
  :    14 NT_167249.2
  : 14857 NT_187361.1
  :    92 NT_187367.1
  :     1 NT_187369.1
  :    13 NT_187381.1
  :    54 NT_187383.1
  :     6 NT_187499.1
  :    46 NT_187502.1
  : 13754 NT_187513.1
  :   611 NT_187517.1
  :     1 NT_187520.1
  :     1 NT_187524.1
  :   249 NT_187526.1
  :    18 NT_187532.1
  :     1 NT_187546.1
  :   886 NT_187562.1
  :     1 NT_187564.1
  :   346 NT_187576.1
  :    13 NT_187600.1
  :     5 NT_187601.1
  :   494 NT_187606.1
  :     1 NT_187607.1
  :    12 NT_187613.1
  :   307 NT_187614.1
  :     1 NT_187625.1
  :   445 NT_187633.1
  :    43 NT_187648.1
  :    18 NT_187649.1
  :     1 NT_187652.1
  :   512 NT_187661.1
  :    18 NT_187678.1
  :    49 NT_187681.1
  :     1 NT_187682.1
  :    18 NT_187688.1
  :    12 NT_187689.1
  :    18 NT_187690.1
  :    18 NT_187691.1
  :   404 NT_187693.1
  :     2 NW_003315952.3
  :     1 NW_003315970.2
  :   203 NW_003571054.1
  :   322 NW_003571055.2
  :    16 NW_003571056.2
  :    16 NW_003571057.2
  :    16 NW_003571058.2
  :    16 NW_003571059.2
  :    16 NW_003571060.1
  :   213 NW_003571061.2
  :     2 NW_009646201.1
  :   322 NW_009646205.1
  :   321 NW_009646206.1
  :   371 NW_012132914.1
  :     1 NW_012132915.1
  :    13 NW_012132918.1
  :     2 NW_013171801.1
  :     1 NW_013171807.1
  :    49 NW_015148966.1
  :    14 NW_015495298.1
  :     2 NW_015495299.1
  :     1 NW_016107298.1
  :     4 NW_017363813.1
  :     2 NW_017852933.1
  :     1 NW_018654722.1
  :    38 NW_021160001.1
  :     1 NW_021160003.1
  :     1 NW_021160007.1
  :     7 NW_021160017.1

***** DONE Regarder la différence avec la version sans les sites non consensuels: ok !
CLOSED: [2022-12-11 Sun 20:11]
#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/tests/debug-commonsnp
sort /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/ID_of_common_snp_not_clinvar_patho.txt  | uniq > notpatho.new
sort /Work/Groups/bisonex/data-alexis/dbSNP/ID_of_common_snp_not_clinvar_patho.txt  | uniq > notpatho.old
comm -13 notpatho.new notpatho.old > notpatho.diff
wc -l
#+end_src

#+RESULTS:
: 528 notpatho.diff

Il manque 528 variants
rs1057520103
#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/tests/debug-commonsnp
PATH=$PATH:$HOME/.nix-profile/bin
 bcftools query -i 'ID=@notpatho.diff' /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/dbSNP_common.vcf.gz -f '%CHROM\n' | sort | uniq -c
 #+end_src

 #+RESULTS:
 : 528 NC_012920.1

 Donc la nouvelle version fonctionne mieux !
 ON vérifie bien qu'ils sont dans l'ancienne version et la nouvelle:
$ grep -w -f notpatho.diff /Work/Groups/bisonex/data-alexis/dbSNP/ID_of_common_snp_not_clinvar_patho.txt  | wc -l
528
$ grep -w -f notpatho.diff  /Work/Groups/bisonex/data/d
bSNP/GRCh38.p13/ID_of_common_snp_not_clinvar_patho.txt
#+end_src
**** DONE Supprimer les sites non consensuels
CLOSED: [2022-12-11 Sun 19:51]
**** DONE Rajouter les mitochondries (vu avec Paul)
CLOSED: [2022-12-13 Tue 17:26]
Ok avec notre version générée. Sur le J: 21155134

$ wc -l dbSNP/GRCh38.p13/ID_of_common_snp_not_clinvar_patho.txt
21155065 dbSNP/GRCh38.p13/ID_of_common_snp_not_clinvar_patho.txt
$ wc -l ../data-alexis/dbSNP/ID_of_common_snp_not_clinvar_patho.txt
21155065 ../data-alexis/dbSNP/ID_of_common_snp_not_clinvar_patho.txt
La différence vient probablement d'une vieille version de clinvar
**** KILL Comprendre la différence nouvelle version et prod
CLOSED: [2023-05-23 Tue 08:46]
#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/tests/debug-commonsnp
#sort /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/ID_of_common_snp_not_clinvar_patho.txt  | uniq > notpatho.new
#sort /Work/Groups/bisonex/data-alexis/dbSNP/ID_of_common_snp_not_clinvar_patho.txt  | uniq > notpatho.old
comm -13 notpatho.new notpatho.old > missing-from-old
comm -23 notpatho.new notpatho.old > missing-from-new
wc -l missing-from-old
wc -l missing-from-new
#+end_src

#+RESULTS:
| 75 | missing-from-old |
|  6 | missing-from-new |

Il manque 75 variants et on a 6 en trop
#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/tests/debug-commonsnp
PATH=$PATH:$HOME/.nix-profile/bin
 bcftools query -i 'ID=@missing-from-old' /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/dbSNP_common.vcf.gz -f '%CHROM\n' | sort | uniq -c
 #+end_src

 #+RESULTS:
 | 16 | NC_000001.11   |
 |  2 | NC_000002.12   |
 | 18 | NC_000003.12   |
 |  7 | NC_000004.12   |
 |  1 | NC_000005.10   |
 |  5 | NC_000006.12   |
 |  3 | NC_000007.14   |
 |  2 | NC_000009.12   |
 |  1 | NC_000010.11   |
 |  5 | NC_000011.10   |
 |  3 | NC_000015.10   |
 |  1 | NC_000016.10   |
 |  4 | NC_000017.11   |
 |  2 | NC_000019.10   |
 |  1 | NC_000020.11   |
 |  3 | NC_000022.11   |
 |  1 | NC_000023.11   |
 |  2 | NT_113891.3    |
 |  2 | NT_167245.2    |
 |  2 | NT_167247.2    |
 |  1 | NT_167248.2    |
 |  2 | NT_167249.2    |
 | 18 | NT_187532.1    |
 |  1 | NT_187562.1    |
 |  1 | NT_187633.1    |
 | 18 | NT_187649.1    |
 | 18 | NT_187678.1    |
 | 18 | NT_187688.1    |
 | 12 | NT_187689.1    |
 | 18 | NT_187690.1    |
 | 18 | NT_187691.1    |
 |  1 | NW_013171807.1 |

 Donc la nouvelle version fonctionne mieux !
 ON vérifie bien qu'ils sont dans l'ancienne version et la nouvelle:
$ grep -w -f notpatho.diff /Work/Groups/bisonex/data-alexis/dbSNP/ID_of_common_snp_not_clinvar_patho.txt  | wc -l
528
$ grep -w -f notpatho.diff  /Work/Groups/bisonex/data/d
bSNP/GRCh38.p13/ID_of_common_snp_not_clinvar_patho.txt
#+end_src
*** DONE Comparer les versions
CLOSED: [2023-01-26 jeu. 17:42]
**** DONE bases de données
CLOSED: [2023-01-26 jeu. 17:42]
***** DONE Genome de référénce:
CLOSED: [2023-01-06 Fri 00:00]
Version calculée
$ find . -type f -name "genomeRef.*" -exec sh -c 'echo {}; sha256sum {}' \;
e0761a7ba5d10de9e7e97fa331667963925531c0199575bcceafbb13c3147e3f  ./genomeRef.fna
d121084c35037763ea58c59726545eaa1c11025a7bf2d75634677c72ddb72fd1  ./genomeRef.dict
0a6e215314659929dbcdffc1881714e311e3d149bdc33978a6f5e28206fcc675  ./genomeRef.fna.fai
45a4aa0d8dc1095d090b13e4df180763b2e24d133ec81f026beaead6d41ebafc  ./bwa/genomeRef.ann
ac6e465f230da6d9f8339ebdf4cb05bcfc47d03f3a3889cae7256983c7809210  ./bwa/genomeRef.bwt
058ffaf8cd38e7bc33c31e86e54e99869a8a4fbabb6737d7420d6b89e8b5988e  ./bwa/genomeRef.sa
f665b64275eb76111463966bcb8e91e550c63c9b58263d43e19bae8552be2815  ./bwa/genomeRef.amb
d0d3731d1203cb4a0d0dd1279c37c85480b85a91da2d1dc543ac391ff927c272  ./bwa/genomeRef.pac

Version de référence
$ find . -type f -exec sh -c 'echo {}; sha256sum {}' \;

e0761a7ba5d10de9e7e97fa331667963925531c0199575bcceafbb13c3147e3f  ./GRCh38_latest_genomic.fna
dd87d628a8179fc1e37a33b99101eece8282bec88dc17b1998a07ff0b912d4a3  ./GRCh38_latest_genomic.fna.index
0a6e215314659929dbcdffc1881714e311e3d149bdc33978a6f5e28206fcc675  ./GRCh38_latest_genomic.fna.fai
974cc313146aedd9ec2ae3f86b382b1317180e078621c1d53e8a803d4ec0d3a9  ./GRCh38_latest_genomic.dict
ac6e465f230da6d9f8339ebdf4cb05bcfc47d03f3a3889cae7256983c7809210  ./GRCh38_latest_genomic.fna.bwt
45a4aa0d8dc1095d090b13e4df180763b2e24d133ec81f026beaead6d41ebafc  ./GRCh38_latest_genomic.fna.ann
f665b64275eb76111463966bcb8e91e550c63c9b58263d43e19bae8552be2815  ./GRCh38_latest_genomic.fna.amb
058ffaf8cd38e7bc33c31e86e54e99869a8a4fbabb6737d7420d6b89e8b5988e  ./GRCh38_latest_genomic.fna.sa
d0d3731d1203cb4a0d0dd1279c37c85480b85a91da2d1dc543ac391ff927c272  ./GRCh38_latest_genomic.fna.pac

Dict ok si on renome le ficdhier d'origine
#+begin_src sh :dir /ssh:meso:/Work/Groups/bisonex/
sed 's/UR:.*/UR:genomeRef.fna/' data-alexis-reference/genome/GRCh38_latest_genomic.dict  > lol.dict
diff lol.dict data/genome/GRCh38.p13/genomeRef.dict
#+end_src
#+RESULTS:
***** DONE dbSNP et dbSNP common: ok
CLOSED: [2023-01-03 Tue 23:17]
sha256sum GCF_000001405.39.gz
452e1112b6339a9b19821c2a226a8a3ba946e92a47e03e6ae464ef8820ee130d  GCF_000001405.39.gz

sha256sum data-alexis-reference/dbSNP/GCF_000001405.39.gz
452e1112b6339a9b19821c2a226a8a3ba946e92a47e03e6ae464ef8820ee130d  data-alexis-reference/dbSNP/GCF_000001405.39.gz"

sha256sum dbSNP_common.vcf.gz
70dfd9be859c39916598d23b5744cc1fbda04add5840cd90a6d0cd005bd3075b  dbSNP_common.vcf.gz

sha256sum data-alexis-reference/dbSNP/dbSNP_common.vcf.gz
70dfd9be859c39916598d23b5744cc1fbda04add5840cd90a6d0cd005bd3075b  data-alexis-reference/dbSNP/dbSNP_common.vcf.gz
***** DONE Clinvar : version différente
CLOSED: [2023-01-06 Fri 22:51]
$ zgrep -v '^#' data-alexis-reference/clinvar/clinvar.vcf.gz | wc -l
1474547
$ zgrep -v '^#' data/clinvar/GRCh38/clinvar.vcf.gz | wc -l
1571404
***** DONE Revérifer checksum de GRCh38 et dbSNP
CLOSED: [2023-01-26 jeu. 17:42]
****** DONE GRCh38
CLOSED: [2023-01-26 jeu. 17:42]
/Work/Projects/bisonex/data-alexis-reference/genome>sha256sum GRCh38_latest_genomic.fna
e0761a7ba5d10de9e7e97fa331667963925531c0199575bcceafbb13c3147e3f  GRCh38_latest_genomic.fna

PS J:\bases_de_donnees\genome> cat .\checksum.txt
e0761a7ba5d10de9e7e97fa331667963925531c0199575bcceafbb13c3147e3f  GRCh38_latest_genomic.fna
****** DONE dbSNP
CLOSED: [2023-01-26 jeu. 17:42]
/Work/Projects/bisonex/data-alexis-reference/dbSNP>sha256sum GCF_000001405.39.gz                 01/26/2023 04:40:48 PM
452e1112b6339a9b19821c2a226a8a3ba946e92a47e03e6ae464ef8820ee130d  GCF_000001405.39.gz

PS J:\bases_de_donnees\dbSNP> cat .\checksums.txt
452e1112b6339a9b19821c2a226a8a3ba946e92a47e03e6ae464ef8820ee130d  GCF_000001405.39.gz

**** DONE Outils
CLOSED: [2023-01-26 jeu. 17:42]
|          |    Prod |         Test |
| VCFtools |  0.1.17 |       0.1.16 |
| bcftools |    1.14 |         1.16 |
| samtools |    1.14 |         1.13 |
| gatk     | 4.2.4.1 |      4.3.0.0 |
| bwa      |       ? | 0.7.17-r1188 |
On a des versions plus vieilles sauf (le plus important) Gatk

*** KILL 63003856_S135
CLOSED: [2023-05-23 Tue 08:45]
**** KILL Notes : bonne reproductibilité sur le cluster mais diff avec la version "de prod"
CLOSED: [2023-01-09 Mon 23:03]
tester sequential puis version spécifique bwa mem
Note: clinvar est plus ancien dans la version d'Alexis, cela explique les 8884 de la version pseudo-prod (via ID not clinvar patho)

Attention : le nombre de lignes = celles sans commentaires (et pas just NC... car il devrait y avoir les mitochondries !)
$ find . -name filter-depth.vcf -exec sh -c 'echo {}; grep -c -v '^#' {}' \;

Attention, on a un vcf.gz pour la version de test !!! ne pas utiliser le vcf
|                     |       prod |      prod |       prod |       prod |       prod |       test |       test |       test |
|---------------------+------------+-----------+------------+------------+------------+------------+------------+------------|
| PC                  |     Karine |    helios |     helios |     helios |     Helios |     Helios |     Helios |     Helios |
| cores               |          4 |         4 |          4 |         24 |          1 |         24 |         24 |          1 |
| gatk                |    4.2.4.1 |   4.2.4.1 |    4.2.4.1 |    4.2.4.1 |    4.2.4.1 |    4.2.4.1 |      4.3.0 |    4.2.3.1 |
| samtools            |       1.10 |      1.10 |       1.13 |       1.13 |       1.13 |       1.13 |            |            |
| clinvar             | 2022-09-03 |           | 2022-09-03 | 2022-09-03 | 2022-09-03 | 2022-12-03 | 2022-12-03 | 2022-12-03 |
|---------------------+------------+-----------+------------+------------+------------+------------+------------+------------|
| bwa mem             |  128077207 | 128077207 |  128077207 |  128077211 |  128077210 |  128077211 |  128077211 |  128077210 |
| cleanSam            |  128077207 |           |            |            |  128077210 |  128077211 |  128077211 |  128077210 |
| applybqsr           |            |           |            |  128077211 |            |            |            |            |
| haplotypecaller     |    1528059 |   1528066 |    1528066 |    1528093 |    1528089 |    1528093 |    1528093 |    1528093 |
| DP_over_30          |      84708 |     84716 |      84716 |      84724 |      84725 |      84724 |      84724 |      84725 |
| not_SNP             |      11362 |     11366 |      11366 |      11377 |      11384 |         NA |         NA |         NA |
| consensual_sequence |       8864 |      8868 |       8868 |       8884 |       8886 |       8898 |       8898 |       8900 |
| tsv                 |       1087 |           |       1121 |            |            |            |            |            |
***** Convention
post_cleanSam = _cleaned.bam
post_markDuplicate = _marked_dup.bam
post_BaseRecalibrator = _recal.table
post_ApplyBQSR = _recalibrated_hg38.bam
post_haplotypecaller = .vcf
post_depth_filter = _DP_over_30.vcf
post_exclude_SNP = _DP_over_30_not_SNP
post_consensual = _DP_over_30_not_SNP_consensual_sequence.vcf
post_technical = _DP_over_30_not_SNP_consensual_sequence_not_technical.vcf.gz

On a vérifié que grep -c et grep | wc -l donnent le même résultat

#
**** KILL Gatk 4.3.0
CLOSED: [2023-01-04 Wed 19:16]
***** KILL Alignement
CLOSED: [2023-01-04 Wed 19:16]
****** DONE Brut
CLOSED: [2022-12-26 Mon 22:03]
Bam alexis
$ samtools view -c /Work/Groups/bisonex/ref_63003856_S135/63003856_S135.bam
128077207

Notre
9f/26cf3d] Cached process > preprocess:BWA_MEM (63003856_S135)
$ samtools view -c work/9f/26cf3deb07b425a3e851be2a7bd782/63003856_S135.bam

On vérifie la sortie
$ samtools view -c out/63003856_S135/preprocessing/mapped/63003856_S135.bam
128077211

Petite différence (< 1e-8) mais selon Alexis, bwa mem est non reproductible. d'autant qu'on utilise une version parallélisée
128077211

****** On vérifie les arguments: ok
#+begin_src
bwa mem \
    -R '@RG\tID:sample\tSM:sample\tPL:ILLUMINA\tPM:Miseq\tCN:CHU_Minjoz\tLB:definition_to_add' \
    -t 24 \
    $INDEX \
    63003856_S135_R1_001.fastq.gz 63003856_S135_R2_001.fastq.gz \
    | samtools sort  --threads 24 -o 63003856_S135.bam -
#+end_src

#+begin_src
bwa mem -R "@RG\tID:$sample\tSM:$sample\tPL:ILLUMINA\tPM:Miseq\tCN:CHU_Minjoz\tLB:definition_to_add" -v 2 -t `nproc` $genomeRef ${fastq[0]} ${fastq[1]} | $samtools sort -@ `nproc` -O BAM -o $tmpDir/$post_bwa
#+end_src
****** DONE Avec gatk 4.2 (version alexis) : idem
$ samtools view -c mapped/63003856_S135.bam
128077211
****** DONE Nettoyé
CLOSED: [2022-12-26 Mon 22:08]

[57/4b5b4c] Cached process > preprocess:GATK4_CLEANSAM (63003856_S135)
$ samtools view -c work/57/4b5b4c647b98bb7099c4d1ba24bd75/63003856_S135.bam
128077211

Et la sortie
$ samtools view -c out/63003856_S135/preprocessing/clean-sam/sorted.bam
128077211

contre
$ samtools view -c /Work/Groups/bisonex/ref_63003856_S135/63003856_S135_cleaned.bam
128077207

On regarde les stats en détails (de la version nettoyée)
$ samtools  flagstat work/57/4b5b4c647b98bb7099c4d1ba24bd75/63003856_S135.bam
128077211 + 0 in total (QC-passed reads + QC-failed reads)
126905130 + 0 primary
0 + 0 secondary
1172081 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
127941054 + 0 mapped (99.89% : N/A)
126768973 + 0 primary mapped (99.89% : N/A)
126905130 + 0 paired in sequencing
63452565 + 0 read1
63452565 + 0 read2
125263664 + 0 properly paired (98.71% : N/A)
126676024 + 0 with itself and mate mapped
92949 + 0 singletons (0.07% : N/A)
979608 + 0 with mate mapped to a different chr
675398 + 0 with mate mapped to a different chr (mapQ>=5)

$ samtools flagstat /Work/Groups/bisonex/ref_63003856_S135/63003856_S135_cleaned.bam
128077207 + 0 in total (QC-passed reads + QC-failed reads)
126905130 + 0 primary
0 + 0 secondary
1172077 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
127941051 + 0 mapped (99.89% : N/A)
126768974 + 0 primary mapped (99.89% : N/A)
126905130 + 0 paired in sequencing
63452565 + 0 read1
63452565 + 0 read2
125263790 + 0 properly paired (98.71% : N/A)
126676026 + 0 with itself and mate mapped
92948 + 0 singletons (0.07% : N/A)
979618 + 0 with mate mapped to a different chr
675412 + 0 with mate mapped to a different chr (mapQ>=5)
****** DONE mark duplicate
CLOSED: [2022-12-26 Mon 22:27]
Alexis
$ samtools view -c /Work/Groups/bisonex/ref_63003856_S135/63003856_S135_marked_dup.bam
128077207

Nous (pas de sortie dans out/)
$ samtools view -c  work/46/bd75b4547452af36ee2c6b45362922/63003856_S135
128077211

logique car on ne supprime pas de donné...

******* Arguments ok
#+begin_src
gatk --java-options "-Xmx3g" MarkDuplicates \
    --INPUT sorted.bam \
    --OUTPUT marked_dups.bam \
    --METRICS_FILE marked_dups.bam.metrics \
    --TMP_DIR . \
    --REFERENCE_SEQUENCE genomeRef.fna \
#+end_src

#+begin_src
			$gatk MarkDuplicates \
			-I $tmpDir/$post_cleanSam \
			-O $tmpDir/$post_markDuplicate \
			-M $tmpDir/"$sample"_marked_dup.metrix \
			--CREATE_INDEX true \
			--VERBOSITY WARNING
#+end_src
****** KILL baserecalibrator
CLOSED: [2023-01-04 Wed 19:15]
#+begin_src
			$gatk BaseRecalibrator \
			-I $tmpDir/$post_markDuplicate \
			-R $genomeRef \
			--known-sites $dbsnpDir/dbSNP_common.vcf.gz \
			-O $tmpDir/$post_BaseRecalibrator
#+end_src

$ cd /Work/Users/apraga/bisonex/out/63003856_S135/preprocessing/baserecalibrator
$ sed 's/sample/63003856_S135/' 63003856_S135.table  > 63003856_S135.table2

Les fichiers n'ont pas le même nombre d'erreurs mais assez proches. Sur le premier table, 3 score
#:GATKTable:3:94:%d:%d:%d:;
#:GATKTable:Quantized:Quality quantization map
QualityScore  Count        QuantizedScore
          11    298878631              11
          25    542282996              25
          34  12846268833              34

vs (référence0)

          11    298877785              11
          25    542282089              25
          34  12846264839              34
******* options ok
#+begin_src
gatk --java-options "-Xmx3g" BaseRecalibrator  \
    --input marked_dups.bam \
    --output 63003856_S135.table \
    --reference genomeRef.fna \
     \
    --known-sites dbSNP_common.vcf.gz \
    --tmp-dir . \
#+end_src
****** KILL applybqsr
CLOSED: [2023-01-04 Wed 19:15]
options ok
#+begin_src
gatk --java-options "-Xmx3g" ApplyBQSR \
    --input marked_dups.bam \
    --output 63003856_S135.bam \
    --reference genomeRef.fna \
    --bqsr-recal-file 63003856_S135.table \
     \
    --tmp-dir . \
#+end_src
#+begin_src
			$gatk ApplyBQSR -R $genomeRef \
			 -I $tmpDir/$post_markDuplicate \
			 --bqsr-recal-file $tmpDir/$post_BaseRecalibrator \
			 -O $bamDir/$post_ApplyBQSR \
			 --verbosity WARNING
#+end_src
missing file
***** KILL Variant caling
CLOSED: [2023-01-04 Wed 19:16]
****** KILL haplotypecaller
CLOSED: [2023-01-04 Wed 19:15]
******** options ok
#+begin_src
gatk --java-options "-Xmx3g" HaplotypeCaller \
    --input 63003856_S135.bam \
    --output 63003856_S135.vcf.gz \
    --reference genomeRef.fna \
    --dbsnp dbSNP.gz \
     \
     \
    --tmp-dir . \
    --max-mnp-distance 2
#+end_src
#+begin_src
		$gatk --java-options "-Xmx32g" HaplotypeCaller \
		-R $genomeRef \
		-I $bamDir/$post_ApplyBQSR \
		-O $vcfDir/$post_haplotypecaller \
		-D "$dbsnpDir"/GCF_000001405.39.gz \
		--max-mnp-distance 2 \
		--verbosity WARNING
#+end_src
******** DONE Nombres lignes gatk 4.3.0 : trop différent
CLOSED: [2023-01-04 Wed 19:11]
$ grep '^NC' out/63003856_S135/variantCalling/haplotypecaller/63003856_S135.vcf | wc -l
1631935

$ grep '^NC' /Work/Groups/bisonex/ref-vcf/63003856_S135 .vcf | wc -l
1506894
******** DONE Regarder les flags d'haplotypecaller : nombreuses différences...
CLOSED: [2023-01-04 Wed 19:02]
| --dbsnp /Work/Users/apraga/bisonex/work/08/fca52ac598f21a2812f866bd590792/dbSNP.gz | --dbsnp /mnt/j/bases_de_donnees/dbSNP/GCF_000001405.39.gz                                       |
| --max-mnp-distance 2                                                               | --max-mnp-distance 2                                                                            |
| --output 63003856_S135.vcf.gz                                                      | --output /mnt/j/working_directory_pipeline_analyse_exome/vcf/63003856_S135.vcf                  |
| --input 63003856_S135.bam                                                          | --input /mnt/j/working_directory_pipeline_analyse_exome/bam/63003856_S135_recalibrated_hg38.bam |
| --reference genomeRef.fna                                                          | --reference /mnt/j/bases_de_donnees/genome/GRCh38_latest_genomic.fna                            |
| --tmp-dir .                                                                        | --verbosity WARNING                                                                             |
| --use-posteriors-to-calculate-qual false                                           | --use-posteriors-to-calculate-qual false                                                        |
| --dont-use-dragstr-priors false                                                    | --dont-use-dragstr-priors false                                                                 |
| --use-new-qual-calculator true                                                     | --use-new-qual-calculator true                                                                  |
| --annotate-with-num-discovered-alleles false                                       | --annotate-with-num-discovered-alleles false                                                    |
| --heterozygosity 0.001                                                             | --heterozygosity 0.001                                                                          |
| --indel-heterozygosity 1.25E-4                                                     | --indel-heterozygosity 1.25E-4                                                                  |
| --heterozygosity-stdev 0.01                                                        | --heterozygosity-stdev 0.01                                                                     |
| --standard-min-confidence-threshold-for-calling 30.0                               | --standard-min-confidence-threshold-for-calling 30.0                                            |
| --max-alternate-alleles 6                                                          | --max-alternate-alleles 6                                                                       |
| --max-genotype-count 1024                                                          | --max-genotype-count 1024                                                                       |
| --sample-ploidy 2                                                                  | --sample-ploidy 2                                                                               |
| --num-reference-samples-if-no-call 0                                               | --num-reference-samples-if-no-call 0                                                            |
| --genotype-assignment-method USE_PLS_TO_ASSIGN                                     | --genotype-assignment-method USE_PLS_TO_ASSIGN                                                  |
| --contamination-fraction-to-filter 0.0                                             | --contamination-fraction-to-filter 0.0                                                          |
| --output-mode EMIT_VARIANTS_ONLY                                                   | --output-mode EMIT_VARIANTS_ONLY                                                                |
| --all-site-pls false                                                               | --all-site-pls false                                                                            |
| --flow-likelihood-parallel-threads 0                                               | --gvcf-gq-bands 1                                                                               |
| --flow-likelihood-optimized-comp false                                             | --gvcf-gq-bands 2                                                                               |
| --flow-use-t0-tag false                                                            | --gvcf-gq-bands 3                                                                               |
| --flow-probability-threshold 0.003                                                 | --gvcf-gq-bands 4                                                                               |
| --flow-remove-non-single-base-pair-indels false                                    | --gvcf-gq-bands 5                                                                               |
| --flow-remove-one-zero-probs false                                                 | --gvcf-gq-bands 6                                                                               |
| --flow-quantization-bins 121                                                       | --gvcf-gq-bands 7                                                                               |
| --flow-fill-empty-bins-value 0.001                                                 | --gvcf-gq-bands 8                                                                               |
| --flow-symmetric-indel-probs false                                                 | --gvcf-gq-bands 9                                                                               |
| --flow-report-insertion-or-deletion false                                          | --gvcf-gq-bands 10                                                                              |
| --flow-disallow-probs-larger-than-call false                                       | --gvcf-gq-bands 11                                                                              |
| --flow-lump-probs false                                                            | --gvcf-gq-bands 12                                                                              |
| --flow-retain-max-n-probs-base-format false                                        | --gvcf-gq-bands 13                                                                              |
| --flow-probability-scaling-factor 10                                               | --gvcf-gq-bands 14                                                                              |
| --flow-order-cycle-length 4                                                        | --gvcf-gq-bands 15                                                                              |
| --flow-number-of-uncertain-flows-to-clip 0                                         | --gvcf-gq-bands 16                                                                              |
| --flow-nucleotide-of-first-uncertain-flow T                                        | --gvcf-gq-bands 17                                                                              |
| --keep-boundary-flows false                                                        | --gvcf-gq-bands 18                                                                              |
| --gvcf-gq-bands 1                                                                  | --gvcf-gq-bands 19                                                                              |
| --gvcf-gq-bands 2                                                                  | --gvcf-gq-bands 20                                                                              |
| --gvcf-gq-bands 3                                                                  | --gvcf-gq-bands 21                                                                              |
| --gvcf-gq-bands 4                                                                  | --gvcf-gq-bands 22                                                                              |
| --gvcf-gq-bands 5                                                                  | --gvcf-gq-bands 23                                                                              |
| --gvcf-gq-bands 6                                                                  | --gvcf-gq-bands 24                                                                              |
| --gvcf-gq-bands 7                                                                  | --gvcf-gq-bands 25                                                                              |
| --gvcf-gq-bands 8                                                                  | --gvcf-gq-bands 26                                                                              |
| --gvcf-gq-bands 9                                                                  | --gvcf-gq-bands 27                                                                              |
| --gvcf-gq-bands 10                                                                 | --gvcf-gq-bands 28                                                                              |
| --gvcf-gq-bands 11                                                                 | --gvcf-gq-bands 29                                                                              |
| --gvcf-gq-bands 12                                                                 | --gvcf-gq-bands 30                                                                              |
| --gvcf-gq-bands 13                                                                 | --gvcf-gq-bands 31                                                                              |
| --gvcf-gq-bands 14                                                                 | --gvcf-gq-bands 32                                                                              |
| --gvcf-gq-bands 15                                                                 | --gvcf-gq-bands 33                                                                              |
| --gvcf-gq-bands 16                                                                 | --gvcf-gq-bands 34                                                                              |
| --gvcf-gq-bands 17                                                                 | --gvcf-gq-bands 35                                                                              |
| --gvcf-gq-bands 18                                                                 | --gvcf-gq-bands 36                                                                              |
| --gvcf-gq-bands 19                                                                 | --gvcf-gq-bands 37                                                                              |
| --gvcf-gq-bands 20                                                                 | --gvcf-gq-bands 38                                                                              |
| --gvcf-gq-bands 21                                                                 | --gvcf-gq-bands 39                                                                              |
| --gvcf-gq-bands 22                                                                 | --gvcf-gq-bands 40                                                                              |
| --gvcf-gq-bands 23                                                                 | --gvcf-gq-bands 41                                                                              |
| --gvcf-gq-bands 24                                                                 | --gvcf-gq-bands 42                                                                              |
| --gvcf-gq-bands 25                                                                 | --gvcf-gq-bands 43                                                                              |
| --gvcf-gq-bands 26                                                                 | --gvcf-gq-bands 44                                                                              |
| --gvcf-gq-bands 27                                                                 | --gvcf-gq-bands 45                                                                              |
| --gvcf-gq-bands 28                                                                 | --gvcf-gq-bands 46                                                                              |
| --gvcf-gq-bands 29                                                                 | --gvcf-gq-bands 47                                                                              |
| --gvcf-gq-bands 30                                                                 | --gvcf-gq-bands 48                                                                              |
| --gvcf-gq-bands 31                                                                 | --gvcf-gq-bands 49                                                                              |
| --gvcf-gq-bands 32                                                                 | --gvcf-gq-bands 50                                                                              |
| --gvcf-gq-bands 33                                                                 | --gvcf-gq-bands 51                                                                              |
| --gvcf-gq-bands 34                                                                 | --gvcf-gq-bands 52                                                                              |
| --gvcf-gq-bands 35                                                                 | --gvcf-gq-bands 53                                                                              |
| --gvcf-gq-bands 36                                                                 | --gvcf-gq-bands 54                                                                              |
| --gvcf-gq-bands 37                                                                 | --gvcf-gq-bands 55                                                                              |
| --gvcf-gq-bands 38                                                                 | --gvcf-gq-bands 56                                                                              |
| --gvcf-gq-bands 39                                                                 | --gvcf-gq-bands 57                                                                              |
| --gvcf-gq-bands 40                                                                 | --gvcf-gq-bands 58                                                                              |
| --gvcf-gq-bands 41                                                                 | --gvcf-gq-bands 59                                                                              |
| --gvcf-gq-bands 42                                                                 | --gvcf-gq-bands 60                                                                              |
| --gvcf-gq-bands 43                                                                 | --gvcf-gq-bands 70                                                                              |
| --gvcf-gq-bands 44                                                                 | --gvcf-gq-bands 80                                                                              |
| --gvcf-gq-bands 45                                                                 | --gvcf-gq-bands 90                                                                              |
| --gvcf-gq-bands 46                                                                 | --gvcf-gq-bands 99                                                                              |
| --gvcf-gq-bands 47                                                                 | --floor-blocks false                                                                            |
| --gvcf-gq-bands 48                                                                 | --indel-size-to-eliminate-in-ref-model 10                                                       |
| --gvcf-gq-bands 49                                                                 | --disable-optimizations false                                                                   |
| --gvcf-gq-bands 50                                                                 | --dragen-mode false                                                                             |
| --gvcf-gq-bands 51                                                                 | --apply-bqd false                                                                               |
| --gvcf-gq-bands 52                                                                 | --apply-frd false                                                                               |
| --gvcf-gq-bands 53                                                                 | --disable-spanning-event-genotyping false                                                       |
| --gvcf-gq-bands 54                                                                 | --transform-dragen-mapping-quality false                                                        |
| --gvcf-gq-bands 55                                                                 | --mapping-quality-threshold-for-genotyping 20                                                   |
| --gvcf-gq-bands 56                                                                 | --max-effective-depth-adjustment-for-frd 0                                                      |
| --gvcf-gq-bands 57                                                                 | --just-determine-active-regions false                                                           |
| --gvcf-gq-bands 58                                                                 | --dont-genotype false                                                                           |
| --gvcf-gq-bands 59                                                                 | --do-not-run-physical-phasing false                                                             |
| --gvcf-gq-bands 60                                                                 | --do-not-correct-overlapping-quality false                                                      |
| --gvcf-gq-bands 70                                                                 | --use-filtered-reads-for-annotations false                                                      |
| --gvcf-gq-bands 80                                                                 | --adaptive-pruning false                                                                        |
| --gvcf-gq-bands 90                                                                 | --do-not-recover-dangling-branches false                                                        |
| --gvcf-gq-bands 99                                                                 | --recover-dangling-heads false                                                                  |
| --floor-blocks false                                                               | --kmer-size 10                                                                                  |
| --indel-size-to-eliminate-in-ref-model 10                                          | --kmer-size 25                                                                                  |
| --disable-optimizations false                                                      | --dont-increase-kmer-sizes-for-cycles false                                                     |
| --dragen-mode false                                                                | --allow-non-unique-kmers-in-ref false                                                           |
| --flow-mode NONE                                                                   | --num-pruning-samples 1                                                                         |
| --apply-bqd false                                                                  | --min-dangling-branch-length 4                                                                  |
| --apply-frd false                                                                  | --recover-all-dangling-branches false                                                           |
| --disable-spanning-event-genotyping false                                          | --max-num-haplotypes-in-population 128                                                          |
| --transform-dragen-mapping-quality false                                           | --min-pruning 2                                                                                 |
| --mapping-quality-threshold-for-genotyping 20                                      | --adaptive-pruning-initial-error-rate 0.001                                                     |
| --max-effective-depth-adjustment-for-frd 0                                         | --pruning-lod-threshold 2.302585092994046                                                       |
| --just-determine-active-regions false                                              | --pruning-seeding-lod-threshold 9.210340371976184                                               |
| --dont-genotype false                                                              | --max-unpruned-variants 100                                                                     |
| --do-not-run-physical-phasing false                                                | --linked-de-bruijn-graph false                                                                  |
| --do-not-correct-overlapping-quality false                                         | --disable-artificial-haplotype-recovery false                                                   |
| --use-filtered-reads-for-annotations false                                         | --enable-legacy-graph-cycle-detection false                                                     |
| --use-flow-aligner-for-stepwise-hc-filtering false                                 | --debug-assembly false                                                                          |
| --adaptive-pruning false                                                           | --debug-graph-transformations false                                                             |
| --do-not-recover-dangling-branches false                                           | --capture-assembly-failure-bam false                                                            |
| --recover-dangling-heads false                                                     | --num-matching-bases-in-dangling-end-to-recover -1                                              |
| --kmer-size 10                                                                     | --error-correction-log-odds -Infinity                                                           |
| --kmer-size 25                                                                     | --error-correct-reads false                                                                     |
| --dont-increase-kmer-sizes-for-cycles false                                        | --kmer-length-for-read-error-correction 25                                                      |
| --allow-non-unique-kmers-in-ref false                                              | --min-observations-for-kmer-to-be-solid 20                                                      |
| --num-pruning-samples 1                                                            | --base-quality-score-threshold 18                                                               |
| --min-dangling-branch-length 4                                                     | --dragstr-het-hom-ratio 2                                                                       |
| --recover-all-dangling-branches false                                              | --dont-use-dragstr-pair-hmm-scores false                                                        |
| --max-num-haplotypes-in-population 128                                             | --pair-hmm-gap-continuation-penalty 10                                                          |
| --min-pruning 2                                                                    | --expected-mismatch-rate-for-read-disqualification 0.02                                         |
| --adaptive-pruning-initial-error-rate 0.001                                        | --pair-hmm-implementation FASTEST_AVAILABLE                                                     |
| --pruning-lod-threshold 2.302585092994046                                          | --pcr-indel-model CONSERVATIVE                                                                  |
| --pruning-seeding-lod-threshold 9.210340371976184                                  | --phred-scaled-global-read-mismapping-rate 45                                                   |
| --max-unpruned-variants 100                                                        | --disable-symmetric-hmm-normalizing false                                                       |
| --linked-de-bruijn-graph false                                                     | --disable-cap-base-qualities-to-map-quality false                                               |
| --disable-artificial-haplotype-recovery false                                      | --enable-dynamic-read-disqualification-for-genotyping false                                     |
| --enable-legacy-graph-cycle-detection false                                        | --dynamic-read-disqualification-threshold 1.0                                                   |
| --debug-assembly false                                                             | --native-pair-hmm-threads 4                                                                     |
| --debug-graph-transformations false                                                | --native-pair-hmm-use-double-precision false                                                    |
| --capture-assembly-failure-bam false                                               | --bam-writer-type CALLED_HAPLOTYPES                                                             |
| --num-matching-bases-in-dangling-end-to-recover -1                                 | --dont-use-soft-clipped-bases false                                                             |
| --error-correction-log-odds -Infinity                                              | --min-base-quality-score 10                                                                     |
| --error-correct-reads false                                                        | --smith-waterman JAVA                                                                           |
| --kmer-length-for-read-error-correction 25                                         | --emit-ref-confidence NONE                                                                      |
| --min-observations-for-kmer-to-be-solid 20                                         | --force-call-filtered-alleles false                                                             |
| --likelihood-calculation-engine PairHMM                                            | --soft-clip-low-quality-ends false                                                              |
| --base-quality-score-threshold 18                                                  | --allele-informative-reads-overlap-margin 2                                                     |
| --dragstr-het-hom-ratio 2                                                          | --smith-waterman-dangling-end-match-value 25                                                    |
| --dont-use-dragstr-pair-hmm-scores false                                           | --smith-waterman-dangling-end-mismatch-penalty -50                                              |
| --pair-hmm-gap-continuation-penalty 10                                             | --smith-waterman-dangling-end-gap-open-penalty -110                                             |
| --expected-mismatch-rate-for-read-disqualification 0.02                            | --smith-waterman-dangling-end-gap-extend-penalty -6                                             |
| --pair-hmm-implementation FASTEST_AVAILABLE                                        | --smith-waterman-haplotype-to-reference-match-value 200                                         |
| --pcr-indel-model CONSERVATIVE                                                     | --smith-waterman-haplotype-to-reference-mismatch-penalty -150                                   |
| --phred-scaled-global-read-mismapping-rate 45                                      | --smith-waterman-haplotype-to-reference-gap-open-penalty -260                                   |
| --disable-symmetric-hmm-normalizing false                                          | --smith-waterman-haplotype-to-reference-gap-extend-penalty -11                                  |
| --disable-cap-base-qualities-to-map-quality false                                  | --smith-waterman-read-to-haplotype-match-value 10                                               |
| --enable-dynamic-read-disqualification-for-genotyping false                        | --smith-waterman-read-to-haplotype-mismatch-penalty -15                                         |
| --dynamic-read-disqualification-threshold 1.0                                      | --smith-waterman-read-to-haplotype-gap-open-penalty -30                                         |
| --native-pair-hmm-threads 4                                                        | --smith-waterman-read-to-haplotype-gap-extend-penalty -5                                        |
| --native-pair-hmm-use-double-precision false                                       | --min-assembly-region-size 50                                                                   |
| --flow-hmm-engine-min-indel-adjust 6                                               | --max-assembly-region-size 300                                                                  |
| --flow-hmm-engine-flat-insertion-penatly 45                                        | --active-probability-threshold 0.002                                                            |
| --flow-hmm-engine-flat-deletion-penatly 45                                         | --max-prob-propagation-distance 50                                                              |
| --pileup-detection false                                                           | --force-active false                                                                            |
| --pileup-detection-enable-indel-pileup-calling false                               | --assembly-region-padding 100                                                                   |
| --num-artificial-haplotypes-to-add-per-allele 5                                    | --padding-around-indels 75                                                                      |
| --artifical-haplotype-filtering-kmer-size 10                                       | --padding-around-snps 20                                                                        |
| --pileup-detection-snp-alt-threshold 0.1                                           | --padding-around-strs 75                                                                        |
| --pileup-detection-indel-alt-threshold 0.5                                         | --max-extension-into-assembly-region-padding-legacy 25                                          |
| --pileup-detection-absolute-alt-depth 0.0                                          | --max-reads-per-alignment-start 50                                                              |
| --pileup-detection-snp-adjacent-to-assembled-indel-range 5                         | --enable-legacy-assembly-region-trimming false                                                  |
| --pileup-detection-bad-read-tolerance 0.0                                          | --interval-set-rule UNION                                                                       |
| --pileup-detection-proper-pair-read-badness true                                   | --interval-padding 0                                                                            |
| --pileup-detection-edit-distance-read-badness-threshold 0.08                       | --interval-exclusion-padding 0                                                                  |
| --pileup-detection-chimeric-read-badness true                                      | --interval-merging-rule ALL                                                                     |
| --pileup-detection-template-mean-badness-threshold 0.0                             | --read-validation-stringency SILENT                                                             |
| --pileup-detection-template-std-badness-threshold 0.0                              | --seconds-between-progress-updates 10.0                                                         |
| --bam-writer-type CALLED_HAPLOTYPES                                                | --disable-sequence-dictionary-validation false                                                  |
| --dont-use-soft-clipped-bases false                                                | --create-output-bam-index true                                                                  |
| --override-fragment-softclip-check false                                           | --create-output-bam-md5 false                                                                   |
| --min-base-quality-score 10                                                        | --create-output-variant-index true                                                              |
| --smith-waterman JAVA                                                              | --create-output-variant-md5 false                                                               |
| --emit-ref-confidence NONE                                                         | --max-variants-per-shard 0                                                                      |
| --force-call-filtered-alleles false                                                | --lenient false                                                                                 |
| --reference-model-deletion-quality 30                                              | --add-output-sam-program-record true                                                            |
| --soft-clip-low-quality-ends false                                                 | --add-output-vcf-command-line true                                                              |
| --allele-informative-reads-overlap-margin 2                                        | --cloud-prefetch-buffer 40                                                                      |
| --smith-waterman-dangling-end-match-value 25                                       | --cloud-index-prefetch-buffer -1                                                                |
| --smith-waterman-dangling-end-mismatch-penalty -50                                 | --disable-bam-index-caching false                                                               |
| --smith-waterman-dangling-end-gap-open-penalty -110                                | --sites-only-vcf-output false                                                                   |
| --smith-waterman-dangling-end-gap-extend-penalty -6                                | --help false                                                                                    |
| --smith-waterman-haplotype-to-reference-match-value 200                            | --version false                                                                                 |
| --smith-waterman-haplotype-to-reference-mismatch-penalty -150                      | --showHidden false                                                                              |
| --smith-waterman-haplotype-to-reference-gap-open-penalty -260                      | --QUIET false                                                                                   |
| --smith-waterman-haplotype-to-reference-gap-extend-penalty -11                     | --use-jdk-deflater false                                                                        |
| --smith-waterman-read-to-haplotype-match-value 10                                  | --use-jdk-inflater false                                                                        |
| --smith-waterman-read-to-haplotype-mismatch-penalty -15                            | --gcs-max-retries 20                                                                            |
| --smith-waterman-read-to-haplotype-gap-open-penalty -30                            | --gcs-project-for-requester-pays                                                                |
| --smith-waterman-read-to-haplotype-gap-extend-penalty -5                           | --disable-tool-default-read-filters false                                                       |
| --flow-assembly-collapse-hmer-size 0                                               | --minimum-mapping-quality 20                                                                    |
| --flow-assembly-collapse-partial-mode false                                        | --disable-tool-default-annotations false                                                        |
| --flow-filter-alleles false                                                        | --enable-all-annotations false                                                                  |
| --flow-filter-alleles-qual-threshold 30.0                                          | --allow-old-rms-mapping-quality-annotation-data false                                           |
| --flow-filter-alleles-sor-threshold 3.0                                            | Version="4.2.4.1",Date="December 3, 2022 at 1:20:38 AM CET">                                    |
| --flow-filter-lone-alleles false                                                   |
| --flow-filter-alleles-debug-graphs false                                           |
| --min-assembly-region-size 50                                                      |
| --max-assembly-region-size 300                                                     |
| --active-probability-threshold 0.002                                               |
| --max-prob-propagation-distance 50                                                 |
| --force-active false                                                               |
| --assembly-region-padding 100                                                      |
| --padding-around-indels 75                                                         |
| --padding-around-snps 20                                                           |
| --padding-around-strs 75                                                           |
| --max-extension-into-assembly-region-padding-legacy 25                             |
| --max-reads-per-alignment-start 50                                                 |
| --enable-legacy-assembly-region-trimming false                                     |
| --interval-set-rule UNION                                                          |
| --interval-padding 0                                                               |
| --interval-exclusion-padding 0                                                     |
| --interval-merging-rule ALL                                                        |
| --read-validation-stringency SILENT                                                |
| --seconds-between-progress-updates 10.0                                            |
| --disable-sequence-dictionary-validation false                                     |
| --create-output-bam-index true                                                     |
| --create-output-bam-md5 false                                                      |
| --create-output-variant-index true                                                 |
| --create-output-variant-md5 false                                                  |
| --max-variants-per-shard 0                                                         |
| --lenient false                                                                    |
| --add-output-sam-program-record true                                               |
| --add-output-vcf-command-line true                                                 |
| --cloud-prefetch-buffer 40                                                         |
| --cloud-index-prefetch-buffer -1                                                   |
| --disable-bam-index-caching false                                                  |
| --sites-only-vcf-output false                                                      |
| --help false                                                                       |
| --version false                                                                    |
| --showHidden false                                                                 |
| --verbosity INFO                                                                   |
| --QUIET false                                                                      |
| --use-jdk-deflater false                                                           |
| --use-jdk-inflater false                                                           |
| --gcs-max-retries 20                                                               |
| --gcs-project-for-requester-pays                                                   |
| --disable-tool-default-read-filters false                                          |
| --minimum-mapping-quality 20                                                       |
| --disable-tool-default-annotations false                                           |
| --enable-all-annotations false                                                     |
| --allow-old-rms-mapping-quality-annotation-data false"                             |
| Version="4.3.0.0",Date="December 16, 2022 at 12:51:03 AM CET">                     |
****** KILL [#B] filterDepth : 21 en trop
CLOSED: [2023-01-04 Wed 19:16]
Nouvelle version (correcton bug markdupicates)
#+begin_src sh
grep '^NC' out/63003856_S135/variantCalling/filter-depth.vcf  |wc -l
82054
#+end_src

Alexis
#+begin_src sh
zgrep '^NC' /Work/Groups/bisonex/ref_63003856_S135/63003856_S135_DP_over_30.vcf  | wc -l
82033
#+end_src

Ne vient pas du filtre sur la profondeur:
bcftools filter -i 'FORMAT/AD[0:1]<=10' 63003856_S135_DP_over_30.vcf
bcftools filter -i 'FORMAT/DP<=30' 63003856_S135_DP_over_30.vcf
Idem pour notre version. Rien ne sort.

On compare le nombre de lignes
#+begin_src sh
bgzip out/63003856_S135/variantCalling/filter-depth.vcf
tabix /Work/Groups/bisonex/ref_63003856_S135/63003856_S135_DP_over_30.vcf.gz
tabix out/63003856_S135/variantCalling/filter-depth.vcf.gz
bcftools isec out/63003856_S135/variantCalling/filter-depth.vcf.gz /Work/Groups/bisonex/ref_63003856_S135/63003856_S135_DP_over_30.vcf.gz -p compare-filter-depth
find compare-filter-depth/ -type f -exec wc -l {} \;

84763 compare-filter-depth/sites.txt
710 compare-filter-depth/0001.vcf
8 compare-filter-depth/README.txt
85339 compare-filter-depth/0002.vcf
85340 compare-filter-depth/0003.vcf
725 compare-filter-depth/0000.vcf
#+end_src
****** KILL exclude SNP + consensual : 34 en trop !!
CLOSED: [2023-01-04 Wed 19:16]
$ grep '^NC' out/63003856_S135/variantCalling/filter-polymorphisms.vcf  | wc -l
8898
vs

$ grep '^NC' /Work/Groups/bisonex/ref_63003856_S135/63003856_S135_DP_over_30_not_SNP_consensual_sequence.vcf | wc -l
8864
****** KILL Filter technical variants
CLOSED: [2023-01-04 Wed 19:16]
**** DONE Gatk 4.2.4 : idem
CLOSED: [2023-01-07 Sat 00:05]
***** DONE Variant calling
CLOSED: [2023-01-07 Sat 00:05]
****** DONE haplotypecaller: mieux mais non identique !
CLOSED: [2023-01-07 Sat 00:05]
******* DONE Nombres lignes gatk 4.2.2 : faible différence
CLOSED: [2023-01-04 Wed 19:18]
$ zgrep '^NC' 63003856_S135.vcf.gz | wc -l
1506931

$ grep '^NC' /Work/Groups/bisonex/ref-vcf/63003856_S135 .vcf | wc -l
1506894

******* DONE Flags la même version de gatk 4.2.2 : ok identique
CLOSED: [2023-01-04 Wed 19:09]

##GATKCommandLine=<ID=HaplotypeCaller,CommandLine="HaplotypeCaller
| ",Version="4.2.4.1",Date="January 4, 2023 at 1:46:41 AM CET">                      | Version="4.2.4.1",Date="December 3, 2022 at 1:20:38 AM CET">                                    |
| --dbsnp /Work/Users/apraga/bisonex/work/5d/feb81028d262d7701bed0a759ff6f6/dbSNP.gz | --dbsnp /mnt/j/bases_de_donnees/dbSNP/GCF_000001405.39.gz                                       |
| --max-mnp-distance 2                                                               | --max-mnp-distance 2                                                                            |
| --output 63003856_S135.vcf.gz                                                      | --output /mnt/j/working_directory_pipeline_analyse_exome/vcf/63003856_S135.vcf                  |
| --input 63003856_S135.bam                                                          | --input /mnt/j/working_directory_pipeline_analyse_exome/bam/63003856_S135_recalibrated_hg38.bam |
| --reference genomeRef.fna                                                          | --reference /mnt/j/bases_de_donnees/genome/GRCh38_latest_genomic.fna                            |
| --tmp-dir .                                                                        | --verbosity WARNING                                                                             |
| --use-posteriors-to-calculate-qual false                                           | --use-posteriors-to-calculate-qual false                                                        |
| --dont-use-dragstr-priors false                                                    | --dont-use-dragstr-priors false                                                                 |
| --use-new-qual-calculator true                                                     | --use-new-qual-calculator true                                                                  |
| --annotate-with-num-discovered-alleles false                                       | --annotate-with-num-discovered-alleles false                                                    |
| --heterozygosity 0.001                                                             | --heterozygosity 0.001                                                                          |
| --indel-heterozygosity 1.25E-4                                                     | --indel-heterozygosity 1.25E-4                                                                  |
| --heterozygosity-stdev 0.01                                                        | --heterozygosity-stdev 0.01                                                                     |
| --standard-min-confidence-threshold-for-calling 30.0                               | --standard-min-confidence-threshold-for-calling 30.0                                            |
| --max-alternate-alleles 6                                                          | --max-alternate-alleles 6                                                                       |
| --max-genotype-count 1024                                                          | --max-genotype-count 1024                                                                       |
| --sample-ploidy 2                                                                  | --sample-ploidy 2                                                                               |
| --num-reference-samples-if-no-call 0                                               | --num-reference-samples-if-no-call 0                                                            |
| --genotype-assignment-method USE_PLS_TO_ASSIGN                                     | --genotype-assignment-method USE_PLS_TO_ASSIGN                                                  |
| --contamination-fraction-to-filter 0.0                                             | --contamination-fraction-to-filter 0.0                                                          |
| --output-mode EMIT_VARIANTS_ONLY                                                   | --output-mode EMIT_VARIANTS_ONLY                                                                |
| --all-site-pls false                                                               | --all-site-pls false                                                                            |
| --gvcf-gq-bands 1                                                                  | --gvcf-gq-bands 1                                                                               |
| --gvcf-gq-bands 2                                                                  | --gvcf-gq-bands 2                                                                               |
| --gvcf-gq-bands 3                                                                  | --gvcf-gq-bands 3                                                                               |
| --gvcf-gq-bands 4                                                                  | --gvcf-gq-bands 4                                                                               |
| --gvcf-gq-bands 5                                                                  | --gvcf-gq-bands 5                                                                               |
| --gvcf-gq-bands 6                                                                  | --gvcf-gq-bands 6                                                                               |
| --gvcf-gq-bands 7                                                                  | --gvcf-gq-bands 7                                                                               |
| --gvcf-gq-bands 8                                                                  | --gvcf-gq-bands 8                                                                               |
| --gvcf-gq-bands 9                                                                  | --gvcf-gq-bands 9                                                                               |
| --gvcf-gq-bands 10                                                                 | --gvcf-gq-bands 10                                                                              |
| --gvcf-gq-bands 11                                                                 | --gvcf-gq-bands 11                                                                              |
| --gvcf-gq-bands 12                                                                 | --gvcf-gq-bands 12                                                                              |
| --gvcf-gq-bands 13                                                                 | --gvcf-gq-bands 13                                                                              |
| --gvcf-gq-bands 14                                                                 | --gvcf-gq-bands 14                                                                              |
| --gvcf-gq-bands 15                                                                 | --gvcf-gq-bands 15                                                                              |
| --gvcf-gq-bands 16                                                                 | --gvcf-gq-bands 16                                                                              |
| --gvcf-gq-bands 17                                                                 | --gvcf-gq-bands 17                                                                              |
| --gvcf-gq-bands 18                                                                 | --gvcf-gq-bands 18                                                                              |
| --gvcf-gq-bands 19                                                                 | --gvcf-gq-bands 19                                                                              |
| --gvcf-gq-bands 20                                                                 | --gvcf-gq-bands 20                                                                              |
| --gvcf-gq-bands 21                                                                 | --gvcf-gq-bands 21                                                                              |
| --gvcf-gq-bands 22                                                                 | --gvcf-gq-bands 22                                                                              |
| --gvcf-gq-bands 23                                                                 | --gvcf-gq-bands 23                                                                              |
| --gvcf-gq-bands 24                                                                 | --gvcf-gq-bands 24                                                                              |
| --gvcf-gq-bands 25                                                                 | --gvcf-gq-bands 25                                                                              |
| --gvcf-gq-bands 26                                                                 | --gvcf-gq-bands 26                                                                              |
| --gvcf-gq-bands 27                                                                 | --gvcf-gq-bands 27                                                                              |
| --gvcf-gq-bands 28                                                                 | --gvcf-gq-bands 28                                                                              |
| --gvcf-gq-bands 29                                                                 | --gvcf-gq-bands 29                                                                              |
| --gvcf-gq-bands 30                                                                 | --gvcf-gq-bands 30                                                                              |
| --gvcf-gq-bands 31                                                                 | --gvcf-gq-bands 31                                                                              |
| --gvcf-gq-bands 32                                                                 | --gvcf-gq-bands 32                                                                              |
| --gvcf-gq-bands 33                                                                 | --gvcf-gq-bands 33                                                                              |
| --gvcf-gq-bands 34                                                                 | --gvcf-gq-bands 34                                                                              |
| --gvcf-gq-bands 35                                                                 | --gvcf-gq-bands 35                                                                              |
| --gvcf-gq-bands 36                                                                 | --gvcf-gq-bands 36                                                                              |
| --gvcf-gq-bands 37                                                                 | --gvcf-gq-bands 37                                                                              |
| --gvcf-gq-bands 38                                                                 | --gvcf-gq-bands 38                                                                              |
| --gvcf-gq-bands 39                                                                 | --gvcf-gq-bands 39                                                                              |
| --gvcf-gq-bands 40                                                                 | --gvcf-gq-bands 40                                                                              |
| --gvcf-gq-bands 41                                                                 | --gvcf-gq-bands 41                                                                              |
| --gvcf-gq-bands 42                                                                 | --gvcf-gq-bands 42                                                                              |
| --gvcf-gq-bands 43                                                                 | --gvcf-gq-bands 43                                                                              |
| --gvcf-gq-bands 44                                                                 | --gvcf-gq-bands 44                                                                              |
| --gvcf-gq-bands 45                                                                 | --gvcf-gq-bands 45                                                                              |
| --gvcf-gq-bands 46                                                                 | --gvcf-gq-bands 46                                                                              |
| --gvcf-gq-bands 47                                                                 | --gvcf-gq-bands 47                                                                              |
| --gvcf-gq-bands 48                                                                 | --gvcf-gq-bands 48                                                                              |
| --gvcf-gq-bands 49                                                                 | --gvcf-gq-bands 49                                                                              |
| --gvcf-gq-bands 50                                                                 | --gvcf-gq-bands 50                                                                              |
| --gvcf-gq-bands 51                                                                 | --gvcf-gq-bands 51                                                                              |
| --gvcf-gq-bands 52                                                                 | --gvcf-gq-bands 52                                                                              |
| --gvcf-gq-bands 53                                                                 | --gvcf-gq-bands 53                                                                              |
| --gvcf-gq-bands 54                                                                 | --gvcf-gq-bands 54                                                                              |
| --gvcf-gq-bands 55                                                                 | --gvcf-gq-bands 55                                                                              |
| --gvcf-gq-bands 56                                                                 | --gvcf-gq-bands 56                                                                              |
| --gvcf-gq-bands 57                                                                 | --gvcf-gq-bands 57                                                                              |
| --gvcf-gq-bands 58                                                                 | --gvcf-gq-bands 58                                                                              |
| --gvcf-gq-bands 59                                                                 | --gvcf-gq-bands 59                                                                              |
| --gvcf-gq-bands 60                                                                 | --gvcf-gq-bands 60                                                                              |
| --gvcf-gq-bands 70                                                                 | --gvcf-gq-bands 70                                                                              |
| --gvcf-gq-bands 80                                                                 | --gvcf-gq-bands 80                                                                              |
| --gvcf-gq-bands 90                                                                 | --gvcf-gq-bands 90                                                                              |
| --gvcf-gq-bands 99                                                                 | --gvcf-gq-bands 99                                                                              |
| --floor-blocks false                                                               | --floor-blocks false                                                                            |
| --indel-size-to-eliminate-in-ref-model 10                                          | --indel-size-to-eliminate-in-ref-model 10                                                       |
| --disable-optimizations false                                                      | --disable-optimizations false                                                                   |
| --dragen-mode false                                                                | --dragen-mode false                                                                             |
| --apply-bqd false                                                                  | --apply-bqd false                                                                               |
| --apply-frd false                                                                  | --apply-frd false                                                                               |
| --disable-spanning-event-genotyping false                                          | --disable-spanning-event-genotyping false                                                       |
| --transform-dragen-mapping-quality false                                           | --transform-dragen-mapping-quality false                                                        |
| --mapping-quality-threshold-for-genotyping 20                                      | --mapping-quality-threshold-for-genotyping 20                                                   |
| --max-effective-depth-adjustment-for-frd 0                                         | --max-effective-depth-adjustment-for-frd 0                                                      |
| --just-determine-active-regions false                                              | --just-determine-active-regions false                                                           |
| --dont-genotype false                                                              | --dont-genotype false                                                                           |
| --do-not-run-physical-phasing false                                                | --do-not-run-physical-phasing false                                                             |
| --do-not-correct-overlapping-quality false                                         | --do-not-correct-overlapping-quality false                                                      |
| --use-filtered-reads-for-annotations false                                         | --use-filtered-reads-for-annotations false                                                      |
| --adaptive-pruning false                                                           | --adaptive-pruning false                                                                        |
| --do-not-recover-dangling-branches false                                           | --do-not-recover-dangling-branches false                                                        |
| --recover-dangling-heads false                                                     | --recover-dangling-heads false                                                                  |
| --kmer-size 10                                                                     | --kmer-size 10                                                                                  |
| --kmer-size 25                                                                     | --kmer-size 25                                                                                  |
| --dont-increase-kmer-sizes-for-cycles false                                        | --dont-increase-kmer-sizes-for-cycles false                                                     |
| --allow-non-unique-kmers-in-ref false                                              | --allow-non-unique-kmers-in-ref false                                                           |
| --num-pruning-samples 1                                                            | --num-pruning-samples 1                                                                         |
| --min-dangling-branch-length 4                                                     | --min-dangling-branch-length 4                                                                  |
| --recover-all-dangling-branches false                                              | --recover-all-dangling-branches false                                                           |
| --max-num-haplotypes-in-population 128                                             | --max-num-haplotypes-in-population 128                                                          |
| --min-pruning 2                                                                    | --min-pruning 2                                                                                 |
| --adaptive-pruning-initial-error-rate 0.001                                        | --adaptive-pruning-initial-error-rate 0.001                                                     |
| --pruning-lod-threshold 2.302585092994046                                          | --pruning-lod-threshold 2.302585092994046                                                       |
| --pruning-seeding-lod-threshold 9.210340371976184                                  | --pruning-seeding-lod-threshold 9.210340371976184                                               |
| --max-unpruned-variants 100                                                        | --max-unpruned-variants 100                                                                     |
| --linked-de-bruijn-graph false                                                     | --linked-de-bruijn-graph false                                                                  |
| --disable-artificial-haplotype-recovery false                                      | --disable-artificial-haplotype-recovery false                                                   |
| --enable-legacy-graph-cycle-detection false                                        | --enable-legacy-graph-cycle-detection false                                                     |
| --debug-assembly false                                                             | --debug-assembly false                                                                          |
| --debug-graph-transformations false                                                | --debug-graph-transformations false                                                             |
| --capture-assembly-failure-bam false                                               | --capture-assembly-failure-bam false                                                            |
| --num-matching-bases-in-dangling-end-to-recover -1                                 | --num-matching-bases-in-dangling-end-to-recover -1                                              |
| --error-correction-log-odds -Infinity                                              | --error-correction-log-odds -Infinity                                                           |
| --error-correct-reads false                                                        | --error-correct-reads false                                                                     |
| --kmer-length-for-read-error-correction 25                                         | --kmer-length-for-read-error-correction 25                                                      |
| --min-observations-for-kmer-to-be-solid 20                                         | --min-observations-for-kmer-to-be-solid 20                                                      |
| --base-quality-score-threshold 18                                                  | --base-quality-score-threshold 18                                                               |
| --dragstr-het-hom-ratio 2                                                          | --dragstr-het-hom-ratio 2                                                                       |
| --dont-use-dragstr-pair-hmm-scores false                                           | --dont-use-dragstr-pair-hmm-scores false                                                        |
| --pair-hmm-gap-continuation-penalty 10                                             | --pair-hmm-gap-continuation-penalty 10                                                          |
| --expected-mismatch-rate-for-read-disqualification 0.02                            | --expected-mismatch-rate-for-read-disqualification 0.02                                         |
| --pair-hmm-implementation FASTEST_AVAILABLE                                        | --pair-hmm-implementation FASTEST_AVAILABLE                                                     |
| --pcr-indel-model CONSERVATIVE                                                     | --pcr-indel-model CONSERVATIVE                                                                  |
| --phred-scaled-global-read-mismapping-rate 45                                      | --phred-scaled-global-read-mismapping-rate 45                                                   |
| --disable-symmetric-hmm-normalizing false                                          | --disable-symmetric-hmm-normalizing false                                                       |
| --disable-cap-base-qualities-to-map-quality false                                  | --disable-cap-base-qualities-to-map-quality false                                               |
| --enable-dynamic-read-disqualification-for-genotyping false                        | --enable-dynamic-read-disqualification-for-genotyping false                                     |
| --dynamic-read-disqualification-threshold 1.0                                      | --dynamic-read-disqualification-threshold 1.0                                                   |
| --native-pair-hmm-threads 4                                                        | --native-pair-hmm-threads 4                                                                     |
| --native-pair-hmm-use-double-precision false                                       | --native-pair-hmm-use-double-precision false                                                    |
| --bam-writer-type CALLED_HAPLOTYPES                                                | --bam-writer-type CALLED_HAPLOTYPES                                                             |
| --dont-use-soft-clipped-bases false                                                | --dont-use-soft-clipped-bases false                                                             |
| --min-base-quality-score 10                                                        | --min-base-quality-score 10                                                                     |
| --smith-waterman JAVA                                                              | --smith-waterman JAVA                                                                           |
| --emit-ref-confidence NONE                                                         | --emit-ref-confidence NONE                                                                      |
| --force-call-filtered-alleles false                                                | --force-call-filtered-alleles false                                                             |
| --soft-clip-low-quality-ends false                                                 | --soft-clip-low-quality-ends false                                                              |
| --allele-informative-reads-overlap-margin 2                                        | --allele-informative-reads-overlap-margin 2                                                     |
| --smith-waterman-dangling-end-match-value 25                                       | --smith-waterman-dangling-end-match-value 25                                                    |
| --smith-waterman-dangling-end-mismatch-penalty -50                                 | --smith-waterman-dangling-end-mismatch-penalty -50                                              |
| --smith-waterman-dangling-end-gap-open-penalty -110                                | --smith-waterman-dangling-end-gap-open-penalty -110                                             |
| --smith-waterman-dangling-end-gap-extend-penalty -6                                | --smith-waterman-dangling-end-gap-extend-penalty -6                                             |
| --smith-waterman-haplotype-to-reference-match-value 200                            | --smith-waterman-haplotype-to-reference-match-value 200                                         |
| --smith-waterman-haplotype-to-reference-gap-open-penalty -260                      | --smith-waterman-haplotype-to-reference-gap-open-penalty -260                                   |
| --smith-waterman-haplotype-to-reference-mismatch-penalty -150                      | --smith-waterman-haplotype-to-reference-mismatch-penalty -150                                   |
| --smith-waterman-haplotype-to-reference-gap-extend-penalty -11                     | --smith-waterman-haplotype-to-reference-gap-extend-penalty -11                                  |
| --smith-waterman-read-to-haplotype-match-value 10                                  | --smith-waterman-read-to-haplotype-match-value 10                                               |
| --smith-waterman-read-to-haplotype-mismatch-penalty -15                            | --smith-waterman-read-to-haplotype-mismatch-penalty -15                                         |
| --smith-waterman-read-to-haplotype-gap-open-penalty -30                            | --smith-waterman-read-to-haplotype-gap-open-penalty -30                                         |
| --smith-waterman-read-to-haplotype-gap-extend-penalty -5                           | --smith-waterman-read-to-haplotype-gap-extend-penalty -5                                        |
| --min-assembly-region-size 50                                                      | --min-assembly-region-size 50                                                                   |
| --max-assembly-region-size 300                                                     | --max-assembly-region-size 300                                                                  |
| --active-probability-threshold 0.002                                               | --active-probability-threshold 0.002                                                            |
| --max-prob-propagation-distance 50                                                 | --max-prob-propagation-distance 50                                                              |
| --force-active false                                                               | --force-active false                                                                            |
| --assembly-region-padding 100                                                      | --assembly-region-padding 100                                                                   |
| --padding-around-indels 75                                                         | --padding-around-indels 75                                                                      |
| --padding-around-snps 20                                                           | --padding-around-snps 20                                                                        |
| --padding-around-strs 75                                                           | --padding-around-strs 75                                                                        |
| --max-extension-into-assembly-region-padding-legacy 25                             | --max-extension-into-assembly-region-padding-legacy 25                                          |
| --max-reads-per-alignment-start 50                                                 | --max-reads-per-alignment-start 50                                                              |
| --enable-legacy-assembly-region-trimming false                                     | --enable-legacy-assembly-region-trimming false                                                  |
| --interval-set-rule UNION                                                          | --interval-set-rule UNION                                                                       |
| --interval-padding 0                                                               | --interval-padding 0                                                                            |
| --interval-exclusion-padding 0                                                     | --interval-exclusion-padding 0                                                                  |
| --interval-merging-rule ALL                                                        | --interval-merging-rule ALL                                                                     |
| --read-validation-stringency SILENT                                                | --read-validation-stringency SILENT                                                             |
| --seconds-between-progress-updates 10.0                                            | --seconds-between-progress-updates 10.0                                                         |
| --disable-sequence-dictionary-validation false                                     | --disable-sequence-dictionary-validation false                                                  |
| --create-output-bam-index true                                                     | --create-output-bam-index true                                                                  |
| --create-output-bam-md5 false                                                      | --create-output-bam-md5 false                                                                   |
| --create-output-variant-index true                                                 | --create-output-variant-index true                                                              |
| --create-output-variant-md5 false                                                  | --create-output-variant-md5 false                                                               |
| --max-variants-per-shard 0                                                         | --max-variants-per-shard 0                                                                      |
| --lenient false                                                                    | --lenient false                                                                                 |
| --add-output-sam-program-record true                                               | --add-output-sam-program-record true                                                            |
| --add-output-vcf-command-line true                                                 | --add-output-vcf-command-line true                                                              |
| --cloud-prefetch-buffer 40                                                         | --cloud-prefetch-buffer 40                                                                      |
| --cloud-index-prefetch-buffer -1                                                   | --cloud-index-prefetch-buffer -1                                                                |
| --disable-bam-index-caching false                                                  | --disable-bam-index-caching false                                                               |
| --sites-only-vcf-output false                                                      | --sites-only-vcf-output false                                                                   |
| --help false                                                                       | --help false                                                                                    |
| --version false                                                                    | --version false                                                                                 |
| --showHidden false                                                                 | --showHidden false                                                                              |
| --verbosity INFO                                                                   | --QUIET false                                                                                   |
| --QUIET false                                                                      | --use-jdk-deflater false                                                                        |
| --use-jdk-deflater false                                                           | --use-jdk-inflater false                                                                        |
| --use-jdk-inflater false                                                           | --gcs-max-retries 20                                                                            |
| --gcs-max-retries 20                                                               | --gcs-project-for-requester-pays                                                                |
| --gcs-project-for-requester-pays                                                   | --disable-tool-default-read-filters false                                                       |
| --disable-tool-default-read-filters false                                          | --minimum-mapping-quality 20                                                                    |
| --minimum-mapping-quality 20                                                       | --disable-tool-default-annotations false                                                        |
| --disable-tool-default-annotations false                                           | --enable-all-annotations false                                                                  |
| --enable-all-annotations false                                                     | --allow-old-rms-mapping-quality-annotation-data false                                           |
| --allow-old-rms-mapping-quality-annotation-data false                              |                                                                                                 |

****** DONE filter depth : Toujours la même différence...
CLOSED: [2023-01-07 Sat 00:05]
$ grep '^NC' filter-depth.vcf | wc -l
82054

$ zgrep '^NC' /Work/Groups/bisonex/ref_63003856_S135/63003856_S135_DP_over_30.vcf.gz | wc -l
82033

Non lié à la profondeur : on teste avec
bcftools filter -i 'FORMAT/DP<=30' filter-depth.vcf
bcftools filter -i 'FORMAT/AD[0:1]<=10' filter-depth.vcf

****** DONE Vérifier qu'en utilsant 2 filtres différents on a bien la même chose : oui
CLOSED: [2023-01-07 Sat 00:05]
$ bcftools filter -e 'FORMAT/DP<=30' 63003856_S135.vcf.gz | bcftools filter -e 'FORMAT/AD[0:1]<=10' -o two-filters.vcf
$ grep '^NC' two-filters.vcf | wc -l
82054
***** DONE Tester bwa en séquentiel
CLOSED: [2023-01-07 Sat 00:05]

**** DONE (save) Version d'Alexis 24 threads, sans télécharger les bases de données, gatk 4.2.4.1
CLOSED: [2023-01-07 Sat 00:06]
***** Bwa mem 24 threads: comme la version test...
$ cd /Work/Users/apraga/bisonex/script/files/tmp_63003856_S135
$ samtools view -c 63003856_S135.bam
128077211

$ samtools view -c /Work/Groups/bisonex/ref_63003856_S135/63003856_S135.bam
128077207

En ne conservant que les mapped reads, minime différence
$ samtools view -c -F 260 /Work/Groups/bisonex/ref_63003856_S135/63003856_S135.bam
127941051
$ samtools view -c -F 260 files/tmp_63003856_S135/63003856_S135.bam
127941054

**** DONE Version prod à la maison
CLOSED: [2023-01-15 Sun 23:22]
preprocessing + variant calling sans alignement
***** DONE GATK nix
CLOSED: [2023-01-15 Sun 23:22]
❯ samtools view -c 63003856_S135_marked_dup.bam
128077211
On a le même nombre
***** DONE GATK compilé: idem
CLOSED: [2023-01-15 Sun 23:22]
script/files-src/tmp_63003856_S135 on  prod [$!?]
❯ samtools view -c 63003856_S135_marked_dup.bam
128077211

Idem...
**** KILL GATK : tests de non régression ??
CLOSED: [2023-05-23 Tue 08:44]
**** DONE Compiler gatk: idem
CLOSED: [2023-01-19 Thu 22:03]
***** DONE Compilation
CLOSED: [2023-01-14 Sat 22:45]
Requirements :
- java8 avec jDK
- git lfs

#+begin_src sh
git clone https://github.com/broadinstitute/gatk
git checkout tags/4.2.4.0 -b 4.2.4.0
nix-shell -p jdk8 git-lfs

# We need the datasets for testing (otherwise it fails)
git lfs install
git lfs pull --include src/main/resources/large
./gradlew bundle

#+end_src
***** DONE Vérifier tests
CLOSED: [2023-01-19 Thu 22:03]
#+begin_src
./gradlew test
#+end_src
267064 tests completed, 444 failed, 1966 skipped
> There were failing tests. See the report at: file:///Home/Users/apraga/gatk/build/reports/tests/test/index.html
BUILD FAILED in 37m 1s
6 actionable tasks: 1 executed, 5 up-to-date

Ceux qui ont planté sont liés à spark + erreurs d'authentification kerberos
***** KILL Lancer calcul : maison
CLOSED: [2023-01-19 Thu 22:03]
 JAVA_HOME=/nix/store/r1r5jr7gv6hcchpiggjmfqjkzbi8y5ja-openjdk-8u322-ga/lib/openjdk PATH=$PATH:$JAVA_HOME/bin ./gatk --version

**** KILL Comparer tsv à la sortie (Alexis)
CLOSED: [2023-05-23 Tue 08:44]
- diff
- Nombre de lignes
**** KILL Version de prod + nix sur Helios
CLOSED: [2023-05-23 Tue 08:45]
***** DONE 24 coeurs : idem
CLOSED: [2023-01-20 Fri 22:49]
****** DONE Preprocessing : idem
CLOSED: [2023-01-20 Fri 22:49]
******* DONE Alignement
CLOSED: [2023-01-19 Thu 22:28]
******** DONE Nombre lignes : idem
CLOSED: [2023-01-19 Thu 22:48]
/Work/Users/apraga/bisonex/script/files/tmp_63003856_S135〉samtools view -c 63003856_S135.bam                                                                                          01/19/2023 10:16:19 PM
128077211

/Work/Projects/bisonex/ref-63003856_S135〉samtools view -c 63003856_S135.bam                                                                                                           01/19/2023 10:11:39 PM
128077207
******** DONE Bwa version ok, arguments ok
CLOSED: [2023-01-20 Fri 22:49]
/Work/Users/apraga/bisonex/script/files/tmp_63003856_S135〉samtools view -H 63003856_S135.bam | grep PN                                                                                01/19/2023 10:46:55 PM
@PG	ID:bwa	PN:bwa	VN:0.7.17-r1188	CL:bwa mem -R @RG\tID:63003856_S135\tSM:63003856_S135\tPL:ILLUMINA\tPM:Miseq\tCN:CHU_Minjoz\tLB:definition_to_add -v 2 -t 24 /Work/Groups/bisonex/data-alexis-reference/genome/GRCh38_latest_genomic.fna files/fastq/63003856_S135_R1_001.fastq.gz files/fastq/63003856_S135_R2_001.fastq.gz
@PG	ID:samtools	PN:samtools	PP:bwa	VN:1.13	CL:samtools sort -@ 24 -O BAM -o files/tmp_63003856_S135/63003856_S135.bam
@PG	ID:samtools.1	PN:samtools	PP:samtools	VN:1.13	CL:samtools view -H 63003856_S135.bam

/Work/Projects/bisonex/ref-63003856_S135〉samtools view -H 63003856_S135.bam | grep PN                                                                                                 01/19/2023 10:49:25 PM
@PG	ID:bwa	PN:bwa	VN:0.7.17-r1188	CL:/bin/bwa mem -R @RG\tID:63003856_S135\tSM:63003856_S135\tPL:ILLUMINA\tPM:Miseq\tCN:CHU_Minjoz\tLB:definition_to_add -t 4 /mnt/j/bases_de_donnees/genome/GRCh38_latest_genomic.fna /mnt/j/working_directory_pipeline_analyse_exome/fastq/2200467051_63003856/63003856_S135_R1_001.fastq.gz /mnt/j/working_directory_pipeline_analyse_exome/fastq/2200467051_63003856/63003856_S135_R2_001.fastq.gz
@PG	ID:samtools	PN:samtools	PP:bwa	VN:1.10	CL:/mnt/h/tools/samtools sort -@ 4 -O BAM -o /mnt/j/working_directory_pipeline_analyse_exome/tmp_63003856_S135/63003856_S135.bam
@PG	ID:samtools.1	PN:samtools	PP:samtools	VN:1.13	CL:samtools view -H 63003856_S135.bam
******* KILL ApplyBQSR
CLOSED: [2023-01-20 Fri 22:48]
/Work/Users/apraga/bisonex/script/files/bam〉samtools view -c 63003856_S135_recalibrated_hg38.bam                                                                                      01/19/2023 10:21:00 PM
128077211

??
****** DONE Variant calling
CLOSED: [2023-01-20 Fri 22:48]
******* DONE Re vérifier flags
CLOSED: [2023-01-19 Thu 22:44]
/Work/Projects/bisonex/ref-63003856_S135〉less 63003856_S135_DP_over_30.vcf.gz
##GATKCommandLine=<ID=HaplotypeCaller,CommandLine="HaplotypeCaller

| Ref                                                                                             | Prod helios                                                                             |
|-------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------|
| --dbsnp /mnt/j/bases_de_donnees/dbSNP/GCF_000001405.39.gz                                       | --dbsnp /Work/Groups/bisonex/data-alexis-reference/dbSNP/GCF_000001405.39.gz            |
| --max-mnp-distance 2                                                                            | --max-mnp-distance 2                                                                    |
| --output /mnt/j/working_directory_pipeline_analyse_exome/vcf/63003856_S135.vcf                  | --output files/vcf/63003856_S135.vcf                                                    |
| --input /mnt/j/working_directory_pipeline_analyse_exome/bam/63003856_S135_recalibrated_hg38.bam | --input files/bam/63003856_S135_recalibrated_hg38.bam                                   |
| --reference /mnt/j/bases_de_donnees/genome/GRCh38_latest_genomic.fna                            | --reference /Work/Groups/bisonex/data-alexis-reference/genome/GRCh38_latest_genomic.fna |
| --verbosity WARNING                                                                             | --verbosity WARNING                                                                     |
| --use-posteriors-to-calculate-qual false                                                        | --use-posteriors-to-calculate-qual false                                                |
| --dont-use-dragstr-priors false                                                                 | --dont-use-dragstr-priors false                                                         |
| --use-new-qual-calculator true                                                                  | --use-new-qual-calculator true                                                          |
| --annotate-with-num-discovered-alleles false                                                    | --annotate-with-num-discovered-alleles false                                            |
| --heterozygosity 0.001                                                                          | --heterozygosity 0.001                                                                  |
| --indel-heterozygosity 1.25E-4                                                                  | --indel-heterozygosity 1.25E-4                                                          |
| --heterozygosity-stdev 0.01                                                                     | --heterozygosity-stdev 0.01                                                             |
| --standard-min-confidence-threshold-for-calling 30.0                                            | --standard-min-confidence-threshold-for-calling 30.0                                    |
| --max-alternate-alleles 6                                                                       | --max-alternate-alleles 6                                                               |
| --max-genotype-count 1024                                                                       | --max-genotype-count 1024                                                               |
| --sample-ploidy 2                                                                               | --sample-ploidy 2                                                                       |
| --num-reference-samples-if-no-call 0                                                            | --num-reference-samples-if-no-call 0                                                    |
| --genotype-assignment-method USE_PLS_TO_ASSIGN                                                  | --genotype-assignment-method USE_PLS_TO_ASSIGN                                          |
| --contamination-fraction-to-filter 0.0                                                          | --contamination-fraction-to-filter 0.0                                                  |
| --output-mode EMIT_VARIANTS_ONLY                                                                | --output-mode EMIT_VARIANTS_ONLY                                                        |
| --all-site-pls false                                                                            | --all-site-pls false                                                                    |
| --gvcf-gq-bands 1                                                                               | --gvcf-gq-bands 1                                                                       |
| --gvcf-gq-bands 2                                                                               | --gvcf-gq-bands 2                                                                       |
| --gvcf-gq-bands 3                                                                               | --gvcf-gq-bands 3                                                                       |
| --gvcf-gq-bands 4                                                                               | --gvcf-gq-bands 4                                                                       |
| --gvcf-gq-bands 5                                                                               | --gvcf-gq-bands 5                                                                       |
| --gvcf-gq-bands 6                                                                               | --gvcf-gq-bands 6                                                                       |
| --gvcf-gq-bands 7                                                                               | --gvcf-gq-bands 7                                                                       |
| --gvcf-gq-bands 8                                                                               | --gvcf-gq-bands 8                                                                       |
| --gvcf-gq-bands 9                                                                               | --gvcf-gq-bands 9                                                                       |
| --gvcf-gq-bands 10                                                                              | --gvcf-gq-bands 10                                                                      |
| --gvcf-gq-bands 11                                                                              | --gvcf-gq-bands 11                                                                      |
| --gvcf-gq-bands 12                                                                              | --gvcf-gq-bands 12                                                                      |
| --gvcf-gq-bands 13                                                                              | --gvcf-gq-bands 13                                                                      |
| --gvcf-gq-bands 14                                                                              | --gvcf-gq-bands 14                                                                      |
| --gvcf-gq-bands 15                                                                              | --gvcf-gq-bands 15                                                                      |
| --gvcf-gq-bands 16                                                                              | --gvcf-gq-bands 16                                                                      |
| --gvcf-gq-bands 17                                                                              | --gvcf-gq-bands 17                                                                      |
| --gvcf-gq-bands 18                                                                              | --gvcf-gq-bands 18                                                                      |
| --gvcf-gq-bands 19                                                                              | --gvcf-gq-bands 19                                                                      |
| --gvcf-gq-bands 20                                                                              | --gvcf-gq-bands 20                                                                      |
| --gvcf-gq-bands 21                                                                              | --gvcf-gq-bands 21                                                                      |
| --gvcf-gq-bands 22                                                                              | --gvcf-gq-bands 22                                                                      |
| --gvcf-gq-bands 23                                                                              | --gvcf-gq-bands 23                                                                      |
| --gvcf-gq-bands 24                                                                              | --gvcf-gq-bands 24                                                                      |
| --gvcf-gq-bands 25                                                                              | --gvcf-gq-bands 25                                                                      |
| --gvcf-gq-bands 26                                                                              | --gvcf-gq-bands 26                                                                      |
| --gvcf-gq-bands 27                                                                              | --gvcf-gq-bands 27                                                                      |
| --gvcf-gq-bands 28                                                                              | --gvcf-gq-bands 28                                                                      |
| --gvcf-gq-bands 29                                                                              | --gvcf-gq-bands 29                                                                      |
| --gvcf-gq-bands 30                                                                              | --gvcf-gq-bands 30                                                                      |
| --gvcf-gq-bands 31                                                                              | --gvcf-gq-bands 31                                                                      |
| --gvcf-gq-bands 32                                                                              | --gvcf-gq-bands 32                                                                      |
| --gvcf-gq-bands 33                                                                              | --gvcf-gq-bands 33                                                                      |
| --gvcf-gq-bands 34                                                                              | --gvcf-gq-bands 34                                                                      |
| --gvcf-gq-bands 35                                                                              | --gvcf-gq-bands 35                                                                      |
| --gvcf-gq-bands 36                                                                              | --gvcf-gq-bands 36                                                                      |
| --gvcf-gq-bands 37                                                                              | --gvcf-gq-bands 37                                                                      |
| --gvcf-gq-bands 38                                                                              | --gvcf-gq-bands 38                                                                      |
| --gvcf-gq-bands 39                                                                              | --gvcf-gq-bands 39                                                                      |
| --gvcf-gq-bands 40                                                                              | --gvcf-gq-bands 40                                                                      |
| --gvcf-gq-bands 41                                                                              | --gvcf-gq-bands 41                                                                      |
| --gvcf-gq-bands 42                                                                              | --gvcf-gq-bands 42                                                                      |
| --gvcf-gq-bands 43                                                                              | --gvcf-gq-bands 43                                                                      |
| --gvcf-gq-bands 44                                                                              | --gvcf-gq-bands 44                                                                      |
| --gvcf-gq-bands 45                                                                              | --gvcf-gq-bands 45                                                                      |
| --gvcf-gq-bands 46                                                                              | --gvcf-gq-bands 46                                                                      |
| --gvcf-gq-bands 47                                                                              | --gvcf-gq-bands 47                                                                      |
| --gvcf-gq-bands 48                                                                              | --gvcf-gq-bands 48                                                                      |
| --gvcf-gq-bands 49                                                                              | --gvcf-gq-bands 49                                                                      |
| --gvcf-gq-bands 50                                                                              | --gvcf-gq-bands 50                                                                      |
| --gvcf-gq-bands 51                                                                              | --gvcf-gq-bands 51                                                                      |
| --gvcf-gq-bands 52                                                                              | --gvcf-gq-bands 52                                                                      |
| --gvcf-gq-bands 53                                                                              | --gvcf-gq-bands 53                                                                      |
| --gvcf-gq-bands 54                                                                              | --gvcf-gq-bands 54                                                                      |
| --gvcf-gq-bands 55                                                                              | --gvcf-gq-bands 55                                                                      |
| --gvcf-gq-bands 56                                                                              | --gvcf-gq-bands 56                                                                      |
| --gvcf-gq-bands 57                                                                              | --gvcf-gq-bands 57                                                                      |
| --gvcf-gq-bands 58                                                                              | --gvcf-gq-bands 58                                                                      |
| --gvcf-gq-bands 59                                                                              | --gvcf-gq-bands 59                                                                      |
| --gvcf-gq-bands 60                                                                              | --gvcf-gq-bands 60                                                                      |
| --gvcf-gq-bands 70                                                                              | --gvcf-gq-bands 70                                                                      |
| --gvcf-gq-bands 80                                                                              | --gvcf-gq-bands 80                                                                      |
| --gvcf-gq-bands 90                                                                              | --gvcf-gq-bands 90                                                                      |
| --gvcf-gq-bands 99                                                                              | --gvcf-gq-bands 99                                                                      |
| --floor-blocks false                                                                            | --floor-blocks false                                                                    |
| --indel-size-to-eliminate-in-ref-model 10                                                       | --indel-size-to-eliminate-in-ref-model 10                                               |
| --disable-optimizations false                                                                   | --disable-optimizations false                                                           |
| --dragen-mode false                                                                             | --dragen-mode false                                                                     |
| --apply-bqd false                                                                               | --apply-bqd false                                                                       |
| --apply-frd false                                                                               | --apply-frd false                                                                       |
| --disable-spanning-event-genotyping false                                                       | --disable-spanning-event-genotyping false                                               |
| --transform-dragen-mapping-quality false                                                        | --transform-dragen-mapping-quality false                                                |
| --mapping-quality-threshold-for-genotyping 20                                                   | --mapping-quality-threshold-for-genotyping 20                                           |
| --max-effective-depth-adjustment-for-frd 0                                                      | --max-effective-depth-adjustment-for-frd 0                                              |
| --just-determine-active-regions false                                                           | --just-determine-active-regions false                                                   |
| --dont-genotype false                                                                           | --dont-genotype false                                                                   |
| --do-not-run-physical-phasing false                                                             | --do-not-run-physical-phasing false                                                     |
| --do-not-correct-overlapping-quality false                                                      | --do-not-correct-overlapping-quality false                                              |
| --use-filtered-reads-for-annotations false                                                      | --use-filtered-reads-for-annotations false                                              |
| --adaptive-pruning false                                                                        | --adaptive-pruning false                                                                |
| --do-not-recover-dangling-branches false                                                        | --do-not-recover-dangling-branches false                                                |
| --recover-dangling-heads false                                                                  | --recover-dangling-heads false                                                          |
| --kmer-size 10                                                                                  | --kmer-size 10                                                                          |
| --kmer-size 25                                                                                  | --kmer-size 25                                                                          |
| --dont-increase-kmer-sizes-for-cycles false                                                     | --dont-increase-kmer-sizes-for-cycles false                                             |
| --allow-non-unique-kmers-in-ref false                                                           | --allow-non-unique-kmers-in-ref false                                                   |
| --num-pruning-samples 1                                                                         | --num-pruning-samples 1                                                                 |
| --min-dangling-branch-length 4                                                                  | --min-dangling-branch-length 4                                                          |
| --recover-all-dangling-branches false                                                           | --recover-all-dangling-branches false                                                   |
| --max-num-haplotypes-in-population 128                                                          | --max-num-haplotypes-in-population 128                                                  |
| --min-pruning 2                                                                                 | --min-pruning 2                                                                         |
| --adaptive-pruning-initial-error-rate 0.001                                                     | --adaptive-pruning-initial-error-rate 0.001                                             |
| --pruning-lod-threshold 2.302585092994046                                                       | --pruning-lod-threshold 2.302585092994046                                               |
| --pruning-seeding-lod-threshold 9.210340371976184                                               | --pruning-seeding-lod-threshold 9.210340371976184                                       |
| --max-unpruned-variants 100                                                                     | --max-unpruned-variants 100                                                             |
| --linked-de-bruijn-graph false                                                                  | --linked-de-bruijn-graph false                                                          |
| --disable-artificial-haplotype-recovery false                                                   | --disable-artificial-haplotype-recovery false                                           |
| --enable-legacy-graph-cycle-detection false                                                     | --enable-legacy-graph-cycle-detection false                                             |
| --debug-assembly false                                                                          | --debug-assembly false                                                                  |
| --debug-graph-transformations false                                                             | --debug-graph-transformations false                                                     |
| --capture-assembly-failure-bam false                                                            | --capture-assembly-failure-bam false                                                    |
| --num-matching-bases-in-dangling-end-to-recover -1                                              | --num-matching-bases-in-dangling-end-to-recover -1                                      |
| --error-correction-log-odds -Infinity                                                           | --error-correction-log-odds -Infinity                                                   |
| --error-correct-reads false                                                                     | --error-correct-reads false                                                             |
| --kmer-length-for-read-error-correction 25                                                      | --kmer-length-for-read-error-correction 25                                              |
| --min-observations-for-kmer-to-be-solid 20                                                      | --min-observations-for-kmer-to-be-solid 20                                              |
| --base-quality-score-threshold 18                                                               | --base-quality-score-threshold 18                                                       |
| --dragstr-het-hom-ratio 2                                                                       | --dragstr-het-hom-ratio 2                                                               |
| --dont-use-dragstr-pair-hmm-scores false                                                        | --dont-use-dragstr-pair-hmm-scores false                                                |
| --pair-hmm-gap-continuation-penalty 10                                                          | --pair-hmm-gap-continuation-penalty 10                                                  |
| --expected-mismatch-rate-for-read-disqualification 0.02                                         | --expected-mismatch-rate-for-read-disqualification 0.02                                 |
| --pair-hmm-implementation FASTEST_AVAILABLE                                                     | --pair-hmm-implementation FASTEST_AVAILABLE                                             |
| --pcr-indel-model CONSERVATIVE                                                                  | --pcr-indel-model CONSERVATIVE                                                          |
| --phred-scaled-global-read-mismapping-rate 45                                                   | --phred-scaled-global-read-mismapping-rate 45                                           |
| --disable-symmetric-hmm-normalizing false                                                       | --disable-symmetric-hmm-normalizing false                                               |
| --disable-cap-base-qualities-to-map-quality false                                               | --disable-cap-base-qualities-to-map-quality false                                       |
| --enable-dynamic-read-disqualification-for-genotyping false                                     | --enable-dynamic-read-disqualification-for-genotyping false                             |
| --dynamic-read-disqualification-threshold 1.0                                                   | --dynamic-read-disqualification-threshold 1.0                                           |
| --native-pair-hmm-threads 4                                                                     | --native-pair-hmm-threads 4                                                             |
| --native-pair-hmm-use-double-precision false                                                    | --native-pair-hmm-use-double-precision false                                            |
| --bam-writer-type CALLED_HAPLOTYPES                                                             | --bam-writer-type CALLED_HAPLOTYPES                                                     |
| --dont-use-soft-clipped-bases false                                                             | --dont-use-soft-clipped-bases false                                                     |
| --min-base-quality-score 10                                                                     | --min-base-quality-score 10                                                             |
| --smith-waterman JAVA                                                                           | --smith-waterman JAVA                                                                   |
| --emit-ref-confidence NONE                                                                      | --emit-ref-confidence NONE                                                              |
| --force-call-filtered-alleles false                                                             | --force-call-filtered-alleles false                                                     |
| --soft-clip-low-quality-ends false                                                              | --soft-clip-low-quality-ends false                                                      |
| --allele-informative-reads-overlap-margin 2                                                     | --allele-informative-reads-overlap-margin 2                                             |
| --smith-waterman-dangling-end-match-value 25                                                    | --smith-waterman-dangling-end-match-value 25                                            |
| --smith-waterman-dangling-end-mismatch-penalty -50                                              | --smith-waterman-dangling-end-mismatch-penalty -50                                      |
| --smith-waterman-dangling-end-gap-open-penalty -110                                             | --smith-waterman-dangling-end-gap-open-penalty -110                                     |
| --smith-waterman-dangling-end-gap-extend-penalty -6                                             | --smith-waterman-dangling-end-gap-extend-penalty -6                                     |
| --smith-waterman-haplotype-to-reference-match-value 200                                         | --smith-waterman-haplotype-to-reference-match-value 200                                 |
| --smith-waterman-haplotype-to-reference-mismatch-penalty -150                                   | --smith-waterman-haplotype-to-reference-mismatch-penalty -150                           |
| --smith-waterman-haplotype-to-reference-gap-open-penalty -260                                   | --smith-waterman-haplotype-to-reference-gap-open-penalty -260                           |
| --smith-waterman-haplotype-to-reference-gap-extend-penalty -11                                  | --smith-waterman-haplotype-to-reference-gap-extend-penalty -11                          |
| --smith-waterman-read-to-haplotype-match-value 10                                               | --smith-waterman-read-to-haplotype-match-value 10                                       |
| --smith-waterman-read-to-haplotype-mismatch-penalty -15                                         | --smith-waterman-read-to-haplotype-mismatch-penalty -15                                 |
| --smith-waterman-read-to-haplotype-gap-open-penalty -30                                         | --smith-waterman-read-to-haplotype-gap-open-penalty -30                                 |
| --smith-waterman-read-to-haplotype-gap-extend-penalty -5                                        | --smith-waterman-read-to-haplotype-gap-extend-penalty -5                                |
| --min-assembly-region-size 50                                                                   | --min-assembly-region-size 50                                                           |
| --max-assembly-region-size 300                                                                  | --max-assembly-region-size 300                                                          |
| --active-probability-threshold 0.002                                                            | --active-probability-threshold 0.002                                                    |
| --max-prob-propagation-distance 50                                                              | --max-prob-propagation-distance 50                                                      |
| --force-active false                                                                            | --force-active false                                                                    |
| --assembly-region-padding 100                                                                   | --assembly-region-padding 100                                                           |
| --padding-around-indels 75                                                                      | --padding-around-indels 75                                                              |
| --padding-around-snps 20                                                                        | --padding-around-snps 20                                                                |
| --padding-around-strs 75                                                                        | --padding-around-strs 75                                                                |
| --max-extension-into-assembly-region-padding-legacy 25                                          | --max-extension-into-assembly-region-padding-legacy 25                                  |
| --max-reads-per-alignment-start 50                                                              | --max-reads-per-alignment-start 50                                                      |
| --enable-legacy-assembly-region-trimming false                                                  | --enable-legacy-assembly-region-trimming false                                          |
| --interval-set-rule UNION                                                                       | --interval-set-rule UNION                                                               |
| --interval-padding 0                                                                            | --interval-padding 0                                                                    |
| --interval-exclusion-padding 0                                                                  | --interval-exclusion-padding 0                                                          |
| --interval-merging-rule ALL                                                                     | --interval-merging-rule ALL                                                             |
| --read-validation-stringency SILENT                                                             | --read-validation-stringency SILENT                                                     |
| --seconds-between-progress-updates 10.0                                                         | --seconds-between-progress-updates 10.0                                                 |
| --disable-sequence-dictionary-validation false                                                  | --disable-sequence-dictionary-validation false                                          |
| --create-output-bam-index true                                                                  | --create-output-bam-index true                                                          |
| --create-output-bam-md5 false                                                                   | --create-output-bam-md5 false                                                           |
| --create-output-variant-index true                                                              | --create-output-variant-index true                                                      |
| --create-output-variant-md5 false                                                               | --create-output-variant-md5 false                                                       |
| --max-variants-per-shard 0                                                                      | --max-variants-per-shard 0                                                              |
| --lenient false                                                                                 | --lenient false                                                                         |
| --add-output-sam-program-record true                                                            | --add-output-sam-program-record true                                                    |
| --add-output-vcf-command-line true                                                              | --add-output-vcf-command-line true                                                      |
| --cloud-prefetch-buffer 40                                                                      | --cloud-prefetch-buffer 40                                                              |
| --cloud-index-prefetch-buffer -1                                                                | --cloud-index-prefetch-buffer -1                                                        |
| --disable-bam-index-caching false                                                               | --disable-bam-index-caching false                                                       |
| --sites-only-vcf-output false                                                                   | --sites-only-vcf-output false                                                           |
| --help false                                                                                    | --help false                                                                            |
| --version false                                                                                 | --version false                                                                         |
| --showHidden false                                                                              | --showHidden false                                                                      |
| --QUIET false                                                                                   | --QUIET false                                                                           |
| --use-jdk-deflater false                                                                        | --use-jdk-deflater false                                                                |
| --use-jdk-inflater false                                                                        | --use-jdk-inflater false                                                                |
| --gcs-max-retries 20                                                                            | --gcs-max-retries 20                                                                    |
| --gcs-project-for-requester-pays                                                                | --gcs-project-for-requester-pays                                                        |
| --disable-tool-default-read-filters false                                                       | --disable-tool-default-read-filters false                                               |
| --minimum-mapping-quality 20                                                                    | --minimum-mapping-quality 20                                                            |
| --disable-tool-default-annotations false                                                        | --disable-tool-default-annotations false                                                |
| --enable-all-annotations false                                                                  | --enable-all-annotations false                                                          |
| --allow-old-rms-mapping-quality-annotation-data false                                           | --allow-old-rms-mapping-quality-annotation-data false"                                  |
| ",Version="4.2.4.1"                                                                             | Version="4.2.4.1",                                                                      |
Prod helios
/Work/Users/apraga/bisonex/script/files/tmp_63003856_S135〉less 63003856_S135_DP_over_30.vcf
##GATKCommandLine=<ID=HaplotypeCaller,CommandLine="HaplotypeCaller
,Date="January 10, 2023 at 12:26:57 AM CET">
******* DONE VCF : même différence
CLOSED: [2023-01-20 Fri 22:48]
/Work/Projects/bisonex/ref-63003856_S135〉ls *.vcf* | insert nblines  {|e| (^zgrep -v '^#' $e.name | wc -l)} | select name nblines                                                     01/14/2023 08:04:02 PM
╭───┬───────────────────────────────────────────────────────────────────────────┬─────────╮
│ # │                                   name                                    │ nblines │
├───┼───────────────────────────────────────────────────────────────────────────┼─────────┤
│ 0 │ 63003856_S135_DP_over_30.vcf.gz                                           │ 84708   │
│ 1 │ 63003856_S135_DP_over_30.vcf.gz.tbi                                       │ 1       │
│ 2 │ 63003856_S135_DP_over_30_not_SNP.recode.vcf                               │ 11362   │
│ 3 │ 63003856_S135_DP_over_30_not_SNP_consensual_sequence.vcf                  │ 8864    │
│ 4 │ 63003856_S135_DP_over_30_not_SNP_consensual_sequence_not_technical.vcf.gz │ 6478    │
╰───┴───────────────────────────────────────────────────────────────────────────┴─────────╯
/Work/Users/apraga/bisonex/script/files/tmp_63003856_S135〉ls *.vcf* | insert nblines  {|e| (^zgrep -v '^#' $e.name | wc -l)} | select name nblines                                    01/14/2023 08:05:23 PM
╭───┬───────────────────────────────────────────────────────────────────────────┬─────────╮
│ # │                                   name                                    │ nblines │
├───┼───────────────────────────────────────────────────────────────────────────┼─────────┤
│ 0 │ 63003856_S135_DP_over_30.vcf                                              │ 84724   │
│ 1 │ 63003856_S135_DP_over_30_not_SNP.recode.vcf                               │ 11377   │
│ 2 │ 63003856_S135_DP_over_30_not_SNP_consensual_sequence.vcf                  │ 8884    │
│ 3 │ 63003856_S135_DP_over_30_not_SNP_consensual_sequence_not_technical.vcf.gz │ 6759    │
╰───┴───────────────────────────────────────────────────────────────────────────┴─────────╯
***** KILL Relancer avec 4 coeurs
CLOSED: [2023-05-23 Tue 08:45]
-c 4
****** DONE Alignement ok !!
CLOSED: [2023-01-20 Fri 23:24]
$ samtools view -c files/tmp_63003856_S135/63003856_S135.bam
128077207

****** KILL Vérifier les flags de chaque étape
CLOSED: [2023-05-23 Tue 08:44]
post_cleanSam = _cleaned.bam
post_markDuplicate = _marked_dup.bam
post_BaseRecalibrator = _recal.table
post_ApplyBQSR = _recalibrated_hg38.bam
post_haplotypecaller = .vcf
post_depth_filter = _DP_over_30.vcf
post_exclude_SNP = _DP_over_30_not_SNP
post_consensual = _DP_over_30_not_SNP_consensual_sequence.vcf
post_technical = _DP_over_30_not_SNP_consensual_sequence_not_technical.vcf.gz

******* DONE Clean sam
CLOSED: [2023-01-20 Fri 23:44]
ref
$ samtools view -H 63003856_S135_cleaned.bam  | grep PN
@PG	ID:samtools	PN:samtools	PP:bwa	VN:1.10	CL:/mnt/h/tools/samtools sort -@ 4 -O BAM -o /mnt/j/working_directory_pipeline_analyse_exome/tmp_63003856_S135/63003856_S135.bam

@PG	ID:samtools	PN:samtools	PP:bwa	VN:1.13	CL:samtools sort -@ 4 -O BAM -o files/tmp_63003856_S135/63003856_S135.bam
******* DONE Markduplicates
CLOSED: [2023-01-20 Fri 23:44]
ref
| @PG	ID:MarkDuplicates	VN:Version:4.2.4.1	CL:MarkDuplicates                                                    | @PG   ID:MarkDuplicates   VN:Version:4.2.4.1  CL:MarkDuplicates                            |   |
| --INPUT /mnt/j/working_directory_pipeline_analyse_exome/tmp_63003856_S135/63003856_S135_cleaned.bam              | --INPUT files/tmp_63003856_S135/63003856_S135_cleaned.bam                                  |   |
| --OUTPUT /mnt/j/working_directory_pipeline_analyse_exome/tmp_63003856_S135/63003856_S135_marked_dup.bam          | --OUTPUT files/tmp_63003856_S135/63003856_S135_marked_dup.bam                              |   |
| --METRICS_FILE /mnt/j/working_directory_pipeline_analyse_exome/tmp_63003856_S135/63003856_S135_marked_dup.metrix | --METRICS_FILE files/tmp_63003856_S135/63003856_S135_marked_dup.metrix                     |   |
| --VERBOSITY WARNING                                                                                              | --VERBOSITY WARNING                                                                        |   |
| --CREATE_INDEX true                                                                                              | --CREATE_INDEX true                                                                        |   |
| --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000                                                                     | --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000                                               |   |
| --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000                                                                        | --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000                                                  |   |
| --SORTING_COLLECTION_SIZE_RATIO 0.25                                                                             | --SORTING_COLLECTION_SIZE_RATIO 0.25                                                       |   |
| --TAG_DUPLICATE_SET_MEMBERS false                                                                                | --TAG_DUPLICATE_SET_MEMBERS false                                                          |   |
| --REMOVE_SEQUENCING_DUPLICATES false                                                                             | --REMOVE_SEQUENCING_DUPLICATES false                                                       |   |
| --TAGGING_POLICY DontTag                                                                                         | --TAGGING_POLICY DontTag                                                                   |   |
| --CLEAR_DT true                                                                                                  | --CLEAR_DT true                                                                            |   |
| --DUPLEX_UMI false                                                                                               | --DUPLEX_UMI false                                                                         |   |
| --ADD_PG_TAG_TO_READS true                                                                                       | --ADD_PG_TAG_TO_READS true                                                                 |   |
| --REMOVE_DUPLICATES false                                                                                        | --REMOVE_DUPLICATES false                                                                  |   |
| --ASSUME_SORTED false                                                                                            | --ASSUME_SORTED false                                                                      |   |
| --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES                                                               | --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES                                         |   |
| --PROGRAM_RECORD_ID MarkDuplicates                                                                               | --PROGRAM_RECORD_ID MarkDuplicates                                                         |   |
| --PROGRAM_GROUP_NAME MarkDuplicates                                                                              | --PROGRAM_GROUP_NAME MarkDuplicates                                                        |   |
| --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values>                       | --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> |   |
| --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100                                                                           | --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100                                                     |   |
| --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000                                                                          | --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000                                                    |   |
| --QUIET false                                                                                                    | --QUIET false                                                                              |   |
| --VALIDATION_STRINGENCY STRICT                                                                                   | --VALIDATION_STRINGENCY STRICT                                                             |   |
| --COMPRESSION_LEVEL 2                                                                                            | --COMPRESSION_LEVEL 2                                                                      |   |
| --MAX_RECORDS_IN_RAM 500000                                                                                      | --MAX_RECORDS_IN_RAM 500000                                                                |   |
| --CREATE_MD5_FILE false                                                                                          | --CREATE_MD5_FILE false                                                                    |   |
| --GA4GH_CLIENT_SECRETS client_secrets.json                                                                       | --GA4GH_CLIENT_SECRETS client_secrets.json                                                 |   |
| --help false                                                                                                     | --help false                                                                               |   |
| --version false                                                                                                  | --version false                                                                            |   |
| --showHidden false                                                                                               | --showHidden false                                                                         |   |
| --USE_JDK_DEFLATER false                                                                                         | --USE_JDK_DEFLATER false                                                                   |   |
| --USE_JDK_INFLATER false                                                                                         | --USE_JDK_INFLATER false                                                                   |   |

| @PG	ID:samtools.1	PN:samtools	PP:samtools	VN:1.13	CL:samtools view -H 63003856_S135_marked_dup.bam |
******* DONE ApplyBQSR
CLOSED: [2023-01-20 Fri 23:48]
Ref
@PG	ID:GATK ApplyBQSR	VN:4.2.4.1	CL:ApplyBQSR
| --output /mnt/j/working_directory_pipeline_analyse_exome/bam/63003856_S135_recalibrated_hg38.bam              | --output files/bam/63003856_S135_recalibrated_hg38.bam                                  |
| --bqsr-recal-file /mnt/j/working_directory_pipeline_analyse_exome/tmp_63003856_S135/63003856_S135_recal.table | --bqsr-recal-file files/tmp_63003856_S135/63003856_S135_recal.table                     |
| --input /mnt/j/working_directory_pipeline_analyse_exome/tmp_63003856_S135/63003856_S135_marked_dup.bam        | --input files/tmp_63003856_S135/63003856_S135_marked_dup.bam                            |
| --reference /mnt/j/bases_de_donnees/genome/GRCh38_latest_genomic.fna                                          | --reference /Work/Groups/bisonex/data-alexis-reference/genome/GRCh38_latest_genomic.fna |
| --verbosity WARNING                                                                                           | --verbosity WARNING                                                                     |
| --preserve-qscores-less-than 6                                                                                | --preserve-qscores-less-than 6                                                          |
| --use-original-qualities false                                                                                | --use-original-qualities false                                                          |
| --quantize-quals 0                                                                                            | --quantize-quals 0                                                                      |
| --round-down-quantized false                                                                                  | --round-down-quantized false                                                            |
| --emit-original-quals false                                                                                   | --emit-original-quals false                                                             |
| --global-qscore-prior -1.0                                                                                    | --global-qscore-prior -1.0                                                              |
| --interval-set-rule UNION                                                                                     | --interval-set-rule UNION                                                               |
| --interval-padding 0                                                                                          | --interval-padding 0                                                                    |
| --interval-exclusion-padding 0                                                                                | --interval-exclusion-padding 0                                                          |
| --interval-merging-rule ALL                                                                                   | --interval-merging-rule ALL                                                             |
| --read-validation-stringency SILENT                                                                           | --read-validation-stringency SILENT                                                     |
| --seconds-between-progress-updates 10.0                                                                       | --seconds-between-progress-updates 10.0                                                 |
| --disable-sequence-dictionary-validation false                                                                | --disable-sequence-dictionary-validation false                                          |
| --create-output-bam-index true                                                                                | --create-output-bam-index true                                                          |
| --create-output-bam-md5 false                                                                                 | --create-output-bam-md5 false                                                           |
| --create-output-variant-index true                                                                            | --create-output-variant-index true                                                      |
| --create-output-variant-md5 false                                                                             | --create-output-variant-md5 false                                                       |
| --max-variants-per-shard 0                                                                                    | --max-variants-per-shard 0                                                              |
| --lenient false                                                                                               | --lenient false                                                                         |
| --add-output-sam-program-record true                                                                          | --add-output-sam-program-record true                                                    |
| --add-output-vcf-command-line true                                                                            | --add-output-vcf-command-line true                                                      |
| --cloud-prefetch-buffer 40                                                                                    | --cloud-prefetch-buffer 40                                                              |
| --cloud-index-prefetch-buffer -1                                                                              | --cloud-index-prefetch-buffer -1                                                        |
| --disable-bam-index-caching false                                                                             | --disable-bam-index-caching false                                                       |
| --sites-only-vcf-output false                                                                                 | --sites-only-vcf-output false                                                           |
| --help false                                                                                                  | --help false                                                                            |
| --version false                                                                                               | --version false                                                                         |
| --showHidden false                                                                                            | --showHidden false                                                                      |
| --QUIET false                                                                                                 | --QUIET false                                                                           |
| --use-jdk-deflater false                                                                                      | --use-jdk-deflater false                                                                |
| --use-jdk-inflater false                                                                                      | --use-jdk-inflater false                                                                |
| --gcs-max-retries 20                                                                                          | --gcs-max-retries 20                                                                    |
| --gcs-project-for-requester-pays                                                                              | --gcs-project-for-requester-pays                                                        |
| --disable-tool-default-read-filters false	PN:GATK ApplyBQSR                                                 | --disable-tool-default-read-filters false     PN:GATK ApplyBQSR                         |

****** KILL Vérifier sha256sum
CLOSED: [2023-01-24 Tue 23:00]
alignment: différent
****** KILL Comparer bam
CLOSED: [2023-01-25 Wed 21:58]

/Work/Users/apraga/bisonex/script/files〉picard CompareSAMs LENIENT_LOW_MQ_ALIGNMENT=true LENIENT_DUP=true tmp_63003856_S135/63003856_S135.bam /Work/Groups/bisonex/ref/tmp_63003856_S135/63003856_S135.bam O=compare-bam.tsv
picard CompareSAMs -LENIENT_LOW_MQ_ALIGNMENT true -LENIENT_DUP true tmp_63003856_S135/63003856_S135.bam /Work/Groups/bisonex/ref/tmp_63003856_S135/63003856_S135.bam -O compare-bam.tsv

VN Program Record attribute differs.
File 1: 1.13
File 2: 1.10

SAM files differ.
[Tue Jan 24 23:12:50 CET 2023] picard.sam.CompareSAMs done. Elapsed time: 7.32 minutes.
***** DONE Relancer avec la même version de samtools
CLOSED: [2023-01-25 Wed 21:58]
Pas d'impact
***** KILL Comparer tsv de sortie
CLOSED: [2023-05-23 Tue 08:45]
***** KILL Regarder où sont les variants différents
CLOSED: [2023-05-23 Tue 08:45]
** TODO GIAB Validation :giab:
https://github.com/ga4gh/benchmarking-tools
Prérequis :
- [[*hap.py][hap.py]]
- [[*NA12878][NA12878]]
*** TODO GIAB : exome :giab:
**** Notes
https://github.com/genome-in-a-bottle/giab_FAQ
**** Résultats résumés :resultats:
***** DONE HG001 :
CLOSED: [2023-04-06 Thu 21:41] SCHEDULED: <2023-04-02 Sun>
| Données | Algorithm | Type    | Recall | Precision |
|---------+-----------+---------+--------+-----------|
| Bisonex | Happy     | SNP     | 0.8552 |    0.9708 |
| Bisonex | vcfeval   | SNP     | 0.8547 |    0.9727 |
| Bisonex | Happy     | INDEL   | 0.7105 |    0.6929 |
| Bisonex | vcfeval   | Non-SNP | 0.7139 |    0.7136 |
|---------+-----------+---------+--------+-----------|
| GIAB    | happy     | INDEL   | 0.7551 |    0.7415 |
| GIAB    | vcfeval   | INDEL   | 0.7598 |    0.7445 |
| GIAB    | happy     | SNP     | 0.8937 |    0.9621 |
| giab    | vcfeval   | SNP     | 0.8937 |    0.9621 |
***** DONE HG002, HG003, HG004
CLOSED: [2023-04-14 Fri 11:36] SCHEDULED: <2023-04-14 Fri>
Capture Agilent
| Patient | Algorithm | Type  |   Recall | Precision |
| HG002   | happy     | INDEL | 0.851495 |  0.923616 |
| HG002   | happy     | SNP   | 0.905926 |  0.992158 |
| HG002   | vcfeval   | indel |   0.8523 |    0.9212 |
| HG002   | vcfeval   | snp   |   0.9054 |    0.9934 |
| HG003   | vcfeval   | indel |   0.8363 |    0.9115 |
| HG003   | vcfeval   | snp   |   0.9069 |    0.9928 |
| HG003   | happy     | INDEL | 0.838521 |  0.917296 |
| HG003   | happy     | SNP   | 0.907466 |  0.991204 |
| HG004   | happy     | INDEL | 0.856835 |  0.925086 |
| HG004   | happy     | SNP   | 0.905067 |  0.992704 |
| HG004   | vcfeval   | indel |   0.8568 |    0.9240 |
| HG004   | vcfeval   | snp   |   0.9048 |    0.9938 |
**** TODO télécharger données avec Nextflow :hg38:
***** DONE Renommer les chromosomes
CLOSED: [2023-02-17 Fri 19:30]
****** DONE Genome de reference NCBI
CLOSED: [2023-02-25 Sat 19:46]
****** DONE Bed avec les exons
CLOSED: [2023-03-29 Wed 23:04]
****** DONE hg19
CLOSED: [2023-02-26 Sun 22:37]
****** DONE hg38
CLOSED: [2023-03-29 Wed 23:04]
- [X] Télécharger hg19 : ok
- [X] convertir bed en interval list
picard BedToIntervalList -I exons_illumina.bed  -O exons_illumina.list -SD  ../../genome/GRCh19/genomeRef.dict
- [X] puis en hg38
picard LiftOverIntervalList -I exons_illumina.list  -O exons_illumina_hg38.list --CHAIN hg19ToHg38.over.chain -SD  ../../genome/GRCh38.p13/genomeRef.dict
- [X] puis en bed

***** KILL VCF de référence
CLOSED: [2023-04-16 Sun 16:32]
****** TODO NA12878 (HG001)
******* DONE Fastq HiSeq
CLOSED: [2023-02-25 Sat 19:46]
On prend le Hiseq, qui est probablement ce qu'utilise Centogène :
https://ftp-trace.ncbi.nih.gov/ReferenceSamples/giab/data/NA12878/Garvan_NA12878_HG001_HiSeq_Exome/
On utilisé les données "trimmés" (https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1069-7), i.e qui ont enlevé les fragments plus petits que la taille d'un read.
Informations:
- https://ftp-trace.ncbi.nih.gov/ReferenceSamples/giab/data/NA12878/Garvan_NA12878_HG001_HiSeq_Exome/Garvan_NA12878_HG001_HiSeq_Exome.README
- Sequencer: HiSeq2500
- kit: Nextera Rapid Capture Exome and Expanded Exome

Il y a 2 samples (NIST7035 et NIST7086), chacun sur 2 lanes -> à concaténer
NB : liste techno illumina https://www.illumina.com/systems/sequencing-platforms.html

Hiseq postérieur nextseq 550
******* TODO Fastq hiseq sans trimming
******* DONE Capture : Exons (bed)
CLOSED: [2023-02-25 Sat 19:46]
https://ftp-trace.ncbi.nih.gov/ReferenceSamples/giab/data/NA12878/Garvan_NA12878_HG001_HiSeq_Exome/nexterarapidcapture_expandedexome_targetedregions.bed.gz
******* DONE Bed, vcf
CLOSED: [2023-02-24 Fri 23:45]
****** DONE Ashkenazy trio HG002, HG003, HGQ004
CLOSED: [2023-04-06 Thu 21:43] SCHEDULED: <2023-04-01 Sat>
****** KILL Chinese trio HG005, 6, 7
CLOSED: [2023-04-16 Sun 16:32]
***** KILL Fastq :fastq:
CLOSED: [2023-04-16 Sun 16:32]
****** DONE NA12878 (HG001)
CLOSED: [2023-02-25 Sat 19:46]
******* DONE Fastq HiSeq
CLOSED: [2023-02-25 Sat 19:46]
On prend le Hiseq, qui est probablement ce qu'utilise Centogène :
https://ftp-trace.ncbi.nih.gov/ReferenceSamples/giab/data/NA12878/Garvan_NA12878_HG001_HiSeq_Exome/
On utilisé les données "trimmés" (https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1069-7), i.e qui ont enlevé les fragments plus petits que la taille d'un read.
Informations:
- https://ftp-trace.ncbi.nih.gov/ReferenceSamples/giab/data/NA12878/Garvan_NA12878_HG001_HiSeq_Exome/Garvan_NA12878_HG001_HiSeq_Exome.README
- Sequencer: HiSeq2500
- kit: Nextera Rapid Capture Exome and Expanded Exome

Il y a 2 samples (NIST7035 et NIST7086), chacun sur 2 lanes -> à concaténer
NB : liste techno illumina https://www.illumina.com/systems/sequencing-platforms.html

Hiseq postérieur nextseq 550
******* DONE Capture : Exons (bed)
CLOSED: [2023-02-25 Sat 19:46]
https://ftp-trace.ncbi.nih.gov/ReferenceSamples/giab/data/NA12878/Garvan_NA12878_HG001_HiSeq_Exome/nexterarapidcapture_expandedexome_targetedregions.bed.gz
****** DONE Ashkenazy trio HG002, HG003, HG004
CLOSED: [2023-04-15 Sat 23:24] SCHEDULED: <2023-04-05 Wed>
******* DONE Capture
CLOSED: [2023-04-15 Sat 23:24]
https://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/analysis/OsloUniversityHospital_Exome_GATK_jointVC_11242015/wex_Agilent_SureSelect_v05_b37.baits.slop50.merged.list
******* DONE Capture Agilent
CLOSED: [2023-04-15 Sat 23:24]
******* DONE Bam à partir des fastq
CLOSED: [2023-04-15 Sat 23:24]
Bam + index + checksum
https://raw.githubusercontent.com/genome-in-a-bottle/giab_data_indexes/master/AshkenazimTrio/alignment.index.AJtrio_OsloUniversityHospital_IlluminaExome_bwamem_GRCh37_11252015

****** KILL Chinese trio
CLOSED: [2023-04-16 Sun 16:32]
Whole exome pour HG005 seulement
******* KILL HG005
CLOSED: [2023-04-16 Sun 16:32]
https://raw.githubusercontent.com/genome-in-a-bottle/giab_data_indexes/master/ChineseTrio/alignment.index.Chinesetrio_HG005_OsloUniversityHospital_IlluminaExome_bwamem_GRCh37_11252015
**** DONE Télécharger FASTQ directement avec aws (via SRA)
CLOSED: [2023-06-30 Fri 22:30] SCHEDULED: <2023-06-27 Tue>
***** Remarques
Numéro d'accession : https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-022-08365-3/tables/1
Fastq disponible via SRA. Avec AWS, on peut accéder au fastq directement.
(Sinon il faut convertir SRA -> Fastq avec le toolkit : compliqué à configurer)
Exemple: https://trace.ncbi.nlm.nih.gov/Traces/?view=run_browser&acc=SRR2962669&display=data-access

Avantage:
- pas de conversion BAM -> fASTQ
- détail des capture
- capture en hg38 sur site du constructeur !!
- capture semblable pour ashkenazi
Inconvénient :
- NA12878 : discordance pour le nombre de paires de bases : NA12878 = 49G (donc 24G de fastq)
- capture non disponible en ligne (site agilent)
- format SRA (le lien pour les fastq n'est pas gratuit): utiliser HTTP ou leur toolkit (télécharge au format SRA puis convertit en fastq). Exemple: pour avoir 2 fastq 
  fastq-dump --split-files --gzip SRR2962669

***** Liste des runs :
https://www.ncbi.nlm.nih.gov/sra

Cherche avec numéro patient. On a le choix entre plusieurs séquenceurs Illumina
- NovaSeq 6000 TruSeq capture SRX11061536
- NovaSeq 6000 IDT capture SRX11061526
- NovaSeq 6000 Agilent SureSelect v7 capture SRX11061516
- HiSeq 4000 TruSeq capture SRX11061506
- HiSeq 4000 IDT capture SRX11061496
- HiSeq 4000 Agilent SureSelect v7 capture SRX11061486

Note: SRX = expérience, SRR = run
          Important:
          - ne pas compresser la sortie avec fasta-dump directement (lent++)
          - Fasterq-dump est plus rapide
          
  Note trueseq non disponible ?
  hg19 : https://www.biostars.org/p/144554/


IDT: lequel
https://www.idtdna.com/pages/products/next-generation-sequencing/workflow/xgen-ngs-hybridization-capture/pre-designed-hyb-cap-panels/exome-hyb-panel-v2

***** DONE HiSeq 4000 + agilent sureselect :sra:
CLOSED: [2023-06-28 Wed 22:06] SCHEDULED: <2023-06-28 Wed>
        - [ ] HG001 with Illumina HiSeq 4000 Agilent SureSelect v7 capture SRX11061486 SRR14724513	
        - [ ] HG002 with Illumina HiSeq 4000 Agilent SureSelect v7 capture SRX11061487 SRR14724512
        - [ ] HG003 with Illumina HiSeq 4000 Agilent SureSelect v7 capture SRX11061488 SRR14724511
        - [ ] HG004 with Illumina HiSeq 4000 Agilent SureSelect v7 capture SRX11061489 SRR14724510
          
        Other
 - HG005 with Illumina HiSeq 4000 Agilent SureSelect v7 capture SRX11061491 SRR14724508
           - HG006 with Illumina HiSeq 4000 Agilent SureSelect v7 capture SRX11061492 SRR14724507
           - HG007 with Illumina HiSeq 4000 Agilent SureSelect v7 capture SRX11061493 SRR14724506

******* DONE Capture agilent sureselect
CLOSED: [2023-06-30 Fri 22:30] SCHEDULED: <2023-06-28 Wed>
**** TODO Lift T2T :T2T:
#+begin_quote
We performed liftover using the GATK release 4.1.9 LiftoverVcf (Picard Version 2.23.3) tool with the default parameters. This successfully lifts over variants that map exactly from GRCh38 to T2T-CHM13v2.0 but does not recover variants with swapped reference and alternative alleles. To recover variants with swapped reference/alternative alleles, we ran LiftoverVCF again, with the RECOVER_SWAPPED_REF_ALT flag. Notably, this feature does not recover multiallelic variants, so to recover these variants, we first separated them into multiple biallelic variants, performed liftover using the RECOVER_SWAPPED_REF_ALT tag, and converted them back to their multiallelic representations.
#+end_quote
***** KILL Liftovervcf avec valeur par défaut
CLOSED: [2023-07-02 Sun 23:09] SCHEDULED: <2023-06-30 Fri>
HG002 : il manque la moitié des valeurs
hg001
[apraga@mesointeractive b946d0e6bc8d0f220eb1ad1649c20d]$ less HG004_GRCh38_1_22_v4.2.1_benchmark.vcf.lifted.vcf.gz
[apraga@mesointeractive b946d0e6bc8d0f220eb1ad1649c20d]$ zgrep -c '^chr' HG004_GRCh38_1_22_v4.2.1_benchmark.vcf.lifted.vcf.gz
2168972
[apraga@mesointeractive b946d0e6bc8d0f220eb1ad1649c20d]$ zgrep -c '^chr' HG004_GRCh38_1_22_v4.2.1_benchmark.vcf.unlifted.vcf.gz
1862374
[apraga@mesointeractive b946d0e6bc8d0f220eb1ad1649c20d]$ zgrep -c '^chr' HG004_GRCh38_1_22_v4.2.1_benchmark.vcf.gz
4031346

***** DONE liftover bed
CLOSED: [2023-07-02 Sun 23:09] SCHEDULED: <2023-06-30 Fri>
792 of 217488 intervals failed (0.364158%) to liftover, encompassing 219109 of 35718732 bases (0.613429%).

 wc -l capture/Agilent_SureSelect_All_Exons_v7_hg38_Regions.bed
217488 capture/Agilent_SureSelect_All_Exons_v7_hg38_Regions.bed

wc -l work/e4/9981dc539a2373c2beeaa0affc3497/Agilent_SureSelect_All_Exons_v7_hg38_Regions_hg38.interval_list

On a donc perdu 1000 zones
***** DONE Liftovervcf avec variant échangé référence/alternative ?
CLOSED: [2023-07-02 Sun 23:09]
**** DONE NA12878 :na12878:hg38:
CLOSED: [2023-06-30 Fri 22:30]
***** DONE Discussion alexis : Mail
CLOSED: [2023-03-29 Wed 22:40]

Avec le patient NA12878 et comparaison avec hap.py du VCF de Genome In A Bottle ("gold" standard), on avait pour rappel
- sensibilité (=recall) 71% pour indel, 85% SNP
- précision  (= VPP) 69 et 97% respectivement

| Type  | TRUTH |    TP |   FN | QUERY |   FP |  UNK | FP.gt | FP.al |   Recall | Precision |
| INDEL |  4871 |  3461 | 1410 |  7048 | 1554 | 1987 |   193 |   346 | 0.710532 |  0.692946 |
| SNP   | 46032 | 39369 | 6663 | 44600 | 1186 | 4041 |   304 |    30 | 0.855253 |  0.970759 |

Les statistiques sur les génomes sont bien meilleurs (cf precisionFDA challenge).
Pour les exome, un article [1] a fait a des meilleures stats sur ce patient avec BWA et GATK mais ils ont moins de variant (on a presque un facteur 2 !).
Je soupçonne qu'on ne travaille pas sur les mêmes zones de capture (pas réussi à récupérer leur .bed)

| Exome | Type  |    TP |   FP |  FN | Sensitivity | Precision | F-Score |   FDR |
|     1 | SNV   | 23689 | 1397 | 613 |       0.975 |     0.944 |   0.959 | 0.057 |
|     2 | SNV   | 23946 |  865 | 356 |       0.985 |     0.965 |   0.975 | 0.036 |
|     1 | indel |  1254 |   72 |  75 |       0.944 |     0.946 |   0.945 | 0.054 |
|     2 | indel |  1309 |   10 |  20 |       0.985 |     0.992 |   0.989 | 0.008 |

Pour essayer d'améliorer les statistiques :

- La version du génome GRC38 vs GRCh38.p13 ne change quasiment rien
- Désactiver dbSNP ne change strictement rien pour le variant calling

J'ai exploré les faux négatifs :
- la grande majorité n'est juste pas vue (ce n'est pas un problème d'haploïde/génotype)
- la répartition par chromosome est relativement homogène, sauf sur le 6 ()
- la majorité est en 5' et 3'UTR (selon Best refseq)

Conclusion: je pense m'arrêter là pour la validation du variant calling par manque de temps. Il faudrait creuser pour savoir pourquoi certains variants ne sont pas vus par GATK mais ce n'est pas la majorité. En tout cas, je peux justifier d'une première analyse pour la thèse.

Ça te va ?

[1]
https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-2928-9
Résultats ici https://static-content.springer.com/esm/art%3A10.1186%2Fs12859-019-2928-9/MediaObjects/12859_2019_2928_MOESM8_ESM.pdf

***** DONE Comparaison
CLOSED: [2023-03-04 Sat 11:14]
HGREF=/Work/Groups/bisonex/data-alexis-reference/genome/GRCh38_latest_genomic.fna ./result/bin/hap.py /Work/Groups/bisonex/NA12878/HG001_GRCh38_1_22_v4.2.1
_benchmark_renamed.vcf.gz script/files/vcf/NA12878_NIST7035_vep_annot.vcf -f /Work/Groups/bison
ex/NA12878/HG001_GRCh38_1_22_v4.2.1_benchmark.bed -o test

na1878.slurm
#+begin_src slurm
#!/bin/bash
#SBATCH -c 4
#SBATCH -p smp
#SBATCH --time=01:00:00
#SBATCH --mem=32G

module load nix/2.11.0
export HGREF=/Work/Groups/bisonex/data-alexis-reference/genome/GRCh38_latest_genomic.fna
dir=/Work/Groups/bisonex/data/NA12878/GRCh38
hap.py ${dir}/HG001_GRCh38_1_22_v4.2.1_benchmark.vcf.gz script/files/vcf/NA12878_NIST7035.vcf -f ${dir}/HG001_GRCh38_1_22_v4.2.1_benchmark.bed -o test
#+end_src
****** KILL beaucoup trop de faux négatifs
CLOSED: [2023-02-17 Fri 19:37]
******* DONE Test 1 : vep annot : beaucoup trop de faux négatif
CLOSED: [2023-02-06 lun. 13:40]
 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision  METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
INDEL    ALL       276768       274    276494         1500       257        968     26     15       0.000990          0.516917        0.645333         0.001976                     NaN                     NaN                   1.483361                   6.129187
INDEL   PASS       276768       274    276494         1500       257        968     26     15       0.000990          0.516917        0.645333         0.001976                     NaN                     NaN                   1.483361                   6.129187
  SNP    ALL      1937706      1193   1936513         3338       106       2037     11      2       0.000616          0.918524        0.610246         0.001231                  2.0785                1.861183                   1.539064                   2.703663
  SNP   PASS      1937706      1193   1936513         3338       106       2037     11      2       0.000616          0.918524        0.610246         0.001231                  2.0785                1.861183                   1.539064                   2.703663
******* KILL Test 3 : indexer vcf de reference
CLOSED: [2023-02-06 lun. 17:19]
Même résultat avec vcfeval, qui a besoin de la version indexée
******* DONE Test 3 sans filtre vep : idem
CLOSED: [2023-02-06 lun. 17:19]
Benchmarking Summary:
 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision  METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
INDEL    ALL       276768     10535    266233        52169     10969      30616   3552   2122       0.038064          0.491069        0.586862         0.070652                     NaN                     NaN                   1.483361                   0.509510
INDEL   PASS       276768     10535    266233        52169     10969      30616   3552   2122       0.038064          0.491069        0.586862         0.070652                     NaN                     NaN                   1.483361                   0.509510
  SNP    ALL      1937706    105753   1831953       357652     74634     177259  35111    797       0.054576          0.586270        0.495619         0.099857                  2.0785                 1.42954                   1.539064                   0.324923
  SNP   PASS      1937706    105753   1831953       357652     74634     177259  35111    797       0.054576          0.586270        0.495619         0.099857                  2.0785                 1.42954                   1.539064                   0.324923
******* DONE Test 4 avec vcfeval sur vep_annot : idem
CLOSED: [2023-02-06 lun. 17:18]
#+begin_src
#!/bin/bash
#SBATCH -c 4
#SBATCH -p smp
#SBATCH --time=01:00:00
#SBATCH --mem=32G
module load nix/2.11.0
export HGREF=/Work/Groups/bisonex/data-alexis-reference/genome/GRCh38_latest_genomic.fna dir=/Work/Groups/bisonex/data/NA12878/GRCh38
rtg vcfeval -b  /Work/Groups/bisonex/data/NA12878/GRCh38/HG001_GRCh38_1_22_v4.2.1_benchmark.vcf.gz  -c files/vcf/NA12878_NIST7035_vep_annot.vcf.gz  -o test-rtg -t /Work/Groups/bisonex/data/genome/GRCh38.p13/genomeRef.sdf
#+end_src

Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
    1.000               2984           2682       1840    3890296     0.5931       0.0008     0.0015
     None               2984           2682       1841    3890296     0.5930       0.0008     0.0015

Exemple du log
2023-02-06 13:50:14 Reference NC_000001.11 baseline contains 307854 variants.
2023-02-06 13:50:14 Reference NC_000001.11 calls contains 426 variants.
2023-02-06 13:50:15 Reference NC_000002.12 baseline contains 325877 variants.
2023-02-06 13:50:15 Reference NC_000002.12 calls contains 320 variants.
******* DONE Regarder quelques variants à la main
CLOSED: [2023-02-07 Tue 22:01]
Ex:
Il manque NC_000001.11    783006  .       A       G       50      PASS
Il y a A -> G et C -> A sur cette position
***** DONE Restreindre genome de référence
CLOSED: [2023-03-04 Sat 11:15]
****** Discussion Alexis
le pipeline prend en compte 5', 3', variant canoniques d'épissage + prédit spip
Le plus simple pour le moment est de restreindre seulement aux exons
GENCODE (version europénne) vs RefSeq: [[https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-16-S8-S2][article 2015]] en faveur de GENCODE mais Alexis conseille Refseq

****** DONE -f, -R ou -T ?
CLOSED: [2023-02-25 Sat 19:47]
Selon la doc : -f avec le bed fourni et -T pour filtrer sor les exons
******* rtg tools
Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
    3.000               1015            910        206      32531     0.8154       0.0303     0.0583
     None               1015            910        206      32531     0.8154       0.0303     0.0583
***** KILL Exons seuls
CLOSED: [2023-04-02 Sun 17:11]
****** DONE BestRefSeq 
CLOSED: [2023-02-19 Sun 12:05]
Dans refseq,[[https://www.ncbi.nlm.nih.gov/genome/annotation_euk/process/][2 types de modèles pour le gène]]
- basé sur refseq (NM_, NP_), curée
- basé sur gnomon (XM_ , XP_), prédite

Les modèles basés sur refseq ont la préférénce (cf lien)
On se restreint donc à bestrefseq
#+begin_src sh
wget https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_genomic.gff.gz
gunzip https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_genomic.gff.gz
#+end_src
On se restrein aux exons codant (NM_)
#+begin_src
awk '/BestRefSeq\texon/ && /transcript_id=NM/ {print $1"\t"$4"\t"$5;}' GRCh38_latest_genomic.gff > exons.csv
#+end_src
Puis intersection
******* DONE Tests après correction bug dans noms de chromosome : precision ~ ok, recall très mauvais -> trop de FN ?
CLOSED: [2023-02-19 Sun 12:05]
 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision
INDEL    ALL         7230       321      6909         1500       290        888     27     18       0.044398          0.526144
INDEL   PASS         7230       321      6909         1500       290        888     27     18       0.044398          0.526144
  SNP    ALL        59052      1653     57399         3338       101       1583     12      2       0.027992          0.942450
  SNP   PASS        59052      1653     57399         3338       101       1583     12      2       0.027992          0.942450

 METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
       0.592000         0.081887                     NaN                     NaN                    1.54733                   6.129187
       0.592000         0.081887                     NaN                     NaN                    1.54733                   6.129187
       0.474236         0.054370                2.433271                1.861183                    1.57523                   2.703663
       0.474236         0.054370                2.433271                1.861183                    1.57523                   2.703663
******* DONE Vérifier exons: on a l'union des exons de tous les transcripts...
CLOSED: [2023-02-19 Sun 12:05]
Il faudrait un .bed d'illumina
On teste Twist for Illumina Exome 2.0 Plus BED File (hg19) sur https://support.illumina.com/downloads/nextera-flex-for-enrichment-BED-files.html
Conversion en hg38 avec ucsc
Renommage des chromosomes
#+begin_src
sed 's:^:s/chr:;s:chrMT:chrM:;s:\s:\\t/:;s:$:\\t/:' ../../genome/GRCh38.p13/chromosome_mapping.txt > pattern.sed
sed -i.bak -f pattern.sed illumina_exons.bed
bedtools intersect -a HG001_GRCh38_1_22_v4.2.1_benchmark.bed -b illumina_exons.bed > HG001_GRCh38_1_22_v4.2.1_benchmark_illumina_exons.bed
#+end_src
Intersection
****** KILL Bed illumina
CLOSED: [2023-02-24 Fri 23:44]
******* KILL Sans filtre vep: Inversion truth et query... on recommence
CLOSED: [2023-02-19 Sun 13:15]
******* KILL Sans filtre vep: mieux mais pas exceptionnel
CLOSED: [2023-02-24 Fri 23:44]
cd work/00/2c72e62400956c96fb101ac7af405e/
$ cat .command.out

 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision
INDEL    ALL          922       490       432          942       439          0     32     56       0.531453          0.533970
INDEL   PASS          922       490       432          942       439          0     32     56       0.531453          0.533970
  SNP    ALL        24618     18689      5929        21257      2571          0    239     17       0.759160          0.879052
  SNP   PASS        24618     18689      5929        21257      2571          0    239     17       0.759160          0.879052


 METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
            0.0         0.532709                     NaN                     NaN                   1.628743                   2.799180
            0.0         0.532709                     NaN                     NaN                   1.628743                   2.799180
            0.0         0.814719                2.899604                2.881526                   1.598479                   1.564378
            0.0         0.814719                2.899604                2.881526                   1.598479                   1.564378
******* KILL Comprendre pour les FN explosent avec vep annot
CLOSED: [2023-02-24 Fri 23:44]
NC_000001.11	924024	.	C	G
non couverte dans bam -> 1er exons de SAMD11 mais non couverte
Il est dans NM_001385641.1 mais pas dans NM_152486.4
****** KILL Obtenir les zones couvertes depuis le bam directement
CLOSED: [2023-02-24 Fri 23:44]
#+begin_src  sh
samtools sort NA12878_chr1.bam -o NA12878_chr1_sorted.bam
bedtools genomecov -ibam NA12878_chr1_sorted.bam  -bg > chr1.bedgraph
head chr1.bedgraph
#+end_src

#+RESULTS:
: NC_000001.11	10001	10021	1
: NC_000001.11	10021	10059	2
: NC_000001.11	10059	10063	1
: NC_000001.11	10063	10087	2
: NC_000001.11	10087	10110	1
: NC_000001.11	10110	10113	3
: NC_000001.11	10113	10121	2

On prend les régions avec 20 reads
awk '$4 > 20 {print $1"\t"$2"\t"$3}' chr1.bedgraph  > tomerge.bed
On fusionne les régions
bedtools merge -i tomerge.bed > exons_from_bam.bed

Problème : on manque parfois le bord des exons...

****** KILL Vérifier un bam de référence de NA12878
CLOSED: [2023-02-24 Fri 23:44]
Notamment pour la définitions des exons, on a des reads qui ne semblent pas alignés sur les bons exons selon IGV
On donne directemet à IGV le bam d'illuma WES ( https://github.com/genome-in-a-bottle/giab_data_indexes )
IGV n'aime pas le ftp apparement...
On récupére 2 échantillons pour ce patient sur HiSeq Exome
#+begin_src sh :dir /ssh:meso:/Work/Groups/bisonex/data/giab/GRCh38
wget https://raw.githubusercontent.com/genome-in-a-bottle/giab_data_indexes/master/NA12878/alignment.index.NA12878_HiSeq_Exome_Garvan_GRCh37_09252015 -O todl.txt
awk '{print $1}' todl.txt | wget -i -
#+end_src
On récupére le premier échantillons en prenant just le chromosome 1
#+begin_src
samtools merge NA12878-NIST7035-HiSeq_Exome_Garan_GRCh37.bam project
.NIST_NIST7035_H7AP8ADXX_TAAGGCGA_1_NA12878.bwa.markDuplicates.bam project.NIST_NIST7035_H7AP8ADXX_TA
AGGCGA_2_NA12878.bwa.markDuplicates.bam
samtools index NA12878-NIST7035-HiSeq_Exome_Garan_GRCh37.bam
#+end_src
******* Étude
NC_000001.11:924024 :
- bed d’illumina : pas correct
- bed refseq : meux
- bam :  exon non ciblé, probablement parce que Mane select n’a pas été utilisé mais refseq (https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:28706)

NC_000001.11:924310: idem
NC_000001.11:924321: idem
NC_000001.11:924533: idem

NC_000001.11:966227: le bed ne couvre pas le bon exon !
https://genome-euro.ucsc.edu/cgi-bin/hgTracks?db=hg38&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr1%3A965863%2D966591&hgsid=295298291_vhDbIv5YIckCKLpGB7UUaZi2cAkr
NC_000001.11:966272

****** KILL Filtrer variants introniques de référence avec vep
CLOSED: [2023-02-24 Fri 23:44]
******* KILL variant calling seulf + seulement -f: nombreux FP
CLOSED: [2023-02-24 Fri 23:44]
/Work/Users/apraga/bisonex/work/68/f1cf72a5a4078fdf743fb3844b369a
 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision
INDEL    ALL          519       284       235        52169     37789      14094     12     64       0.547206          0.007511
INDEL   PASS          519       284       235        52169     37789      14094     12     64       0.547206          0.007511
  SNP    ALL        22131     17434      4697       357652    305313      34904    189     32       0.787764          0.054020
  SNP   PASS        22131     17434      4697       357652    305313      34904    189     32       0.787764          0.054020

 METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
       0.270160         0.014820                     NaN                     NaN                   1.775956                   0.509510
       0.270160         0.014820                     NaN                     NaN                   1.775956                   0.509510
       0.097592         0.101108                2.971834                1.429533                   1.579776                   0.324923
       0.097592         0.101108                2.971834                1.429533                   1.579776                   0.324923

****** DONE Bed donnés par GIAB (en hg19) : résultats corrects :resultats:
CLOSED: [2023-03-09 Thu 22:42] SCHEDULED: <2023-03-02 Thu>
Téléchargé à la main et converti via UCSCS. À automatiser, voir : *hg38
Sans oublier de renommer les chromosomes !

| Type  | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision |
|-------+-------------+----------+----------+-------------+----------+-----------+-------+-------+---------------+------------------|
| INDEL |        4871 |     3461 |     1410 |        7048 |     1554 |      1987 |   193 |   346 |      0.710532 |         0.692946 |
| SNP   |       46032 |    39369 |     6663 |       44600 |     1186 |      4041 |   304 |    30 |      0.855253 |         0.970759 |

 Type   Ssensibilité    VPP
 INDEL       0.710532   0.692946 
 SNP         0.855253   0.970759



 | METRIC.Frac_NA | METRIC.F1_Score | TRUTH.TOTAL.TiTv_ratio | QUERY.TOTAL.TiTv_ratio | TRUTH.TOTAL.het_hom_ratio | QUERY.TOTAL.het_hom_ratio |
 |----------------+-----------------+------------------------+------------------------+---------------------------+---------------------------|
 |       0.281924 |        0.701629 |                        |                        |        1.6174985978687606 |        3.0674091441969518 |
 |       0.090605 |        0.909353 |      2.529551552318896 |     2.4019675131548843 |        1.6206857273037931 |        1.6274012964054214 |

****** DONE Re-tester avec exons Refseq et option -T: résultats un peu moins bons
CLOSED: [2023-03-09 Thu 22:42] SCHEDULED: <2023-03-08 Wed>
On utilise directement les coordonées données par refseq

 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision
INDEL    ALL         7226      3417      3809         6978      1599       1918    228    353       0.472876          0.683992
  SNP    ALL        59052     37825     21227        43480      1913       3740    675     35       0.640537          0.951862

 METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
       0.274864         0.559171                     NaN                     NaN                   1.547733                   2.756151
       0.086017         0.765767                2.433271                2.350281                   1.575230                   1.492346
****** DONE Vérifier que la zone de capture a vraiment besoin d'un liftover...
CLOSED: [2023-04-02 Sun 14:13]
D'après la documentation, oui
https://support.illumina.com/sequencing/sequencing_kits/nextera-rapid-capture-exome-kit/downloads.html
***** DONE D'où vient la différence ?
CLOSED: [2023-04-02 Sun 17:11] SCHEDULED: <2023-03-09 Thu>
****** DONE Statistiques
CLOSED: [2023-03-15 mer. 13:41] SCHEDULED: <2023-03-04 Sat>
On confirnme le nombre de SNP 6665
- 304 ont un problème d’haploide/allèle différente
- 28 avec une différence de génotype
- La majorité ne sont pas vu 6361

[1] "ALl FN 6665"
[1] "Genotype mismatch 28"
[1] "haplotype mismatch 304"
[1] "Missing  6333"
****** DONE Majorité des FN pour SPN sont peu couverts
CLOSED: [2023-03-16 Thu 16:54]
Chromosome 1 : majorité des FN ne sont pas vus...
cf figure
#+attr_html: :width 500px
[[file:reads.png]]
 1623412 : 5’UTR
 1953616 : intronique mais dans bed...
 2589991
 3488573
 3488732
 3780326
 3884683
 3914055
 4711993
 4712657

****** DONE Alignement
CLOSED: [2023-03-15 mer. 13:41]
******* DONE Impact de la version mineure du genome: non
CLOSED: [2023-03-09 Thu 23:13]
******** DONE GHC38 + version alexis + exons refseq
CLOSED: [2023-03-22 Wed 00:02]
 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision
INDEL    ALL         7226      3417      3809         6979      1599       1919    228    353       0.472876          0.683992
  SNP    ALL        59052     37827     21225        43483      1913       3741    676     35       0.640571          0.951865

 METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
       0.274968         0.559171                     NaN                     NaN                   1.547733                   2.756698
       0.086034         0.765792                2.433271                2.350254                   1.575230                   1.492375

******** DONE GHC38.p13 + notre version alexis + exons refseq
CLOSED: [2023-03-22 Wed 00:02]
 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision
INDEL    ALL         7226      3417      3809         6978      1599       1918    228    353       0.472876          0.683992
  SNP    ALL        59052     37825     21227        43480      1913       3740    675     35       0.640537          0.951862

 METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
       0.274864         0.559171                     NaN                     NaN                   1.547733                   2.756151
       0.086017         0.765767                2.433271                2.350281                   1.575230                   1.492346

****** DONE Vérifier si coordonnées génomiques (vcf)
CLOSED: [2023-03-09 Thu 23:05]
****** DONE Variant calling
CLOSED: [2023-03-22 Wed 23:11]
******* DONE Désactiver dbSNP : idem
CLOSED: [2023-03-22 Wed 15:00]
Après haplotycaller
$ sha256sum work/94/d4ae5999ca59a25469b1ed221b7b1b/NA12878_NIST.vcf.gz
193fb37e2a581bb2855ae6fd11638f3b97958515def0e4295005e6a60c9dc650  work/94/d4ae5999ca59a25469b1ed221b7b1b/NA12878_NIST.vcf.gz

Fichier utilisé par hap.py
$ sha256sum work/5b/199360963641524a076d06b621e7e0/NA12878_NIST.vcf.gz
193fb37e2a581bb2855ae6fd11638f3b97958515def0e4295005e6a60c9dc650  work/5b/199360963641524a076d06b621e7e0/NA12878_NIST.vcf.gz

On a bien le même que [[*Bed donnés par GIAB (en hg19) : résultats corrects][Bed donnés par GIAB (en hg19) : résultats corrects]]
 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision
INDEL    ALL         4871      3461      1410         7048      1554       1987    193    346       0.710532          0.692946
  SNP    ALL        46032     39367      6665        44599      1186       4042    304     30       0.855209          0.970757

 METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
       0.281924         0.701629                     NaN                     NaN                   1.617499                   3.067409
       0.090630         0.909327                2.529552                2.402151                   1.620686                   1.627342

****** DONE Définition des exons
CLOSED: [2023-03-22 Wed 23:16]
******* KILL Vérifier qu'il ne manque pas des exons (avec bam ?)
CLOSED: [2023-03-16 Thu 16:54] SCHEDULED: <2023-03-05 Sun>
******* KILL Vérifier la couverture des variants introniques
CLOSED: [2023-03-22 Wed 15:01]
NC_000001.11  23344310  : intronique mais la région dans le bed couvre en partie l'exon
Profondeur sur le variant : 14 reads sur l'alt
******* Notes: UTR = exon - CDS
intron dans le gene = mRNA - exon
On le vérifie avec
  25 │ NC_000001.11  BestRefSeq  exon      65520   65573
  27 │ NC_000001.11  BestRefSeq  CDS       65565   65573
  https://genome-euro.ucsc.edu/cgi-bin/hgTracks?db=hg38&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr1%3A65520%2D65573&hgsid=295007972_J1rAQiur4dC2KLadnaOCw27WUihG

  mRNA :
  https://genome-euro.ucsc.edu/cgi-bin/hgTracks?db=hg38&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr1%3A65419%2D71585&hgsid=295007972_J1rAQiur4dC2KLadnaOCw27WUihG

******* DONE Statistiques : reads par type (introns, UTR, exons)
CLOSED: [2023-03-22 Wed 23:16]
- la répartition par chromosome est relativement homogène, sauf sur le 6 ()
- la majorité est en 5' et 3'UTR (selon Best refseq)
#+begin_src julia
using CSV
using DataFrames, DataFramesMeta
using XAM, GenomicFeatures

function exonFromBestRefSeq(f)
    println("Reading exons from $(f)")
    cols = ["seqname", "source", "feature", "start", "end", "score", "strand", "frame", "attribute"]
    d = DataFrame(CSV.File(f ,header=cols, delim="\t",comment="#"))
    @chain d begin
        @rsubset :feature in ["exon", "CDS", "mRNA"] :source .== "BestRefSeq"
        @select :seqname :source :feature :start :end
        @distinct
    end
end

# Get exons from refseq (Bestrefeq, gnomon or just refseq)
function getExon(f)
    if isfile(f)
        println("Reading exons from preprocessed $(f)")
        d = DataFrame(CSV.File(f))
    else
        d = exonFromBestRefSeq("GRCh38_latest_genomic.gff.gz")
        CSV.write(f, d)
    end
    return d
end

# Return false negative and TRUTH annotation (genotype, type...)
# Read vcf manually: it's a tab-separated file and we extract the TRUTH column manually
#BD = Decision for call (TP/FP/FN/N)
#BK = tSub-type for decision (match/mismatch type)
#BI = Additional comparison information
#QQ = Variant quality for ROC creation.
#BV = High-level variant type (SNP|INDEL).
#BL = High-level location type (het|homref|hetalt|homalt|nocall).
function getFN(f)
    cols = ["CHROM", "POS", "ID", "REF", "ALT", "QUAL", "FILTER", "INFO", "FORMAT", "TRUTH", "QUERY"]
    vcf = DataFrame(CSV.File(f,comment="#", delim="\t",
                             header = cols))
    # Extract genothype and other subcolumn fronm FORMAT
    format = split(vcf[1,:FORMAT], ":")
    @chain vcf begin
        @rtransform $format=split(:TRUTH, ':')
        @subset :BD .== "FN"
        @select :type=:BVT :CHROM :POS :REF :ALT :GT :BK :BLT
    end
end

# Search a variant in refseq exons
# Return the smallest interval (CDS)
function searchVariant(chrom, pos, exons)
    res = @subset exons :seqname .== chrom :start .<= pos :end .>= pos
    if isempty(res)
        return "intronicOutsideGene"
    elseif "CDS" in res.feature
        return "CDS"
    elseif "exon" in res.feature
        # UTR = exon - CDS
        return "UTR"
    elseif "mRNA" in res.feature
        # intron = mRNA - exon
        return "intronicGene"
    else
        return "intronicOutsideGene"
    end
end

# Once the bam file has been opened, count the reads for a single variant
function countReadsVariant(reader, chrom, pos)
    n = 0
    for record in eachoverlap(reader, chrom,pos:pos)
        n += 1
    end
    return n
end

# For all variants, get the number of reads from the bam file
function countReads(d)
    bamDir = "../script/files/bam"
    bam = bamDir * "/NA12878_NIST7035_recalibrated_hg38.bam"
    bai = bam * ".bai"
    reader = open(BAM.Reader, bam, index=bai)
    d2 = @transform(d, @byrow :reads = countReadsVariant(reader, :CHROM, :POS))
    close(reader)
    return d2
end

function exonicStatus(d, exons)
    # Check variants are in exon: for each variant, search in all exons
    @transform(d, @byrow :match = searchVariant(:CHROM, :POS, exons))
end

exons = getExon("GRCh38_latest_genomic_exon.csv")
d = getFN("../out/test-bed/test-allchr.vcf") |>
    x -> exonicStatus(x, exons) |>
    countReads
# # bed = DataFrame(CSV.File("../out/test-bed/exons_illumina_hg38.bed",comment="#", delim="\t", header=false))
CSV.write("falseNegatives.csv", d)
#+end_src

Discordance nombre de reads avec igv mais cohérent avec samtools : vu avec Alexis : probablement du à IVG, on reste sur samtools

$ samtools view   NA12878_NIST7035_recalibrated_hg38.bam NC_000001.11:1734812-1734812  -F "0x200" | wc -l
164

****** KILL Comprendre pourquoi le chromosome 6 a plus de FN
CLOSED: [2023-03-29 Wed 22:39]
Vérifier le graphe fait en julia
$ grep "FN.*SNP" test-allchr.vcf | grep NC_000006 -c
1264

@subset d :type .== "SNP" :CHROM .== "NC_000006.12"
1264×10 DataFrame

Région HLA: difficile car nombreux allèles alternatifs.
Mais dans nos données, sont principalement non vus.
La répartition des reads est similaire aux autres régions...
#+begin_src julia
vcf = vcfHappy("test-allchr.vcf")
d = DataFrame(CSV.File("data/falseNegatives.csv"))
d2 = @subset d :POS .>= 29600000 :POS .<= 32000000 :type .== "SNP" :CHROM .=="NC_000006.12"
#+end_src

#+RESULTS:
: julia> nrow(@subset d2 :BK .== ".")
: 612
: julia> nrow(@subset d2 :BK .!= ".")
: 6
: julia> nrow(@subset d2 :BK .!= "am")
: 612

****** KILL comparer avec Kumaran 2021
CLOSED: [2023-03-22 Wed 23:16]
Bed probablement différent, on a presque un facteur 2 sur le nombre de variant (TP..)
****** KILL Comparer avec stats de NA12878 dans example/happy sur chr21 (exons fournis)
CLOSED: [2023-03-04 Sat 11:01]
******* DONE DP_over_30_not_SNP_consensual_sequence.vcf: horrible
CLOSED: [2023-02-25 Sat 19:47]
déplorable
 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision  METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
INDEL    ALL          519        16       503         1579      1032        531      2      2       0.030829          0.015267        0.336289         0.020421                     NaN                     NaN                   1.775956                   4.768382
INDEL   PASS          519        16       503         1579      1032        531      2      2       0.030829          0.015267        0.336289         0.020421                     NaN                     NaN                   1.775956                   4.768382
  SNP    ALL        22131      1191     20940         4346      2342        813      4      1       0.053816          0.337107        0.187069         0.092815                2.971834                1.967235                   1.579776                   1.492828
  SNP   PASS        22131      1191     20940         4346      2342        813      4      1       0.053816          0.337107        0.187069         0.092815                2.971834                1.967235                   1.579776                   1.492828
******* DONE DP _over_30 : mieux
CLOSED: [2023-03-04 Sat 11:01]
/Work/Users/apraga/bisonex/work/69/cc1391f5a0fbf4db958a370ec17a19
 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision
INDEL    ALL           76        20        56           36         6         10      1      3       0.263158          0.769231
INDEL   PASS           76        20        56           36         6         10      1      3       0.263158          0.769231
  SNP    ALL          582       312       270          370         9         49      0      0       0.536082          0.971963
  SNP   PASS          582       312       270          370         9         49      0      0       0.536082          0.971963

******** DONE NC_000021.9:45515163 TA -> T : référence non normalisée ??
CLOSED: [2023-03-04 Sat 11:01]
NC_000021.9     45515163        .       TA      T       50      PASS    BS=45515163;Regions=CONF,TS_contained   GT:BD:BK:BI:BVT:BLT:QQ  0/1:FN:lm:d1_5:INDEL:het:.      ./.:.:.:.:NOCALL:nocall:0
NC_000021.9     45515163        .       TAA     T       694.02  PASS    BS=45515163;Regions=CONF,TS_contained   GT:BD:BK:BI:BVT:BLT:QQ  ./.:.:.:.:NOCALL:nocall:.       0/1:FP:lm:d1_5:INDEL:het:694.02

Mais dans notre vcf
NC_000021.9     45515163        rs11284347      TAA     T,TA

La séquence est TAAAA donc notre TAA -> TA aurait du être reconnu comme identique. Faut-il utiliser un autre fasta ?

La décomposition de xcmp est
45515163        TAA -> T
45515164        AA -> A
NB: testée avec pre.py
#+begin_src
grep '^#' NA12878_NIST7035_DP_over_30.vcf > NA12878_NIST7035_DP_over_30_chr21.vcf
grep '^NC_000021' NA12878_NIST7035_DP_over_30.vcf >> NA12878_NIST7035_DP_over_30_chr21.vcf

pre.py NA12878_NIST7035_DP_over_30_chr21.vcf out.vcf.gz -r GCA_000001405.15_GRCh38_no_alt_analysis_set.fasta
#+end_src
Selon l'algorithm, on devrait avoir TA -> T et non AA -> A (non left-align)
Si on débug:
#+begin_src
grep "^#" NA12878_NIST7035_DP_over_30_chr21.vcf > test_align.vcf
grep 45515163 NA12878_NIST7035_DP_over_30_chr21.vcf  >> test_align.vcf.gz
bgzip test_align.vcf.gz
bcftools index test_align.vcf.gz
multimerge test_align.vcf.gz -o out.vcf.gz -r GCA_000001405.15_GRCh38_no_alt_analysis_set.fasta
#+end_src
Idem... Si on utile un genome de référence plus récent ?
#+begin_src
multimerge test_align.vcf.gz -o out.vcf.gz -r /Work/Groups/bisonex/data/genome/GRCh38.p13/genomeRef.fna --process-full=1
#+end_src
Idem. Et vcfeval ?
#+begin_src
tabix HG001_GRCh38_1_22_v4.2.1_benchmark_chr21.vcf.gz
tabix test_align.vcf.gz
rtg vcfeval -b HG001_GRCh38_1_22_v4.2.1_benchmark_chr21.vcf.gz -c test_align.vcf.gz -t /Work/Groups/bisonex/data/genome/GRCh38.p13/genomeRef.sdf -o vcfeval-test
#+end_src

Selected score threshold using: maximized F-measure
Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
   99.000                  0              0          1      54828     0.0000       0.0000     0.0000
     None                  0              0          1      54828     0.0000       0.0000     0.0000

     On essaie les différentes étapes
#+begin_src
multimerge test_align.vcf.gz -o out.vcf.gz -r /Work/Groups/bisonex/data/genome/GRCh38.p13/genomeRef.fna --homref-split 1 --homref-vcf-out 1 --trimalleles 1 --splitalleles 1
#+end_src
1er changement
TAA T
TAA TA

Après réflexion, c'est la référence qui n'est pas normalisée ! (left-trimmed)
******** DONE NC_000021.9     14108836  T -> C
CLOSED: [2023-03-04 Sat 11:01]
Dans le bam mais filtré par DP
******* DONE variant calling seul : meilleur score pour l'instant (77% recall, 95% precision)
CLOSED: [2023-03-04 Sat 11:01]
tests/chr21-alexis
 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision
INDEL    ALL           76        43        33           82        18         20      3      5       0.565789          0.709677
  SNP    ALL          582       448       134          530        25         57      6      1       0.769759          0.947146

 METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
       0.243902         0.629617                     NaN                     NaN                   1.181818                   1.612903
       0.107547         0.849289                3.098592                2.925926                   1.530435                   1.774869

******** NC_000021.9:14144627 : FN, dans le bam mais pas dans le vcf (2 reads/9)
 On récupére tout le vcf: pas dedans
 Dans le bam : 9/2
 Idem pour le bam dans notre pipeline
 - base qualité 33 sur les 2 reads
https://gatk.broadinstitute.org/hc/en-us/articles/360043491652-When-HaplotypeCaller-and-Mutect2-do-not-call-an-expected-variant
https://gatk.broadinstitute.org/hc/en-us/articles/360035891111-Expected-variant-at-a-specific-site-was-not-called
On debug
#+begin_src
cd /Work/Users/apraga/bisonex/out/NA12878_NIST7035/preprocessing/applybqsr
samtools view -b NA12878_NIST7035.bam NC_000021.9 -o NA12878_NIST7035_chr21.bam
samtools index NA12878_NIST7035_chr21.bam
#+end_src
********* --debug
#+begin_src
gatk --java-options "-Xmx3g" HaplotypeCaller --input NA12878_NIST7035_chr21.bam \
    --output debug_chr1.vcf.gz \
    --reference /Work/Groups/bisonex/data/genome/GRCh38.p13/genomeRef.fna \
    --dbsnp /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/dbSNP.gz \
    --tmp-dir . \
    --max-mnp-distance 2 --debug &> lol.txt

#+end_src
Les allèles sont bien retrouvées
#+begin_quote
14:41:05.530 INFO  EventMap - === Best Haplotypes ===
14:41:05.530 INFO  EventMap - AATTTAATTTTCTTACCTTTCTGGGTATGTAAGTGATTTTA
14:41:05.530 INFO  EventMap - > Cigar = 41M
14:41:05.530 INFO  EventMap - >> Events = EventMap{}
14:41:05.530 INFO  EventMap - AATTTAATTTTCTTACCTTTTTGGGTATGTAAGTGATTTTA
14:41:05.530 INFO  EventMap - > Cigar = 41M
14:41:05.530 INFO  EventMap - >> Events = EventMap{NC_000021.9:14144627-14144627 [C*, T],}
14:41:05.530 INFO  HaplotypeCallerGenotypingEngine - Genotyping event at 14144627 with alleles = [C*, T]
#+end_quote
NB: même en filtranrt sur le chromosome 21 avec julia, haplotypecaller parcourt tous les chromosomes
********* DONE --linked-de-bruijn-graph : idem
CLOSED: [2023-02-26 Sun 17:26]
********* DONE examine sortie --bamout : non présent
CLOSED: [2023-02-26 Sun 19:53]

#+begin_src
cd test/chr21-alexis
gatk --java-options "-Xmx3g" HaplotypeCaller \
    --input /Work/Users/apraga/bisonex/script/files/bam/NA12878_chr21.bam \
    --output debug_chr1.vcf.gz \
    --reference /Work/Groups/bisonex/data/genome/GRCh38.p13/genomeRef.fna \
    --dbsnp /Work/Groups/bisonex/data/dbSNP/GRCh38.p13/dbSNP.gz \
    --tmp-dir . \
    --max-mnp-distance 2 -bamout debug.bam

#+end_src
Pas de reads

 #+begin_quote
If you see nothing overlapping your region, then it might not have been flagged as active, or could have failed to assemble.
 #+end_quote
********* 14582339: FN mais pas de reads...
********* 14583327 idem
********* 17512551 idem
********* 17567111: difference d'haplotype
********* 17567621 pas de reads
****** DONE Comparer avec sortie du variant calling vcf donné par GIAB
CLOSED: [2023-04-02 Sun 17:11]
******* DONE vcfeval
CLOSED: [2023-04-01 Sat 11:59] SCHEDULED: <2023-04-01 Sat>
#+begin_src sh
 nextflow run workflows/test.nf -profile standard,helios -resume --test.vcfeval --test.giabVCF --outdir=test-giabVCF
cat test-giabVCF/vcfeval/output/summary.txt
#+end_src

Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
    1.000              44818          44818       2892       6087     0.9394       0.8804     0.9089
     None              44819          44819       2896       6086     0.9393       0.8804     0.9089

Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
    1.000              44818          44818       2892       6087     0.9394       0.8804     0.9089
     None              44819          44819       2896       6086     0.9393       0.8804     0.9089
******* DONE happy
CLOSED: [2023-04-01 Sat 11:56]
 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.PrecisioN
INDEL   PASS         4871      3678      1193         7036      1299       2011    208    217       0.755081          0.741493
  SNP   PASS        46032     41138      4894        47694      1622       4930    362     31       0.893683          0.962071


 METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
       0.285816         0.748225                     NaN                     NaN                   1.617499                   2.524051
       0.103367         0.926617                2.529552                2.412446                   1.620686                   1.688868
****** DONE Statistiques avec vcfeval
CLOSED: [2023-04-02 Sun 17:10] SCHEDULED: <2023-04-01 Sat>
***** DONE Résultats finaux
CLOSED: [2023-04-14 Fri 09:53]
:PROPERTIES:
:ID:       cd79a77c-a0b6-4bb1-9e08-fe08dc89e3aa
:END:
Version GIAB avec hap.py + vcfeval:
#+begin_src sh
NXF_OPTS=-D"user.name=${USER}" nextflow run workflows/compareVCF.nf -profile standard,helios -resume --outdir=compareNA12878-giab --test.compare=happy,vcfeval  --test.query=giab --test.id=HG001
#+end_src

Notre version avec hap.py + vcfeval
#+begin_src sh
NXF_OPTS=-D"user.name=${USER}" nextflow run workflows/compareVCF.nf -profile standard,helios -resume --outdir=compareNA12878 --test.vcfeval --test.query="out/NA12878_NIST/variantCalling/haplotypecaller/NA12878_NIST.vcf.gz" --test.happy
#+end_src

On concatene les csv avec une colonne indicant le type
# awk '{if (NR==1) {print "Data,Algorithm" $0} else {print "bisonx,happy,"$0}}' compareNA12878/happy/NA12878.summary.csv

compareNA12878/happy/NA12878.summary.csv
| Type  | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision | METRIC.Frac_NA | METRIC.F1_Score | TRUTH.TOTAL.TiTv_ratio | QUERY.TOTAL.TiTv_ratio | TRUTH.TOTAL.het_hom_ratio | QUERY.TOTAL.het_hom_ratio |
| INDEL | ALL    |        4871 |     3461 |     1410 |        7048 |     1554 |      1987 |   193 |   346 |      0.710532 |         0.692946 |       0.281924 |        0.701629 |                        |                        |        1.6174985978687606 |        3.0674091441969518 |
| INDEL | PASS   |        4871 |     3461 |     1410 |        7048 |     1554 |      1987 |   193 |   346 |      0.710532 |         0.692946 |       0.281924 |        0.701629 |                        |                        |        1.6174985978687606 |        3.0674091441969518 |
| SNP   | ALL    |       46032 |    39367 |     6665 |       44599 |     1186 |      4042 |   304 |    30 |      0.855209 |         0.970757 |        0.09063 |        0.909327 |      2.529551552318896 |      2.402150701647346 |        1.6206857273037931 |        1.6273423688862698 |
| SNP   | PASS   |       46032 |    39367 |     6665 |       44599 |     1186 |      4042 |   304 |    30 |      0.855209 |         0.970757 |        0.09063 |        0.909327 |      2.529551552318896 |      2.402150701647346 |        1.6206857273037931 |        1.6273423688862698 |

compareNA12878/vcfeval/NA12878.summary.txt
| Threshold | True-pos-baseline | True-pos-call | False-pos | False-neg | Precision | Sensitivity | F-measure |
|-----------+-------------------+---------------+-----------+-----------+-----------+-------------+-----------|
| 3.000     |             42789 |         42416 |      2598 |      8080 |    0.9423 |      0.8412 |    0.8889 |
| None      |             42798 |         42425 |      2616 |      8071 |    0.9419 |      0.8413 |    0.8888 |

Indel avec le plus petit seuil : zcat NA12878.non_snp_roc.tsv.gz
Attention à inverser precision et recall !
 zcat NA12878.non_snp_roc.tsv.gz  | tail -n 1 | awk '{print $7 $6}'
0.71390.7136

SNP avec le plus petit seuil : zcat NA12878.non_snp_roc.tsv.gz
Attention à inverser precision et recall !
$ zcat NA12878.snp_roc.tsv.gz  | tail -n 1 | awk '{print $7 $6}'
0.85470.9727

compareNA12878-giab/vcfeval/NA12878.summary.txt
| Threshold | True-pos-baseline | True-pos-call | False-pos | False-neg | Precision | Sensitivity | F-measure |
| 1.000     |             44812 |         44812 |      2878 |      6057 |    0.9397 |      0.8809 |    0.9093 |
| None      |             44813 |         44813 |      2882 |      6056 |    0.9396 |      0.8809 |    0.9093 |

SNP:
$ zcat NA12878.snp_roc.tsv.gz  | tail -n 1 | awk '{print $7 $6}'
0.89370.9621

indel
$ zcat NA12878.non_snp_roc.tsv.gz  | tail -n 1 | awk '{print $7 $6}'
0.75980.7445

compareNA12878-giab/happy/NA12878.summary.csv
| Type  | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision | METRIC.Frac_NA | METRIC.F1_Score | TRUTH.TOTAL.TiTv_ratio | QUERY.TOTAL.TiTv_ratio | TRUTH.TOTAL.het_hom_ratio | QUERY.TOTAL.het_hom_ratio |
|-------+--------+-------------+----------+----------+-------------+----------+-----------+-------+-------+---------------+------------------+----------------+-----------------+------------------------+------------------------+---------------------------+---------------------------|
| INDEL | ALL    |        4871 |     3678 |     1193 |        7036 |     1299 |      2011 |   208 |   217 |      0.755081 |         0.741493 |       0.285816 |        0.748225 |                        |                        |        1.6174985978687606 |        2.5240506329113925 |
| INDEL | PASS   |        4871 |     3678 |     1193 |        7036 |     1299 |      2011 |   208 |   217 |      0.755081 |         0.741493 |       0.285816 |        0.748225 |                        |                        |        1.6174985978687606 |        2.5240506329113925 |
| SNP   | ALL    |       46032 |    41138 |     4894 |       47694 |     1622 |      4930 |   362 |    31 |      0.893683 |         0.962071 |       0.103367 |        0.926617 |      2.529551552318896 |     2.4124463519313304 |        1.6206857273037931 |        1.6888675840288743 |
| SNP   | PASS   |       46032 |    41138 |     4894 |       47694 |     1622 |      4930 |   362 |    31 |      0.893683 |         0.962071 |       0.103367 |        0.926617 |      2.529551552318896 |     2.4124463519313304 |        1.6206857273037931 |         1.688867584028874 |
***** KILL Résultats sans trimming
CLOSED: [2023-06-25 Sun 15:53] SCHEDULED: <2023-06-26 Mon>
***** DONE Refaire : HiSeq4000 + agilent sureselect + génome "prêt à l'emploi"
CLOSED: [2023-06-30 Fri 22:08] SCHEDULED: <2023-06-25 Sun>
#+begin_src
nextflow run workflows/compareVCF.nf -profile standard,helios --outdir=out/HG001-SRX11061486_SRR14724513-GRCh38 --query=out/HG001-SRX11061486_SRR14724513-GRCh38/callVariant/haplotypecaller/HG001-SRX11061486_SRR14724513-GRCh38.vcf.gz --compare=vcfeval,happy -lib lib --capture=capture/Agilent_SureSelect_All_Exons_v7_hg38_Regions.bed  --id=HG001
#+end_src

Meilleurs résultats !
| Type  | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision | METRIC.Frac_NA | METRIC.F1_Score | TRUTH.TOTAL.TiTv_ratio | QUERY.TOTAL.TiTv_ratio | TRUTH.TOTAL.het_hom_ratio | QUERY.TOTAL.het_hom_ratio |
| INDEL | ALL    |         549 |      489 |       60 |         899 |       64 |       340 |     8 |    17 |       0.89071 |          0.88551 |       0.378198 |        0.888102 |                        |                        |          1.86096256684492 |         2.247272727272727 |
| INDEL | PASS   |         549 |      489 |       60 |         899 |       64 |       340 |     8 |    17 |       0.89071 |          0.88551 |       0.378198 |        0.888102 |                        |                        |          1.86096256684492 |         2.247272727272727 |
| SNP   | ALL    |       21973 |    21462 |      511 |       26285 |      563 |      4263 |    68 |    16 |      0.976744 |         0.974435 |       0.162184 |        0.975588 |      3.007110300820419 |       2.78468624064479 |        1.5918102430965306 |        1.8161449399656946 |
| SNP   | PASS   |       21973 |    21462 |      511 |       26285 |      563 |      4263 |    68 |    16 |      0.976744 |         0.974435 |       0.162184 |        0.975588 |      3.007110300820419 |       2.78468624064479 |        1.5918102430965306 |        1.8161449399656946 |

***** DONE Refaire : HiSeq4000 + agilent sureselect + génome "prêt à l'emploi" gatk-4.4
CLOSED: [2023-08-03 Thu 23:55] SCHEDULED: <2023-06-25 Sun>
#+begin_src
 ID="HG001-SRX11061486_SRR14724513-GRCh38"; nextflow run workflows/compareVCF.nf -profile standard,helios --outdir=out/${ID} --query=out/${ID}/callVariant/haplotypecaller/${ID}.vcf.gz --compare=vcfeval,hay -lib lib --capture=capture/Agilent_SureSelect_All_Exons_v7_hg38_Regions.bed  --id=HG001 --genome=GRCh38 -bg
#+end_src

***** KILL Utiliser d'autres données brutes ?
CLOSED: [2023-06-25 Sun 15:58]
https://zenodo.org/record/3597727
Capture en hg37 également. Serait intéressant mais pas le temps..
***** KILL Comparer avec UCSCS liftover
CLOSED: [2023-06-26 Mon 19:02] SCHEDULED: <2023-06-25 Sun>
Picard liftoverinterval est basé sur UCSCS
Mais on n'aurait pas la différence pour NA12878 qu'on voit...
**** DONE HG002 :hg002:hg38:
CLOSED: [2023-07-30 Sun 14:25] SCHEDULED: <2023-07-25 Tue>
#+begin_src
    NXF_OPTS=-D"user.name=${USER}" nextflow run workflows/giabFastq.nf -profile standard,helios
    NXF_OPTS=-D"user.name=${USER}" nextflow run main.nf -profile standard,helios -resume --input="/Work/Groups/bisonex/data/giab/GRCh38/HG002_{1,2}.fq.gz --test.id=HG002

Only the capture file differs. Results are better using the capture file given by Agilent, stored in data/

    NXF_OPTS=-D"user.name=${USER}" nextflow run workflows/compareVCF.nf -profile standard,helios -resume --outdir=compareHG002 --test.id=HG002 --test.query=out/HG002_1/variantCalling/haplotypecaller/HG002_1.vcf.gz  --test.compare=vcfeval,happy --test.capture=data/AgilentSureSelectv05_hg38.bed
#
#+end_src
***** DONE Mauvais résultats
CLOSED: [2023-04-14 Fri 09:42]
avec vcfeval

Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
    0.000              24585          24390      10060      39415     0.7080       0.3841     0.4980
     None              24585          24390      10060      39415     0.7080       0.3841     0.4980

La sortie du variantCalling est celle d'happy ???
On relance...
***** DONE Vérifier vcf en hg38
CLOSED: [2023-04-12 Wed 10:33] SCHEDULED: <2023-04-12 Wed>
***** KILL Capture en hg19 ?
CLOSED: [2023-04-13 Thu 09:46] SCHEDULED: <2023-04-12 Wed>
***** KILL Vraiment fichier de capture ou zone d'intérêt ?
CLOSED: [2023-04-13 Thu 09:45] SCHEDULED: <2023-04-12 Wed>
"target region" +/- 50bp
[[https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/giab/data/AshkenazimTrio/analysis/OsloUniversityHospital_Exome_GATK_jointVC_11242015/README.txt][README]]
 list file describing the variant calling regions (target regions extended with 50 bp on each end)
***** DONE .bed fourni par AGilent: sensbilité très mauvaise
CLOSED: [2023-04-13 Thu 09:46] SCHEDULED: <2023-04-13 Thu>
Agilent SureSelect Human All Exon V5 kit
Disponible en hg38

Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
    0.000              19653          19501       6410      21657     0.7526       0.4757     0.5830
     None              19653          19501       6410      21657     0.7526       0.4757     0.5830
***** DONE Trier par nom avec samtools sort : bons résultats
CLOSED: [2023-04-14 Fri 09:25] SCHEDULED: <2023-04-13 Thu>
Avec capture fourni par GIAB

vcf eval
Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
    5.000              57443          57032        984       6557     0.9830       0.8975     0.9383
     None              57457          57046       1009       6543     0.9826       0.8978     0.9383

Happy
| Type  | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision | METRIC.Frac_NA | METRIC.F1_Score | TRUTH.TOTAL.TiTv_ratio | QUERY.TOTAL.TiTv_ratio | TRUTH.TOTAL.het_hom_ratio | QUERY.TOTAL.het_hom_ratio |
|-------+--------+-------------+----------+----------+-------------+----------+-----------+-------+-------+---------------+------------------+----------------+-----------------+------------------------+------------------------+---------------------------+---------------------------|
| INDEL | ALL    |        6150 |     5007 |     1143 |        6978 |      556 |      1346 |   151 |   168 |      0.814146 |         0.901278 |       0.192892 |          0.8555 |                        |                        |        1.5434221840068787 |        1.9467178175618074 |
| INDEL | PASS   |        6150 |     5007 |     1143 |        6978 |      556 |      1346 |   151 |   168 |      0.814146 |         0.901278 |       0.192892 |          0.8555 |                        |                        |        1.5434221840068787 |        1.9467178175618074 |
| SNP   | ALL    |       57818 |    52464 |     5354 |       56016 |      500 |      3046 |    90 |    30 |      0.907399 |         0.990561 |       0.054377 |        0.947158 |     2.4892012548262548 |      2.426824047458871 |        1.5904527117884357 |        1.6107795598657217 |
| SNP   | PASS   |       57818 |    52464 |     5354 |       56016 |      500 |      3046 |    90 |    30 |      0.907399 |         0.990561 |       0.054377 |        0.947158 |     2.4892012548262548 |      2.426824047458871 |        1.5904527117884357 |        1.6107795598657217 |
***** DONE Capture agilent légment meilleur que celui fourni par GIAB (padding ?)
CLOSED: [2023-04-14 Fri 09:48]
GIAB:

vcf eval
Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
    5.000              57443          57032        984       6557     0.9830       0.8975     0.9383
     None              57457          57046       1009       6543     0.9826       0.8978     0.9383

Happy
| Type  | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision | METRIC.Frac_NA | METRIC.F1_Score | TRUTH.TOTAL.TiTv_ratio | QUERY.TOTAL.TiTv_ratio | TRUTH.TOTAL.het_hom_ratio | QUERY.TOTAL.het_hom_ratio |
|-------+--------+-------------+----------+----------+-------------+----------+-----------+-------+-------+---------------+------------------+----------------+-----------------+------------------------+------------------------+---------------------------+---------------------------|
| INDEL | ALL    |        6150 |     5007 |     1143 |        6978 |      556 |      1346 |   151 |   168 |      0.814146 |         0.901278 |       0.192892 |          0.8555 |                        |                        |        1.5434221840068787 |        1.9467178175618074 |
| INDEL | PASS   |        6150 |     5007 |     1143 |        6978 |      556 |      1346 |   151 |   168 |      0.814146 |         0.901278 |       0.192892 |          0.8555 |                        |                        |        1.5434221840068787 |        1.9467178175618074 |
| SNP   | ALL    |       57818 |    52464 |     5354 |       56016 |      500 |      3046 |    90 |    30 |      0.907399 |         0.990561 |       0.054377 |        0.947158 |     2.4892012548262548 |      2.426824047458871 |        1.5904527117884357 |        1.6107795598657217 |
| SNP   | PASS   |       57818 |    52464 |     5354 |       56016 |      500 |      3046 |    90 |    30 |      0.907399 |         0.990561 |       0.054377 |        0.947158 |     2.4892012548262548 |      2.426824047458871 |        1.5904527117884357 |        1.6107795598657217 |

Agilent
Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
    6.000              37241          36965        449       4069     0.9880       0.9015     0.9428
     None              37248          36972        461       4062     0.9877       0.9017     0.9427

| Type  | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision | METRIC.Frac_NA | METRIC.F1_Score | TRUTH.TOTAL.TiTv_ratio | QUERY.TOTAL.TiTv_ratio | TRUTH.TOTAL.het_hom_ratio | QUERY.TOTAL.het_hom_ratio |
| INDEL | ALL    |        2909 |     2477 |      432 |        3229 |      207 |       519 |    52 |    50 |      0.851495 |         0.923616 |       0.160731 |        0.886091 |                        |                        |        1.4964850615114236 |        1.8339222614840989 |
| INDEL | PASS   |        2909 |     2477 |      432 |        3229 |      207 |       519 |    52 |    50 |      0.851495 |         0.923616 |       0.160731 |        0.886091 |                        |                        |        1.4964850615114236 |        1.8339222614840989 |
| SNP   | ALL    |       38406 |    34793 |     3613 |       36935 |      275 |      1868 |    37 |    15 |      0.905926 |         0.992158 |       0.050575 |        0.947083 |     2.6247759222568168 |     2.5752854654538417 |         1.588953331534934 |        1.6192536889897844 |
| SNP   | PASS   |       38406 |    34793 |     3613 |       36935 |      275 |      1868 |    37 |    15 |      0.905926 |         0.992158 |       0.050575 |        0.947083 |     2.6247759222568168 |     2.5752854654538417 |         1.588953331534934 |        1.6192536889897844 |
***** DONE Refaire : HiSeq4000 + agilent sureselect + génome "prêt à l'emploi"
CLOSED: [2023-07-30 Sun 14:24] SCHEDULED: <2023-07-23 Sun>
| Type  | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision | METRIC.Frac_NA | METRIC.F1_Score | TRUTH.TOTAL.TiTv_ratio | QUERY.TOTAL.TiTv_ratio | TRUTH.TOTAL.het_hom_ratio | QUERY.TOTAL.het_hom_ratio |
| INDEL | ALL    |         626 |      536 |       90 |         900 |       23 |       334 |     6 |    14 |       0.85623 |         0.959364 |       0.371111 |        0.904868 |                        |                        |        1.5696202531645569 |         2.406844106463878 |
| INDEL | PASS   |         626 |      536 |       90 |         900 |       23 |       334 |     6 |    14 |       0.85623 |         0.959364 |       0.371111 |        0.904868 |                        |                        |        1.5696202531645569 |         2.406844106463878 |
| SNP   | ALL    |       23136 |    22346 |      790 |       26289 |      103 |      3845 |    77 |     4 |      0.965854 |         0.995411 |       0.146259 |         0.98041 |     2.9282199219412863 |     2.7752583237657866 |        1.6348301800775018 |        1.8423330808354075 |
| SNP   | PASS   |       23136 |    22346 |      790 |       26289 |      103 |      3845 |    77 |     4 |      0.965854 |         0.995411 |       0.146259 |         0.98041 |     2.9282199219412863 |     2.7752583237657866 |        1.6348301800775018 |        1.8423330808354075 |

***** DONE Refaire : HiSeq4000 + agilent sureselect + génome "prêt à l'emploi" gatk-4.4
CLOSED: [2023-08-03 Thu 23:24] SCHEDULED: <2023-08-03 Thu>
#+begin_src sh
ID="HG002-SRX11061487_SRR14724512-GRCh38" ; nextflow run workflows/compareVCF.nf -profile standard,helios --outdir=out/${ID} --query=out/${ID}/callVariant/haplotypecaller/${ID}.vcf.gz --compare=vcfeval,happy -lib lib --capture=capture/Agilent_SureSelect_All_Exons_v7_hg38_Regions.bed  --id=HG002 --genome=GRCh38
#+end_src
**** DONE HG003 :hg003:hg38:
CLOSED: [2023-07-30 Sun 14:26]
***** Notes
#+begin_src sh
NXF_OPTS=-D"user.name=${USER}" nextflow run main.nf -profile standard,helios  --input /Work/Groups/bisonex/data/giab/GRCh38/HG003_{1,2}.fq.gz -bg
#+end_src
#+begin_src  sh
NXF_OPTS=-D"user.name=${USER}" nextflow run workflows/compareVCF.nf -profile standard,helios -resume --outdir=compareHG003  --test.id=HG003 --test.query=out/HG003_1/variantCalling/haplotypecaller/HG003_1.vcf.gz  --test.compare=vcfeval,happy --test.capture=data/AgilentSureSelectv05_hg38.bed
#+end_src
vcfeval
Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
    5.000              36745          36473        486       3988     0.9869       0.9021     0.9426
     None              36748          36476        495       3985     0.9866       0.9022     0.9425

$ zcat NA12878.snp_roc.tsv.gz  | tail -n 1 | awk '{print $7 $6}'

happy
Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision  METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
INDEL    ALL         2731      2290       441         3092       208        577     62     53       0.838521          0.917296        0.186611         0.876141                     NaN                     NaN                   1.505145                   1.888993
INDEL   PASS         2731      2290       441         3092       208        577     62     53       0.838521          0.917296        0.186611         0.876141                     NaN                     NaN                   1.505145                   1.888993
  SNP    ALL        37997     34481      3516        36861       306       2074     33     13       0.907466          0.991204        0.056265         0.947488                2.611269                2.565915                   1.555780                   1.621727
  SNP   PASS        37997     34481      3516        36861       306       2074     33     13       0.907466          0.991204        0.056265         0.947488                2.611269                2.5659
***** DONE Refaire : HiSeq4000 + agilent sureselect + génome "prêt à l'emploi"
CLOSED: [2023-07-30 Sun 14:25] SCHEDULED: <2023-07-23 Sun>
| Type  | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision | METRIC.Frac_NA | METRIC.F1_Score | TRUTH.TOTAL.TiTv_ratio | QUERY.TOTAL.TiTv_ratio | TRUTH.TOTAL.het_hom_ratio | QUERY.TOTAL.het_hom_ratio |
| INDEL | ALL    |         644 |      538 |      106 |         914 |       32 |       337 |     7 |    19 |      0.835404 |         0.944541 |       0.368709 |        0.886626 |                        |                        |        1.7444933920704846 |        2.3138686131386863 |
| INDEL | PASS   |         644 |      538 |      106 |         914 |       32 |       337 |     7 |    19 |      0.835404 |         0.944541 |       0.368709 |        0.886626 |                        |                        |        1.7444933920704846 |        2.3138686131386863 |
| SNP   | ALL    |       23126 |    22271 |      855 |       26405 |      135 |      4002 |    90 |    20 |      0.963029 |         0.993974 |       0.151562 |        0.978257 |      2.949462182004439 |     2.7766657134686876 |        1.6080333972695475 |        1.8465106245280984 |
| SNP   | PASS   |       23126 |    22271 |      855 |       26405 |      135 |      4002 |    90 |    20 |      0.963029 |         0.993974 |       0.151562 |        0.978257 |      2.949462182004439 |     2.7766657134686876 |        1.6080333972695475 |        1.8465106245280984 |

***** DONE Refaire : HiSeq4000 + agilent sureselect + génome "prêt à l'emploi" gatk-4.4
CLOSED: [2023-08-03 Thu 23:24] SCHEDULED: <2023-08-03 Thu>
#+begin_src
ID="HG003-SRX11061488_SRR14724511-GRCh38" ; nextflow run workflows/compareVCF.nf -profile standard,helios --outdir=out/${ID} --query=out/${ID}/callVariant/haplotypecaller/${ID}.vcf.gz --compare=vcfeval,happy -lib lib --capture=capture/Agilent_SureSelect_All_Exons_v7_hg38_Regions.bed  --id=HG003 --genome=GRCh38
#+end_src
**** TODO HG004 :hg38:hg004:
***** Notes
#+begin_src sh
NXF_OPTS=-D"user.name=${USER}" nextflow run main.nf -profile standard,helios  --input /Work/Groups/bisonex/data/giab/GRCh38/HG004_{1,2}.fq.gz -bg
#+end_src

vcfeval
Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
    6.000              36938          36678        421       4040     0.9887       0.9014     0.9430
     None              36942          36682        432       4036     0.9884       0.9015     0.9429


happy
 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision  METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
INDEL    ALL         2787      2388       399         3183       195        580     53     38       0.856835          0.925086        0.182218         0.889654                     NaN                     NaN                   1.507834                   1.848649
INDEL   PASS         2787      2388       399         3183       195        580     53     38       0.856835          0.925086        0.182218         0.889654                     NaN                     NaN                   1.507834                   1.848649
  SNP    ALL        38185     34560      3625        36921       254       2107     46      7       0.905067          0.992704        0.057068         0.946862                2.589175                2.553546                   1.632595                   1.653534
  SNP   PASS        38185     34560      3625        36921       254       2107     46      7       0.905067          0.992704        0.057068         0.946862                2.589175                2.553546                   1.632595                   1.653534
***** DONE Refaire : HiSeq4000 + agilent sureselect + génome "prêt à l'emploi"
CLOSED: [2023-07-30 Sun 14:39] SCHEDULED: <2023-07-23 Sun>
| Type  | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision | METRIC.Frac_NA | METRIC.F1_Score | TRUTH.TOTAL.TiTv_ratio | QUERY.TOTAL.TiTv_ratio | TRUTH.TOTAL.het_hom_ratio | QUERY.TOTAL.het_hom_ratio |
| INDEL | ALL    |         588 |      511 |       77 |         873 |       15 |       332 |     7 |     8 |      0.869048 |         0.972274 |       0.380298 |        0.917767 |                        |                        |        1.6111111111111112 |                 2.3984375 |
| INDEL | PASS   |         588 |      511 |       77 |         873 |       15 |       332 |     7 |     8 |      0.869048 |         0.972274 |       0.380298 |        0.917767 |                        |                        |        1.6111111111111112 |                 2.3984375 |
| SNP   | ALL    |       23041 |    22225 |      816 |       26302 |      114 |      3972 |    87 |     5 |      0.964585 |         0.994895 |       0.151015 |        0.979505 |       2.91407709288504 |     2.7368645271229766 |        1.6883829538820783 |        1.8632748665431964 |
| SNP   | PASS   |       23041 |    22225 |      816 |       26302 |      114 |      3972 |    87 |     5 |      0.964585 |         0.994895 |       0.151015 |        0.979505 |       2.91407709288504 |     2.7368645271229766 |        1.6883829538820783 |        1.8632748665431964 |

***** DONE Refaire : HiSeq4000 + agilent sureselect + génome "prêt à l'emploi" gatk4.4
CLOSED: [2023-08-03 Thu 23:24] SCHEDULED: <2023-08-03 Thu>
#+begin_src sh
ID="HG004-SRX11061489_SRR14724510-GRCh38" ; nextflow run workflows/compareVCF.nf -profile standard,helios --outdir=out/${ID} --query=out/${ID}/callVariant/haplotypecaller/${ID}.vcf.gz --compare=vcfeval,happy -lib lib --capture=capture/Agilent_SureSelect_All_Exons_v7_hg38_Regions.bed  --id=HG003 --genome=GRCh38
#+end_src
**** DONE Refaire la figure pour HG001,2,3,4
CLOSED: [2023-08-05 Sat 09:10] SCHEDULED: <2023-08-04 Fri>
**** DONE Regarder impact changement de version gatk 4.3 -> 4.4: aucun
CLOSED: [2023-08-05 Sat 09:07] SCHEDULED: <2023-08-05 Sat>

**** STRT HG001 :hg001:T2T:
Avec liftover : 10x moins de variants...

Type,Filter,TRUTH.TOTAL,TRUTH.TP,TRUTH.FN,QUERY.TOTAL,QUERY.FP,QUERY.UNK,FP.gt,FP.al,METRIC.Recall,METRIC.Precision,METRIC.Frac_NA,METRIC.F1_Score,TRUTH.TOTAL.TiTv_ratio,QUERY.TOTAL.TiTv_ratio,TRUTH.TOTAL.het_hom_ratio,QUERY.TOTAL.het_hom_ratio
INDEL,ALL,413,246,167,751,289,215,2,93,0.595642,0.460821,0.286285,0.519629,,,2.4285714285714284,2.4651162790697674
INDEL,PASS,413,246,167,751,289,215,2,93,0.595642,0.460821,0.286285,0.519629,,,2.4285714285714284,2.4651162790697674
SNP,ALL,11236,10985,251,23597,9771,2841,26,58,0.977661,0.529245,0.120397,0.686734,3.1146100329549617,2.857049501715406,3.640644361833953,2.1146328578975173
SNP,PASS,11236,10985,251,23597,9771,2841,26,58,0.977661,0.529245,0.120397,0.686734,3.1146100329549617,2.857049501715406,3.640644361833953,2.1146328578975173

***** TODO Comprendre mauvais résultat
****** DONE 2x moins de variants, beaucoup trop de FP : erreur de FILTER
CLOSED: [2023-07-08 Sat 11:17] SCHEDULED: <2023-07-07 Fri>
******* DONE [#A] Certains FP sont des vrais FP: erreur de flag et non d'happy
CLOSED: [2023-07-08 Sat 10:43] SCHEDULED: <2023-07-07 Fri>

******** Ex: chr1:408793
G	C	600.64	.	BS=408793;Regions=CONF,TS_contained	GT:BD:BK:BI:BVT:BLT:QQ	./.:.:.:.:NOCALL:nocall:.	0/1:FP:.:tv:SNP:het:600.64
CLOSED: [2023-07-07 Fri 20:03]
********* DONE Vérifier que l'on est bien dans la zone "high confidence" définie dans le bed
oui:
chr1	408514	414397	.	500	+
********* DONE Vérifier que le variant est bien dans le truth et query
CLOSED: [2023-07-07 Fri 20:14]
truth
chr1    408793  .       G       C       50      MismatchedRefAllele     AttemptedAlleles=C*->G;AttemptedLocus=chr1:408793-408793;SwappedAlleles;callable=CS_HiSeqPE300xGATK_callable,CS_10XLRGATK_callable,CS_CCS15kb_20kbGATK4_callable,CS_CGnormal_callable,CS_HiSeqPE300xfreebayes_callable;callsetnames=HiSeqPE300xGATK,CCS15kb_20kbGATK4,CGnormal,HiSeqPE300xfreebayes,CCS15kb_20kbDV,10XLRGATK;callsets=6;datasetnames=HiSeqPE300x,CCS15kb_20kb,CGnormal,10XChromiumLR;datasets=4;datasetsmissingcall=IonExome,SolidSE75bp;filt=CS_CCS15kb_20kbDV_filt,CS_CCS15kb_20kbGATK4_filt;platformnames=Illumina,PacBio,CG,10X;platforms=4       GT:AD:ADALL:DP:GQ       1/0:153,159:141,132:689:99

query
chr1    408793  .       G       C       600.64  .       AC=1;AF=0.500;AN=2;BaseQRankSum=-1.390;DP=64;ExcessHet=0.0000;FS=0.961;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=9.53;ReadPosRankSum=-3.146;SOR=0.906     GT:AD:DP:GQ:PL  0/1:33,30:63:99:608,0,798
********* DONE Tester happy sur ce variant: non retrouvé
CLOSED: [2023-07-08 Sat 10:42]
Il faut au moins 2 variant + les index (sinon "no chromosome in common")
Donc on augmente la taille de la région
#+begin_src sh :dir ~/code/bisonex/out
bcftools view HG001_GRCh38_1_22_v4_lifted_merged.vcf.gz  -o lifted_test.vcf.gz chr1:408793-408900
bcftools view HG001-SRX11061486_SRR14724513-T2T.vcf.gz -o run_test.vcf.gz chr1:408793-408900
bcftools index lifted_test.vcf.gz
bcftools index run_test.vcf.gz
#+end_src

 #+begin_src sh
 hap.py lifted_test.vcf.gz run_test.vcf.gz --reference ../data/chm13v2.0.fa -o test
 #+end_src
On trouve bien les 2 variants. L'indel est retrouvé mais pas le SNP
chr1    408793  .       G       C       600.64  .       BS=408793       GT:BD:BK:BI:BVT:BLT:QQ  ./.:.:.:.:NOCALL:nocall:.       0/1:FP:.:tv:SNP:het:600.64
chr1    408874  .       CGAA    C       1476.6  .       BS=408874       GT:BD:BK:BI:BVT:BLT:QQ  0/1:TP:gm:d1_5:INDEL:het:1476.6 0/1:TP:gm:d1_5:INDEL:het:1476.6
********* DONE Changer flag : mismatch -> PASS: OUI
CLOSED: [2023-07-08 Sat 10:42]
********* DONE Changer haplotype 1/0 -> 0/1 truth: idem
CLOSED: [2023-07-08 Sat 10:35]
********* DONE Tester avec vcfeval : idemnt
CLOSED: [2023-07-08 Sat 10:43]
******* DONE Comparer hg38 et T2T: 2x moinsr de variants, trop de FP et FN
CLOSED: [2023-07-07 Fri 18:38]
T2T
| Type  | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision | METRIC.Frac_NA | METRIC.F1_Score | TRUTH.TOTAL.TiTv_ratio | QUERY.TOTAL.TiTv_ratio | TRUTH.TOTAL.het_hom_ratio | QUERY.TOTAL.het_hom_ratio |
| INDEL | ALL    |         413 |      246 |      167 |         751 |      289 |       215 |     2 |    93 |      0.595642 |         0.460821 |       0.286285 |        0.519629 |                    NaN |                    NaN |                  2.428571 |                  2.465116 |
| INDEL | PASS   |         413 |      246 |      167 |         751 |      289 |       215 |     2 |    93 |      0.595642 |         0.460821 |       0.286285 |        0.519629 |                    NaN |                    NaN |                  2.428571 |                  2.465116 |
| SNP   | ALL    |       11236 |    10985 |      251 |       23597 |     9771 |      2841 |    26 |    58 |      0.977661 |         0.529245 |       0.120397 |        0.686734 |                3.11461 |                2.85705 |                  3.640644 |                  2.114633 |
| SNP   | PASS   |       11236 |    10985 |      251 |       23597 |     9771 |      2841 |    26 |    58 |      0.977661 |         0.529245 |       0.120397 |        0.686734 |                3.11461 |                2.85705 |                  3.640644 |                  2.114633 |

Hg38
| Type  | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision | METRIC.Frac_NA | METRIC.F1_Score | TRUTH.TOTAL.TiTv_ratio | QUERY.TOTAL.TiTv_ratio | TRUTH.TOTAL.het_hom_ratio | QUERY.TOTAL.het_hom_ratio |
| INDEL | ALL    |         549 |      489 |       60 |         899 |       64 |       340 |     8 |    17 |      0.890710 |         0.885510 |       0.378198 |        0.888102 |                    NaN |                    NaN |                  1.860963 |                  2.247273 |
| INDEL | PASS   |         549 |      489 |       60 |         899 |       64 |       340 |     8 |    17 |      0.890710 |         0.885510 |       0.378198 |        0.888102 |                    NaN |                    NaN |                  1.860963 |                  2.247273 |
| SNP   | ALL    |       21973 |    21462 |      511 |       26285 |      563 |      4263 |    68 |    16 |      0.976744 |         0.974435 |       0.162184 |        0.975588 |                3.00711 |               2.784686 |                  1.591810 |                  1.816145 |
| SNP   | PASS   |       21973 |    21462 |      511 |       26285 |      563 |      4263 |    68 |    16 |      0.976744 |         0.974435 |       0.162184 |        0.975588 |                3.00711 |               2.784686 |                  1.591810 |                  1.816145 |

******** Résumé
T2T
| Type  | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision |
| INDEL |         413 |      246 |      167 |         751 |      289 |       215 |     2 |    93 |      0.595642 |         0.460821 |
| SNP   |       11236 |    10985 |      251 |       23597 |     9771 |      2841 |    26 |    58 |      0.977661 |         0.529245 |

Hg38
| Type  | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision |
| INDEL |         549 |      489 |       60 |         899 |       64 |       340 |     8 |    17 |      0.890710 |         0.885510 |
| SNP   |       21973 |    21462 |      511 |       26285 |      563 |      4263 |    68 |    16 |      0.976744 |         0.974435 |
****** DONE Interesection des bed: similaire
CLOSED: [2023-07-04 Tue 23:11]
HG38
 #+begin_src sh
 bedtools intersect -a capture/Agilent_SureSelect_All_Exons_v7_hg38_Regions.bed -b /Work/Groups/bisonex/data/giab/GRCh38/HG001_GRCh38_1_22_v4.2.1_benchmark.bed  | wc -l
 #+end_src
 204280

 T2T
 #+begin_src sh
 bedtools intersect -a /Work/Groups/bisonex/data/giab/T2T/Agilent_SureSelect_All_Exons_v7_hg38_Regions_hg38_T2T.bed -b /Work/Groups/bisonex/data/giab/T2T/HG001_GRCh38_1_22_v4.2.1_benchmark_hg38_T2T.bed  | wc -l
 #+end_src
 204021
****** DONE Vérifier la ligne de commande
CLOSED: [2023-07-04 Tue 23:38]
#+begin_src sh
hap.py \
    HG001_GRCh38_1_22_v4_lifted_merged.vcf.gz \
    HG001-SRX11061486_SRR14724513-T2T.vcf.gz \
     \
    --reference chm13v2.0.fa \
    --threads 6 \
     \
    -T Agilent_SureSelect_All_Exons_v7_hg38_Regions_hg38_T2T.bed \
    --false-positives HG001_GRCh38_1_22_v4.2.1_benchmark_hg38_T2T.bed \
     \
    -o HG001
#+end_src
****** DONE Corriger FILTER : mieux mais toujours trop de négatifs. 3/4 SNP retrouvés
CLOSED: [2023-07-08 Sat 15:19] SCHEDULED: <2023-07-08 Sat>
 Type Filter  TRUTH.TOTAL  TRUTH.TP  TRUTH.FN  QUERY.TOTAL  QUERY.FP  QUERY.UNK  FP.gt  FP.al  METRIC.Recall  METRIC.Precision  METRIC.Frac_NA  METRIC.F1_Score  TRUTH.TOTAL.TiTv_ratio  QUERY.TOTAL.TiTv_ratio  TRUTH.TOTAL.het_hom_ratio  QUERY.TOTAL.het_hom_ratio
INDEL    ALL          413       246       167          751       289        215      2     98       0.595642          0.460821        0.286285         0.519629                     NaN                     NaN                   2.428571                   2.465116
INDEL   PASS          413       246       167          751       289        215      2     98       0.595642          0.460821        0.286285         0.519629                     NaN                     NaN                   2.428571                   2.465116
  SNP    ALL        15883     15479       404        23597      5277       2841     46     44       0.974564          0.745760        0.120397         0.844947                3.017198                 2.85705                   5.560099                   2.114633
  SNP   PASS        15883     15479       404        23597      5277       2841     46     44       0.974564          0.745760        0.120397         0.844947                3.017198                 2.85705                   5.560099                   2.114633
******* DONE Vérifier qu'il ne reste plus de filtre autre que PASS
CLOSED: [2023-07-08 Sat 15:19]
#+begin_src
$ zgrep -c 'PASS' HG001_GRCh38_1_22_v4_lifted_merged.vcf.gz
3730505
$ zgrep -c '^chr' HG001_GRCh38_1_22_v4_lifted_merged.vcf.gz
3730506
#+end_src
****** TODO 1/4 SNP manquant ?
******* DONE Regarder avec Julia si ce sont vraiment des FP: 61/5277 qui ne le sont pas
CLOSED: [2023-07-09 Sun 12:09]
******* DONE Examiner les FP
CLOSED: [2023-07-30 Sun 22:05]
******* DONE Tester un FP
CLOSED: [2023-07-30 Sun 22:05]
  2 │ chr1        608765  A           G           ./.:.:.:.:NOCALL:nocall:.  1/1:FP:.:ti:SNP:homalt:188
  liftDown UCSC: rien en GIAB : vrai FP

 3 │ chr1        762943  A           G           ./.:.:.:.:NOCALL:nocall:.  1/1:FP:.:ti:SNP:homalt:287
 4 │ chr1        762945  A           T           ./.:.:.:.:NOCALL:nocall:.  1/1:FP:.:tv:SNP:homalt:287
 Remaniements complexes ? Pas dans le gène en HG38
******* DONE La plupart des FP (4705/5566) sont homozygotes: erreur de référence ?
CLOSED: [2023-07-12 Wed 21:10] SCHEDULED: <2023-07-09 Sun>
Sur les 2 premiers variants, ils montrent en fait la différence entre T2T et GRCh38
Erreur à l'alignement ?
******** KILL relancer l'alignement
CLOSED: [2023-07-09 Sun 17:36]
******** DONE vérifier reads identiques hg38 et T2T: oui
CLOSED: [2023-07-09 Sun 16:36]
T2T CHR1608765
38   	chr1:1180168-1180168 (

SRR14724513.24448214
SRR14724513.24448214
******* DONE Vérifier quelques variants sur IGV
CLOSED: [2023-07-09 Sun 17:36]
******* KILL Répartition des FP : cluster ?
CLOSED: [2023-07-09 Sun 17:36]
****** DONE Examiner les FP restant après correction selon séquence de référence
CLOSED: [2023-08-12 Sat 15:57]
****** HOLD Examiner les variants supprimé
****** TODO Enlever les FP qui correspondent à un changement dans le génome
******* Condition:
- pas de variation à la position en GRCh38
- variantion homozygote
- la varation en T2T correspond au changement de pair de base GRC38 -> T2T
  pour les SNP:
  alt_T2T[i] = DNA_GRC38[j]
  avec i la position en T2T et j la position en GRCh38

  Note: définir un ID n'est pas correct car les variants peuvent être modifié par happy !
******* Idée
 - Pour chaque FP, c'est un "faux" FP si
     - REF en hg38 == ALT en T2T
     - et REF en hg38 != REF en T2T
     - et variant homozygote

Comment obtenir les séquences de réferences ?
1. liftover
2. blat sur la séquence autour du variant
3. identifier quelques reads contenant le variant et regarder leur aligneement en hg38

Après discussion avec Alexis: solution 3
******* Algorithme
1. Extraire les coordonnées en T2T des faux positifs *homozygote*
2. Pour chaque faux positif
   1. lister 10 reads contenant le variant
   2. pour chacun de ces reads, récupérer la séquence en T2T et GRCh38 via le nom du read dans le bam
   3. si la séquence en T2T modifiée par le variant est "identique" à celle en GRCh38, alors on ignore ce faux positif

Note: on ignore les reads qui ont changé de chromosome entre les version
******* DONE Résultat préliminaire
CLOSED: [2023-07-23 Sun 14:30]
cf [[file:~/roam/research/bisonex/code/giab/giab-corrected.csv][script julia]]
3498 faux positifs en moins, soit 0.89 sensibilité

julia> tp=15479
julia> fp=5277
julia> tp/(tp+fp)
0.7457602620928888

julia> tp/(tp+(fp-3498))
0.8969173716537258

On est toujours en dessous des 97%
******* HOLD Corriger proprement VCF ou résultats Happy
******* TODO Adapter pour gérer plusieurs variants par read
****** DONE Méthodologie du pangenome
CLOSED: [2023-10-03 Tue 21:28]
Voir biblio[cite:@liao2023]  mais ont aligné sur GRCH38
******* DONE Mail alexis
CLOSED: [2023-10-03 Tue 21:28]
****** DONE Méthodologie T2T
CLOSED: [2023-10-16 Mon 19:42]
Mail alexis
SCHEDULED: <2023-10-04 Wed>

***** TODO Rendre simplement le nombre de vrais positifs
SCHEDULED: <2023-10-23 Mon>
***** KILL Mail Yannis
CLOSED: [2023-07-08 Sat 10:44]
***** DONE Mail GIAB pour version T2T
CLOSED: [2023-07-07 Fri 18:37]
**** TODO HG002 :hg002:T2T:
**** TODO HG003 :hg003:T2T:
**** TODO HG004 :hg004:T2T:
**** DONE Plot : ashkenazim trio :hg38:
CLOSED: [2023-07-30 Sun 16:49] SCHEDULED: <2023-07-30 Sun 15:00>
:LOGBOOK:
CLOCK: [2023-07-30 Sun 16:06]--[2023-07-30 Sun 16:35] =>  0:29
CLOCK: [2023-07-30 Sun 15:39]--[2023-07-30 Sun 15:40] =>  0:01
:END:
/Entered on/ [2023-04-16 Sun 17:29]
Refaire résultats
**** DONE Mail Paul sur les résultat ashkenazim +/- centogene
CLOSED: [2023-08-06 Sun 20:24] SCHEDULED: <2023-08-06 Sun>
**** DONE Relancer comparaison GIAB avec GATK 4.4.0
CLOSED: [2023-08-12 Sat 15:55]
/Entered on/ [2023-08-03 Thu 12:42]
*** TODO Platinum genome :platinum:
https://emea.illumina.com/platinumgenomes.html
**** TODO Tester sur la zone couverte par l'exome centogène
SCHEDULED: <2023-10-23 Mon>
*** DONE Séquencer NA12878 :cento:hg001:
CLOSED: [2023-10-07 Sat 17:59]
Discussion avec Paul : sous-traitant ne nous donnera pas les données, il faut commander l'ADN
**** DONE ADN commandé
CLOSED: [2023-06-30 Fri 22:29]
**** DONE Sauvegarder les données brutes
CLOSED: [2023-07-30 Sun 14:22] SCHEDULED: <2023-07-19 Wed>
K, scality, S
**** KILL Récupérer le fichier de capture
CLOSED: [2023-07-30 Sun 14:25] SCHEDULED: <2023-07-23 Sun>
Candidats donnés dans publication https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8354858/
#+begin_quote
In short, the Nextera Rapid Capture Exome Kit (Illumina, San Diego, CA), the SureSelect Human All Exon kit (Agilent, Santa Clara, CA) or the Twist Human Core Exome was used for enrichment, and a Nextseq500, HiSeq4000, or Novoseq 6000 (Illumina) instrument was used for the actual sequencing, with the average coverage targeted to at least 100× or at least 98% of the target DNA covered 20×.
#+end_quote
Par défaut, on utilisera https://www.twistbioscience.com/products/ngs/alliance-panels#tab-3
ANnonce récente pour nouveau panel Twist : https://www.centogene.com/news-events/news/newsdetails/twist-bioscience-and-centogene-launch-three-panels-to-advance-rare-disease-and-hereditary-cancer-research-and-support-diagnostics
Masi pas de fichier BED
***** DONE Mail centogène
CLOSED: [2023-07-30 Sun 14:22] DEADLINE: <2023-07-23 Sun>
**** DONE Tester Nextera Rapid Capture Exome v1.2 (hg19) :giab:
CLOSED: [2023-08-06 Sun 19:05] SCHEDULED: <2023-08-03 Thu 19:00>
https://support.illumina.com/downloads/nextera-rapid-capture-exome-v1-2-product-files.html
***** DONE Liftover capture
CLOSED: [2023-08-06 Sun 18:30] SCHEDULED: <2023-08-06 Sun>
#+begin_src sh
 nextflow run -profile standard,helios workflows/lift-nextera-capture.nf  -lib lib
#+end_src
Vérification rapide : ok
***** DONE Run
CLOSED: [2023-08-06 Sun 19:05] SCHEDULED: <2023-08-06 Sun>
#+begin_src sh
 nextflow run workflows/compareVCF.nf -profile standard,helios --query=out/2300346867_NA12878-63118093_S260-GRCh38/callVariant/haplotypecaller/2300346867_NA12878-63118093_S260-GRCh38.vcf.gz --outdir=out/2300346867_NA12878-63118093_S260-GRCh38/happy-nextera-lifted/ --compare=happy -lib lib --capture=capture/nexterarapidcapture_exome_targetedregions_v1.2-nochrM_lifted.bed  --id=HG001 --genome=GRCh38
#+end_src
**** DONE Tester Agilent SureSelect All Exon V8 (hg38) :giab:
CLOSED: [2023-07-31 Mon 23:09] SCHEDULED: <2023-07-31 Mon>
https://earray.chem.agilent.com/suredesign/index.htm
"Find design"
"Agilent catalog"
Fichiers:
- Regions.bed: Targeted exon intervals, curated and targeted by Agilent Technologies
- MergedProbes.bed: Merged probes for targeted enrichment of exons described in Regions.bed
- Covered.bed: Merged probes and sequences with 95% homology or above
- Padded.bed: Merged probes and sequences with 95% homology or above extended 50 bp at each side
- AllTracks.bed: Targeted regions and covered tracks

 #+begin_src sh
nextflow run workflows/compareVCF.nf -profile standard,helios --query=out/2300346867_63118093_NA12878-GRCh38/callVariant/haplotypecaller/2300346867_63118093_NA12878-GRCh38.vcf.gz --outdir=out/2300346867_63118093_NA12878-GRCh38/happy/ --compare=happy -lib lib --capture=capture/Agilent_SureSelect_All_Exons_v8_hg38_Regions.bed  --id=HG001 --genome=GRCh38
 #+end_src

| Type  | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision | METRIC.Frac_NA | METRIC.F1_Score | TRUTH.TOTAL.TiTv_ratio | QUERY.TOTAL.TiTv_ratio | TRUTH.TOTAL.het_hom_ratio | QUERY.TOTAL.het_hom_ratio |
| INDEL | ALL    |         423 |      395 |       28 |         915 |      108 |       405 |     4 |    13 |      0.933806 |         0.788235 |       0.442623 |        0.854868 |                        |                        |        1.7012987012987013 |        2.7916666666666665 |
| INDEL | PASS   |         423 |      395 |       28 |         915 |      108 |       405 |     4 |    13 |      0.933806 |         0.788235 |       0.442623 |        0.854868 |                        |                        |        1.7012987012987013 |        2.7916666666666665 |
| SNP   | ALL    |       20984 |    20600 |      384 |       26080 |      780 |      4703 |    62 |    10 |        0.9817 |         0.963512 |        0.18033 |        0.972521 |     3.0499710592321048 |     2.7596541786743516 |          1.58256372367935 |        1.8978207694018234 |
| SNP   | PASS   |       20984 |    20600 |      384 |       26080 |      780 |      4703 |    62 |    10 |        0.9817 |         0.963512 |        0.18033 |        0.972521 |     3.0499710592321048 |     2.7596541786743516 |          1.58256372367935 |        1.8978207694018234 |

**** DONE Test Twist Human core Exome (hg38):giab:
CLOSED: [2023-08-01 Tue 23:16] SCHEDULED: <202 3-08-02 Wed>
https://www.twistbioscience.com/resources/data-files/ngs-human-core-exome-panel-bed-file
#+begin_src
nextflow run workflows/compareVCF.nf -profile standard,helios --query=out/2300346867_63118093_NA12878-GRCh38/callVariant/haplotypecaller/2300346867_63118093_NA12878-GRCh38.vcf.gz --outdir=out/2300346867_63118093_NA12878-GRCh38/happy-twist-exome-core/ --compare=happy -lib lib --capture=capture/Twist_Exome_Core_Covered_Targets_hg38.bed  --id=HG001 --genome=GRCh38 -bg
#+end_src

| Type  | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | FP.al | METRIC.Recall | METRIC.Precision | METRIC.Frac_NA | METRIC.F1_Score | TRUTH.TOTAL.TiTv_ratio | QUERY.TOTAL.TiTv_ratio | TRUTH.TOTAL.het_hom_ratio | QUERY.TOTAL.het_hom_ratio |
| INDEL | ALL    |         328 |      313 |       15 |         722 |       95 |       309 |     4 |    13 |      0.954268 |         0.769976 |       0.427978 |        0.852273 |                        |                        |        1.8584070796460177 |        2.8967391304347827 |
| INDEL | PASS   |         328 |      313 |       15 |         722 |       95 |       309 |     4 |    13 |      0.954268 |         0.769976 |       0.427978 |        0.852273 |                        |                        |        1.8584070796460177 |        2.8967391304347827 |
| SNP   | ALL    |       19198 |    18962 |      236 |       23381 |      684 |      3738 |    48 |    10 |      0.987707 |         0.965178 |       0.159873 |        0.976313 |     3.1034188034188035 |      2.859264147830391 |        1.5669565217391304 |        1.8578767123287672 |
| SNP   | PASS   |       19198 |    18962 |      236 |       23381 |      684 |      3738 |    48 |    10 |      0.987707 |         0.965178 |       0.159873 |        0.976313 |     3.1034188034188035 |      2.859264147830391 |        1.5669565217391304 |        1.8578767123287672 |

**** DONE Test Twist Human core Exome (hg38):giab:
CLOSED: [2023-08-05 Sat 09:25] SCHEDULED: <2023-08-03 Thu 20:00>
#+begin_src sh
ID="2300346867_NA12878-63118093_S260-GRCh38"; nextflow run workflows/compareVCF.nf -profile standard,helios --query=out/${ID}/callVariant/haplotypecaller/${ID}.vcf.gz --outdir=out/${ID}/happy-twist-exome-core/ --compare=happy -lib lib --capture=capture/Twist_Exome_Core_Covered_Targets_hg38.bed  --id=HG001 --genome=GRCh38 -bg
#+end_src
**** DONE Tester Agilen SureSelect All Exon V8 (hg38) GATK-4.4:giab:
CLOSED: [2023-08-05 Sat 09:25] SCHEDULED: <2023-08-03 Thu 20:00>
**** DONE Vérifier l'impact gatk 4.3 - 4.4 : aucun
CLOSED: [2023-08-05 Sat 09:25]
**** DONE Figure comparant les 3 capture :hg001:
CLOSED: [2023-08-06 Sun 20:24] SCHEDULED: <2023-08-06 Sun>
**** DONE Mail Paul sur  les 3 capture :hg001:
CLOSED: [2023-08-06 Sun 20:24] SCHEDULED: <2023-08-06 Sun>
**** KILL Tester si le panel Twist Alliance VCGS Exome suffit
CLOSED: [2023-07-31 Mon 22:31] SCHEDULED: <2023-07-30 Sun>
**** DONE Mail cento pour demande le type de capture
CLOSED: [2023-10-07 Sat 17:59]
/Entered on/ [2023-08-07 Mon 20:40]
Twist exome
*** PROJ Comparer happy et happy-vcfeval :giab:
** TODO Données CHM13 :chm:
https://github.com/lh3/CHM-eval
*** TODO Run ERR1341793
SCHEDULED: <2023-10-22 Sun>
(raw reads ERR1341793_1.fastq.gz and ERR1341793_2.fastq.gz downloaded from https://www.ebi.ac.uk/ena/browser/view/ERR1341793)
*** TODO Run ERR1341796
SCHEDULED: <2023-10-22 Sun>
** TODO Insilico :cento:
*** TODO tous les variants centogène
**** DONE Extraire liste des SNVs
CLOSED: [2023-04-22 Sat 17:32] SCHEDULED: <2023-04-17 Mon>
***** DONE Corriger manquant à la main
CLOSED: [2023-04-22 Sat 17:31]
La sortie est sauvegardé dans git-annex : variants_success.csv
***** DONE Automatique
CLOSED: [2023-04-22 Sat 17:31]
**** DONE Convert SNVs : transcript -> génomique
CLOSED: [2023-06-03 Sat 17:16]
***** DONE Variant_recoder
CLOSED: [2023-04-26 Wed 21:21] SCHEDULED: <2023-04-22 Sat>
****** KILL Haskell: 160 manquant : recoded-success.csv
CLOSED: [2023-04-25 Tue 18:32]
La liste des variants a été générée en Haskel   l et nettoyée à la main.
On générer une liste de variant pour variant_rec            oder et on soumet tout d'un coup.
[[file:~/recherche/bisonex/parsevariants/app/Main.hs][parsevariant]]
#+begin_src haskell
recodeVariant = do
  prepareVariantRecod   er "variant_success.csv" "renamed.csv"
  runVariantRecoder "renamed.csv" "recoded.json"
#+end_src

#+RESULTS:
: <interactive>:4:3-19: error:
:     Variable not in scope: runVariantRecoder :: String -> String -> t
: gh

Problème : 160 n'ont pas pu être lu sur 820, probablement à cause du numéro mineur de transcrit
La sortie est sauvegardé dans git-annex : variants-recoded-raw.json.

****** KILL Julia
CLOSED: [2023-04-25 Tue 18:32]
On regénère la liste de variant et on passe à Julia pour préparer l'appel en parallèle à variant recoder
[[file:~/recherche/bisonex/parsevariants/variantRecoder.jl][variantRecoder.jl]]
#+begin_src julia
setupVariantRecoder(unique(init), n)
#+end_src

Puis
#+begin_src sh
parallel -a parallel-recoder.sh --jobs 10
#+end_src
On récupère les résultats
#+begin_src julia
(fails, success) = mergeVariantRecoder(n)
CSV.write(fSuccess, success)
CSV.write(fFailures, fails)
#+end_src

Certains variants ne sont pas trouvé, donc on prépare un nouveau job en enlevant les versionrs mineures des transcrits
#+begin_src julia

# Cleanup json and txt
if isfile(fSuccess) && isfile(fFailures)
    foreach(rm, variantRecoderInput())
    foreach(rm, variantRecoderOutput())
end
redoFails(fFailures)
#+end_src
Puis
#+begin_src sh
parallel -a parallel-recoder.sh --jobs 3
#+end_src
Il manque encore 70 transcrits
***** DONE Julia avec mobidetails: recode-failures-mobidetails.csv
CLOSED: [2023-04-25 Tue 18:58]
Nouvelle stratégie : on essaie une fois variant recoder.
Pour tous les échecs, on utilise mobidetails (~170).
Si l'ID n'est pas trouvé, on incrémente le numéro de version 2 fois

***** DONE Reste une dizaine à corriger à la main
CLOSED: [2023-04-26 Wed 21:21]
- [X] certains transcrits ont juste été supprimé
- [X] Erreur de parsing, manque souvent un -
#+begin_src julia
lastTryMobidetails("recoded-failures-mobidetails.csv")
#+end_src
***** DONE Fusionner données
CLOSED: [2023-04-26 Wed 22:35]
#+begin_src julia
function mergeAllGenomic()
    dNew = mergeAll("recoded-success.csv",
                    "recoded-failures-mobidetails.csv",
                    "recoded-failures-mobidetails-redo.csv")

    dInit = @chain DataFrame(CSV.File("variant_success.csv")) begin
        @transform :transcript = :transcript .* ":" .* :coding .* :codingPos .* :codingChange
        @select :file :transcript :classification :zygosity
        @rename :classificationCento = :classification
    end
    dTmp = outerjoin(dInit, dNew, on = :transcript)
    CSV.write("variant_genomic.csv", dTmp)
end

fSuccess = "recoded-success.csv"
fFailures = "recoded-failures.csv"

# variantRecoder(fSuccess, fFailures)
# mobidetailsOnFailures(fFailures)
# lastTryMobidetails("recoded-failures-mobidetails.csv")
mergeAllGenomic()


#+end_src
***** DONE Formatter donner pour simuscop
CLOSED: [2023-04-28 Fri 11:55] SCHEDULED: <2023-04-26 Wed>
**** KILL Extraire liste des CNVs
CLOSED: [2023-08-12 Sat 15:54]

SCHEDULED: <2023-04-17 Mon>
**** KILL Simuscop :simuscop:
CLOSED: [2023-08-12 Sat 15:54]
***** DONE Entrainer le modèle sur 63003856/
CLOSED: [2023-04-29 Sat 19:56]
Relancer le modèle pour être sûr
***** DONE Générer fastq avec simuscop (del et ins seulement) 20x
CLOSED: [2023-04-28 Fri 23:35] SCHEDULED: <2023-04-22 Sat>
****** DONE Génerer un profile avec bed de centogène
CLOSED: [2023-04-28 Fri 11:54] SCHEDULED: <2023-04-22 Sat>
NA12878 mais à refaire avec un vrai séquencage
Voir [[*Centogène][Bed Centogène]] pour choix
****** DONE Générer les données en 20x
CLOSED: [2023-04-28 Fri 11:54] SCHEDULED: <2023-04-22 Sat>
capture de cento
****** DONE Regénérer en supprimant les doublons
CLOSED: [2023-04-28 Fri 17:28]
***** DONE Quelle couverture ?
CLOSED: [2023-04-29 Sat 18:26]
ex sur chr11:16,014,966 où on a 11 reads dans la simulation contre 200 !
****** 200 est la plus proche
#+attr_html: :width 500px
[[./simuscop-200-chr1-1.png]]

#+attr_html: :width 500px
[[./simuscop-200-chr1-2.png]]

****** DONE 20x
CLOSED: [2023-04-29 Sat 15:38]
****** DONE 50x
CLOSED: [2023-04-29 Sat 15:38]
****** DONE 100x
CLOSED: [2023-04-29 Sat 15:39]
****** DONE 200x
CLOSED: [2023-04-29 Sat 15:39]
***** DONE Reads mal centrés sur des petits exons seuls
CLOSED: [2023-04-29 Sat 19:56] SCHEDULED: <2023-04-29 Sat>
Capture ok : [[https://genome-euro.ucsc.edu/cgi-bin/hgTracks?db=hg38&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr1%3A153817168%2D153817824&hgsid=296556270_F4fkENLPXHXidi2oALXls2jxNH9l][UCSC]] (track noire)
Mais mauvaise répartitiopn
#+attr_html: :width 800px
[[./simuscop-error.png]]

À tester
- Problème de profile ?
  - mauvais patient ?
  - mauvaise génération ? -> comparer avec ceux donnés sur github
- nom des chromosomes ?
****** DONE [#A] Tester sur exon 6 GATAD2B pour NC_000001.11:g.153817496A>T
CLOSED: [2023-04-29 Sat 19:56] SCHEDULED: <2023-04-29 Sat>
******* DONE Configuration + Profile 63003856.profile: idem, mal centré
CLOSED: [2023-04-29 Sat 19:18]
Téléchargement des données
#+begin_src sh :dir ~/code/bisonex/test-simuscop
scp meso:/Work/Projects/bisonex/data/genome/GRCh38.p14/genomeRef.fna .
scp meso:Work/Projects/bisonex/data/simuscop/*.profile .
scp -r meso:/Work/Projects/bisonex/data/genome/GRCh38.p13/bwa .
#+end_src

On récupère l'exon (NB: org-mode ne lance pas le code...)
#+begin_src julia
using CSV,DataFramesMeta
d = CSV.read("VCGS_Exome_Covered_Targets_hg38_40.1MB_renamed.bed", header=false, delim="\t", DataFrame)
@subset d :Column1 .== "NC_000001.11" :Column2 .<= 153817496 :Column3 .>= 153817496
#+end_src

NC_000001.11  153817371  153817542

Génération du bed
#+begin_src sh :dir ~/code/bisonex/test-simuscop
echo -e "NC_000001.11\t153817371\t153817542" > gatad2b-exon6.bed
#+end_src

#+RESULTS:

Génération d'un variant
#+begin_src sh :dir ~/code/bisonex/test-simuscop
echo -e "s\tsingle\tNC_000001.11\t153817496\tA\tT\thet"> variant.txt
#+end_src

#+RESULTS:

Génération du fichier de config
#+begin_src sh :dir ~/code/bisonex/test-simuscop
cat > config_wes.txt << EOL
ref = genomeRef.fna
profile = ./63003856.profile
variation = ./variant.txt
target = ./gatad2b-exon6.bed
layout = PE
threads = 1
name = single
output = test-gatad2b
coverage = 20
EOL
#+end_src

#+RESULTS:

On démarre la simulation
#+begin_src sh :dir ~/code/bisonex/test-simuscop
simuReads config_wes.txt
#+end_src

#+RESULTS:

Alignement
#+begin_src sh :dir ~/code/bisonex/test-simuscop
bwa mem -R '@RG\tID:sample\tSM:sample\tPL:ILLUMINA\tPM:Miseq\tCN:lol\tLB:definition_to_add' bwa/genomeRef test-gatad2b/single_1.fq  test-gatad2b/single_2.fq | samtools sort  -o single.bam
#+end_src

#+RESULTS:
******* DONE Profile github  HiSeq2000
CLOSED: [2023-04-29 Sat 19:56]
#+begin_src sh :dir ~/code/bisonex/test-simuscop :result file
wget https://raw.githubusercontent.com/qasimyu/simuscop/master/testData/Illumina_HiSeq2000.profile
#+end_src

#+RESULTS:

#+begin_src sh :dir ~/code/bisonex/test-simuscop
cat > config_wes.txt << EOL
ref = genomeRef.fna
profile = ./Illumina_HiSeq2000.profile
variation = ./variant.txt
target = ./gatad2b-exon6.bed
layout = PE
threads = 1
name = single
output = test-gatad2b-hiseq2000
coverage = 20
EOL
simuReads config_wes.txt
bwa mem -R '@RG\tID:sample\tSM:sample\tPL:ILLUMINA\tPM:Miseq\tCN:lol\tLB:definition_to_add' bwa/genomeRef test-gatad2b-hiseq2000/single_1.fq  test-gatad2b-hiseq2000/single_2.fq | samtools sort  -o single-hiseq2000.bam
samtools index single-hiseq2000.bam
#+end_src

#+RESULTS:
******* KILL Tester exemple sur github
CLOSED: [2023-04-29 Sat 19:56]
#+begin_src sh
git clone https://github.com/qasimyu/simuscop/
cd simuscop
simuReads configFiles/config_test_wes.txt
#+end_src
******* KILL Centrer la fenêtre sur les zones de capture
CLOSED: [2023-04-30 Sun 13:28] SCHEDULED: <2023-04-29 Sat>
1000bp par défaut, ce qui est plus grand que les zones de captures...
Changer fragzip ne fonctionne pas

Si on rajoute un offset sur l'exon: 200bp, est encore plus allongé
NC_000001.11 153817371 153817542 ->
NC_000001.11 153817171 153817742

Si on désactive les target ?
Regarder les target sur le chromosome 1
#+begin_src sh :dir ~/code/bisonex/test-simuscop :results silent
scp meso:/Work/Projects/bisonex/data/simuscop/VCGS_Exome_Covered_Targets_hg38_40.1MB_renamed.bed .
#+end_src

#+begin_src sh :dir ~/code/bisonex/test-simuscop :results silent
head -n 100 VCGS_Exome_Covered_Targets_hg38_40.1MB_renamed.bed > 100exons.bed
echo -e "s\tsingle\tNC_000001.11\t153817496\tA\tT\thet"> variant.txt
cat > config_wes.txt << EOL
ref = genomeRef.fna
profile = ./63003856.profile
variation = ./variant.txt
layout = PE
threads = 4
target = 100exons.bed
name = single
output = test-gatad2b
coverage = 200
EOL
./simuscop/bin/simuReads config_wes.txt
bwa mem bwa/genomeRef test-gatad2b/single_1.fq  test-gatad2b/single_2.fq | samtools sort  -o single.bam
samtools index single.bam
#+end_src

***** KILL Vérifier tous les variants sont retrouvés en 200x: hg38
CLOSED: [2023-06-12 Mon 23:25]
****** DONE Après alignement
CLOSED: [2023-04-29 Sat 18:27] SCHEDULED: <2023-04-28 Fri>
******* DONE SNV: avec doublons
CLOSED: [2023-04-28 Fri 18:12]
On utilise [[file:~/recherche/bisonex/simuscop/checkBam.jl][checkBam.jl]]
#+begin_src julia
d = prepareVariant("../parsevariants/variant_genomic.csv")

root = "/home/alex/code/bisonex/simuscop-cento/cento"
bam = root * "/preprocessing/applybqsr/cento.bam"
bai = root * "/preprocessing/recalibrated/cento.bam.bai"

snv = getSNV(d, bam, bai)
#+end_src
Nombreux faux homozygouteS
Vérification avec checkFalseHemizygous(snv) : nombreux doublons dans le fichier pour simuscop...
******* DONE SNV sans doublons
CLOSED: [2023-04-29 Sat 18:27]
******** DONE 18 faux homozygote mais avec peu de reads
CLOSED: [2023-04-29 Sat 18:27]

julia> @subset snv :refCount .== 0 :altCount .> 0 :zygosity .== "heterozygous"

18×10 DataFrame
 Row │ chrom         pos        variant         variantType  zygosity      ref        alt        refCount  altCount  readsCount
     │ SubStrin…?    Int64      SubStrin…?      String?      String15      SubStrin…  SubStrin…  Int64     Int64     Int64
─────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ NC_000022.11   42213078  g.42213078T>G   snv          heterozygous  T          G                 0         1           1
   2 │ NC_000012.12  101680427  g.101680427C>A  snv          heterozygous  C          A                 0         3           3
   3 │ NC_000014.9   105385684  g.105385684G>C  snv          heterozygous  G          C                 0         4           4
   4 │ NC_000011.10  125978299  g.125978299C>T  snv          heterozygous  C          T                 0         3           3
   5 │ NC_000023.11   77998618  g.77998618C>T   snv          heterozygous  C          T                 0         2           2
   6 │ NC_000015.10   66703292  g.66703292C>T   snv          heterozygous  C          T                 0         3           3
   7 │ NC_000010.11   87961118  g.87961118G>A   snv          heterozygous  G          A                 0         3           3
   8 │ NC_000012.12  112477719  g.112477719A>G  snv          heterozygous  A          G                 0         2           2
   9 │ NC_000020.11    6778406  g.6778406C>T    snv          heterozygous  C          T                 0         3           3
  10 │ NC_000023.11   68192943  g.68192943G>A   snv          heterozygous  G          A                 0         2           2
  11 │ NC_000004.12     987858  g.987858C>T     snv          heterozygous  C          T                 0         3           4
  12 │ NC_000015.10   66435145  g.66435145G>A   snv          heterozygous  G          A                 0         1           2
  13 │ NC_000002.12   47809595  g.47809595C>T   snv          heterozygous  C          T                 0         2           2
  14 │ NC_000003.12  136477305  g.136477305C>G  snv          heterozygous  C          G                 0         4           4
  15 │ NC_000005.10  157285458  g.157285458C>T  snv          heterozygous  C          T                 0         3           3
  16 │ NC_000012.12   23604413  g.23604413T>G   snv          heterozygous  T          G                 0         5           5
  17 │ NC_000019.10   52219703  g.52219703C>T   snv          heterozygous  C          T                 0         1           1
  18 │ NC_000016.10   88856757  g.88856757C>T   snv          heterozygous  C          T                 0         8           8

******** DONE 8 non retrouvé => probablement hors de la zjone de capture
CLOSED: [2023-04-28 Fri 19:49]
julia> @subset snv :refCount .== 0 :altCount .== 0
8×10 DataFrame
 Row │ chrom         pos        variant         variantType  zygosity      ref        alt        refCount  altCount  readsCount
     │ SubStrin…?    Int64      SubStrin…?      String?      String15      SubStrin…  SubStrin…  Int64     Int64     Int64
─────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ NC_000015.10   74343027  g.74343027C>T   snv          heterozygous  C          T                 0         0           0
   2 │ NC_000011.10   20638345  g.20638345A>G   snv          heterozygous  A          G                 0         0           0
   3 │ NC_000004.12  139370252  g.139370252C>T  snv          heterozygous  C          T                 0         0           2
   4 │ NC_000017.11   61966475  g.61966475G>T   snv          heterozygous  G          T                 0         0           0
   5 │ NC_000019.10   54144058  g.54144058G>A   snv          heterozygous  G          A                 0         0           0
   6 │ NC_000023.11   77635947  g.77635947A>G   snv          hemizygous    A          G                 0         0           0
   7 │ NC_000005.10    1258495  g.1258495G>A    snv          heterozygous  G          A                 0         0           0
   8 │ NC_000012.12    2449086  g.2449086C>G    snv          heterozygous  C          G                 0         0           0

****** KILL Après haplotypecaller
CLOSED: [2023-06-12 Mon 23:24]
******* KILL 20x
CLOSED: [2023-04-29 Sat 15:39]
Manque 183 sur 766
[[file:~/recherche/bisonex/simuscop/checkVCF.jl][checkVCF.jl]]
#+begin_src julia
@subset leftjoin(d2, dHaplo2, on=:genomic) ismissing.(:Column1)
#+end_src
Problème de profondeur ?
Ex: chr13 nombre de 101081606
NC_000011.10   16014966  g.16014966G>A
1 read sur 11 pour allèle alternative
Sur le patient de référence, 202 reads!

Celui-ci n'est pas le fichier de capture (ni dans le bam !)
ex: NC_000015.10   74343027  g.74343027C>T

Pour les autres, on devrait les retrouver...

Vérifier le nombre de reads sur 63003856
Vérifier la paramétrisation du modèle également
******* DONE [#B] 200x
CLOSED: [2023-05-18 Thu 11:04] SCHEDULED: <2023-04-30 Sun>
120 manquants (99 sans doublon)!

On vérifie dans IGV (vcf + bam après alignement) :
******** snv NC_000015.10   74343027
- rien d'appelé
- pas une région répétée
- base quality (voir [[*Phred score][Phred score]] ) à 37 donc ok
- variant retrouvé à 26/42
- Bam après aplybqsr: base qualità 35 donc ok

chr15 également à 89318565, variant retrouvé à 25/33 avec basequal de 37


Sans oublier de charger les instructions avx
#+begin_src sh
module load gcc@11.3.0/gcc-12.1.0
#+end_src

On coupe le .bam par chromosome pour débugger (sur le mesocentre)
#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/simuscop-cento-200x/cento/testing :results silent
ln -s ../preprocessing/applybqsr/cento.bam .
ln -s ../preprocessing/recalibrated/cento.bam.bai .
ln -s /Work/Projects/bisonex/data/dbSNP/GRCh38.p13/dbSNP.gz .
ln -s /Work/Projects/bisonex/data/dbSNP/GRCh38.p13/dbSNP.gz.tbi .
ln -s /Work/Projects/bisonex/data/genome/GRCh38.p13/genomeRef.dict .
ln -s /Work/Projects/bisonex/data/genome/GRCh38.p13/genomeRef.fna .
ln -s /Work/Projects/bisonex/data/genome/GRCh38.p13/genomeRef.fna.fai .
#+end_src

On doit lancer à la main (org-mode ne connait pas le chemin de samtools)
samtools view -b cento.bam NC_000015.10 > cento_chr15.bam
samtools index cento_chr15.bam

Puis on se restreint au chronmosome 15
samtools faidx genomeRef.fna NC_000015.10 > genomeRef_chr15.fa
samtools faidx genomeRef_chr15.fa
gatk CreateSequenceDictionary -R genomeRef_chr15.fa -O genomeRef_chr15.dict

On restreint au chromosome 15 avec l'option -L (dure = 1min)
gatk --java-options "-Xmx3072M" HaplotypeCaller --input cento_chr15.bam \
    --output test.vcf.gz --reference genomeRef.fna --dbsnp dbSNP.gz --tmp-dir . --max-mnp-distance 2 -L NC_000015.10

******** DONE Tutorial haplotycaller
CLOSED: [2023-05-01 Mon 19:58]
Procédure : https://gatk.broadinstitute.org/hc/en-us/articles/360043491652-When-HaplotypeCaller-and-Mutect2-do-not-call-an-expected-variant
********* DONE Supprimer --max-mnp-distance = 2: idem
CLOSED: [2023-04-30 Sun 15:42]
********* DONE --debug &> run.log : Non appelé...
CLOSED: [2023-04-30 Sun 15:52]
********* DONE --linked-de-bruijn-graph: idem
CLOSED: [2023-04-30 Sun 15:55]
********* DONE --recover-all-dangling-branches
CLOSED: [2023-04-30 Sun 16:01]
********* DONE --min-pruning 0 : plus mais pas celui là
CLOSED: [2023-04-30 Sun 15:59]
********* DONE --bam-output
CLOSED: [2023-04-30 Sun 16:50]
********** DONE : rien !
CLOSED: [2023-04-30 Sun 16:08]
********** DONE + --recover-all-dangling-branches : rien !
CLOSED: [2023-04-30 Sun 16:08]
********* DONE Données filtrées ? apparement non
CLOSED: [2023-04-30 Sun 16:41]
183122 read(s) filtered by: MappingQualityReadFilter
3674 read(s) filtered by: NotDuplicateReadFilter
********** DONE --disable-read-filter MappingQualityReadFilter: idem
CLOSED: [2023-04-30 Sun 16:34]
On a bien  - 0 read(s) filtered by: MappingQualityAvailableReadFilter
********** DONE --disable-read-filter NotDuplicateReadFilter: idem
CLOSED: [2023-04-30 Sun 16:40]

********* DONE Essayer freebayes : idem
CLOSED: [2023-04-30 Sun 16:22]
freebayes -f genomeRef.fna -r NC_000015.10 cento_chr15.bam > freebayes-test-chr15.vcf
********* DONE Avec toutes les options : idem
--linked-de-bruijn-graph --recover-all-dangling-branches --min-pruning 0 --bam-output debug.bam
CLOSED: [2023-04-30 Sun 16:50]
********* DONE Vérifier qu'on regarde le même bam : oui
CLOSED: [2023-04-30 Sun 16:50]
********* DONE Désactiver dbSNP : idem
CLOSED: [2023-04-30 Sun 16:52]
********* DONE Changer kmer size : idem
CLOSED: [2023-04-30 Sun 16:56]
par exemple[[https://gatk.broadinstitute.org/hc/en-us/community/posts/360075653152-REAL-Variant-not-called-by-HaplotypeCaller][forum gatk]] --kmer-size 18 --kmer-size 22
********* DONE --adaptive-pruning true
CLOSED: [2023-05-01 Mon 19:57]

******** DONE Mapping quality : est à 0 !!!!
CLOSED: [2023-05-01 Mon 19:58]

******* KILL Comparer VCF avec vcfeval :haplotypecaller:
CLOSED: [2023-06-12 Mon 23:24]
On prépare les données en julia
#+begin_src ~/recherche/bisonex/simuscop
julia --project=. toVCF.jl
#+end_src
Puis on export sur le mésocentre
#+begin_src
scp variants_for_vcfeval.tsv.gz* meso:cento_variants/
#+end_src
#+begin_src
z bis
cd simuscop-200x
rtg vcfeval -b ~/cento_variants/variants_for_vcfeval.tsv.gz -c cento/variantCalling/haplotypecaller/cento.vcf.gz -o compare-haplotypecaller -t /Work/Groups/bisonex/data/giab/GRCh38/genomeRef.sdf
#+end_src
Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
   82.000                540            540         60         45     0.9000       0.9231     0.9114
     None                546            546        329         39     0.6240       0.9333     0.7479

******* KILL Comparer avec hap.py :haplotypecaller:
CLOSED: [2023-06-12 Mon 23:24]
 NXF_OPTS=-D"user.name=${USER}" nextflow run workflows/checkInserted.nf -profile standard,helios --outdir=compare-simuscop-200x  --query=out/simuscop-cento-200x/cento/callVariant/haplotypecaller/cento.vcf.gz --truth=cento_variants/variants_for_vcfeval.tsv.gz --id=simuscop-200x-check
******* DONE Méthode naïve 549/585
CLOSED: [2023-05-04 Thu 21:57]
Haplotypecaller: Nb reference SNV 692 vs found 585
Variant calling, filter technical: reference SNV 692 vs found 521

****** KILL Avant annotation
CLOSED: [2023-06-12 Mon 23:25] SCHEDULED: <2023-04-28 Fri>
#+begin_src
cd cento/variantCalling
bgzip filter-technical.vcf
tabix -p vcf filter-technical.vcf.gz -f
#+end_src
Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
   12.000                519            519         55         66     0.9042       0.8872     0.8956
     None                519            519         55         66     0.9042       0.8872     0.8956
******* DONE Méthode naïve 521/585
CLOSED: [2023-05-04 Thu 21:57]
Haplotypecaller: Nb reference SNV 692 vs found 585
Variant calling, filter technical: reference SNV 692 vs found 521
******* KILL Comparer avec hap.py
CLOSED: [2023-06-12 Mon 23:24]
****** KILL Après filtre annotation
CLOSED: [2023-06-12 Mon 23:25]
******* DONE Méthode naïve : 493/585
CLOSED: [2023-05-04 Thu 22:09]
******* KILL Comparer avec hap.py
CLOSED: [2023-06-12 Mon 23:25]
******* KILL VCf eval
CLOSED: [2023-06-12 Mon 23:25]
 cd cento/annotation/
 bgzip postvep-filter.vcf
 tabix postvep-filter.vcf.gz
 cd ../..
 rtg vcfeval -b ~/cento_variants/variants_for_vcfeval.tsv.gz -c cento/annotation/postvep-filter.vcf.gz  -o compare-vepfilter -t /Work/Groups/bisonex/data/giab/GRCh38/genomeRef.sdf
 Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
   12.000                491            491         50         94     0.9076       0.8393     0.8721
     None                491            491         50         94     0.9076       0.8393     0.8721

***** KILL Vérifier tous les variants sont retrouvés en 200x: hg38 :T2T:
CLOSED: [2023-08-12 Sat 15:54]
****** DONE Après alignement
CLOSED: [2023-04-29 Sat 18:27] SCHEDULED: <2023-04-28 Fri>
******* DONE SNV: avec doublons
CLOSED: [2023-04-28 Fri 18:12]
On utilise [[file:~/recherche/bisonex/simuscop/checkBam.jl][checkBam.jl]]
#+begin_src julia
d = prepareVariant("../parsevariants/variant_genomic.csv")

root = "/home/alex/code/bisonex/simuscop-cento/cento"
bam = root * "/preprocessing/applybqsr/cento.bam"
bai = root * "/preprocessing/recalibrated/cento.bam.bai"

snv = getSNV(d, bam, bai)
#+end_src
Nombreux faux homozygouteS
Vérification avec checkFalseHemizygous(snv) : nombreux doublons dans le fichier pour simuscop...
******* DONE SNV sans doublons
CLOSED: [2023-04-29 Sat 18:27]
******** DONE 18 faux homozygote mais avec peu de reads
CLOSED: [2023-04-29 Sat 18:27]

julia> @subset snv :refCount .== 0 :altCount .> 0 :zygosity .== "heterozygous"

18×10 DataFrame
 Row │ chrom         pos        variant         variantType  zygosity      ref        alt        refCount  altCount  readsCount
     │ SubStrin…?    Int64      SubStrin…?      String?      String15      SubStrin…  SubStrin…  Int64     Int64     Int64
─────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ NC_000022.11   42213078  g.42213078T>G   snv          heterozygous  T          G                 0         1           1
   2 │ NC_000012.12  101680427  g.101680427C>A  snv          heterozygous  C          A                 0         3           3
   3 │ NC_000014.9   105385684  g.105385684G>C  snv          heterozygous  G          C                 0         4           4
   4 │ NC_000011.10  125978299  g.125978299C>T  snv          heterozygous  C          T                 0         3           3
   5 │ NC_000023.11   77998618  g.77998618C>T   snv          heterozygous  C          T                 0         2           2
   6 │ NC_000015.10   66703292  g.66703292C>T   snv          heterozygous  C          T                 0         3           3
   7 │ NC_000010.11   87961118  g.87961118G>A   snv          heterozygous  G          A                 0         3           3
   8 │ NC_000012.12  112477719  g.112477719A>G  snv          heterozygous  A          G                 0         2           2
   9 │ NC_000020.11    6778406  g.6778406C>T    snv          heterozygous  C          T                 0         3           3
  10 │ NC_000023.11   68192943  g.68192943G>A   snv          heterozygous  G          A                 0         2           2
  11 │ NC_000004.12     987858  g.987858C>T     snv          heterozygous  C          T                 0         3           4
  12 │ NC_000015.10   66435145  g.66435145G>A   snv          heterozygous  G          A                 0         1           2
  13 │ NC_000002.12   47809595  g.47809595C>T   snv          heterozygous  C          T                 0         2           2
  14 │ NC_000003.12  136477305  g.136477305C>G  snv          heterozygous  C          G                 0         4           4
  15 │ NC_000005.10  157285458  g.157285458C>T  snv          heterozygous  C          T                 0         3           3
  16 │ NC_000012.12   23604413  g.23604413T>G   snv          heterozygous  T          G                 0         5           5
  17 │ NC_000019.10   52219703  g.52219703C>T   snv          heterozygous  C          T                 0         1           1
  18 │ NC_000016.10   88856757  g.88856757C>T   snv          heterozygous  C          T                 0         8           8

******** DONE 8 non retrouvé => probablement hors de la zjone de capture
CLOSED: [2023-04-28 Fri 19:49]
julia> @subset snv :refCount .== 0 :altCount .== 0
8×10 DataFrame
 Row │ chrom         pos        variant         variantType  zygosity      ref        alt        refCount  altCount  readsCount
     │ SubStrin…?    Int64      SubStrin…?      String?      String15      SubStrin…  SubStrin…  Int64     Int64     Int64
─────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ NC_000015.10   74343027  g.74343027C>T   snv          heterozygous  C          T                 0         0           0
   2 │ NC_000011.10   20638345  g.20638345A>G   snv          heterozygous  A          G                 0         0           0
   3 │ NC_000004.12  139370252  g.139370252C>T  snv          heterozygous  C          T                 0         0           2
   4 │ NC_000017.11   61966475  g.61966475G>T   snv          heterozygous  G          T                 0         0           0
   5 │ NC_000019.10   54144058  g.54144058G>A   snv          heterozygous  G          A                 0         0           0
   6 │ NC_000023.11   77635947  g.77635947A>G   snv          hemizygous    A          G                 0         0           0
   7 │ NC_000005.10    1258495  g.1258495G>A    snv          heterozygous  G          A                 0         0           0
   8 │ NC_000012.12    2449086  g.2449086C>G    snv          heterozygous  C          G                 0         0           0

****** KILL Après haplotypecaller
CLOSED: [2023-06-12 Mon 23:24]
******* KILL 20x
CLOSED: [2023-04-29 Sat 15:39]
Manque 183 sur 766
[[file:~/recherche/bisonex/simuscop/checkVCF.jl][checkVCF.jl]]
#+begin_src julia
@subset leftjoin(d2, dHaplo2, on=:genomic) ismissing.(:Column1)
#+end_src
Problème de profondeur ?
Ex: chr13 nombre de 101081606
NC_000011.10   16014966  g.16014966G>A
1 read sur 11 pour allèle alternative
Sur le patient de référence, 202 reads!

Celui-ci n'est pas le fichier de capture (ni dans le bam !)
ex: NC_000015.10   74343027  g.74343027C>T

Pour les autres, on devrait les retrouver...

Vérifier le nombre de reads sur 63003856
Vérifier la paramétrisation du modèle également
******* DONE [#B] 200x
CLOSED: [2023-05-18 Thu 11:04] SCHEDULED: <2023-04-30 Sun>
120 manquants (99 sans doublon)!

On vérifie dans IGV (vcf + bam après alignement) :
******** snv NC_000015.10   74343027
- rien d'appelé
- pas une région répétée
- base quality (voir [[*Phred score][Phred score]] ) à 37 donc ok
- variant retrouvé à 26/42
- Bam après aplybqsr: base qualità 35 donc ok

chr15 également à 89318565, variant retrouvé à 25/33 avec basequal de 37


Sans oublier de charger les instructions avx
#+begin_src sh
module load gcc@11.3.0/gcc-12.1.0
#+end_src

On coupe le .bam par chromosome pour débugger (sur le mesocentre)
#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/simuscop-cento-200x/cento/testing :results silent
ln -s ../preprocessing/applybqsr/cento.bam .
ln -s ../preprocessing/recalibrated/cento.bam.bai .
ln -s /Work/Projects/bisonex/data/dbSNP/GRCh38.p13/dbSNP.gz .
ln -s /Work/Projects/bisonex/data/dbSNP/GRCh38.p13/dbSNP.gz.tbi .
ln -s /Work/Projects/bisonex/data/genome/GRCh38.p13/genomeRef.dict .
ln -s /Work/Projects/bisonex/data/genome/GRCh38.p13/genomeRef.fna .
ln -s /Work/Projects/bisonex/data/genome/GRCh38.p13/genomeRef.fna.fai .
#+end_src

On doit lancer à la main (org-mode ne connait pas le chemin de samtools)
samtools view -b cento.bam NC_000015.10 > cento_chr15.bam
samtools index cento_chr15.bam

Puis on se restreint au chronmosome 15
samtools faidx genomeRef.fna NC_000015.10 > genomeRef_chr15.fa
samtools faidx genomeRef_chr15.fa
gatk CreateSequenceDictionary -R genomeRef_chr15.fa -O genomeRef_chr15.dict

On restreint au chromosome 15 avec l'option -L (dure = 1min)
gatk --java-options "-Xmx3072M" HaplotypeCaller --input cento_chr15.bam \
    --output test.vcf.gz --reference genomeRef.fna --dbsnp dbSNP.gz --tmp-dir . --max-mnp-distance 2 -L NC_000015.10

******** DONE Tutorial haplotycaller
CLOSED: [2023-05-01 Mon 19:58]
Procédure : https://gatk.broadinstitute.org/hc/en-us/articles/360043491652-When-HaplotypeCaller-and-Mutect2-do-not-call-an-expected-variant
********* DONE Supprimer --max-mnp-distance = 2: idem
CLOSED: [2023-04-30 Sun 15:42]
********* DONE --debug &> run.log : Non appelé...
CLOSED: [2023-04-30 Sun 15:52]
********* DONE --linked-de-bruijn-graph: idem
CLOSED: [2023-04-30 Sun 15:55]
********* DONE --recover-all-dangling-branches
CLOSED: [2023-04-30 Sun 16:01]
********* DONE --min-pruning 0 : plus mais pas celui là
CLOSED: [2023-04-30 Sun 15:59]
********* DONE --bam-output
CLOSED: [2023-04-30 Sun 16:50]
********** DONE : rien !
CLOSED: [2023-04-30 Sun 16:08]
********** DONE + --recover-all-dangling-branches : rien !
CLOSED: [2023-04-30 Sun 16:08]
********* DONE Données filtrées ? apparement non
CLOSED: [2023-04-30 Sun 16:41]
183122 read(s) filtered by: MappingQualityReadFilter
3674 read(s) filtered by: NotDuplicateReadFilter
********** DONE --disable-read-filter MappingQualityReadFilter: idem
CLOSED: [2023-04-30 Sun 16:34]
On a bien  - 0 read(s) filtered by: MappingQualityAvailableReadFilter
********** DONE --disable-read-filter NotDuplicateReadFilter: idem
CLOSED: [2023-04-30 Sun 16:40]

********* DONE Essayer freebayes : idem
CLOSED: [2023-04-30 Sun 16:22]
freebayes -f genomeRef.fna -r NC_000015.10 cento_chr15.bam > freebayes-test-chr15.vcf
********* DONE Avec toutes les options : idem
--linked-de-bruijn-graph --recover-all-dangling-branches --min-pruning 0 --bam-output debug.bam
CLOSED: [2023-04-30 Sun 16:50]
********* DONE Vérifier qu'on regarde le même bam : oui
CLOSED: [2023-04-30 Sun 16:50]
********* DONE Désactiver dbSNP : idem
CLOSED: [2023-04-30 Sun 16:52]
********* DONE Changer kmer size : idem
CLOSED: [2023-04-30 Sun 16:56]
par exemple[[https://gatk.broadinstitute.org/hc/en-us/community/posts/360075653152-REAL-Variant-not-called-by-HaplotypeCaller][forum gatk]] --kmer-size 18 --kmer-size 22
********* DONE --adaptive-pruning true
CLOSED: [2023-05-01 Mon 19:57]

******** DONE Mapping quality : est à 0 !!!!
CLOSED: [2023-05-01 Mon 19:58]

******* KILL Comparer VCF avec vcfeval :haplotypecaller:
CLOSED: [2023-06-12 Mon 23:24]
On prépare les données en julia
#+begin_src ~/recherche/bisonex/simuscop
julia --project=. toVCF.jl
#+end_src
Puis on export sur le mésocentre
#+begin_src
scp variants_for_vcfeval.tsv.gz* meso:cento_variants/
#+end_src
#+begin_src
z bis
cd simuscop-200x
rtg vcfeval -b ~/cento_variants/variants_for_vcfeval.tsv.gz -c cento/variantCalling/haplotypecaller/cento.vcf.gz -o compare-haplotypecaller -t /Work/Groups/bisonex/data/giab/GRCh38/genomeRef.sdf
#+end_src
Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
   82.000                540            540         60         45     0.9000       0.9231     0.9114
     None                546            546        329         39     0.6240       0.9333     0.7479

******* KILL Comparer avec hap.py :haplotypecaller:
CLOSED: [2023-06-12 Mon 23:24]
 NXF_OPTS=-D"user.name=${USER}" nextflow run workflows/checkInserted.nf -profile standard,helios --outdir=compare-simuscop-200x  --query=out/simuscop-cento-200x/cento/callVariant/haplotypecaller/cento.vcf.gz --truth=cento_variants/variants_for_vcfeval.tsv.gz --id=simuscop-200x-check
******* DONE Méthode naïve 549/585
CLOSED: [2023-05-04 Thu 21:57]
Haplotypecaller: Nb reference SNV 692 vs found 585
Variant calling, filter technical: reference SNV 692 vs found 521

****** KILL Avant annotation
CLOSED: [2023-06-12 Mon 23:25] SCHEDULED: <2023-04-28 Fri>
#+begin_src
cd cento/variantCalling
bgzip filter-technical.vcf
tabix -p vcf filter-technical.vcf.gz -f
#+end_src
Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
   12.000                519            519         55         66     0.9042       0.8872     0.8956
     None                519            519         55         66     0.9042       0.8872     0.8956
******* DONE Méthode naïve 521/585
CLOSED: [2023-05-04 Thu 21:57]
Haplotypecaller: Nb reference SNV 692 vs found 585
Variant calling, filter technical: reference SNV 692 vs found 521
******* KILL Comparer avec hap.py
CLOSED: [2023-06-12 Mon 23:24]
****** KILL Après filtre annotation
CLOSED: [2023-06-12 Mon 23:25]
******* DONE Méthode naïve : 493/585
CLOSED: [2023-05-04 Thu 22:09]
******* KILL Comparer avec hap.py
CLOSED: [2023-06-12 Mon 23:25]
******* KILL VCf eval
CLOSED: [2023-06-12 Mon 23:25]
 cd cento/annotation/
 bgzip postvep-filter.vcf
 tabix postvep-filter.vcf.gz
 cd ../..
 rtg vcfeval -b ~/cento_variants/variants_for_vcfeval.tsv.gz -c cento/annotation/postvep-filter.vcf.gz  -o compare-vepfilter -t /Work/Groups/bisonex/data/giab/GRCh38/genomeRef.sdf
 Threshold  True-pos-baseline  True-pos-call  False-pos  False-neg  Precision  Sensitivity  F-measure
----------------------------------------------------------------------------------------------------
   12.000                491            491         50         94     0.9076       0.8393     0.8721
     None                491            491         50         94     0.9076       0.8393     0.8721

**** KILL NEAT : trop lent :neat:
CLOSED: [2023-04-29 Sat 22:06]
***** KILL Génération fastq sur exno 5 GATAD2B
CLOSED: [2023-04-29 Sat 22:06]
Trop lent : pour 1 exon : 1500 secondes !
#+begin_src sh
samtools faidx genomeRef.fna NC_000001.11 | save -f genomeRef_chr1.fna
python gen_reads.py  -r ../test-simuscop/genomeRef_chr1.fna -o lol  -tr ../test-simuscop/gatad2b-exon6.bed  -R 147 --pe 150 10
#+end_src
**** KILL ReSeq : exome avec exons comme fasta mais ne gère pas des exons trop petits :reseq:
CLOSED: [2023-04-30 Sun 19:44] SCHEDULED: <2023-04-29 Sat>
#+begin_quote
Can I simulate exome sequencing? Yes. You need to use a reference that only contains the exons as individual scaffolds. Using --refBiasFile you can specify the coverage of individual exons. To simulate intron contamination you can add the whole reference to the reference containing the exons and strongly reduce the coverage for these scaffolds using --refBiasFile.
#+end_quote
Par contre, rapide
***** DONE Fasta pour exons seuls
CLOSED: [2023-04-30 Sun 19:25]
Depuis le GFF
#+begin_src sh :dir ~/code/bisonex/test-reseq :results silent
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.39_GRCh38.p13/GCF_000001405.39_GRCh38.p13_genomic.gff.gz
#+end_src

#+begin_src sh :dir ~/code/bisonex/test-reseq  :results silent
gunzip -c GCF_000001405.39_GRCh38.p13_genomic.gff.gz | grep -w "exon" > exons.gff
#+end_src

On génère les exons
#+begin_src sh :dir ~/code/bisonex/test-reseq
bedtools getfasta -fi ../test-simuscop/genomeRef.fna -bed exons.gff -fo exons.fna
#+end_src

A tester avec un profile déjà fait :
https://github.com/schmeing/ReSeq-profiles/tree/master/profiles

On cherche l'exons qui nous intéresse
 NC_000001.11 g.153817496 A>T
N'y est pas ??

****** DONE On test sur les 2 premiers : exec
CLOSED: [2023-04-30 Sun 18:39]

#+begin_src
head exons.fa -n 2 > 2exons.fna
#+end_src


#+begin_src sh
../ReSeq/bin/reseq illuminaPE -j 32 -R exons.fa -s Ec-Hi2000-TruSeq.reseq --ipfIterations 0 -1 reseq-sim_1.fq reseq_sim_2.fq
#+end_src

#+begin_quote
error: All reference sequences are too short for simulating. They should have at least 1991 bases
#+end_quote

#+begin_src sh
grep '^>NC_000001.10' exons.fa  | sed 's/:/,/;s/-/,/;s/^>//' > exons.csv
#+end_src

****** DONE Sur 200 premiers exons du chr1
CLOSED: [2023-04-30 Sun 19:17]
#+begin_src sh :dir ~/code/bisonex/test-reseq  :results silent
head -n200 exons.fna > exons-200.fna
 bwa index exons-200.fna
 #+end_src

Simulation avec 30x
#+begin_src sh :dir ~/code/bisonex/test-reseq  :results silent
 ../ReSeq/bin/reseq illuminaPE -R exons-200.fna -s Ec-Hi2000-TruSeq.reseq --ipfIterations 0 -1 reseq1.fq -2 reseq2.fq -c 30
 #+end_src

 Attention, pour l'alignement, il faut le nfa complet ! Sinon erreur du type
 Erreurs:::sam_hdr_create] Duplicated sequence "NC_000001.10:762970-763155" in file "-"
 Et pas de bam avec
 samtools sort: failed to change sort order header to 'coordinate'
 #+begin_src
 bwa mem ../test-simuscop/bwa/genomeRef.fna reseq1.fq reseq2.fq | samtools sort -o reseq.bam
 #+end_src

 Manque des exons et l'allure ne correspond pas...
****** DONE Utiliser le fichier de capture : exons trop petits
CLOSED: [2023-04-30 Sun 19:25]
Comme pour ART
Trop court avec
echo -e "NC_000001.11\t153817371\t153817542" > gatad2b-exon6.bed
Donc on ajoute 1000 de chaque côté
#+begin_src sh :dir ~/code/bisonex/test-reseq :results silent
echo -e "NC_000001.11\t153816371\t153818542" > gatad2b-exon6.bed
bedtools getfasta -fi ../test-simuscop/genomeRef.fna -bed gatad2b-exon6.bed -fo gatad2b-exon6.fna
bwa index gatad2b-exon6.bed
 ../ReSeq/bin/reseq illuminaPE -R gatad2b-exon6.fna -s Ec-Hi2000-TruSeq.reseq --ipfIterations 0 -1 reseq1.fq -2 reseq2.fq -c 30
 bwa mem ../test-simuscop/bwa/genomeRef.fna reseq1.fq reseq2.fq | samtools sort -o reseq.bam
 samtools index reseq.bam
#+end_src
***** KILL Sur le chromosome 15 puis trier à la main sur les zones de capture ?
CLOSED: [2023-04-30 Sun 19:44]
#+begin_src sh :dir ~/code/bisonex/test-reseq :results silent
samtools faidx ../test-simuscop/genomeRef.fna NC_000015.10 > chr15.fna
 ../ReSeq/bin/reseq illuminaPE -R chr15.fna -s Ec-Hi2000-TruSeq.reseq --ipfIterations 0 -1 reseq1.fq -2 reseq2.fq -c 30
#+end_src
**** DONE ART : fonctionne très mal en targeted
CLOSED: [2023-04-30 Sun 11:49]
***** DONE Génération de reads
CLOSED: [2023-04-30 Sun 11:49]
****** DONE Avec seulement les exons en séquence
CLOSED: [2023-04-30 Sun 10:24]
 head -n6 exons.fa | save three-exons.fna

../art_bin_MountRainier/art_illumina -ss HS25 -i three-exons.fna -o ./paired_end_com -l 150 -f 10 -p -m 500 -s 10 -sam

Le sam n'est pas visible sur igv mais si on aligne avec bwa mem, on a quelques reads
****** DONE Extraire une zone de capture dans le fasta
CLOSED: [2023-04-30 Sun 11:49]
 NC_000001.11 g.153817496 A>T
******* DONE Essai 1: ne dépasse pas la zone
CLOSED: [2023-04-30 Sun 10:49]
#+begin_src sh :dir ~/code/bisonex/test-art :results silent
echo -e "NC_000001.11\t153817371\t153817542" > gatad2b-exon6.bed
bedtools getfasta -fi ../test-simuscop/genomeRef.fna -bed gatad2b-exon6.bed -fo gatad2b-exon6.fa
#+end_src

-ss HS25 : nom du profile illumina
-l 150 : reads de 150
-f 10 : coverage de 10
-p : paired end
-m 500 : longueur moyenne des fragment d'ADN
-s 10 : déviation standard
#+begin_src sh :dir ~/code/bisonex/test-art :results silent
../art_bin_MountRainier/art_illumina -ss HS25 -i gatad2b-exon6.fa -o ./gatad2b -l 150 -f 100 -p -m 500 -s 10
#+end_src
#+begin_src sh :dir ~/code/bisonex/test-art :results silent
bwa mem ../test-simuscop/bwa/genomeRef gatad2b1.fq gatad2b2.fq | samtools sort -o gatad2b.b
am
samtools index gatad2b.bam
#+end_src

#+attr_html: :width 800px
[[./art-capture-1.png]]
******* Avec offset
50bp idem
NC_000001.11 153817371  153817542 -> NC_000001.11 153817321  153817592
On essaie 1000
NC_000001.11 153817371  153817542 -> NC_000001.11 153816371  153818542 ->
#+begin_src sh :dir ~/code/bisonex/test-art :results silent
echo -e "NC_000001.11\t153816371\t153818542" > gatad2b-exon6-offset.bed
bedtools getfasta -fi ../test-simuscop/genomeRef.fna -bed gatad2b-exon6-offset.bed -fo gatad2b-exon6-offset.fa
../art_bin_MountRainier/art_illumina -ss HS25 -i gatad2b-exon6-offset.fa -o ./gatad2b -l 150 -f 100 -p -m 500 -s 10
bwa mem ../test-simuscop/bwa/genomeRef gatad2b1.fq gatad2b2.fq | samtools sort -o gatad2b.bam
samtools index gatad2b.bam
#+end_src

mieux mais trop large
#+attr_html: :width 800px
[[./art-exon6-offset1000.png]]

Sur les vraies données, on a une large de 500bp environ
#+attr_html: :width 800px
[[./illumina-ref-exon6.png]]


Ici l'exon fait 250bp donc on rajouter 125bp de chaque côté
NC_000001.11 153817371  153817542
NC_000001.11 153817246  153817667
Résulats incohérents :on a 2 colonnes séparées !

Essai avec +200bp de chaque côt
NC_000001.11 153817371  153817542
NC_000001.11 153817171  153817742
#+begin_src sh :dir ~/code/bisonex/test-art :results silent
echo -e "NC_000001.11\t153817171\t153817742"> gatad2b-exon6-offset.bed
bedtools getfasta -fi ../test-simuscop/genomeRef.fna -bed gatad2b-exon6-offset.bed -fo gatad2b-exon6-offset.fa
../art_bin_MountRainier/art_illumina -ss HS25 -i gatad2b-exon6-offset.fa -o ./gatad2b -l 150 -f 100 -p -m 500 -s 50
bwa mem ../test-simuscop/bwa/genomeRef gatad2b1.fq gatad2b2.fq | samtools sort -o gatad2b.bam
samtools index gatad2b.bam
#+end_src

En changeant la longueur des reads
#+begin_src sh :dir ~/code/bisonex/test-art :results silent
echo -e "NC_000001.11\t153817171\t153817742"> gatad2b-exon6-offset.bed
bedtools getfasta -fi ../test-simuscop/genomeRef.fna -bed gatad2b-exon6-offset.bed -fo gatad2b-exon6-offset.fa
../art_bin_MountRainier/art_illumina -ss HS25 -i gatad2b-exon6-offset.fa -o ./gatad2b -l 126 -f 100 -p -m 500 -s 50
bwa mem ../test-simuscop/bwa/genomeRef gatad2b1.fq gatad2b2.fq | samtools sort -o gatad2b.bam
samtools index gatad2b.bam
#+end_src
Idem, fait 2 "tas" séparés de chaque côté de l'exon

+/- 100
#+begin_src sh :dir ~/code/bisonex/test-art :results silent
echo -e "NC_000001.11\t153817271\t153817642" > gatad2b-exon6-offset.bed
bedtools getfasta -fi ../test-simuscop/genomeRef.fna -bed gatad2b-exon6-offset.bed -fo gatad2b-exon6-offset.fa
../art_bin_MountRainier/art_illumina -ss HS25 -i gatad2b-exon6-offset.fa -o ./gatad2b -l 125 -f 100 -p -m 500 -s 30
bwa mem ../test-simuscop/bwa/genomeRef gatad2b1.fq gatad2b2.fq | samtools sort -o gatad2b.bam
samtools index gatad2b.bam
#+end_src
**** KILL NGSNG : plante
CLOSED: [2023-05-01 Mon 09:27]
https://github.com/RAHenriksen/NGSNGS
***** KILL Essai avec fasta
CLOSED: [2023-05-01 Mon 09:27]
#+begin_src sh :dir ~/code/bisonex/test-ngsngs :results silent
echo -e "NC_000001.11\t153817371\t153817542" > gatad2b-exon6.bed
bedtools getfasta -fi ../test-simuscop/genomeRef.fna -bed gatad2b-exon6.bed -fo gatad2b-exon6.fa
#+end_src

-ss HS25 : nom du profile illumina
-l 150 : reads de 150
-f 10 : coverage de 10
-p : paired end
-m 500 : longueur moyenne des fragment d'ADN
-s 10 : déviation standard
#+begin_src sh :dir ~/code/bisonex/test-art :results silent
../art_bin_MountRainier/art_illumina -ss HS25 -i gatad2b-exon6.fa -o ./gatad2b -l 150 -f 100 -p -m 500 -s 10
#+end_src
#+begin_src sh :dir ~/code/bisonex/test-art :results silent
bwa mem ../test-simuscop/bwa/genomeRef gatad2b1.fq gatad2b2.fq | samtools sort -o gatad2b.b
am
samtools index gatad2b.bam
#+end_src
**** KILL Script maison :generate:
CLOSED: [2023-05-13 Sat 18:29] SCHEDULED: <2023-05-01 Mon>
***** KILL SNV
CLOSED: [2023-05-13 Sat 18:29] SCHEDULED: <2023-05-01 Mon>
****** DONE Script python: ok seulement pour reads corrigé, trop long sinon
CLOSED: [2023-05-01 Mon 13:32]
Même sur un seul chromosome (15)...
****** DONE Couper les données avec bedtools intersect ?
CLOSED: [2023-05-01 Mon 20:33]
-wa pour avoir les reads qui corresponds
-v pour ceux qui n'intersecte pass
ok sur un exemple
******* DONE Test simple : ok
CLOSED: [2023-05-01 Mon 13:43]

#+attr_html: :width 500px
[[./test-intersect-chr15.png]]
#+attr_html: :width 500px
[[./test-nointersect-chr15.png]]
******* DONE Chromosome 15 : vérifier BAM
CLOSED: [2023-05-01 Mon 20:33] SCHEDULED: <2023-05-01 Mon>
#+begin_src
samtools view -b `63003856_S135.bam`  NC_000015.10 > `63003856_S135_chr15.bam`
#+end_src
On génère un python avec les dépendances
#+begin_src
nix-build
#+end_src
Puis on lance un script julia qui va couper le bam en 2, lancer le script python sur l'intersection et fusionner le résultat
#+begin_src
julia -Jbisonex.so --project=. insertVariants.jl `63003856_S135_chr15.bam` test_new.bam
#+end_src
NB: pour accélerer l'exécution, générer une sysimage :

#+begin_src julia
(1.7) pkg> activate .
julia> create_sysimage(; sysimage_path="bisonex.so")
#+end_src

3 variants: Ok sur le nombre de reads et les variants
Ok pour homozygote

#+attr_html: :width 500px
[[./check-snv-chr15.png]]

******* DONE Chromosome 15 Vérifier VAF avec checkBam.jl: ok
julia> @subset snv :chrom .== "NC_000015.10"
26×10 DataFrame
 Row │ chrom         pos       variant        variantType  zygosity      ref        alt        refCount  altCount  readsCount
     │ SubStrin…?    Int64     SubStrin…?     String?      String15      SubStrin…  SubStrin…  Int64     Int64     Int64
─────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ NC_000015.10  74343027  g.74343027C>T  snv          heterozygous  C          T                61        58         120
   2 │ NC_000015.10  75400778  g.75400778C>G  snv          heterozygous  C          G               108        79         187
   3 │ NC_000015.10  89327201  g.89327201C>T  snv          heterozygous  C          T               243       241         486
   4 │ NC_000015.10  48767448  g.48767448A>C  snv          heterozygous  A          C                72        70         142
   5 │ NC_000015.10  75411685  g.75411685T>C  snv          heterozygous  T          C                79        81         160
   6 │ NC_000015.10  66703292  g.66703292C>T  snv          heterozygous  C          T                70        60         130
   7 │ NC_000015.10  89325639  g.89325639G>A  snv          heterozygous  G          A               257       267         524
   8 │ NC_000015.10  89330184  g.89330184G>A  snv          heterozygous  G          A               258       287         548
   9 │ NC_000015.10  89330184  g.89330184G>A  snv          heterozygous  G          A               258       287         548
  10 │ NC_000015.10  89325639  g.89325639G>A  snv          heterozygous  G          A               257       267         524
  11 │ NC_000015.10  42401752  g.42401752G>A  snv          homozygous    G          A                61       212         273
  12 │ NC_000015.10  89327201  g.89327201C>T  snv          heterozygous  C          T               243       241         486
  13 │ NC_000015.10  38339896  g.38339896G>A  snv          heterozygous  G          A                56        86         144
  14 │ NC_000015.10  26869324  g.26869324A>T  snv          heterozygous  A          T                62        49         113
  15 │ NC_000015.10  66435145  g.66435145G>A  snv          heterozygous  G          A                98        95         193
  16 │ NC_000015.10  60514655  g.60514655G>A  snv          heterozygous  G          A                94        99         194
  17 │ NC_000015.10  42410947  g.42410947A>G  snv          heterozygous  A          G               153       123         276
  18 │ NC_000015.10  75430368  g.75430368C>T  snv          heterozygous  C          T                80        62         142
  19 │ NC_000015.10  25375494  g.25375494T>C  snv          heterozygous  T          C               103       104         207
  20 │ NC_000015.10  60497497  g.60497497C>A  snv          heterozygous  C          A                61        65         126
  21 │ NC_000015.10  74891539  g.74891539C>T  snv          heterozygous  C          T               118       124         242
  22 │ NC_000015.10  48488433  g.48488433A>G  snv          heterozygous  A          G               367       122         492
  23 │ NC_000015.10  89318565  g.89318565A>G  snv          heterozygous  A          G               303        98         404
  24 │ NC_000015.10  89323426  g.89323426C>G  snv          heterozygous  C          G                93       109         202
  25 │ NC_000015.10  89318595  g.89318595T>C  snv          heterozygous  T          C               321       128         453
  26 │ NC_000015.10  48488437  g.48488437T>C  snv          heterozygous  T          C               356       132         488
CLOSED: [2023-05-01 Mon 17:18]
****** KILL Chromosome1 15 :Test haplotype caller : échec car CIGARE non mis à jour
CLOSED: [2023-05-13 Sat 18:29] SCHEDULED: <2023-05-01 Mon>

#+begin_src
julia -Jbisonex.so --project=. insertVariants.jl `63003856_S135_chr15.bam` 63003856_S135_chr15_inserted.bam
scp 63003856_S135_chr15_inserted.bam* meso:/Work/Users/apraga/bisonex/tests/synthetic/

#+end_src

#+begin_src sh :dir /ssh:meso:/Work/Users/apraga/bisonex/tests/synthetic :results silent
ln -s /Work/Projects/bisonex/data/dbSNP/GRCh38.p13/dbSNP.gz .
ln -s /Work/Projects/bisonex/data/dbSNP/GRCh38.p13/dbSNP.gz.tbi
ln -s /Work/Projects/bisonex/data/genome/GRCh38.p13/genomeRef.dict .
ln -s /Work/Projects/bisonex/data/genome/GRCh38.p13/genomeRef.fna .
ln -s /Work/Projects/bisonex/data/genome/GRCh38.p13/genomeRef.fna.fai .
#+end_src

puis
#+begin_src
gatk --java-options "-Xmx3072M" HaplotypeCaller --input 63003856_S135_chr15_inserted.bam --output testchr15.vcf.gz --reference genomeRef.fna  --tmp-dir . -L NC_000015.10
#+end_src

scp meso:/Work/Users/apraga/bisonex/tests/synthetic/testchr15.vcf.gz haplotypecaller-chr15.vcf.gz

Aucun variant inséré
- base quality ok
  -
******* DONE bam out : non appelé
CLOSED: [2023-05-01 Mon 21:57]
gatk --java-options "-Xmx3072M" HaplotypeCaller --input 63003856_S135_chr15_inserted.bam     --output haplotypecaller-chr15.vcf.gz --reference genomeRef.f
na  --tmp-dir . -L NC_000015.10  --bam-output debug.bam
******* DONE --linked-de-bruijn-graph : idem
CLOSED: [2023-05-01 Mon 21:57]
readlink testchr15.vcf.gz -f^C
[apraga@mesointeractive synthetic]$ gatk --java-options "-Xmx3072M" HaplotypeCaller --input 63003856_S135_chr15_inserted.bam     --output haplotypecaller-chr15.vcf.gz --reference genomeRef.fna  --tmp-dir . -L NC_000015.10  --linked-de-bruijn-graph
******* KILL regénérer fastq
CLOSED: [2023-05-13 Sat 18:29]
Non
****** KILL Générer bam données pour tous les chromosomes
CLOSED: [2023-05-13 Sat 18:29]
 timeit julia -Jbisonex.so --project=. insertVariants.jl ~/code/bisonex/out/63003856/preprocessing/63003856_S135.bam 63003856_S135_inserted.bam

40min 516ms 835µs 405ns
Avertissement:
 [W::bam_hdr_read] EOF marker is absent. The input is probably truncated

Inserted.bam et excluded.bam (fichier avant le merge)  ont l'air ok...

On réessaie à la main : ça passe
#+begin_src
samtools merge test-all.bam inserted.bam excluded.bam
❯ mv test-all.bam `63003856_S135_inserted.bam` -f
❯ mv test-all.bam.bai `63003856_S135_chr15_inserted.bam.bai` -f
#+end_src
****** DONE BAm2fastq pour avoir CIGAR à jour : échec (variants "cachés")
CLOSED: [2023-05-04 Thu 20:30] SCHEDULED: <2023-05-01 Mon>

On lance la génération de bam depuis le mesocentro (la copie plante via le VPN)
#+begin_src sh
cd /Work/Users/apraga/recherche/bisonex/generate
julia --project=. insertVariants.jl  ../../../bisonex/out/63003856_S135/preprocessing/applybqsr/63003856_S135.bam 63003856_S135_inserted.bam
#+end_src
Workflow après avec désactivé storeDir pour SAMTOOLS_BAM2FQ dans nextflow.config (pourquoi ??)
#+begin_src nextflow
include { SAMTOOLS_BAM2FQ }                            from "${params.modulesDir}/samtools/bam2fq/main"
include { SAMTOOLS_SORT as sortBamByName }             from "${params.modulesDir}/samtools/sort/main"

workflow {
    f = Channel.fromPath("${params.dataDir}/synthetic/63003856_S135_inserted.bam",
                         checkIfExists: true).map{it -> [["id": "synthetic_63003856"], it]}
    // Important: use "-n" option !!
    sortBamByName(f)
    SAMTOOLS_BAM2FQ(sortBamByName.out.bam, true)
}
#+end_src
Puis
#+begin_src
cp work/34/fb2fc136f6f6d7f42d0960512f06de/*.fq.gz /Work/Groups/bisonex/data/synthetic/
#+end_src
****** KILL Lancer pipeline
CLOSED: [2023-05-04 Thu 20:30] SCHEDULED: <2023-05-01 Mon>
NXF_OPTS=-D"user.name=apraga" nextflow run   main.nf -c nextflow.config  -profile standard,helios -bg --input="/Work/Groups/bisonex/data/synthetic/synthetic_63003856_{1,2}.fq.gz" --outdir out/synthetic_63003856
**** HOLD Bamsurgeon :bamsurgeon:
***** HOLD Package nix
1. Patcher la recherche du génome de référence pour bien trouver les index (en utilisant une regexp comme nf-core)
2. Rajouter le chemin de picard dans les arguments
3. Option -O3 pour performance
****** DONE Erreur ValueError: quality and sequence mismatch
CLOSED: [2023-05-19 Fri 18:44]
******* DONE Idem avec dernière version sur github
CLOSED: [2023-05-18 Thu 14:36]
******* DONE Version 1.3: ok mais
CLOSED: [2023-05-19 Fri 18:44]
Test sur chr22: variants ok mais VAF=1...
S'exécute "normalement" (échec selon nextflow) mais le bam de sorstie quasiment vide
La fusion du bam avec les variants et du fichier de référence n'a fonctionné correctement.
******** DONE Lancer replacereads.py à la main
CLOSED: [2023-05-18 Thu 21:41]
Dans /Work/Users/apraga/bisonex/work/f3/ce044f80ca91016d68d1bc4f4f5301
#+begin_src sh
 /nix/store/xw277la6w4sjqlsvw9h32cvrlacrfkgm-python3-3.10.9-env/bin/python3.10 /nix/store/abzangf0q8k37053p776cfkw181dzjn3-bamsurgeon-1.3/bin/bamsurgeon/replacereads.py -b cento.bam -r addsnv.e14561be-4fdd-45cc-9989-048ab6da6cc6.muts.bam -o snv-manual.bam
#+end_src
Puis
#+begin_src
samtools sort snv-manual.bam -o snv-manual-sorted.bam
#+end_src
A l'air de fonctionne
****** HOLD Corriger run nextflow pour éviter les erreurs
Trop de message d'erreur en sortie ?
****** DONE Test sur mini-bam: échec
CLOSED: [2023-05-14 Sun 21:12]
❯ samtools view -h ~/code/bisonex/simuscop-cento-200x/cento/preprocessing/mapped/cento.bam | head -n1000 | samtools view -Sb - > mini.bam

❯ samtools index mini.bam

Sans spécfier le variant:
#+begin_quote
NC_000001.11	17651	17651
#+end_quote
./result/bin/addsnv -v snv.txt -f mini.bam -r ../data/genomeRef.fna -o test.bam

****** DONE Test chr22
CLOSED: [2023-05-15 Mon 23:24]
Pas assez de reads, on prend le chromosome 22
#+begin_src sh
samtools view ../simuscop-cento-200x/cento/preprocessing/mapped/cento.bam NC_000022.11 -b -o chr22.bam
samtools index chr22.bam
#+end_src
Mésocentre
dans tests/bamsurgeno
#+begin_src
addsnv -v snv.txt -f chr22.bam -r ../genomeRef.fna -o test.bam --aligner mem
#+end_src
******* DONE SNV aléatoire:
CLOSED: [2023-05-15 Mon 23:13]
NC_000022.11	17499704	17499704    0.2
On retrouve bien un variant à cette position A > T
******* DONE SNV avec ALT prédéfini : retrouvée dans IGV (mais pas dans pileup)
CLOSED: [2023-05-15 Mon 23:13]
NC_000022.11	17499704	17499704    0.2 G
******* DONE Variants patients chr22: ok IGV
CLOSED: [2023-05-15 Mon 23:23]
Fichier non trié donc
samtools sort test.bam -o test-sorted.bam
samtools index test-sorted.bam
******* DONE Vérifier qu'il faut POS et POS+1: non
CLOSED: [2023-05-14 Sun 21:21]
***** HOLD Variants cento
****** STRT SNV
Attention à la mémoire: 32G ne semble pas suffire avec 12 threads
#+begin_src sh
NXF_OPTS=-D"user.name=${USER}" nextflow run workflows/bamsurgeon.nf -profile standard,helios --input=tests/bamsurgeon/snv-cento.tsv -bg
#+end_src

ET
#+begin_src nextflow
workflow {
    f = Channel.fromPath(params.input, checkIfExists: true)
    bam = Channel.fromPath("simuscop-cento-200x/cento/preprocessing/mapped/cento.bam",
                           checkIfExists: true)
    bamIndex = bam.map { it -> it + ".bai" }

    downloadGenome | indexGenome
    indexGenome.out.index | view
    addSNV(f, bam, bamIndex, downloadGenome.out, indexGenome.out.index, indexGenome.out.dict, indexGenome.out.fai)
}
#+end_src
******* DONE v1.3: Lancer le pipeline pour vérifier qu'on retrouve les variants
CLOSED: [2023-05-19 Fri 18:41] SCHEDULED: <2023-05-18 Thu>
******* HOLD Corrigier position pour avoir une bonne VAF
POS POS+1 VAF ALT
Attention, la base corrigée est à POS+1...
******* DONE Comparaison manuelle avec julia (VAF = 1...)
CLOSED: [2023-05-19 Fri 21:58]
552 found over 585
****** TODO del
****** TODO ins
**** KILL [[id:966a298c-948a-4694-a6f5-c326b1046a05][XAMscissors.jl]] :xamscissors:
CLOSED: [2023-08-12 Sat 15:59]
***** KILL Test SNV
CLOSED: [2023-08-12 Sat 15:59]
****** DONE Phase 1 : chr22, VAF=1
CLOSED: [2023-05-29 Mon 15:36]
******* DONE 1 SNV  : ok !
CLOSED: [2023-05-20 Sat 19:35] SCHEDULED: <2023-05-20 Sat>

#+begin_src
 make run READS="tests/bamscissors/corrected_{1,2}.fq.gz"
Puis on lance le pipeline sur correct_1
- [X] Variant visible dans IGV
- [X] Variant visible après alignement
- [X] Variant visible après appel de variant
******* KILL Tester SNV chromosome 22
CLOSED: [2023-05-29 Mon 15:36] SCHEDULED: <2023-05-20 Sat>
****** KILL PHase 2 : chr22, VAF variable
CLOSED: [2023-06-12 Mon 23:27] SCHEDULED: <2023-06-03 Sat>
******* DONE de nombreux reads sont perdus -> ok sur un SNV en alignant sur chromosome 22
CLOSED: [2023-05-29 Mon 15:38]
Problème dansr le BAM car sans les insertion
******** DONE Filtrer les reads sans pair : idem
CLOSED: [2023-05-23 Tue 00:01]
FOUND:Found 16662 unpaired mates
	at htsjdk.samtools.SAMUtils.processValidationError(SAMUtils.java:470)
	at picard.sam.SamToFastq.doWork(SamToFastq.java:224)
	at picard.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:308)
	at picard.cmdline.PicardCommandLine.instanceMain(PicardCommandLine.java:103)
	at picard.cmdline.PicardCommandLine.main(PicardCommandLine.java:113)

ex: chr22:42213078

On essaie
samtools view ~/code/bisonex/out/63003856/preprocessing/mapped/63003856_S135.bam NC_000022.11  -hf 0x2 -o 63003856_chr22.bam

Ne rale pas...
SCHEDULED: <2023-05-22 Mon>
******** DONE Problème de la conversion BAM -> fastq ?
CLOSED: [2023-05-24 Wed 23:19]
********* DONE Test bamtofastq sur le BAM originel: idem
CLOSED: [2023-05-23 Tue 01:16]
Dans ~/recherche/code/bisonex/Bamscissors.jl
samtools view ~/code/bisonex/out/63003856/preprocessing/mapped/63003856_S135.bam NC_000022.11  -hf 0x2 -o 63003856_chr22.bam
On trie bien par nom de read
samtools sort -n 63003856_chr22.bam -o 63003856_chr22_sorted.bam

Conversion en fastq. Attention à bien prendre le *fichier trié* !
samtools bam2fq  -1 63003856_chr22_init1.fq.gz -2 63003856_chr22_init2.fq.gz  -n 63003856_chr22_sorted.bam
[M::bam2fq_mainloop] discarded 0 singletons
[M::bam2fq_mainloop] processed 2592110 reads

On envoie sur le mésocentre
scp 63003856_chr22_init{1,2}.fq.gz meso:/Work/Users/apraga/bisonex/tests/bamscissors
On démarre un job avec 24 coeurs pour aller vite

srun -c 24 -p smp -t 1:00:00 --pty bash
bwa mem -t 24 /Work/Projects/bisonex/data/genome/GRCh38.p13/bwa/genomeRef 63003856_chr22_init1.fq.gz 63003856_chr22_init2.fq.gz | samtools sort -@24 -o output.bam -

Dans /home/alex/recherche/bisonex/code/BamScissors.jl/aligned
❯ scp meso:/Work/Users/apraga/bisonex/tests/bamscissors/output.bam*
********* DONE Avec samtools fastq
CLOSED: [2023-05-23 Tue 01:16]
samtools sort -n 63003856_chr22.bam -o 63003856_chr22_sorted.bam
samtools fastq 63003856_chr22_1.fq.gz -2 63003856_chr22_2.fq.gz -0 /dev/null -s /dev/null -o 63003856_chr22_sorted.bam
scp 63003856_chr22_{1,2}.fq.gz  meso:/Work/Users/apraga/bisonex/tests/bamscissors/

mesocentre
 bwa mem -t 24 /Work/Projects/bisonex/data/genome/GRCh38.p13/bwa/genomeRef 63003856_chr22_1.fq.gz 63003856_chr22_2.fq.gz | samtools sort -@24 -o output.bam

 scp meso:/Work/Users/apraga/bisonex/tests/bamscissors/output.bam\* aligned/
********* DONE En rajoutant .fna dans le doserrie (+samtools fastq): idem
CLOSED: [2023-05-23 Tue 01:16]
ln -s /Work/Projects/bisonex/data/genome/GRCh38.p13/bwa/genomeRef* .
ln -s /Work/Projects/bisonex/data/genome/GRCh38.p13/genomeRef* .

bwa mem -t 24 genomeRef 63003856_chr22_1.fq.gz 63003856_chr22_2.fq.gz | samtools sort -@24 -o output.bam
********* DONE Test picard : idem
CLOSED: [2023-05-23 Tue 01:16]
Dans bisonex/code/BamScissors.jl
❯ picard SamToFastq -I 63003856_chr22_sorted.bam -F testpicard1.fastq -F2 testpicard2.fastq

bwa mem -t 24 /Work/Projects/bisonex/data/genome/GRCh38.p13/bwa/genomeRef testpicard1.fastq.gz testpicard2.fastq.gz | samtools sort -@24 -o testpicard.bam -
********* DONE Il ne manque pas des reads dans les fastq :
CLOSED: [2023-05-23 Tue 01:26]
bisonex/code/BamScissors.jl on  bamscissors [!?] via ஃ v1.9.0
❯ samtools  flagstat aligned/output.bam
1306273 + 0 in total (QC-passed reads + QC-failed reads)
1296055 + 0 primary
0 + 0 secondary
10218 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
1304871 + 0 mapped (99.89% : N/A)
1294653 + 0 primary mapped (99.89% : N/A)
0 + 0 paired in sequencing
0 + 0 read1
0 + 0 read2
0 + 0 properly paired (N/A : N/A)
0 + 0 with itself and mate mapped
0 + 0 singletons (N/A : N/A)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)

bisonex/code/BamScissors.jl on  bamscissors [!?] via ஃ v1.9.0
❯ samtools flagstat 63003856_chr22.bam
2609214 + 0 in total (QC-passed reads + QC-failed reads)
2592110 + 0 primary
0 + 0 secondary
17104 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
2609214 + 0 mapped (100.00% : N/A)
2592110 + 0 primary mapped (100.00% : N/A)
2592110 + 0 paired in sequencing
1296055 + 0 read1
1296055 + 0 read2
2592110 + 0 properly paired (100.00% : N/A)
2592110 + 0 with itself and mate mapped
0 + 0 singletons (0.00% : N/A)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)

❯ echo $(zcat 63003856_chr22_init2.fq.gz | wc -l)/4 | bc
1296055

bisonex/code/BamScissors.jl on  bamscissors [!?] via ஃ v1.9.0 took 2s
❯ echo $(zcat 63003856_chr22_init1.fq.gz | wc -l)/4 | bc
1296055
❯ samtools  flagstat 63003856_chr22_sorted.bam
2609214 + 0 in total (QC-passed reads + QC-failed reads)
2592110 + 0 primary
0 + 0 secondary
17104 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
2609214 + 0 mapped (100.00% : N/A)
2592110 + 0 primary mapped (100.00% : N/A)
2592110 + 0 paired in sequencing
1296055 + 0 read1
1296055 + 0 read2
2592110 + 0 properly paired (100.00% : N/A)
2592110 + 0 with itself and mate mapped
0 + 0 singletons (0.00% : N/A)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)
********* DONE Avec sort -O BAM idem
CLOSED: [2023-05-23 Tue 01:21]
********* DONE Singleton ou non mapped ? nonVirtPosition
CLOSED: [2023-05-23 Tue 01:22]
❯ samtools bam2fq  -1 63003856_chr22_init1.fq.gz -2 63003856_chr22_init2.fq.gz -0 both -s single.fq.gz  -n 63003856_chr22_sorted.bam


bisonex/code/BamScissors.jl on  bamscissors [!?] via ஃ v1.9.0
❯ wc -l 63003856_chr22_init* both single.fq.gz
   403540 63003856_chr22_init1.fq.gz
   404788 63003856_chr22_init2.fq.gz
        0 both
        0 single.fq.gz
   808328 total
********* Problème d'aligner car les reads sont bien dans le .fastq ?
Ex:
 zgrep "A00853:477:HMLWYDSX3:2:2624:2826:18630" 63003856_chr22_{1,2}.fq.gz
63003856_chr22_1.fq.gz:@A00853:477:HMLWYDSX3:2:2624:2826:18630
63003856_chr22_2.fq.gz:@A00853:477:HMLWYDSX3:2:2624:2826:18630
******** DONE Refaire la manip sur bam chr22 non modifié + mail Alexis
CLOSED: [2023-05-23 Tue 23:27]

[1]_samtools view ~/code/bisonex/out/63003856/preprocessing/mapped/63003856_S135.bam NC_000022.11  -f 0x2 -o 63003856_chr22.bam

 Les reads sont bien tous mappé
samtools flagstat 63003856_chr22.bam
2609214 + 0 in total (QC-passed reads + QC-failed reads)
2592110 + 0 primary
0 + 0 secondary
17104 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
2609214 + 0 mapped (100.00% : N/A)

[2] samtools sort -n 63003856_chr22.bam -o 63003856_chr22_sorted.bam
flagstat idem

[3] samtools fastq -1 63003856_chr22_1.fq.gz -2 63003856_chr22_2.fq.gz -0 /dev/null -s /dev/null -n 63003856_chr22_sorted.bam
[M::bam2fq_mainloop] discarded 0 singletons
[M::bam2fq_mainloop] processed 2592110 reads

Nombre de reads ok (= la moitié) et les séquences ne sont pas identiques pour le premier read (= on n'a pas 2x la même chose)
 echo $(zcat 63003856_chr22_1.fq.gz | wc -l)/4 | bc
1296055
 echo $(zcat 63003856_chr22_2.fq.gz | wc -l)/4 | bc
1296055

 [4]
 bwa mem -t 24 /Work/Projects/bisonex/data/genome/GRCh38.p13/bwa/genomeRef 63003856_chr22_1.fq.gz 63003856_chr22_2.fq.gz  -o wtf.bam
 [M::bwa_idx_load_from_disk] read 0 ALT contigs
[M::process] read 1752492 sequences (240000014 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 702821, 0, 18)
[M::mem_pestat] analyzing insert size distribution for orientation FF...
[M::mem_pestat] (25, 50, 75) percentile: (54, 197, 268)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 696)
[M::mem_pestat] mean and std.dev: (163.92, 129.71)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 910)
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (129, 175, 233)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 441)
[M::mem_pestat] mean and std.dev: (185.56, 75.37)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 545)
[M::mem_pestat] skip orientation RF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation RR...
[M::mem_pestat] (25, 50, 75) percentile: (56, 192, 239)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 605)
[M::mem_pestat] mean and std.dev: (158.00, 110.30)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 788)
[M::mem_pestat] skip orientation FF
[M::mem_pestat] skip orientation RR
[M::process] read 839618 sequences (114952336 bp)...
[M::mem_process_seqs] Processed 1752492 reads in 375.714 CPU sec, 17.645 real sec
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 336379, 0, 5)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (128, 174, 232)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 440)
[M::mem_pestat] mean and std.dev: (184.73, 74.63)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 544)
[M::mem_pestat] skip orientation RF as there are not enough pairs
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_process_seqs] Processed 839618 reads in 183.039 CPU sec, 7.961 real sec
[main] Version: 0.7.17-r1188
[main] CMD: bwa mem -t 24 -o wtf.bam /Work/Projects/bisonex/data/genome/GRCh38.p13/bwa/genomeRef 63003856_chr22_1.fq.gz 63003856_chr22_2.fq.gz
[main] Real time: 38.278 sec; CPU: 565.821 sec

Bon nombre de reads pourtant
 samtools flagstat wtf.bam
2611059 + 0 in total (QC-passed reads + QC-failed reads)
2592110 + 0 primary
0 + 0 secondary
18949 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
2611058 + 0 mapped (100.00% : N/A)
2592109 + 0 primary mapped (100.00% : N/A)
2592110 + 0 paired in sequencing
1296055 + 0 read1
1296055 + 0 read2
2590970 + 0 properly paired (99.96% : N/A)
2592108 + 0 with itself and mate mapped
1 + 0 singletons (0.00% : N/A)
458 + 0 with mate mapped to a different chr
63 + 0 with mate mapped to a different chr (mapQ>=5)

$ samtools sort -@24 -o wtf_sorted.bam wtf.sam
[bam_sort_core] merging from 0 files and 24 in-memory blocks...

 samtools flagstat wtf.bam
2611059 + 0 in total (QC-passed reads + QC-failed reads)
2592110 + 0 primary
0 + 0 secondary
18949 + 0 supplementary
0 + 0 duplicates
0 + 0 primary duplicates
2611058 + 0 mapped (100.00% : N/A)
2592109 + 0 primary mapped (100.00% : N/A)
2592110 + 0 paired in sequencing
1296055 + 0 read1
1296055 + 0 read2
2590970 + 0 properly paired (99.96% : N/A)
2592108 + 0 with itself and mate mapped
1 + 0 singletons (0.00% : N/A)
458 + 0 with mate mapped to a different chr
63 + 0 with mate mapped to a different chr (mapQ>=5)

Effectivement, ce n'est pas un problème d'IGV

❯ samtools mpileup 63003856_chr22.bam -r NC_000022.11:42213078-42213078
[mpileup] 1 samples in 1 input files
NC_000022.11	42213078	N	19	TTtTTtTTTtTTtTtTttt	kkk_kkkkFkFF_FkFkQk

bisonex/code/BamScissors.jl on  bamscissors [!?] via ஃ v1.9.0
❯ samtools mpileup aligned/wtf_sorted.bam -r NC_000022.11:42213078-42213078
[mpileup] 1 samples in 1 input files
NC_000022.11	42213078	N	5	TTtTT	_FkFF
******** DONE Regarder où ont été aligné les reads (nouveau run)
CLOSED: [2023-05-24 Wed 23:18]
********* DONE Préparation
CLOSED: [2023-05-24 Wed 21:59]
On relance le pipeline pour avoir un BAM propre
On garde les reads non mappé à partsir de la sortie d'applybqsr
#+begin_src sh
NXF_OPTS=-D"user.name=apraga" nextflow run main.nf -c nextflow.config  -profile standard,helios --input="/Work/Projects/bisonex/cento/fastq/2200467051_63003856/63003856_S135_R{1,2}_001.fastq.gz" --outdir=out -bg
cd out/63003856_S135_R/preprocessing/applybqsr/
samtools view 63003856_S135_R.bam NC_000022.11  -f 0x2 -o 63003856_chr22.bam
samtools sort -n 63003856_chr22.bam -o 63003856_chr22_sorted.bam
samtools fastq -1 63003856_chr22_1.fq.gz -2 63003856_chr22_2.fq.gz -0 /dev/null -s /dev/null -n 63003856_chr22_sorted.bam
make run BG= READS="out/63003856_S135_R/preprocessing/applybqsr/63003856_chr22_{1,2}.fq.gz"
cd out/63003856_chr22/preprocessing/mapped/
samtools index 63003856_chr22.bam
samtools mpileup 63003856_chr22.bam -r NC_000022.11:42213078-42213078
#+end_src

On récupère les 2 bam dans
#+begin_src
cd /home/alex/recherche/bisonex/code/BamScissors.jl/
rsync -avz meso:/Work/Users/apraga/bisonex/out/63003856_chr22/preprocessing/mapped data
rsync -avz meso:/Work/Users/apraga/bisonex/out/63003856_S135_R/preprocessing/applybqsr/ data/init/
#+end_src
********* Vérification que le reads est ailleurs
On cherche un read manquant dans le second alignement
#+begin_src sh
samtools view data/init/63003856_chr22.bam | rg "A00853:477:HMLWYDSX3:1:1413:4390:28573"
#+end_src

#+RESULTS:
: A00853:477:HMLWYDSX3:1:1413:4390:28573	163	NC_000022.11	42212845	0	151M	=	42212883	189	CCCAGGGGCCCCAGTGGGGATTTTCTAATAGAGACCCAATGCTTTTGTTCAGAAGGCCCCTGCTAGCTAATTCATTGGTTTGACTAACCAAGACATTGGGCCTTGTGGTTCCTTCTAGGCTACCAGCCATCCCCTGATGCTCTTGAGTACT	ACC+FBCDCBBBAEAEDEEBBCCCECACBAEBEBDCCBCBFDCCCCFACEBEBCEEDCCCCFDCAEDCACBCEBBCFEACCFBDCACDCBCEBDBBCFEEDCCCFAFEACECCCECAEEDCADCBEDC7BEBCCCFBAFDCECCFBEAACA	MC:Z:151M	MD:Z:151	PG:Z:MarkDuplicates	RG:Z:sample	NM:i:0	AS:i:151	XS:i:151
: A00853:477:HMLWYDSX3:1:1413:4390:28573	83	NC_000022.11	42212883	0	151M	=	42212845	-189	ATGCTTTTGTTCAGAAGGCCCCTGCTAGCTAATTCATTGGTTTGACTAACCAAGACATTGGGCCTTGTGGTTCCTTCTAGGCTACCAGCCATCCCCTGATGCTCTTGAGTACTCCTAGAATATCTCCTGTCAGGGTGGTGGTGGTAACCCT	AADECCCBDCBFCE<?CDEEEEBDEACDEAC;:BFBCBCDCCBEAEACAEFCCEAFBCBCCDEECBDBCECBEECCEACDEEBBFGDEFGCCFFFFCFCCEFBFDCFCDAAEBEE:CECBABBEBEE;DBFCCCDBCDBCCBBC?@BEEDA	MC:Z:151M	MD:Z:151	PG:Z:MarkDuplicates	RG:Z:sample	NM:i:0	AS:i:151	XS:i:151

#+begin_src sh
samtools view data/mapped/63003856_chr22.bam | rg "A00853:477:HMLWYDSX3:1:1413:4390:28573"
#+end_src

#+RESULTS:
: A00853:477:HMLWYDSX3:1:1413:4390:28573	163	NW_014040930.1	115017	0	151M	=	115055	189	CCCAGGGGCCCCAGTGGGGATTTTCTAATAGAGACCCAATGCTTTTGTTCAGAAGGCCCCTGCTAGCTAATTCATTGGTTTGACTAACCAAGACATTGGGCCTTGTGGTTCCTTCTAGGCTACCAGCCATCCCCTGATGCTCTTGAGTACT	ACC+FBCDCBBBAEAEDEEBBCCCECACBAEBEBDCCBCBFDCCCCFACEBEBCEEDCCCCFDCAEDCACBCEBBCFEACCFBDCACDCBCEBDBBCFEEDCCCFAFEACECCCECAEEDCADCBEDC7BEBCCCFBAFDCECCFBEAACA	NM:i:0	MD:Z:151	MC:Z:151M	AS:i:151	XS:i:151	RG:Z:sample
: A00853:477:HMLWYDSX3:1:1413:4390:28573	83	NW_014040930.1	115055	0	151M	=	115017	-189	ATGCTTTTGTTCAGAAGGCCCCTGCTAGCTAATTCATTGGTTTGACTAACCAAGACATTGGGCCTTGTGGTTCCTTCTAGGCTACCAGCCATCCCCTGATGCTCTTGAGTACTCCTAGAATATCTCCTGTCAGGGTGGTGGTGGTAACCCT	AADECCCBDCBFCE<?CDEEEEBDEACDEAC;:BFBCBCDCCBEAEACAEFCCEAFBCBCCDEECBDBCECBEECCEACDEEBBFGDEFGCCFFFFCFCCEFBFDCFCDAAEBEE:CECBABBEBEE;DBFCCCDBCDBCCBBC?@BEEDA	NM:i:0	MD:Z:151	MC:Z:151M	AS:i:151	XS:i:151	RG:Z:sample

Effectivement, on aligne sur une zonne supprimée !
******** DONE Corriger la qualité: non
CLOSED: [2023-05-24 Wed 22:19]
********* DONE Comparaison avec le fastq de référénce : qualité !!
CLOSED: [2023-05-24 Wed 22:17]
#+begin_src sh
cd /Work/Users/apraga/bisonex/work/6e/8548fc90263830bf677f36585f11dc
zgrep -A 3 "A00853:477:HMLWYDSX3:1:1413:4390:28573" 63003856_chr22_1.fq.gz
#+end_src
@A00853:477:HMLWYDSX3:1:1413:4390:28573
AGGGTTACCACCACCACCCTGACAGGAGATATTCTAGGAGTACTCAAGAGCATCAGGGGATGGCTGGTAGCCTAGAAGGAACCACAAGGCCCAATGTCTTGGTTAGTCAAACCAATGAATTAGCTAGCAGGGGCCTTCTGAACAAAAGCAT
+
ADEEB@?CBBCCBDCBDCCCFBD;EEBEBBABCEC:EEBEAADCFCDFBFECCFCFFFFCCGFEDGFBBEEDCAECCEEBCECBDBCEEDCCBCBFAECCFEACAEAEBCCDCBCBFB:;CAEDCAEDBEEEEDC?<ECFBCDBCCCEDAA

#+begin_src
zgrep -A 3 "A00853:477:HMLWYDSX3:1:1413:4390:28573" /Work/Projects/bisonex/cento/fastq/2200467051_63003856/63003856_S135_R1_001.fastq.gz
#+end_src
AGGGTTACCACCACCACCCTGACAGGAGATATTCTAGGAGTACTCAAGAGCATCAGGGGATGGCTGGTAGCCTAGAAGGAACCACAAGGCCCAATGTCTTGGTTAGTCAAACCAATGAATTAGCTAGCAGGGGCCTTCTGAACAAAAGCAT
: +
: FFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF::FFFFFFFFFFFFFFF::FFFFFFFFFFFFFF
********* DONE Regarder la qualité après bwa mem vs applybqsr: différente
CLOSED: [2023-05-24 Wed 22:18]
Sur le mésocentre, dans /Work/Users/apraga/bisonex/out/63003856_S135_R/preprocessing
$ samtools view mapped/63003856_S135_R.bam NC_000022.11 | rg "A00853:477:HMLWYDSX3:1:1413:4390:28573"
A00853:477:HMLWYDSX3:1:1413:4390:28573	163	NC_000022.11	42212845	0	151M	=	42212883	189	CCCAGGGGCCCCAGTGGGGATTTTCTAATAGAGACCCAATGCTTTTGTTCAGAAGGCCCCTGCTAGCTAATTCATTGGTTTGACTAACCAAGACATTGGGCCTTGTGGTTCCTTCTAGGCTACCAGCCATCCCCTGATGCTCTTGAGTACT	FFF,FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFF	NM:i:0	MD:Z:151	MC:Z:151M	AS:i:151	XS:i:151	RG:Z:sample
A00853:477:HMLWYDSX3:1:1413:4390:28573	83	NC_000022.11	42212883	0	151M	=	42212845	-189	ATGCTTTTGTTCAGAAGGCCCCTGCTAGCTAATTCATTGGTTTGACTAACCAAGACATTGGGCCTTGTGGTTCCTTCTAGGCTACCAGCCATCCCCTGATGCTCTTGAGTACTCCTAGAATATCTCCTGTCAGGGTGGTGGTGGTAACCCT	FFFFFFFFFFFFFF::FFFFFFFFFFFFFFF::FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFF	NM:i:0	MD:Z:151	MC:Z:151M	AS:i:151	XS:i:151	RG:Z:sample

samtools view applybqsr/63003856_S135_R.bam NC_000022.11 | rg "A00853:477:HMLWYDSX3:1:1413:4390:28573"
A00853:477:HMLWYDSX3:1:1413:4390:28573	163	NC_000022.11	42212845	0	151M	=	42212883	189	CCCAGGGGCCCCAGTGGGGATTTTCTAATAGAGACCCAATGCTTTTGTTCAGAAGGCCCCTGCTAGCTAATTCATTGGTTTGACTAACCAAGACATTGGGCCTTGTGGTTCCTTCTAGGCTACCAGCCATCCCCTGATGCTCTTGAGTACT	ACC+FBCDCBBBAEAEDEEBBCCCECACBAEBEBDCCBCBFDCCCCFACEBEBCEEDCCCCFDCAEDCACBCEBBCFEACCFBDCACDCBCEBDBBCFEEDCCCFAFEACECCCECAEEDCADCBEDC7BEBCCCFBAFDCECCFBEAACA	MC:Z:151M	MD:Z:151	PG:Z:MarkDuplicates	RG:Z:sample	NM:i:0	AS:i:151	XS:i:151
A00853:477:HMLWYDSX3:1:1413:4390:28573	83	NC_000022.11	42212883	0	151M	=	42212845	-189	ATGCTTTTGTTCAGAAGGCCCCTGCTAGCTAATTCATTGGTTTGACTAACCAAGACATTGGGCCTTGTGGTTCCTTCTAGGCTACCAGCCATCCCCTGATGCTCTTGAGTACTCCTAGAATATCTCCTGTCAGGGTGGTGGTGGTAACCCT	AADECCCBDCBFCE<?CDEEEEBDEACDEAC;:BFBCBCDCCBEAEACAEFCCEAFBCBCCDEECBDBCECBEECCEACDEEBBFGDEFGCCFFFFCFCCEFBFDCFCDAAEBEE:CECBABBEBEE;DBFCCCDBCDBCCBBC?@BEEDA	MC:Z:151M	MD:Z:151	PG:Z:MarkDuplicates	RG:Z:sample	NM:i:0	AS:i:151	XS:i:151
********* DONE Réaligner à partir de la sortie de bwa mem
CLOSED: [2023-05-24 Wed 22:32]
#+begin_src sh
cd out/63003856_S135_R/preprocessing/mapped/
samtools view 63003856_S135_R.bam NC_000022.11  -f 0x2 -o 63003856_chr22.bam
samtools sort -n 63003856_chr22.bam -o 63003856_chr22_sorted.bam
samtools fastq -1 63003856_chr22_1.fq.gz -2 63003856_chr22_2.fq.gz -0 /dev/null -s /dev/null -n 63003856_chr22_sorted.bam
#+end_src

ON vérifie la qualité
#+begin_src
zgrep -A 3 "A00853:477:HMLWYDSX3:1:1413:4390:28573" 63003856_chr22_1.fq.gz
#+end_src

#+RESULTS:
: @A00853:477:HMLWYDSX3:1:1413:4390:28573
: AGGGTTACCACCACCACCCTGACAGGAGATATTCTAGGAGTACTCAAGAGCATCAGGGGATGGCTGGTAGCCTAGAAGGAACCACAAGGCCCAATGTCTTGGTTAGTCAAACCAATGAATTAGCTAGCAGGGGCCTTCTGAACAAAAGCAT
: +
: FFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF::FFFFFFFFFFFFFFF::FFFFFFFFFFFFFF


#+begin_src
NXF_OPTS=-D"user.name=apraga" nextflow run main.nf -c nextflow.config  -profile standard,helios -resume  --input="out/63003856_S135_R/preprocessing/mapped/63003856_chr22_{1,2}.fq.gz" --outdir=out/63003856_chr22-from-mapped
#+end_src
Puis ::
#+begin_src
cd /Work/Users/apraga/bisonex/out/63003856_chr22-from-mapped/63003856_chr22/preprocessing/mapped
samtools view 63003856_chr22.bam | rg "A00853:477:HMLWYDSX3:1:1413:4390:28573"
#+end_src

#+RESULTS:
: A00853:477:HMLWYDSX3:1:1413:4390:28573	163	NW_014040930.1	115017	0	151M	=	115055	189	CCCAGGGGCCCCAGTGGGGATTTTCTAATAGAGACCCAATGCTTTTGTTCAGAAGGCCCCTGCTAGCTAATTCATTGGTTTGACTAACCAAGACATTGGGCCTTGTGGTTCCTTCTAGGCTACCAGCCATCCCCTGATGCTCTTGAGTACT	FFF,FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFF	NM:i:0	MD:Z:151	MC:Z:151M	AS:i:151	XS:i:151	RG:Z:sample
: A00853:477:HMLWYDSX3:1:1413:4390:28573	83	NW_014040930.1	115055	0	151M	=	115017	-189	ATGCTTTTGTTCAGAAGGCCCCTGCTAGCTAATTCATTGGTTTGACTAACCAAGACATTGGGCCTTGTGGTTCCTTCTAGGCTACCAGCCATCCCCTGATGCTCTTGAGTACTCCTAGAATATCTCCTGTCAGGGTGGTGGTGGTAACCCT	FFFFFFFFFFFFFF::FFFFFFFFFFFFFFF::FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFF	NM:i:0	MD:Z:151	MC:Z:151M	AS:i:151	XS:i:151	RG:Z:sample
******** DONE Aligner sur génome de référence limité au chromosome 22
CLOSED: [2023-05-24 Wed 23:18]
********* KILL Test données non modifiées
CLOSED: [2023-05-24 Wed 23:18]
/Work/Users/apraga/bisonex/tests/bamscissors
#+begin_src
cd /Work/Groups/bisonex/data/genome/GRCh38.p13/
mkdir chr22/
samtools faidx genomeRef.fna NC_000022.11 > chr22/chr22.fna
cd chr22
samtools faidx chr22.fna
bwa index chr22.fna
#+end_src

#+begin_src
cd /Work/Users/apraga/bisonex/tests/bamscissors
ln -s ../   ../out/63003856_S135_R/preprocessing/applybqsr/63003856_chr22_{1,2}.fq.gz .

srun -c 24 -p smp -t 1:00:00 --pty bash
bwa mem -t 24 /Work/Projects/bisonex/data/genome/GRCh38.p13/chr22/chr22.fna 63003856_chr22_1.fq.gz 63003856_chr22_1.fq.gz -o smallref.sam
#+end_src
********* DONE Test données modifiées: ok
CLOSED: [2023-05-24 Wed 23:18]
Données dans data/init
#+begin_src sh
time julia insertVariant.jl
rsync -avz data/init/*.fq.gz meso:/Work/Users/apraga/bisonex/tests/bamscissors/
#+end_src
#+begin_src
srun -c 24 -p smp -t 1:00:00 --pty bash
bwa mem -t 24 /Work/Projects/bisonex/data/genome/GRCh38.p13/chr22/chr22.fna 63003856_chr22_1.fq.gz 63003856_chr22_1.fq.gz | samtools sort -@24 - -o smallref.bam
#+end_src

#+begin_src
rsync -avz meso:/Work/Users/apraga/bisonex/tests/bamscissors/smallref.bam mapped/
#+end_src
******* DONE Test haplotypecaller 1 variant
CLOSED: [2023-05-29 Mon 15:38]
******* DONE Test haplotypecaller tous les variants
******* DONE Comprendre pourquoi la répartiton ne suit pas la loi normale
CLOSED: [2023-06-01 Thu 21:44]
Certains hétérozygote soint à 0.01 ou 1...
******** DONE augmenter le nombre d'échantillions: idem
CLOSED: [2023-05-31 Wed 22:24]
******** DONE Vérifier le nombre de reads marqué vs édité
CLOSED: [2023-06-01 Thu 21:44]
******** DONE vérifier que 100 appel à rand(d, 1)[1] est semblable à un appel de rand(d, 100)
CLOSED: [2023-05-31 Wed 22:24]

julia> df = vcat(DataFrame(:y => z, :type => "z"), DataFrame(:y => y, :type => "y"));
julia> y = [rand(d, 1)[1] for x in 1:1000];
julia> z = rand(d,1000);
julia> df = vcat(DataFrame(:y => z, :type => "z"), DataFrame(:y => y, :type => "y"));
 draw(data(df)*histogram(bins=100)*mapping(:y, color=:type,dodge=:type))
******* DONE Améliorer les performances
CLOSED: [2023-06-02 Fri 23:39]
#+begin_src julia
@time include("xamscissors.jl")
#+end_src

430s pour chromosome 22. Majorité dans l'édition de reads:
******** DONE Inserér tous les variants d'un reads d'un coup
CLOSED: [2023-06-01 Thu 23:09]
Ne change rien
******** DONE Test avec -t4: idem
CLOSED: [2023-06-01 Thu 23:17]
******** DONE Test mésocentre : idem
CLOSED: [2023-06-01 Thu 23:40]
348s
******** Changer la structure de données des
Dataframe -> dict = les performances horribles ont disparuse
******* KILL Refaire le test avec la nouvelle version
CLOSED: [2023-06-12 Mon 23:27]
******** DONE Génération des données
CLOSED: [2023-06-02 Fri 23:40]
Mésocentre
#+begin_src sh
cd /Work/Users/apraga/bisonex/out/63003856_S135_R/preprocessing/mapped
samtools view 63003856_S135_R.bam NC_000022.11 -o 63003856_S135_R_chr22.bam
samtools index 63003856_S135_R_chr22.bam
cp 63003856_S135_R_chr22.bam* /Work/Users/apraga/bisonex/tests/xamscissors/
cd /Work/Users/apraga/bisonex/tests/xamscissors
#+end_src
On génère les données
#+begin_src julia
using XAMScissors

insertSNV("./63003856_S135_R_chr22.bam", "snvs_chr22.csv", "out")
#+end_src
Puis
#+begin_src sh
julia xamscissors.jl
#+end_src
******** DONE Run
CLOSED: [2023-06-03 Sat 18:26]
NXF_OPTS=-D"user.name=${USER}" nextflow run workflows/runInserted.nf -profile standard,helios  --input="tests/xamscissors/out/inserted_{1,2}.fq.gz"
******** DONE Après haplotypecaller : ok
CLOSED: [2023-06-03 Sat 18:27] SCHEDULED: <2023-06-03 Sat>
******** KILL Après filtre vep
CLOSED: [2023-06-12 Mon 23:27] SCHEDULED: <2023-06-03 Sat>
****** KILL PHase 3 : tous les SNV
CLOSED: [2023-06-12 Mon 23:28] SCHEDULED: <2023-06-03 Sat>
******* DONE Générer les données
CLOSED: [2023-06-03 Sat 20:16] SCHEDULED: <2023-06-03 Sat>
#+begin_src julia
using XAMScissors

insertSNV("../../out/63003856_S135_R/preprocessing/mapped/63003856_S135_R.bam", "snvs.csv", "out")
#+end_src

temps d'exécution 73min
#+begin_src sh
nohup bash -c 'time julia xamscissors.jl' &
xamscissors-63003856/*.fq.gz /Work/Groups/bisonex/data/xamscissors/
#+end_src
******* DONE Regénérer avec @time pour avoir les performaces
CLOSED: [2023-06-03 Sat 21:45]

markReads 6.265202 seconds (1.36 M allocations: 137.090 MiB, 1.00% gc time, 9.79% compilation time)
editReads 1327.701623 seconds (1.03 G allocations: 81.996 GiB, 0.59% gc time, 0.03% compilation time)
samtools index 117.743727 seconds (53 allocations: 1.883 KiB)
samtools sort 2820.074930 seconds (66 allocations: 2.789 KiB)
bam2fastq 134.148952 seconds (794 allocations: 40.539 KiB, 0.01% compilation time)

real	73m33.273s
user	77m38.194s
sys	1m26.684s
[bam_sort_core] merging from 60 files and 1 in-memory blocks...
[M::bam2fq_mainloop] discarded 0 singletons
[M::bam2fq_mainloop] processed 126905130 reads

real	73m6.934s
user	77m25.397s
sys	1m21.339s

******* KILL Après haplotypecaller 556/590, majorité = échec alignement
CLOSED: [2023-06-12 Mon 23:27] SCHEDULED: <2023-06-04 Sun>
Haplotypecaller 556 found over 590
Amongst 34 missed variant, 2 have a mapping quality > 0
2×7 DataFrame
 Row │ chrom         pos        ref      alt      zygosity  meanQual   stdQual
     │ String15      Int64      String1  String1  String7   Float64    Float64
─────┼─────────────────────────────────────────────────────────────────────────
   1 │ NC_000017.11   39672244  G        A        het       60.0       0.0
   2 │ NC_000001.11  155235252  A        G        het        0.258065  2.48868

NC_000017.11   39672244  G        A        het => ok, problème de représentation car 2 variant côte à cote
NC_000001.11  155235252  A        G        het => peu de reads alternatifs (9/93 donc ok)

Position: chromoe 1 et 6 surtout
34×7 DataFrame
 Row │ chrom         pos        ref      alt      zygosity
     │ String15      Int64      String1  String1  String7
─────┼──────────────────────────────────────────────────────
   1 │ NC_000001.11  153817496  A        T        het
   2 │ NC_000001.11  155235252  A        G        het
   3 │ NC_000001.11  155236268  G        A        het
   4 │ NC_000001.11  155290591  C        T        het
   5 │ NC_000001.11  155291918  G        A        het
   6 │ NC_000001.11  155294358  G        T        het
   7 │ NC_000002.12  149010343  C        T        het
   8 │ NC_000006.12   32039426  T        A        het
   9 │ NC_000006.12   32040110  G        T        het
  10 │ NC_000006.12   32040723  G        A        het
  11 │ NC_000006.12   32041006  C        T        het
  12 │ NC_000006.12   32041147  G        A        het
  13 │ NC_000006.12   33443054  G        T        het
  14 │ NC_000006.12   33451815  C        T        het
  15 │ NC_000006.12  170283230  C        A        het
  16 │ NC_000006.12  170283754  G        A        het
  17 │ NC_000006.12  170285637  T        C        het
  18 │ NC_000006.12  170289678  A        C        het
  19 │ NC_000010.11   87961118  G        A        het
  20 │ NC_000012.12    2449086  C        G        het
  21 │ NC_000015.10   74343027  C        T        het
  22 │ NC_000016.10   16163078  G        A        het
  23 │ NC_000016.10   21262032  C        G        het
  24 │ NC_000016.10     21962506  C        T        homo
  25 │ NC_000017.11    7513122  C        T        het
  26 │ NC_000017.11    7513752  C        T        het
  27 │ NC_000017.11   39672244  G        A        het
  28 │ NC_000017.11   46018710  C        T        het
  29 │ NC_000019.10   54144058  G        A        het
  30 │ NC_000021.9    43063074  A        G        het
  31 │ NC_000021.9    43426167  C        T        het
  32 │ NC_000022.11   18918421  A        G        het
  33 │ NC_000022.11   42087168  T        A        homo
  34 │ NC_000022.11   42213078  T        G        het
******* DONE Voir où est l'alignement alternatif: sur NW_ (zone supprimée)
CLOSED: [2023-06-04 Sun 22:15] SCHEDULED: <2023-06-04 Sun>
ex chr15 74343027
  A00853:477:HMLWYDSX3:2:2444:22354:28870

#+begin_src
cd /Work/Groups/bisonex/data/xamscissors
zgrep -A4 "A00853:477:HMLWYDSX3:2:2444:22354:28870" *.fq.gz
 #+end_src
63003856_xamscissors_1.fq.gz:@A00853:477:HMLWYDSX3:2:2444:22354:28870
63003856_xamscissors_1.fq.gz:CACCGTGTCCACCCCTCCTGCCGGCATCTCTGTGACGTTGGCCTTGATGTCCTTGAAGGACATCTTGCTGTCTCCCAGGAGTCTGTAGAGGATGCCACGGTAATCGTGGTGAACACTTCCTTTCTGTC
63003856_xamscissors_1.fq.gz:+
63003856_xamscissors_1.fq.gz:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFF:FFFFFFFFFF::FFFFFFFFFFF:FFFFFFFFFFFFFF:FFFFFFF,FFFFFF,FFFFFFFFFFFF:FF::FF

63003856_xamscissors_2.fq.gz:@A00853:477:HMLWYDSX3:2:2444:22354:28870
63003856_xamscissors_2.fq.gz:GACAGAAAGGAAGTGTTCACCACGATTACCGTGGCATCCTCTACAGACTCCTGGGAGACAGCAAGATGTCCTTCGAGGACATCAAGGCCAACGTCACAGAGATGCCGGCAGGAGGGGTGGACACGGTG
63003856_xamscissors_2.fq.gz:+
63003856_xamscissors_2.fq.gz:FF:FFF:FF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFF:F:FF:FFFFFFFFFFFFFF:FFFFFFFFFFFFFFFF,:FFF,FFFFFF:FFFFFFFFFFFFF

******** DONE Avec BLAT: sur _fix
CLOSED: [2023-06-04 Sun 21:07]
1er =

   ACTIONS      QUERY   SCORE START   END QSIZE IDENTITY  CHROM                 STRAND  START       END   SPAN
--------------------------------------------------------------------------------------------------------------
browser details YourSeq   124     1   128   128    98.5%  chr15_ML143370v1_fix  +      172243    172370    128   What is chrom_fix?
browser details YourSeq   124     1   128   128    98.5%  chr15                 +    74342974  74343101    128
browser details YourSeq    23     1    25   128    96.0%  chr19                 -    33396097  33396121     25

Second
--------------------------------------------------------------------------------------------------------------
browser details YourSeq   126     1   128   128    99.3%  chr15_ML143370v1_fix  -      172243    172370    128   What is chrom_fix?
browser details YourSeq   126     1   128   128    99.3%  chr15                 -    74342974  74343101    128
browser details YourSeq    23   104   128   128    96.0%  chr19                 +    33396097  33396121     25

******** DONE Bwa mem à la main GRCh38.p13 : on est dans une zone NW
CLOSED: [2023-06-04 Sun 21:51]
On met les 2 reads dans des fichiers séparés puis
#+begin_src sh
cd /Work/Users/apraga/bisonex/tests/xamscissors/align
bwa mem /Work/Groups/bisonex/data/genome/GRCh38.p13/bwa/genomeRef test1.fq test2.fq
#+end_src

A00853:477:HMLWYDSX3:2:2444:22354:28870	97	NW_021160016.1	172243	0	128M	=	172243	128	CACCGTGTCCACCCCTCCTGCCGGCATCTCTGTGACGTTGGCCTTGATGTCCTTGAAGGACATCTTGCTGTCTCCCAGGAGTCTGTAGAGGATGCCACGGTAATCGTGGTGAACACTTCCTTTCTGTC	FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFF:FFFFFFFFFF::FFFFFFFFFFF:FFFFFFFFFFFFFF:FFFFFFF,FFFFFF,FFFFFFFFFFFF:FF::FF	NM:i:2	MD:Z:22A30C7MC:Z:128M	AS:i:118	XS:i:118	XA:Z:NC_000015.10,+74342974,128M,2;
A00853:477:HMLWYDSX3:2:2444:22354:28870	145	NW_021160016.1	172243	0	128M	=	172243	-128	CACCGTGTCCACCCCTCCTGCCGGCATCTCTGTGACGTTGGCCTTGATGTCCTCGAAGGACATCTTGCTGTCTCCCAGGAGTCTGTAGAGGATGCCACGGTAATCGTGGTGAACACTTCCTTTCTGTC	FFFFFFFFFFFFF:FFFFFF,FFF:,FFFFFFFFFFFFFFFF:FFFFFFFFFFFFFF:FF:F:FFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FF:FFF:FF	NM:i:1	MD:Z:22A105	MC:Z:128M	AS:i:123	XS:i:123	XA:Z:NC_000015.10,-74342974,128M,1;

******** DONE GRCh38.p14: idem
CLOSED: [2023-06-04 Sun 21:51]
A00853:477:HMLWYDSX3:2:2444:22354:28870	97	NW_021160016.1	172243	0	128M	=	172243	128	CACCGTGTCCACCCCTCCTGCCGGCATCTCTGTGACGTTGGCCTTGATGTCCTTGAAGGACATCTTGCTGTCTCCCAGGAGTCTGTAGAGGATGCCACGGTAATCGTGGTGAACACTTCCTTTCTGTC	FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFF:FFFFFFFFFF::FFFFFFFFFFF:FFFFFFFFFFFFFF:FFFFFFF,FFFFFF,FFFFFFFFFFFF:FF::FF	NM:i:2	MD:Z:22A30C7MC:Z:128M	AS:i:118	XS:i:118	XA:Z:NC_000015.10,+74342974,128M,2;
A00853:477:HMLWYDSX3:2:2444:22354:28870	145	NW_021160016.1	172243	0	128M	=	172243	-128	CACCGTGTCCACCCCTCCTGCCGGCATCTCTGTGACGTTGGCCTTGATGTCCTCGAAGGACATCTTGCTGTCTCCCAGGAGTCTGTAGAGGATGCCACGGTAATCGTGGTGAACACTTCCTTTCTGTC	FFFFFFFFFFFFF:FFFFFF,FFF:,FFFFFFFFFFFFFFFF:FFFFFFFFFFFFFF:FF:F:FFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FF:FFF:FF	NM:i:1	MD:Z:22A105	MC:Z:128M	AS:i:123	XS:i:123	XA:Z:NC_000015.10,-74342974,128M,1;

******** DONE GRCh38 : ok
CLOSED: [2023-06-04 Sun 22:15]
 bwa mem /Work/Projects/bisonex/data/genome/GRCh38/GCA_000001405.15_GRCh38_full_analysis_set.fna test1.fq test2.fq
******* DONE Vérifier que les reads ont la même qualité sur les fichiers d'origine: oui
CLOSED: [2023-06-04 Sun 21:07]
******* DONE Supprimer les NW_ ?
CLOSED: [2023-06-10 Sat 10:40] SCHEDULED: <2023-06-04 Sun>


@A00853:477:HMLWYDSX3:3:2114:14742:8860
CAGGCCAGCCGCTCAGCCCGCTCCTTTCACCCTCTGCAGGAGAGCCTCGTGGCAGGCCAGTGGAGGGACATGATGGACTACATGCTCCAAGGGGTGGCGCAGCCGAGCATGGAAGAGGGCTCTGGACAGCTCCTGGAAGGGCACTTGCAC
+
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF

@A00853:477:HMLWYDSX3:3:2114:14742:8860
CTTTTGCTTGTCCCCAGGACGCACCTCAGGGTGGTGAAGCAAAAAAACCACGGCCCAGGAGAGGGTGGGTGCTGTGGTCTCAGTGCCACCGATCAGGAGGTCCACTGCAGCCATGTGCAAGTGCCCTTCCAGGAGCTGTCCAGAGCCCTCT
+
FFFFFFFFFFFFFFFFFFFFFFF:FFF:FFFFFFFFFFFFF,FFFFFFFFFFFF:F:FFFF:FFFFF,,FFF:FFFFFFFFFF,FFFFFFF,FFFFFFFFFFF,FFFFFFFFF:FFFF,F:FFFFF:FFFFFFFFF:FFFF,FFFFFFFFF
******* DONE Supprimer NW_ et NT_
****** KILL Phase 2 : chr22, vaf variable :T2T:
CLOSED: [2023-08-12 Sat 15:59]
****** KILL Phase 3 : tous SNV, vaf variable :T2T:
CLOSED: [2023-08-12 Sat 15:59]
***** KILL Test Indel
CLOSED: [2023-08-12 Sat 15:59]
**** Divers
***** DONE Vérifier nombre de reads fastq - bam
CLOSED: [2022-10-09 Sun 22:31]
*** KILL Liste varants "clinically relevent" (Clinge - CT-R d)
CLOSED: [2023-06-25 Sun 15:53] SCHEDULED: <2023-06-25 Sun>
[cite:@wilcox2021]
Vu avec alexis: pas notre cas d'usage
*** TODO Variant cento confirmé en Sanger :xamscissors:
**** DONE Générer la liste
CLOSED: [2023-08-13 Sun 22:26] SCHEDULED: <2023-08-13 Sun 12:00>
**** DONE 2 variants chr20 dans HG002 avec XAMscissors: homozygote
CLOSED: [2023-08-15 Tue 00:13] SCHEDULED: <2023-08-13 Sun 13:00>
***** DONE Insertion
CLOSED: [2023-08-13 Sun 23:31] SCHEDULED: <2023-08-13 Sun>
On récupère le BAM
#+begin_src sh :dir /home/alex/code/bisonex/out
scp /Work/Users/apraga/bisonex/out/HG002-151002_7001448_0359_AC7F6GANXX-GRCh38/preprocessing/mapped/HG002-151002_7001448_0359_AC7F6GANXX-GRCh38.bam
#+end_src
On formate les données et on génère les fastq
#+begin_src sh :dir /home/alex/roam/research/bisonex/code/sanger
julia --project=.. xamscissors.jl
#+end_src
On upload
#+begin_src sh :dir /home/alex/roam/research/bisonex/code/sanger
cd out
mv inserted_1.fq.gz HG002-sanger-inserted-chr20_1.fq.gz
mv inserted_2.fq.gz HG002-sanger-inserted-chr20_2.fq.gz
#+end_src
On upload
#+begin_src sh :dir /home/alex/roam/research/bisonex/code/sanger
scp HG002-sanger-inserted-chr20_*.fq.gz meso:/Work/Users/apraga/bisonex/data/
#+end_src
Fichier de configuration

patient,sample,fastq1,fastq2
HG002,sanger-chr20,data/HG002-sanger-inserted-chr20_1.fq.gz,data/HG002-sanger-inserted-chr20_2.fq.gz

On lance la simulation
#+begin_src
nextflow run main.nf -profile standard,helios --input=samples-synthetic.csv --genome=GRCh38  -bg
#+end_src
***** DONE Vérifier post alignement
CLOSED: [2023-08-13 Sun 23:44] SCHEDULED: <2023-08-13 Sun>
***** DONE Vérifier haplotypecaller
CLOSED: [2023-08-15 Tue 00:13] SCHEDULED: <2023-08-13 Sun>
***** DONE Vérifier post-filtre
CLOSED: [2023-08-15 Tue 00:13]
**** DONE 2 variant chr20 dans HG002 heterozygote
CLOSED: [2023-08-15 Tue 13:45] SCHEDULED: <2023-08-15 Tue>
La zygosité n'était pas correcte...
**** DONE chr20 hétérozygote 2 variants  avec XAMscissors à jour
CLOSED: [2023-08-19 Sat 20:45] SCHEDULED: <2023-08-19 Sat>
***** DONE Insertion
CLOSED: [2023-08-19 Sat 20:08]
XAMscissors avec dépendence git
NA12878 après bwa mem
Sur le mésocentre, dans code/sanger/
#+begin_src
ln -s  /Work/Users/apraga/bisonex/out/2300346867_NA12878-63118093_S260-full-GRCh38/preprocessing/mapped/2300346867_NA12878-63118093_S260-GRCh38.bam
julia --project=.
include("xamscissors.jl")
#+end_src
Renommer out-chr20/inserted -> data/data/NA12878-sanger-inserted-chr20

On lance le calcul avec samples-synthetic.csv
NA12878,sanger-chr20,data/NA12878-sanger-inserted-chr20_1.fq.gz,data/NA12878-sanger-inserted-chr20_2.fq.gz
#+begin_src
 nextflow run main.nf -profile standard,helios --input=samples-synthetic.csv --genome=GRCh38  -bg
#+end_src
***** DONE Vérifier après appel de variant
CLOSED: [2023-08-19 Sat 20:42] SCHEDULED: <2023-08-19 Sat>
#+begin_src
cd /Work/Users/apraga/bisonex/out/NA12878-sanger-chr20-GRCh38/callVariant/haplotypecaller
zgrep "46043645\|62334188" NA12878-sanger-chr20-GRCh38.vcf.gz
#+end_src

#+RESULTS:
: chr20   46043645        rs1332625463    G       A       933.64  .       AC=1;AF=0.500;AN=2;BaseQRankSum=-1.624;DB;DP=96;ExcessHet=0.0000;FS=1.654;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=9.83;ReadPosRankSum=1.833;SOR=0.516  GT:AD:DP:GQ:PL 0/1:49,46:95:99:941,0,1103
: chr20   62334188        .       G       A       910.64  .       AC=1;AF=0.500;AN=2;BaseQRankSum=-0.975;DP=87;ExcessHet=0.0000;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=10.59;ReadPosRankSum=-1.566;SOR=0.685   GT:AD:DP:GQ:PL0/1:39,47:86:99:918,0,757
***** DONE Vérifier après filtre depth
CLOSED: [2023-08-19 Sat 20:44] SCHEDULED: <2023-08-19 Sat>
#+begin_src sh
cd /Work/Users/apraga/bisonex/out/NA12878-sanger-chr20-GRCh38/callVariant
 zgrep "46043645\|62334188" filter-depth.vcf
#+end_src

#+RESULTS:
: chr20   46043645        rs1332625463    G       A       933.64  PASS    AC=1;AF=0.5;AN=2;BaseQRankSum=-1.624;DB;DP=96;ExcessHet=0;FS=1.654;MLEAC=1;MLEAF=0.5;MQ=60;MQRankSum=0;QD=9.83;ReadPosRankSum=1.833;SOR=0.516  GT:AD:DP:GQ:PL  0/1:49,46:95:99:941,0,1103
: chr20   62334188        .       G       A       910.64  PASS    AC=1;AF=0.5;AN=2;BaseQRankSum=-0.975;DP=87;ExcessHet=0;FS=0;MLEAC=1;MLEAF=0.5;MQ=60;MQRankSum=0;Q
***** DONE Vérifier après filtre polymorphisme
CLOSED: [2023-08-19 Sat 20:44] SCHEDULED: <2023-08-19 Sat>
 zgrep "46043645\|62334188" filter-polymorphisms.vcf
chr20   46043645        rs1332625463    G       A       933.64  PASS    .       GT:AD:DP:GQ:PL  0/1:49,46:95:99:941,0,1103
chr20   62334188        .       G       A       910.64  PASS    .       GT:AD:DP:GQ:PL  0/1:39,47:86:99:918,0,757
***** DONE vérifier après filter vep
CLOSED: [2023-08-19 Sat 20:45]
#+begin_src sh
cd /Work/Users/apraga/bisonex/out/NA12878-sanger-chr20-GRCh38/annotate
zgrep "46043645\|62334188" postvep-filter.tsv
#+end_src

#+RESULTS:
: rs1332625463    chr20:46043645  A       ENSG00000124140 ENST00000243964 Transcript      missense_variant        1380  1250     417     G/D     gGt/gAt rs1332625463    G       MODERATE        -       1       -       YES     -       Ensembl-       1       probably_damaging(0.981)        ENST00000243964.7:c.1250G>A     ENSP00000243964.4:p.Gly417Asp   -     chr20:g.46043645G>A      0       0       0       0       0       0       0       0       0       -       -       -     --       -       -       rs1332625463    NTR     6.8     [03.77%-11.14%]
: chr20_62334188_G/A      chr20:62334188  A       ENSG00000130702 ENST00000252999 Transcript      stop_gained,splice_region_variant      2804    2737    913     Q/*     Cag/Tag COSV99440955    G       HIGH    -       -1      -       YES   -Ensembl -       1       -       ENST00000252999.7:c.2737C>T     ENSP00000252999.3:p.Gln913Ter   -       chr20:g.62334188G>A    -       -       -       -       -       -       -       -       -       -       1       1       -       -     --       chr20:62334188-62334188 Alter by SPiCE + Alter ESR      69.33   [61.29%-76.59%]
**** KILL [#B] Tout insérer dans HG002 avec XAMscissors: erreur patient..
CLOSED: [2023-08-16 Wed 20:21] SCHEDULED: <2023-08-14 Mon>
***** DONE Insertion
CLOSED: [2023-08-16 Wed 20:21]
On récupère le BAM
#+begin_src sh :dir /home/alex/code/bisonex/out
scp /Work/Users/apraga/bisonex/out/HG002-151002_7001448_0359_AC7F6GANXX-GRCh38/preprocessing/mapped/HG002-151002_7001448_0359_AC7F6GANXX-GRCh38.bam
#+end_src
#+begin_src sh :dir /home/alex/roam/research/bisonex/code/sanger
julia --project=.. xamscissors.jl
cd out-all
mv inserted_1.fq.gz HG002-sanger-inserted-_1.fq.gz
mv inserted_2.fq.gz HG002-sanger-inserted-_2.fq.gz
scp HG002-sanger-inserted-all_*.fq.gz meso:/Work/Users/apraga/bisonex/data/
#+end_src
Fichier de configuration

patient,sample,fastq1,fastq2
HG002,sanger-all,data/HG002-sanger-inserted-all_1.fq.gz,data/HG002-sanger-inserted-all_2.fq.gz

On lance la simulation
#+begin_src
nextflow run main.nf -profile standard,helios --input=samples-synthetic.csv --genome=GRCh38  -bg
#+end_src
***** DONE Résultat après haplopecaller: 3 reads manquants mais logique
CLOSED: [2023-08-16 Wed 19:13]
Comparaison naive: Haplotypecaller 146 found over 149
Manquant:
 Row │ chrom    pos       ref      alt      zygosity  meanQual  stdQual  depth
     │ String7  Int64     String1  String1  String7   Float64   Float64  Int64
─────┼─────────────────────────────────────────────────────────────────────────
   1 │ chr12    13720138  C        T        het           60.0    NaN        1
   2 │ chr17    10296150  T        A        het           60.0      0.0      3
   3 │ chr21    43426167  C        T        het            0.0      0.0     88

   Donc pas assez de reads pour les 2 premiers et le dernier a des reads de mauvaise qualité (mais bien inséré).
   On est limité par le système d'un patient....
***** DONE Résultat après filter depth: 30 variants perdu mais logique
CLOSED: [2023-08-16 Wed 20:18] SCHEDULED: <2023-08-16 Wed>
On perd 30 variant
13×8 DataFrame
 Row │ chrom    pos        ref      alt      variant              meanQual  stdQual    depth
     │ String7  Int64      String1  String1  String               Float64   Float64    Int64
─────┼───────────────────────────────────────────────────────────────────────────────────────
   1 │ chr3      71112628  C        T        chr3:g.71112628C>T    60.0       0.0         62
   2 │ chr12     13720138  C        T        chr12:g.13720138C>T   60.0     NaN            1
   3 │ chr12     40367710  A        G        chr12:g.40367710A>G   59.3333    3.30289     45
   4 │ chr14     58458545  G        A        chr14:g.58458545G>A   60.0       0.0          9
   5 │ chr15     66703292  C        T        chr15:g.66703292C>T   60.0       0.0         33
   6 │ chr16     30965737  C        A        chr16:g.30965737C>A   60.0       0.0         18
   7 │ chr17     10296150  T        A        chr17:g.10296150T>A   60.0       0.0          3
   8 │ chr17     61968202  A        C        chr17:g.61968202A>C   60.0       0.0         46
   9 │ chr21     43426167  C        T        chr21:g.43426167C>T    0.0       0.0         88
  10 │ chrX     124056226  T        G        chrX:g.124056226T>G   60.0       0.0         40
  11 │ chrX      24737739  G        T        chrX:g.24737739G>T    60.0       0.0         16
  12 │ chrX      40591349  C        T        chrX:g.40591349C>T    60.0       0.0         37
  13 │ chrX      53193275  G        A        chrX:g.53193275G>A    60.0       0.0         32

 Le filtre est sur DP (read depth) et non la profondeur. Sur 1 et 3, DP=29 et 28...
 Si on est hétérozygote, le DP est la moitié de la profondeur donc c'est logique...
***** DONE Résultat après filter polymorphism : idem
CLOSED: [2023-08-16 Wed 20:18]
***** Résultat après filtre VEP

**** KILL [#B] Tout insérer dans NA12878 avec XAMscissors: reads manquants
CLOSED: [2023-08-19 Sat 20:04] SCHEDULED: <2023-08-14 Mon>
***** KILL Insertion
CLOSED: [2023-08-19 Sat 20:03]
On récupère le BAM
#+begin_src sh :dir /home/alex/code/bisonex/out
rsync -avz  meso:/Work/Users/apraga/bisonex/out/2300346867_NA12878-63118093_S260-GRCh38/preprocessing/mapped 2300346867_NA12878-63118093_S260-GRCh38/preprocessing/
#+end_src
#+begin_src sh :dir /home/alex/roam/research/bisonex/code/sanger
julia --project=.. xamscissors.jl
cd out-all
mv inserted_1.fq.gz NA12878-sanger-inserted-all_1.fq.gz
mv inserted_2.fq.gz NA12878-sanger-inserted-all_2.fq.gz

rsync -avz NA12878-sanger-inserted_* meso:/Work/Users/apraga/bisonex/data/

#+end_src
Fichier de configuration

patient,sample,fastq1,fastq2
NA12878,sanger-all,data/NA12878-sanger-inserted-all_1.fq.gz,data/NA12878-sanger-inserted-all_2.fq.gz

On lance la simulation
#+begin_src
nextflow run main.nf -profile standard,helios --input=samples-synthetic.csv --genome=GRCh38  -bg
#+end_src
***** DONE Résultat après haplotyecaller: 3 varinat perdus => ok
CLOSED: [2023-08-17 Thu 19:13]
Haplotypecaller 143 found over 146
3×3 DataFrame
| variant             | meanQual | depth |
|---------------------+----------+-------|
| chr12:g.13720138C>T |     60.0 |     1 |
| chr17:g.10296150T>A |     60.0 |     3 |
| chr21:g.43426167C>T |      0.0 |    88 |
Pas assez de read (1,2) et problème d'alignement (3)
***** KILL Résultat après filtre depth : +10 variants perduis
CLOSED: [2023-08-19 Sat 20:04] SCHEDULED: <2023-08-18 Fri>
filter depth : another 10 missed variants
10×3 DataFrame
| variant             | meanQual | depth |
|---------------------+----------+-------|
| chr3:g.71112628C>T  |     60.0 |    62 |
| chr12:g.40367710A>G |  58.0435 |    46 |
| chr14:g.58458545G>A |     60.0 |     9 |
| chr15:g.66703292C>T |     60.0 |    33 |
| chr16:g.30965737C>A |     60.0 |    18 |
| chr17:g.61968202A>C |     60.0 |    46 |
| chrX:g.124056226T>G |     60.0 |    40 |
| chrX:g.24737739G>T  |     60.0 |    16 |
| chrX:g.40591349C>T  |     60.0 |    37 |
| chrX:g.53193275G>A  |     60.0 |    32 |
|                     |          |       |
S'ils sont hétérozygotes, 0.5*depth est effectivement < 30 (notre filtre...)
****** KILL Problème d'inserstion des reads: on en perd de nombreux ! -> regénérer données
CLOSED: [2023-08-19 Sat 20:04] SCHEDULED: <2023-08-18 Fri>
Ex: chrX:g.124056226T>G : on passe de 65 reads à 1
***** DONE Résultat après filtre common variant: +0 ok
CLOSED: [2023-08-17 Thu 19:32]
***** KILL Résultat après filtre VEP : +23 perdus ??
CLOSED: [2023-08-19 Sat 20:04] SCHEDULED: <2023-08-18 Fri>
filter vep : another 23 missed variants
23×3 DataFrame
 Row │ variant               meanQual  depth
     │ String                Float64   Int64
─────┼───────────────────────────────────────
   1 │ chr1:g.183222115C>T    60.0       168
   2 │ chr1:g.39388062C>T     60.0       285
   3 │ chr2:g.240719197G>C    60.0        77
   4 │ chr3:g.41227353G>C     60.0       105
   5 │ chr4:g.15536991T>G     60.0        41
   6 │ chr5:g.14474096G>A     60.0       191
   7 │ chr8:g.43122149C>T     60.0       237
   8 │ chr9:g.128603589A>C    60.0       304
   9 │ chr9:g.137452819G>C    60.0       107
  10 │ chr10:g.129957338T>C   60.0       116
  11 │ chr10:g.247389T>G      60.0        56
  12 │ chr11:g.61313668G>A    60.0        83
  13 │ chr12:g.45850467C>T    60.0       291
  14 │ chr14:g.64216315C>G    60.0       263
  15 │ chr15:g.60514655G>A    60.0       259
  16 │ chr17:g.61966475G>T    60.0       144
  17 │ chr17:g.7852503T>C     60.0       190
  18 │ chr19:g.13230158G>A    60.0       172
  19 │ chr19:g.38523211C>G    60.0        93
  20 │ chr19:g.4110557G>C     59.9929    425
  21 │ chr20:g.62334188G>A    60.0        62
  22 │ chrX:g.47575255G>A     60.0       244
  23 │ chrX:g.53409112G>A     60.0       136

**** DONE [#A] Tout insérer dans NA12878 avec XAMscissors (XAMScissors à jour)
CLOSED: [2023-08-20 Sun 13:45] SCHEDULED: <2023-08-19 Sat>
***** DONE Insertion
CLOSED: [2023-08-20 Sun 09:15]
***** DONE Vérifier après haplotypecaller: 3 variants manquant mais ok
CLOSED: [2023-08-20 Sun 09:18] SCHEDULED: <2023-08-20 Sun>

3×3 DataFrame
 Row │ variant              meanQual  depth
     │ String               Float64   Int64
─────┼──────────────────────────────────────
   1 │ chr12:g.13720138C>T      60.0      1
   2 │ chr17:g.10296150T>A      60.0      1
   3 │ chr21:g.43426167C>T       0.0     59
Manque de profondeur sur 2 et mauvaise qualité sur 3

***** DONE Vérifier après filterdepth: 0 perdus en plus
CLOSED: [2023-08-20 Sun 09:18] SCHEDULED: <2023-08-20 Sun>
***** DONE Vérifier après filterpolymorphis : 0 perdus en plus
CLOSED: [2023-08-20 Sun 09:18] SCHEDULED: <2023-08-20 Sun>
***** DONE Vérifier après filter vep: 2 perdus en plus
CLOSED: [2023-08-20 Sun 12:37] SCHEDULED: <2023-08-20 Sun>
2×3 DataFrame
 Row │ variant             meanQual  depth
     │ String              Float64   Int64
─────┼─────────────────────────────────────
   1 │ chr17:g.7852503T>C      60.0     96
   2 │ chrX:g.47575255G>A      60.0    145
***** DONE 1ere correction spip: meilleur nombre de variants en sortie mais manque toujours ces 2
CLOSED: [2023-08-20 Sun 11:38]
***** DONE --pick : résout le problème
CLOSED: [2023-08-20 Sun 12:37]
chrX:g.47575255G>A est rendu downstream_gene_variant avec l'option --pick
Or il n'est pas en5' dans les transcrits refseq...
https://genome-euro.ucsc.edu/cgi-bin/hgTracks?db=hg38&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chrX%3A47575242%2D47575268&hgsid=301211823_xpelPqPJije7wSIhg070JeGH5ZwV

https://mobidetails.iurc.montp.inserm.fr/MD/api/variant/238296/browser/

Idem pour l'autre
   chr17:g.7852503T>C
https://mobidetails.iurc.montp.inserm.fr/MD/api/variant/182993/browser/

Note:

VEP chooses one block of annotation per variant, using an ordered set of criteria. This order may be customised using --pick_order.

    MANE Select transcript status
    MANE Plus Clinical transcript status
    canonical status of transcript
    APPRIS isoform annotation
    transcript support level
    biotype of transcript ("protein_coding" preferred)
    CCDS status of transcript
    consequence rank according to this table
    translated, transcript or feature length (longer preferred)

"Wherever possible we would discourage you from summarising data in this way. "

**** DONE Mail alexis
CLOSED: [2023-08-20 Sun 13:45] SCHEDULED: <2023-08-20 Sun>
**** TODO Données simuscop 200x
SCHEDULED: <2023-10-22 Sun>
**** DONE En T2T avec liftover (filtre = spip) : ok mais lent et trop de variants :tests:
CLOSED: [2023-09-17 Sun 17:13] SCHEDULED: <2023-09-17 Sun>
1. Conversion en bed
#+begin_src sh :dir:~/code/sanger
open snvs-cento-sanger.csv | select chrom pos | insert pos2 {$in.pos } | to csv --separator="\t" | save snvs-cento-sanger.bed -f
#+end_src
2. Liftover avec UCSC (en ligne)
NB: vérifié sur le premier résultat en cherche le read contenant le variant (samtools view -r puis samtools view | grep en T2T) et avec l'aide d'IGV, on a un variant qui correspond en
chr1:10757746
3. En supposant que l'ordre des variants n'a pas changé, on ajoute simplement REF et ALT avec annotateLifted.jl

Annotation spip *très lente* : 1h13 !

Résultat:
2×3 DataFrame
 Row │ variant              meanQual  depth
     │ String               Float64   Int64
─────┼──────────────────────────────────────
   1 │ chr12:g.13594572      60.0      1
   2 │ chr17:g.10204026      60.0      1
144 found over 146
filter depth : another 0 missed variants
filter poly : another 0 missed variants
filter vep   : another 0 missed variants

Et on a trop de variants en sortie (7330 !)
**** DONE Mail Paul avec résultats filtre en T2T + nouveau schéma
CLOSED: [2023-09-17 Sun 23:15] SCHEDULED: <2023-09-17 Sun>
** TODO Medically relevant genes
SCHEDULED: <2023-10-25 Wed>
/Entered on/ [2023-10-18 Wed 22:37]
* Ré-interprétation :reanalysis:
** DONE Lancer tests sur données brutes [225/250] <(samples.csv)>  <(runs.waiting)>
CLOSED: [2023-10-14 Sat 11:58] SCHEDULED: <2023-10-08 Sun>
- [X] 100222_63015289
- [X] 1600304839_63051311
- [X] 1900007827_62913191
- [X] 1900398899_62999500
- [X] 1900486799_62913197
- [X] 2100422923_62952677
- [X] 2100458888_62933047
- [X] 2100601558_62903840
- [X] 2100609288_62905768
- [X] 2100609501_62905776
- [X] 2100614493_62951074
- [X] 2100622566_62908067
- [X] 2100622601_62908060
- [X] 2100622705_62908063
- [X] 2100640027_62911936
- [X] 2100645285_62913212
- [X] 2100661411_62914081
- [X] 2100661462_62914086
- [X] 2100708257_62921596
- [X] 2100738732_62926501
- [X] 2100738850_62926509
- [X] 2100746751_62926505
- [X] 2100746797_62926506
- [X] 2100782349_62931722
- [X] 2100782416_62931561
- [X] 2100782559_62931718
- [X] 2100799204_62934768
- [X] 2200010202_62940284
- [X] 2200023600_62940631
- [X] 2200024348_62999591
- [X] 2200027505_62942457
- [X] 2200038776_62943412
- [X] 2200041919_62943405
- [X] 2200088014_62951326
- [X] 2200146652_62959388
- [X] 2200151850_62960953
- [X] 2200160014_62959475
- [X] 2200160070_62959478
- [X] 2200201368_62967471
- [X] 2200201400_62967470
- [X] 2200265558_62976332
- [X] 2200265605_62976401
- [X] 2200267046_62975192
- [X] 2200273878_62999530
- [X] 2200279708_62977002
- [X] 2200284408_62979102
- [X] 2200293987_62979116
- [X] 2200294359_62979118
- [X] 2200306299_62982217
- [X] 2200306539_62982193
- [X] 220030671_62982211
- [X] 2200307058_62982231
- [X] 2200307108_62982196
- [X] 2200307136_62982221
- [X] 2200307199_62982239
- [X] 2200307230_62982234
- [X] 2200307262_62982219
- [X] 2200307297_62982227
- [X] 2200324510_62985453
- [X] 2200324549_62985478
- [X] 2200324573_62985445
- [X] 2200324594_62985467
- [X] 2200324606_62985463
- [X] 2200324614_62985459
- [X] 2200338306_62985430
- [X] 2200343880_62989407
- [X] 2200343910_62989460
- [X] 2200343938_62989451
- [X] 2200343966_62989456
- [X] 2200343993_62989440
- [X] 2200344013_62989464
- [X] 2200349749_62989465
- [X] 2200363462_62988848
- [X] 2200377880_62991993
- [X] 2200378032_62991991
- [X] 2200383996_62993828
- [X] 2200384015_62993796
- [X] 2200384046_62993822
- [X] 2200384117_62993808
- [X] 2200384187_62993825
- [X] 2200384231_62992898
- [X] 2200385658_63060260
- [X] 2200394260_62994732
- [X] 2200395817_62994742
- [X] 2200396731_62994737
- [X] 2200424073_62999579
- [X] 2200424207_62999632
- [X] 2200426178_62999630
- [X] 2200426243_62999635
- [X] 2200426466_62999605
- [X] 2200426642_62999627
- [X] 2200427406_62999649
- [X] 2200427512_62999639
- [X] 2200428953_62999572
- [X] 2200428981_62999600
- [X] 2200428999_62999592
- [X] 2200441970_63000868
- [X] 2200441989_63000882
- [X] 2200442135_63000864
- [X] 2200442216_63000886
- [X] 2200442257_63000951
- [X] 2200451801_63003573
- [X] 2200451862_63004218
- [X] 2200451894_63004210
- [X] 2200456165_63051294
- [X] 2200459865_63004933
- [X] 2200459968_63004937
- [X] 2200460073_63004943
- [X] 2200460121_63004684
- [X] 2200467051_63003856
- [X] 2200467225_63004940
- [X] 2200467261_63004930
- [X] 2200467338_63004925
- [X] 2200470099_63004485
- [X] 2200470142_63004480
- [X] 2200471780_63004362
- [X] 2200480910_63006466
- [X] 2200495073_63010427
- [X] 2200495510_63009152
- [X] 2200508677_63060252
- [X] 2200510531_63012582
- [X] 2200510628_63012549
- [X] 2200510657_63012554
- [X] 2200511249_63012533
- [X] 2200511274_63012586
- [X] 2200517952_63060399
- [X] 2200519525_63060439
- [X] 2200524009_63014044
- [X] 2200524609_63014046
- [X] 2200524616_63014048
- [X] 2200533429_63060425
- [X] 2200539735_63060406
- [X] 2200549908_63019339
- [X] 2200549965_63019349
- [X] 2200550414_63019357
- [X] 2200550471_63020031
- [X] 2200550490_63019351
- [X] 2200550505_63019340
- [X] 2200555565_63018614
- [X] 2200559438_63020029
- [X] 2200559682_63020030
- [X] 2200559713_63019623
- [X] 2200559739_63019626
- [X] 2200569969_63019991
- [X] 2200570001_63021580
- [X] 2200570025_63021490
- [X] 2200570035_63021491
- [X] 2200570042_63021493
- [X] 2200570050_63021494
- [X] 2200579897_63024910
- [X] 2200583995_63024866
- [X] 2200584035_63024905
- [X] 2200584069_63024888
- [X] 2200584126_63024810
- [X] 2200589507_63026712
- [X] 2200597365_63027994
- [X] 2200597480_63027988
- [X] 2200597752_63026853
- [X] 2200597778_63027992
- [X] 22005977_63026903
- [X] 2200609031_63026527
- [X] 2200614198_63113928
- [X] 2200620372_63030821
- [X] 2200620442_63030810
- [X] 2200620498_63030816
- [X] 2200620628_63031031
- [X] 2200622310_63030984
- [X] 2200622355_63030956
- [X] 2200625369_63028699
- [X] 2200625410_63028697
- [X] 2200625536_63028694
- [X] 2200630189_63030665
- [X] 2200635149_63033182
- [X] 2200644544_63037731
- [X] 2200644594_63037725
- [X] 2200650089_63038093
- [X] 2200666292_63076568
- [X] 2200669188_63036688
- [X] 2200669320_63040259
- [X] 2200669383_63040254
- [X] 2200669414_63040257
- [X] 2200669446_63040251
- [X] 2200680342_63105271
- [X] 2200694535_63042853
- [X] 2200694789_63042862
- [X] 2200694858_63042702
- [X] 2200694917_63042696
- [X] 2200699290_63043047
- [X] 2200699345_63040238
- [X] 2200699383_63043050
- [X] 2200699412_63040731
- [X] 220071551_63048935
- [X] 2200731515_63048963
- [X] 2200748145_63051198
- [X] 2200748171_63051213
- [X] 2200751046_63051249
- [X] 2200751101_63051234
- [X] 2200766471_63054590
- [X] 2200767731_63054595
- [X] 2200767822_63054464
- [X] 2200775505_63060410
- [X] 2200850441_63019345
- [X] 220597589_63026879
- [X] 2300003253_63060430
- [X] 2300005679_63060370
- [X] 2300009914_63060390
- [X] 2300028784_63060001
- [X] 2300036815_63063357
- [X] 2300055382_63061874
- [X] 2300055421_63061871
- [X] 2300055440_63061880
- [X] 230006894_63064950
- [X] 2300071111_63070356
- [X] 2300083434_63071675
- [X] 2300103609_63076239
- [X] 2300104572_63076232
- [X] 2300109602_63076765
- [X] 2300109665_63076770
- [X] 2300119721_63078732
- [X] 2300137773_63078133
- [X] 2300137834_63078123
- [X] 2300167821_63086183
- [X] 2300172698_63113453
- [X] 2300188216_63090609
- [X] 2300188281_63090632
- [ ] 2300188800_63090616
- [ ] 2300193193645_63090623
- [ ] 2300193668_63090611
- [ ] 2300195426_63090608
- [ ] 2300201017_63089636
- [ ] 2300227479_63098330
- [ ] 2300232688_63130821
- [ ] 2300292749_63109239
- [ ] 230029277_63109247
- [ ] 2300294712_63109236
- [ ] 2300308032_63111581
- [ ] 2300323537_63114209
- [ ] 2300334609_63115535
- [ ] 2300346867_63118093
- [ ] 2300346867_63118093_NA12878
- [ ] 2300348940_63118099
- [ ] 2300359806_63119915
- [ ] 2300380476_63123963
- [ ] 2300382582_63123749
- [ ] 2300384269_63126867
- [ ] 2300407581_63130826
- [ ] 2300407626_63130842
- [ ] 2300409593_63130874
- [ ] 2300409612_63130980
- [ ] 2300417623_63131524
** TODO Variants manqués :missed:
SCHEDULED: <2023-10-21 Sat>
*** DONE 63012582: chr10:g.102230760 filtré par AD :63012582:
CLOSED: [2023-10-08 Sun 23:24] SCHEDULED: <2023-10-08 Sun>
Il est en sortie d'haplotypecaller !
Attention à la position : POS=102230753  noté CG->C
GT:AD:DP:GQ:PL	0/1:26,8:34:99:146,0,671

Filtré par la condition AD <= 10 (porté par 8 reads seulement)

Non confirméen sanger, rendu vous
**** KILL image BAM cento
CLOSED: [2023-10-08 Sun 23:13]
**** DONE image BAM bisonex
CLOSED: [2023-10-08 Sun 23:23] SCHEDULED: <2023-10-08 Sun>
**** DONE Mail Paul
CLOSED: [2023-10-08 Sun 23:24] SCHEDULED: <2023-10-08 Sun>
*** DONE 63060439: chr15:g.26869324 = Problème de profondeur DP=15 :63060439:
CLOSED: [2023-10-08 Sun 23:24] SCHEDULED: <2023-10-08 Sun>
GABRA5
Rendu VOUS avec un variant patho MDB5 pour même patient (VOUS- même)
Non confirmé en Sanger
GT:AD:DP:GQ:PL 0/1:9,6:15:99:103,0,213
**** DONE image BAM bisonex
CLOSED: [2023-10-08 Sun 22:56]
**** DONE Mail Paul
CLOSED: [2023-10-08 Sun 23:24] SCHEDULED: <2023-10-08 Sun>
** TODO Comparer variants cento à sortie bisonex :checkpipeline:
SCHEDULED: <2023-10-21 Sat>
*** TODO Ajouter négatifs dans la liste des variants
SCHEDULED: <2023-10-21 Sat>
*** TODO Mettre à jour liste des variants
SCHEDULED: <2023-10-21 Sat>
*** KILL Lifter coordonées variants cento génomique en GRCh38
CLOSED: [2023-10-21 Sat 22:47]
*** TODO Un seul exécutable pour toutes les étapes
SCHEDULED: <2023-10-21 Sat>
Un utilitaire en ligne de commande qui appel les différentes étapes.
On utilise une structure unique pour toutes les étapes mais qui sera remplie au fur et à mesure. En stockant dans un csv à chaque étape
**** DONE parse variants
CLOSED: [2023-10-21 Sat 23:29] SCHEDULED: <2023-10-21 Sat>
**** TODO Un seul type de données
SCHEDULED: <2023-10-21 Sat>
**** TODO annote avec mobidetails
SCHEDULED: <2023-10-21 Sat>
**** TODO compare chaque variant avec la sortie du pipeline
SCHEDULED: <2023-10-21 Sat>
**** TODO confirme ou non en Sanger
SCHEDULED: <2023-10-21 Sat>
* Résultats
** TODO Speed-up BWA-mem
SCHEDULED: <2023-10-22 Sun>

** TODO Speed-up Hapotypecaller
SCHEDULED: <2023-10-22 Sun>
* Communication
** DONE Mail NGS-diag
CLOSED: [2023-10-06 Fri 08:04] SCHEDULED: <2023-10-06 Fri>
/Entered on/ [2023-10-04 Wed 19:33]
