# Classification ACMG
#moleculaire

| Classe            | Critère 1                    | AND one of one of               |  
|-------------------|------------------------------|--------------------------|
| Pathogenic        | 1 Very strong                | ≥1 Strong  
|                   |                              | ≥2 Moderate 
|                   |                              | 1 Moderate and 1 supporting
|                   |                              | ≥2 Supporting 
|                   |   ≥2 Strong                  | 
|                   |  1 Strong                    | ≥3 Moderate  
|                   |                              | 2 Moderate  AND ≥2 Supporting  
|                   |                              | 1 Moderate  AND ≥4 supporting 
| Likely pathogenic |  1 Very strong               | 1 moderate  
|                   |  1 Strong                    | 1–2 moderate  
|                   |  1 Strong                    | ≥2 supporting  
|                   |  ≥3 Moderate                 |
|                   |  2 Moderate                  | ≥2 supporting  
|                   |  1 Moderate                  |  ≥4 supporting 
| Benign            |  1 Stand-alone  
|                   |  ≥2 Strong 
| Likely benign     |1 Strong  and                 | 1 supporting  
|                   |  ≥2 Supporting 

Interpretation of sequence variants | RICHARDS et al ACMG Standards and Guidelines
large frequency differences, and the Mendelian disease under
study is early onset. Patients referred to a clinical laboratory
for testing are likely to include individuals sent to “rule out”
a disorder, and thus they may not qualify as well-phenotyped
cases. When using a general population as a control cohort,
the presence of individuals with subclinical disease is always a
possibility. In both of these scenarios, however, a case–control
comparison will be underpowered with respect to detecting a
difference; as such, showing a statistically significant difference
can still be assumed to provide supportive evidence for pathogenicity,
as noted above. By contrast, the absence of a statistical
difference, particularly with extremely rare variants and less
penetrant phenotypes, should be interpreted cautiously.
Odds ratios (ORs) or relative risk is a measure of association
between a genotype (i.e., the variant is present in the
genome) and a phenotype (i.e., affected with the disease/
outcome) and can be used for either Mendelian diseases or
complex traits. In this guideline we are addressing only its
use in Mendelian disease. While relative risk is different from
the OR, relative risk asymptotically approaches ORs for small
probabilities. An OR of 1.0 means that the variant does not
affect the odds of having the disease, values above 1.0 mean
there is an association between the variant and the risk of disease,
and those below 1.0 mean there is a negative association
between the variant and the risk of disease. In general, variants
with a modest Mendelian effect size will have an OR of
3 or greater, whereas highly penetrant variants will have very
high ORs; for example, APOE E4/E4 homozygotes compared
with E3/E3 homozygotes have an OR of 13 (https://www.tgen.
org/home/education-outreach/past-summer-interns/2012-
summer-interns/erika-kollitz.aspx#.VOSi3C7G_vY).
However, the confidence interval (CI) around the OR is as
important as the measure of association itself. If the CI includes
1.0 (e.g., OR = 2.5, CI = 0.9–7.4), there is little confidence in
Figure 1 Evidence framework. This chart organizes each of the criteria by the type of evidence as well as the strength of the criteria for a benign (left side)
or pathogenic (right side) assertion. Evidence code descriptions can be found in Tables 3 and 4. BS, benign strong; BP, benign supporting; FH, family history;
LOF, loss of function; MAF, minor allele frequency; path., pathogenic; PM, pathogenic moderate; PP, pathogenic supporting; PS, pathogenic strong; PVS,
pathogenic very strong.

|             | B. Strong                                                                                                 | B. Supporting | P. Supporting | P. Moderate | P. Strong                           | P. Very strong                                                    |
| --            | -- | -- | -- | -- | -- | -- |
| Population  |  MAF is too high for disorder BA1/BS1 OR observation in controls inconsistent with disease penetrance BS2 |               |               |             |  Absent in population databases PM2 | Prevalence in affecteds statistically increased over controls PS4 |

Computational |
Segregation |
Other


Predicted null
variant in a gene
where LOF is a
known
mechanism of
disease
PVS1
De novo (paternity and
maternity confirmed)
PS2
Well-established
functional studies
show a deleterious
effect PS3
Novel missense change
at an amino acid residue
where a different
pathogenic missense
change has been seen
before PM5
Protein length changing
variant PM4
Multiple lines of
computational
evidence support a
deleterious effect
on the gene /gene
product PP3
De novo (without
paternity & maternity
confirmed) PM6
Nonsegregation
with disease BS4
Patient’s phenotype or
FH highly specific for
gene PP4
For recessive
disorders, detected
in trans with a
pathogenic variant
PM3
Found in case with
an alternate cause
BP5
Multiple lines of
computational evidence
suggest no impact on gene
/gene product BP4
Missense in gene where
only truncating cause
disease BP1
Silent variant with non
predicted splice impact BP7
In-frame indels in repeat
w/out known function BP3
Well-established
functional studies show
no deleterious effect
BS3
Mutational hot spot
or well-studied
functional domain
without benign
variation PM1
Same amino acid
change as an
established
pathogenic variant
PS1
Observed in trans with
a dominant variant BP2
Observed in cis with a
pathogenic variant BP2
Functional
data
Cosegregation with
disease in multiple
affected family
members PP1
De novo
data
Allelic data
Strong
Reputable source w/out
shared data = benign BP6
Supporting Supporting Moderate Strong Very strong
Reputable source
= pathogenic PP5
Missense in gene with
low rate of benign
missense variants and
path. missenses
common PP2
Other data
Benign Pathogenic
Increased segregation data
Genetics in medicine
